diff --git a/sympy/simplify/simplify.py b/sympy/simplify/simplify.py
--- a/sympy/simplify/simplify.py
+++ b/sympy/simplify/simplify.py
@@ -1455,10 +1455,7 @@
     sympy.core.function.nfloat
 
     """
-    try:
-        return sympify(as_int(expr))
-    except (TypeError, ValueError):
-        pass
+    expr = sympify(expr)
     expr = sympify(expr).xreplace({
         Float('inf'): S.Infinity,
         Float('-inf'): S.NegativeInfinity,
         })