diff --git a/lib/matplotlib/axis.py b/lib/matplotlib/axis.py
--- a/lib/matplotlib/axis.py
+++ b/lib/matplotlib/axis.py
@@ -2288,6 +2288,9 @@
         - "unknown" otherwise.
         """
         representative_ticks = []
+        if not isinstance(self.get_major_locator(), NullLocator):
+            representative_ticks.append(self.majorTicks[0])
+
         if all(tick.tick1line.get_visible()
                and not tick.tick2line.get_visible()
                and tick.label1.get_visible()
