diff --git a/pylint/lint/parallel.py b/pylint/lint/parallel.py
--- a/pylint/lint/parallel.py
+++ b/pylint/lint/parallel.py
@@ -60,7 +60,7 @@ def _worker_initialize(
     enable = copy(_worker_linter.config.enable)
     disable = copy(_worker_linter.config.disable)
     _worker_linter._reset_msgs_state()
-    _worker_linter.deregister_checkers(
+    _worker_linter._deregister_checkers(
         _worker_linter._registered_dynamic_plugin_checkers
     )
     _worker_linter.load_plugin_modules(_worker_linter._dynamic_plugins, force=True)
