diff --git a/astropy/wcs/wcsapi/high_level_api.py b/astropy/wcs/wcsapi/high_level_api.py
--- a/astropy/wcs/wcsapi/high_level_api.py
+++ b/astropy/wcs/wcsapi/high_level_api.py
@@ -298,7 +298,11 @@
     # arrays, not e.g. Quantity. Note that we deliberately use type(w) because
     # we don't want to match Numpy subclasses.
     for w in world_values:
-        if not isinstance(w, numbers.Number) and not type(w) == np.ndarray:
+        if (
+            not isinstance(w, numbers.Number)
+            and not type(w) == np.ndarray
+            and not type(w) == MaskedNDArray
+        ):
             raise TypeError(
                 f"Expected world coordinates as scalars or plain Numpy "
                 f"arrays (got {type(w)})"
