diff --git a/sympy/polys/densetools.py b/sympy/polys/densetools.py
--- a/sympy/polys/densetools.py
+++ b/sympy/polys/densetools.py
@@ -926,8 +926,8 @@ def dmp_shift(f, a, u, K):
 
     a0, a1 = a[0], a[1:]
 
-    f = [ dmp_shift(c, a1, u-1, K) for c in f ]
-    n = len(f) - 1
+    if any(a1):
+        f = [ dmp_shift(c, a1, u-1, K) for c in f ]
 
     for i in range(n, 0, -1):
         for j in range(0, i):

