diff --git a/sympy/polys/densetools.py b/sympy/polys/densetools.py
--- a/sympy/polys/densetools.py
+++ b/sympy/polys/densetools.py
@@ -928,13 +928,9 @@
 
     if any(a1):
         f = [ dmp_shift(c, a1, u-1, K) for c in f ]
-
-    for i in range(n, 0, -1):
-        for j in range(0, i):
-            afj = dmp_mul_ground(f[j], a0, u-1, K)
-            f[j + 1] = dmp_add(f[j + 1], afj, u-1, K)
-
-    return f
+    else:
+        f = list(f)
+
 
 
 def dup_transform(f, p, q, K):
