diff --git a/xarray/computation/computation.py b/xarray/computation/computation.py
--- a/xarray/computation/computation.py
+++ b/xarray/computation/computation.py
@@ -657,11 +657,11 @@ def where(cond, x, y, keep_attrs=None):
     y : scalar, array, Variable, DataArray or Dataset
         values to choose from where `cond` is False
     keep_attrs : bool or {"drop", "identical", "no_conflicts", "drop_conflicts", "override"} or callable, optional
-        - 'drop' or False (default): empty attrs on returned xarray object.
+        - 'override' or True (default): skip comparing and copy attrs from `x` to the result.
+        - 'drop' or False: empty attrs on returned xarray object.
         - 'identical': all attrs must be the same on every object.
         - 'no_conflicts': attrs from all objects are combined, any that have the same name must also have the same value.
         - 'drop_conflicts': attrs from all objects are combined, any that have the same name but different values are dropped.
-        - 'override' or True: skip comparing and copy attrs from `x` to the result.
 
     Returns
     -------
@@ -701,7 +701,8 @@ def where(cond, x, y, keep_attrs=None):
     Coordinates:
       * lat      (lat) int64 80B 0 1 2 3 4 5 6 7 8 9
 
-    Preserve the attrs on `cond` by using `keep_attrs="drop_conflicts"`
+    If `x` is a scalar (and therefore has no attrs), preserve the
+    attrs on `cond` by using `keep_attrs="drop_conflicts"`
     >>> xr.where(x < 0.5, 1, 0, keep_attrs="drop_conflicts")
     <xarray.DataArray 'sst' (lat: 10)> Size: 80B
     array([1, 1, 1, 1, 1, 0, 0, 0, 0, 0])
