diff --git a/xarray/structure/alignment.py b/xarray/structure/alignment.py
--- a/xarray/structure/alignment.py
+++ b/xarray/structure/alignment.py
@@ -200,7 +200,7 @@
         for k, idx in indexes.items():
             if not isinstance(idx, Index):
                 if getattr(idx, "dims", (k,)) != (k,):
-                    raise ValueError(
+                    raise AlignmentError(
                         f"Indexer has dimensions {idx.dims} that are different "
                         f"from that to be indexed along '{k}'"
                     )
@@ -231,7 +231,7 @@
             elif exclude_dims:
                 excl_dims_str = ", ".join(str(d) for d in exclude_dims)
                 incl_dims_str = ", ".join(str(d) for d in all_dims - exclude_dims)
-                raise ValueError(
+                raise AlignmentError(
                     f"cannot exclude dimension(s) {excl_dims_str} from alignment because "
                     "these are used by an index together with non-excluded dimensions "
                     f"{incl_dims_str}"
@@ -272,7 +272,7 @@
             for dim_sizes in all_indexes_dim_sizes.values():
                 for dim, sizes in dim_sizes.items():
                     if len(sizes) > 1:
-                        raise ValueError(
+                        raise AlignmentError(
                             "cannot align objects with join='override' with matching indexes "
                             f"along dimension {dim!r} that don't have the same size"
                         )
