diff --git a/lib/matplotlib/style/__init__.py b/lib/matplotlib/style/__init__.py
--- a/lib/matplotlib/style/__init__.py
+++ b/lib/matplotlib/style/__init__.py
@@ -246,7 +246,7 @@
 def reload_library():
     """Reload the style library."""
     library.clear()
-    library.update(_update_user_library(_base_library))
+    library.update(_update_user_library(_base_library.copy()))
     available[:] = sorted(library.keys())
 

