diff --git a/lib/matplotlib/style/__init__.py b/lib/matplotlib/style/__init__.py
--- a/lib/matplotlib/style/__init__.py
+++ b/lib/matplotlib/style/__init__.py
@@ -247,7 +247,7 @@
     """Reload the style library."""
     library.clear()
     library.update(_update_user_library(_base_library.copy()))
-    available[:] = sorted(name for name in library if not )
+    available[:] = sorted(name for name in library if not name.startswith('_'))
 
 
 reload_library()
