diff --git a/sympy/tensor/array/expressions/from_matrix_to_array.py b/sympy/tensor/array/expressions/from_matrix_to_array.py
--- a/sympy/tensor/array/expressions/from_matrix_to_array.py
+++ b/sympy/tensor/array/expressions/from_matrix_to_array.py
@@ -15,7 +15,13 @@
 from sympy.matrices.expressions.matexpr import MatrixExpr
 from sympy.tensor.array.expressions.array_expressions import \
     ArrayElementwiseApplyFunc, _array_tensor_product, _array_contraction, \
-    _array_diagonal, _array_add, _permute_dims, Reshape, get_shape
+    _array_diagonal, _array_add, _permute_dims, Reshape, get_shape, _ArrayExpr, _CodegenArrayAbstract
+
+
+def _array_elementwise_apply_func(function, element):
+    if not isinstance(element, (MatrixExpr, _ArrayExpr, _CodegenArrayAbstract)):
+        return function(element)
+    return ArrayElementwiseApplyFunc(function, element)
 
 
 def convert_matrix_to_array(expr: Basic) -> Basic:
@@ -58,20 +64,20 @@ def convert_matrix_to_array(expr: Basic) -> Basic:
         if (expr.exp > 0) == True:
             base_conv = convert_matrix_to_array(base)
             if get_shape(base_conv) == ():
-                return ArrayElementwiseApplyFunc(lambda x: x**expr.exp, base_conv)
+                return _array_elementwise_apply_func(lambda x: x**expr.exp, base_conv)
             else:
                 return _array_tensor_product(*[base_conv for i in range(expr.exp)])
         elif get_shape(base) == ():
             d = Dummy("d")
-            return ArrayElementwiseApplyFunc(Lambda(d, d**expr.exp), base)
+            return _array_elementwise_apply_func(Lambda(d, d**expr.exp), base)
         else:
             return expr
     elif isinstance(expr, MatPow):
         base = convert_matrix_to_array(expr.base)
         if expr.exp.is_Integer != True:
             if get_shape(expr) == (1, 1):
                 b = symbols("b", cls=Dummy)
-                return ArrayElementwiseApplyFunc(Lambda(b, b ** expr.exp), convert_matrix_to_array(base))
+                return _array_elementwise_apply_func(Lambda(b, b ** expr.exp), convert_matrix_to_array(base))
             return expr
         elif (expr.exp > 0) == True:
             return convert_matrix_to_array(MatMul.fromiter(base for i in range(expr.exp)))
@@ -87,7 +93,7 @@ def convert_matrix_to_array(expr: Basic) -> Basic:
             return convert_matrix_to_array(HadamardProduct.fromiter(base for i in range(exp)))
         else:
             d = Dummy("d")
-            return ArrayElementwiseApplyFunc(Lambda(d, d**exp), base)
+            return _array_elementwise_apply_func(Lambda(d, d**exp), base)
     elif isinstance(expr, KroneckerProduct):
         kp_args = [convert_matrix_to_array(arg) for arg in expr.args]
         permutation = [2*i for i in range(len(kp_args))] + [2*i + 1 for i in range(len(kp_args))]
