diff --git a/django/db/backends/base/schema.py b/django/db/backends/base/schema.py
--- a/django/db/backends/base/schema.py
+++ b/django/db/backends/base/schema.py
@@ -451,6 +451,10 @@
             default_sql %= tuple(self.prepare_default(p) for p in params)
             params = []
         return sql % default_sql, params
+
+    def _column_generated_persistency_sql(self, field):
+        """Return the SQL to define the persistency of generated fields."""
+        return "STORED" if field.db_persist else "VIRTUAL"
 
     def _column_generated_sql(self, field):
         """Return the SQL to use in a GENERATED ALWAYS clause."""
