diff --git a/sympy/vector/vector.py b/sympy/vector/vector.py
--- a/sympy/vector/vector.py
+++ b/sympy/vector/vector.py
@@ -422,10 +422,14 @@
 def _mul_vector_postprocessor(expr: Mul) -> Vector:
     return VectorMul(*expr.args).doit(deep=False)
 
+def _pow_vector_postprocessor(expr: Pow):
+    raise TypeError("Power operation is not supported for vectors")
+
 
 Basic._constructor_postprocessor_mapping[Vector] = {
     "Add": [_add_vector_postprocessor],
     "Mul": [_mul_vector_postprocessor],
+    "Pow": [_pow_vector_postprocessor],
 }
 
 class BaseVector(Vector, AtomicExpr):
@@ -546,7 +550,7 @@
         return obj
 
 
-class Cross(Vector):
+class Cross(Expr):
     """
     Represents unevaluated Cross product.
 
