diff --git a/doc/development/tutorials/examples/autodoc_intenum.py b/doc/development/tutorials/examples/autodoc_intenum.py
--- a/doc/development/tutorials/examples/autodoc_intenum.py
+++ b/doc/development/tutorials/examples/autodoc_intenum.py
@@ -37,9 +37,9 @@ def add_line(self, line: str, source: str = '', *lineno: int, indent: str) -> No
         analyzer_source = '' if self.analyzer is None else self.analyzer.srcname
         source_name = _docstring_source_name(props=self.props, source=analyzer_source)
         if line.strip():  # not a blank line
-            self.directive.result.append(indent + line, source_name, *lineno)
+            self.result.append(indent + line, source_name, *lineno)
         else:
-            self.directive.result.append('', source_name, *lineno)
+            self.result.append('', source_name, *lineno)
 
     def add_directive_header(self, *, indent: str) -> None:
         super().add_directive_header(indent=indent)
