diff --git a/lib/matplotlib/transforms.py b/lib/matplotlib/transforms.py
--- a/lib/matplotlib/transforms.py
+++ b/lib/matplotlib/transforms.py
@@ -2928,7 +2928,7 @@
     return _nonsingular(vmin, vmax, expander, tiny, increasing)
 
 
-def interval_contains(interval, val):
+def _interval_contains(interval, val):
     """
     Check, inclusively, whether an interval includes a given value.
 
@@ -2948,6 +2948,11 @@
     if a > b:
         a, b = b, a
     return a <= val <= b
+
+
+@_api.deprecated("3.11")
+def interval_contains(interval, val):
+    return _interval_contains(interval, val)
 
 
 def _interval_contains_close(interval, val, rtol=1e-10):
