diff --git a/lib/matplotlib/transforms.py b/lib/matplotlib/transforms.py
--- a/lib/matplotlib/transforms.py
+++ b/lib/matplotlib/transforms.py
@@ -2984,7 +2984,7 @@
     return a - rtol <= val <= b + rtol
 
 
-def interval_contains_open(interval, val):
+def _interval_contains_open(interval, val):
     """
     Check, excluding endpoints, whether an interval includes a given value.
 
@@ -3002,6 +3002,11 @@
     """
     a, b = interval
     return a < val < b or a > val > b
+
+
+@_api.deprecated("3.11")
+def interval_contains_open(interval, val):
+    return _interval_contains_open(interval, val)
 
 
 def offset_copy(trans, fig=None, x=0.0, y=0.0, units='inches'):
