diff --git a/lib/matplotlib/transforms.py b/lib/matplotlib/transforms.py
--- a/lib/matplotlib/transforms.py
+++ b/lib/matplotlib/transforms.py
@@ -2980,7 +2980,7 @@ def _interval_contains_close(interval, val, rtol=1e-10):
     return a - rtol <= val <= b + rtol
 
 
-def interval_contains_open(interval, val):
+def _interval_contains_open(interval, val):
     """
     Check, excluding endpoints, whether an interval includes a given value.
 
@@ -3000,6 +3000,12 @@ def interval_contains_open(interval, val):
     return a < val < b or a > val > b
 
 
+@_api.deprecated("3.11")
+def interval_contains_open(interval, val):
+    return _interval_contains_open(interval, val)
+_interval_contains_open.__doc__ = _interval_contains_open.__doc__
+
+
 def offset_copy(trans, fig=None, x=0.0, y=0.0, units='inches'):
     """
     Return a new transform with an added offset.
