diff --git a/sklearn/inspection/_plot/decision_boundary.py b/sklearn/inspection/_plot/decision_boundary.py
--- a/sklearn/inspection/_plot/decision_boundary.py
+++ b/sklearn/inspection/_plot/decision_boundary.py
@@ -498,7 +498,7 @@
             )
 
         prediction_method = _check_boundary_response_method(
-            estimator, response_method, class_of_interest
+            estimator, response_method
         )
         try:
             response, _, response_method_used = _get_response_values(
