diff --git a/lib/matplotlib/_mathtext.py b/lib/matplotlib/_mathtext.py
--- a/lib/matplotlib/_mathtext.py
+++ b/lib/matplotlib/_mathtext.py
@@ -1326,6 +1326,7 @@ def __init__(self, elements: T.Sequence[Node], w: float = 0.0,
         if do_kern:
             self.kern()
         self.hpack(w=w, m=m)
+        self.is_phantom = False
 
     def is_char_node(self) -> bool:
         # See description in Node.is_char_node.
@@ -1716,6 +1717,11 @@ def ship(box: Box, xy: tuple[float, float] = (0, 0)) -> Output:
     off_v = oy + box.height
     output = Output(box)
 
+    phantom: list[bool] = []
+    def render(node, *args):
+        if not any(phantom):
+            node.render(*args)
+
     def clamp(value: float) -> float:
         return -1e9 if value < -1e9 else +1e9 if value > +1e9 else value
 
@@ -1729,9 +1735,11 @@ def hlist_out(box: Hlist) -> None:
         base_line = cur_v
         left_edge = cur_h
 
+        phantom.append(box.is_phantom)
+
         for p in box.children:
             if isinstance(p, Char):
-                p.render(output, cur_h + off_h, cur_v + off_v)
+                render(p, output, cur_h + off_h, cur_v + off_v)
                 cur_h += p.width
             elif isinstance(p, Kern):
                 cur_h += p.width
@@ -1762,9 +1770,9 @@ def hlist_out(box: Hlist) -> None:
                     rule_depth = box.depth
                 if rule_height > 0 and rule_width > 0:
                     cur_v = base_line + rule_depth
-                    p.render(output,
-                             cur_h + off_h, cur_v + off_v,
-                             rule_width, rule_height)
+                    render(p, output,
+                           cur_h + off_h, cur_v + off_v,
+                           rule_width, rule_height)
                     cur_v = base_line
                 cur_h += rule_width
             elif isinstance(p, Glue):
@@ -1782,6 +1790,8 @@ def hlist_out(box: Hlist) -> None:
                 rule_width += cur_g
                 cur_h += rule_width
 
+        phantom.pop()
+
     def vlist_out(box: Vlist) -> None:
         nonlocal cur_v, cur_h
 
@@ -1821,9 +1831,9 @@ def vlist_out(box: Vlist) -> None:
                 rule_height += rule_depth
                 if rule_height > 0 and rule_depth > 0:
                     cur_v += rule_height
-                    p.render(output,
-                             cur_h + off_h, cur_v + off_v,
-                             rule_width, rule_height)
+                    render(p, output,
+                           cur_h + off_h, cur_v + off_v,
+                           rule_width, rule_height)
             elif isinstance(p, Glue):
                 glue_spec = p.glue_spec
                 rule_height = glue_spec.width - cur_g
@@ -2159,6 +2169,10 @@ def csnames(group: str, names: Iterable[str]) -> Regex:
 
         p.customspace = cmd(r"\hspace", "{" + p.float_literal("space") + "}")
 
+        p.phantom = cmd(r"\phantom", p.optional_group("value"))
+        p.llap = cmd(r"\llap", p.optional_group("value"))
+        p.rlap = cmd(r"\rlap", p.optional_group("value"))
+
         p.accent = (
             csnames("accent", [*self._accent_map, *self._wide_accents])
             - p.named_placeable("sym"))
@@ -2225,7 +2239,8 @@ def csnames(group: str, names: Iterable[str]) -> Regex:
             r"\boldsymbol", "{" + ZeroOrMore(p.simple)("value") + "}")
 
         p.placeable     <<= (
-            p.accent     # Must be before symbol as all accents are symbols
+            p.phantom | p.llap | p.rlap
+            | p.accent   # Must be before symbol as all accents are symbols
             | p.symbol   # Must be second to catch all named symbols and single
                          # chars not in a group
             | p.function
@@ -2419,6 +2434,16 @@ def symbol(self, s: str, loc: int,
     def unknown_symbol(self, s: str, loc: int, toks: ParseResults) -> T.Any:
         raise ParseFatalException(s, loc, f"Unknown symbol: {toks['name']}")
 
+    def phantom(self, toks: ParseResults) -> T.Any:
+        toks["value"].is_phantom = True
+        return toks["value"]
+
+    def llap(self, toks: ParseResults) -> T.Any:
+        return [Hlist([Kern(-toks["value"].width), toks["value"]])]
+
+    def rlap(self, toks: ParseResults) -> T.Any:
+        return [Hlist([toks["value"], Kern(-toks["value"].width)])]
+
     _accent_map = {
         r'hat':            r'\circumflexaccent',
         r'breve':          r'\combiningbreve',