diff --git a/django/core/serializers/base.py b/django/core/serializers/base.py
--- a/django/core/serializers/base.py
+++ b/django/core/serializers/base.py
@@ -290,11 +290,11 @@ def __repr__(self):
     def save(self, save_m2m=True, using=None, **kwargs):
         # Call save on the Model baseclass directly. This bypasses any
         # model-defined save. The save is also forced to be raw.
-        # raw=True is passed to any pre/post_save signals.
+        # raw=True is passed to any pre/post_save and m2m_changed signals.
         models.Model.save_base(self.object, using=using, raw=True, **kwargs)
         if self.m2m_data and save_m2m:
             for accessor_name, object_list in self.m2m_data.items():
-                getattr(self.object, accessor_name).set(object_list)
+                getattr(self.object, accessor_name).set_base(object_list, raw=True)
 
         # prevent a second (possibly accidental) call to save() from saving
         # the m2m data twice.
diff --git a/django/db/models/fields/related_descriptors.py b/django/db/models/fields/related_descriptors.py
--- a/django/db/models/fields/related_descriptors.py
+++ b/django/db/models/fields/related_descriptors.py
@@ -1262,15 +1262,16 @@ def count(self):
             else:
                 return super().count()
 
-        def add(self, *objs, through_defaults=None):
-            self._remove_prefetched_objects()
-            db = router.db_for_write(self.through, instance=self.instance)
+        def _add_base(self, *objs, through_defaults=None, using=None, raw=False):
+            db = using or router.db_for_write(self.through, instance=self.instance)
             with transaction.atomic(using=db, savepoint=False):
                 self._add_items(
                     self.source_field_name,
                     self.target_field_name,
                     *objs,
                     through_defaults=through_defaults,
+                    using=db,
+                    raw=raw,
                 )
                 # If this is a symmetrical m2m relation to self, add the mirror
                 # entry in the m2m table.
@@ -1280,8 +1281,15 @@ def add(self, *objs, through_defaults=None):
                         self.source_field_name,
                         *objs,
                         through_defaults=through_defaults,
+                        using=db,
+                        raw=raw,
                     )
 
+        def add(self, *objs, through_defaults=None):
+            self._remove_prefetched_objects()
+            db = router.db_for_write(self.through, instance=self.instance)
+            self._add_base(*objs, through_defaults=through_defaults, using=db)
+
         add.alters_data = True
 
         async def aadd(self, *objs, through_defaults=None):
@@ -1291,9 +1299,16 @@ async def aadd(self, *objs, through_defaults=None):
 
         aadd.alters_data = True
 
+        def _remove_base(self, *objs, using=None, raw=False):
+            db = using or router.db_for_write(self.through, instance=self.instance)
+            self._remove_items(
+                self.source_field_name, self.target_field_name, *objs, using=db, raw=raw
+            )
+
         def remove(self, *objs):
             self._remove_prefetched_objects()
-            self._remove_items(self.source_field_name, self.target_field_name, *objs)
+            db = router.db_for_write(self.through, instance=self.instance)
+            self._remove_base(*objs, using=db)
 
         remove.alters_data = True
 
@@ -1302,8 +1317,8 @@ async def aremove(self, *objs):
 
         aremove.alters_data = True
 
-        def clear(self):
-            db = router.db_for_write(self.through, instance=self.instance)
+        def _clear_base(self, using=None, raw=False):
+            db = using or router.db_for_write(self.through, instance=self.instance)
             with transaction.atomic(using=db, savepoint=False):
                 signals.m2m_changed.send(
                     sender=self.through,
@@ -1313,8 +1328,8 @@ def clear(self):
                     model=self.model,
                     pk_set=None,
                     using=db,
+                    raw=raw,
                 )
-                self._remove_prefetched_objects()
                 filters = self._build_remove_filters(super().get_queryset().using(db))
                 self.through._default_manager.using(db).filter(filters).delete()
 
@@ -1326,25 +1341,34 @@ def clear(self):
                     model=self.model,
                     pk_set=None,
                     using=db,
+                    raw=raw,
                 )
 
+        def clear(self):
+            self._remove_prefetched_objects()
+            db = router.db_for_write(self.through, instance=self.instance)
+            self._clear_base(using=db)
+
         clear.alters_data = True
 
         async def aclear(self):
             return await sync_to_async(self.clear)()
 
         aclear.alters_data = True
 
-        def set(self, objs, *, clear=False, through_defaults=None):
+        def set_base(self, objs, *, clear=False, through_defaults=None, raw=False):
             # Force evaluation of `objs` in case it's a queryset whose value
             # could be affected by `manager.clear()`. Refs #19816.
             objs = tuple(objs)
 
             db = router.db_for_write(self.through, instance=self.instance)
             with transaction.atomic(using=db, savepoint=False):
+                self._remove_prefetched_objects()
                 if clear:
-                    self.clear()
-                    self.add(*objs, through_defaults=through_defaults)
+                    self._clear_base(using=db, raw=raw)
+                    self._add_base(
+                        *objs, through_defaults=through_defaults, using=db, raw=raw
+                    )
                 else:
                     old_ids = set(
                         self.using(db).values_list(
@@ -1364,8 +1388,13 @@ def set(self, objs, *, clear=False, through_defaults=None):
                         else:
                             new_objs.append(obj)
 
-                    self.remove(*old_ids)
-                    self.add(*new_objs, through_defaults=through_defaults)
+                    self._remove_base(*old_ids, using=db, raw=raw)
+                    self._add_base(
+                        *new_objs, through_defaults=through_defaults, using=db, raw=raw
+                    )
+
+        def set(self, objs, *, clear=False, through_defaults=None):
+            self.set_base(objs, clear=clear, through_defaults=through_defaults)
 
         set.alters_data = True
 
@@ -1516,7 +1545,13 @@ def _get_add_plan(self, db, source_field_name):
             )
 
         def _add_items(
-            self, source_field_name, target_field_name, *objs, through_defaults=None
+            self,
+            source_field_name,
+            target_field_name,
+            *objs,
+            through_defaults=None,
+            using=None,
+            raw=False,
         ):
             # source_field_name: the PK fieldname in join table for the source
             # object target_field_name: the PK fieldname in join table for the
@@ -1527,7 +1562,7 @@ def _add_items(
 
             through_defaults = dict(resolve_callables(through_defaults or {}))
             target_ids = self._get_target_ids(target_field_name, objs)
-            db = router.db_for_write(self.through, instance=self.instance)
+            db = using or router.db_for_write(self.through, instance=self.instance)
             can_ignore_conflicts, must_send_signals, can_fast_add = self._get_add_plan(
                 db, source_field_name
             )
@@ -1559,6 +1594,7 @@ def _add_items(
                         model=self.model,
                         pk_set=missing_target_ids,
                         using=db,
+                        raw=raw,
                     )
                 # Add the ones that aren't there already.
                 self.through._default_manager.using(db).bulk_create(
@@ -1584,9 +1620,12 @@ def _add_items(
                         model=self.model,
                         pk_set=missing_target_ids,
                         using=db,
+                        raw=raw,
                     )
 
-        def _remove_items(self, source_field_name, target_field_name, *objs):
+        def _remove_items(
+            self, source_field_name, target_field_name, *objs, using=None, raw=False
+        ):
             # source_field_name: the PK colname in join table for the source
             # object target_field_name: the PK colname in join table for the
             # target object *objs - objects to remove. Either object instances,
@@ -1603,7 +1642,7 @@ def _remove_items(self, source_field_name, target_field_name, *objs):
                 else:
                     old_ids.add(obj)
 
-            db = router.db_for_write(self.through, instance=self.instance)
+            db = using or router.db_for_write(self.through, instance=self.instance)
             with transaction.atomic(using=db, savepoint=False):
                 # Send a signal to the other end if need be.
                 signals.m2m_changed.send(
@@ -1614,6 +1653,7 @@ def _remove_items(self, source_field_name, target_field_name, *objs):
                     model=self.model,
                     pk_set=old_ids,
                     using=db,
+                    raw=raw,
                 )
                 target_model_qs = super().get_queryset()
                 if target_model_qs._has_filters():
@@ -1633,6 +1673,7 @@ def _remove_items(self, source_field_name, target_field_name, *objs):
                     model=self.model,
                     pk_set=old_ids,
                     using=db,
+                    raw=raw,
                 )
 
     return ManyRelatedManager