diff --git a/django/dispatch/dispatcher.py b/django/dispatch/dispatcher.py
--- a/django/dispatch/dispatcher.py
+++ b/django/dispatch/dispatcher.py
@@ -423,7 +423,10 @@
             self.receivers = [
                 r
                 for r in self.receivers
-                if not (isinstance(r[1], weakref.ReferenceType) and r[1]() is None)
+                if (
+                    not (isinstance(r[1], weakref.ReferenceType) and r[1]() is None)
+                    and not (r[2] is not None and r[2]() is None)
+                )
             ]
 
     def _live_receivers(self, sender):
