diff --git a/django/dispatch/dispatcher.py b/django/dispatch/dispatcher.py
--- a/django/dispatch/dispatcher.py
+++ b/django/dispatch/dispatcher.py
@@ -470,11 +470,12 @@
                 receiver = receiver()
                 if receiver is None:
                     continue
+            if sender_ref is not None and sender_ref() is None:
+                continue
+            if is_async:
+                non_weak_async_receivers.append(receiver)
             else:
-                if is_async:
-                    non_weak_async_receivers.append(receiver)
-                else:
-                    non_weak_sync_receivers.append(receiver)
+                non_weak_sync_receivers.append(receiver)
         return non_weak_sync_receivers, non_weak_async_receivers
 
     def _flag_dead_receivers(self, reference=None):
