diff --git a/astropy/units/core.py b/astropy/units/core.py
--- a/astropy/units/core.py
+++ b/astropy/units/core.py
@@ -807,15 +807,17 @@ def __pow__(self, p: Real) -> CompositeUnit:
                 ) from None
             return CompositeUnit(1, [self], [sanitize_power(p)], _error_check=False)
 
-    def __truediv__(self, m):
-        if isinstance(m, (bytes, str)):
-            m = Unit(m)
 
+    def __truediv__(self, m):
         if isinstance(m, UnitBase):
             if m.is_unity():
                 return self
             return CompositeUnit(1, [self, m], [1, -1], _error_check=False)
 
+        if isinstance(m, (bytes, str)):
+            self._warn_about_operation_with_deprecated_type("divisions", m)
+            return self / Unit(m)
+
         try:
             # Cannot handle this as Unit, re-try as Quantity
             from .quantity import Quantity
