diff --git a/sklearn/utils/stats.py b/sklearn/utils/stats.py
--- a/sklearn/utils/stats.py
+++ b/sklearn/utils/stats.py
@@ -137,72 +137,74 @@
     n_percentiles = percentile_rank.shape[0]
     result = xp.empty((n_features, n_percentiles), dtype=floating_dtype, device=device)
 
-    adjusted_percentile_rank = percentile_rank / 100 * weight_cdf[..., -1]
-
-    # Ignore leading `sample_weight=0` observations when `percentile_rank=0` (#20528)
-    mask = adjusted_percentile_rank == 0
-    adjusted_percentile_rank[mask] = xp.nextafter(
-        adjusted_percentile_rank[mask], adjusted_percentile_rank[mask] + 1
-    )
-    # For each feature with index j, find sample index i of the scalar value
-    # `adjusted_percentile_rank[j]` in 1D array `weight_cdf[j]`, such that:
-    # weight_cdf[j, i-1] < adjusted_percentile_rank[j] <= weight_cdf[j, i].
-    # Note `searchsorted` defaults to equality on the right, whereas Hyndman and Fan
-    # reference equation has equality on the left.
-    percentile_indices = xp.stack(
-        [
-            xp.searchsorted(
-                weight_cdf[feature_idx, ...], adjusted_percentile_rank[feature_idx]
+    for p_idx, p_rank in enumerate(percentile_rank):
+        adjusted_percentile_rank = p_rank / 100 * weight_cdf[..., -1]
+
+        # Ignore leading `sample_weight=0` observations
+        # when `percentile_rank=0` (#20528)
+        mask = adjusted_percentile_rank == 0
+        adjusted_percentile_rank[mask] = xp.nextafter(
+            adjusted_percentile_rank[mask], adjusted_percentile_rank[mask] + 1
+        )
+        # For each feature with index j, find sample index i of the scalar value
+        # `adjusted_percentile_rank[j]` in 1D array `weight_cdf[j]`, such that:
+        # weight_cdf[j, i-1] < adjusted_percentile_rank[j] <= weight_cdf[j, i].
+        # Note `searchsorted` defaults to equality on the right, whereas Hyndman and Fan
+        # reference equation has equality on the left.
+        percentile_indices = xp.stack(
+            [
+                xp.searchsorted(
+                    weight_cdf[feature_idx, ...], adjusted_percentile_rank[feature_idx]
+                )
+                for feature_idx in range(weight_cdf.shape[0])
+            ],
+        )
+        # `percentile_indices` may be equal to `sorted_idx.shape[0]` due to floating
+        # point error (see #11813)
+        max_idx = sorted_idx.shape[0] - 1
+        percentile_indices = xp.clip(percentile_indices, 0, max_idx)
+
+        col_indices = xp.arange(array.shape[1], device=device)
+        percentile_in_sorted = sorted_idx[percentile_indices, col_indices]
+
+        if average:
+            # From Hyndman and Fan (1996), `fraction_above` is `g`
+            fraction_above = (
+                weight_cdf[col_indices, percentile_indices] - adjusted_percentile_rank
             )
-            for feature_idx in range(weight_cdf.shape[0])
-        ],
-    )
-    # `percentile_indices` may be equal to `sorted_idx.shape[0]` due to floating
-    # point error (see #11813)
-    max_idx = sorted_idx.shape[0] - 1
-    percentile_indices = xp.clip(percentile_indices, 0, max_idx)
-
-    col_indices = xp.arange(array.shape[1], device=device)
-    percentile_in_sorted = sorted_idx[percentile_indices, col_indices]
-
-    if average:
-        # From Hyndman and Fan (1996), `fraction_above` is `g`
-        fraction_above = (
-            weight_cdf[col_indices, percentile_indices] - adjusted_percentile_rank
-        )
-        is_fraction_above = fraction_above > xp.finfo(floating_dtype).eps
-        percentile_plus_one_indices = xp.clip(percentile_indices + 1, 0, max_idx)
-        percentile_plus_one_in_sorted = sorted_idx[
-            percentile_plus_one_indices, col_indices
-        ]
-        # Handle case when next index ('plus one') has sample weight of 0
-        zero_weight_cols = col_indices[
-            sample_weight[percentile_plus_one_in_sorted, col_indices] == 0
-        ]
-        for col_idx in zero_weight_cols:
-            cdf_val = weight_cdf[col_idx, percentile_indices[col_idx]]
-            # Search for next index where `weighted_cdf` is greater
-            next_index = xp.searchsorted(
-                weight_cdf[col_idx, ...], cdf_val, side="right"
+            is_fraction_above = fraction_above > xp.finfo(floating_dtype).eps
+            percentile_plus_one_indices = xp.clip(percentile_indices + 1, 0, max_idx)
+            percentile_plus_one_in_sorted = sorted_idx[
+                percentile_plus_one_indices, col_indices
+            ]
+            # Handle case when next index ('plus one') has sample weight of 0
+            zero_weight_cols = col_indices[
+                sample_weight[percentile_plus_one_in_sorted, col_indices] == 0
+            ]
+            for col_idx in zero_weight_cols:
+                cdf_val = weight_cdf[col_idx, percentile_indices[col_idx]]
+                # Search for next index where `weighted_cdf` is greater
+                next_index = xp.searchsorted(
+                    weight_cdf[col_idx, ...], cdf_val, side="right"
+                )
+                # Handle case where there are trailing 0 sample weight samples
+                # and `percentile_indices` is already max index
+                if next_index >= max_idx:
+                    # use original `percentile_indices` again
+                    next_index = percentile_indices[col_idx]
+
+                percentile_plus_one_in_sorted[col_idx] = sorted_idx[next_index, col_idx]
+
+            result = xp.where(
+                is_fraction_above,
+                array[percentile_in_sorted, col_indices],
+                (
+                    array[percentile_in_sorted, col_indices]
+                    + array[percentile_plus_one_in_sorted, col_indices]
+                )
+                / 2,
             )
-            # Handle case where there are trailing 0 sample weight samples
-            # and `percentile_indices` is already max index
-            if next_index >= max_idx:
-                # use original `percentile_indices` again
-                next_index = percentile_indices[col_idx]
-
-            percentile_plus_one_in_sorted[col_idx] = sorted_idx[next_index, col_idx]
-
-        result = xp.where(
-            is_fraction_above,
-            array[percentile_in_sorted, col_indices],
-            (
-                array[percentile_in_sorted, col_indices]
-                + array[percentile_plus_one_in_sorted, col_indices]
-            )
-            / 2,
-        )
-    else:
-        result = array[percentile_in_sorted, col_indices]
+        else:
+            result = array[percentile_in_sorted, col_indices]
 
     return result[0] if n_dim == 1 else result
