diff --git a/xarray/core/groupby.py b/xarray/core/groupby.py
--- a/xarray/core/groupby.py
+++ b/xarray/core/groupby.py
@@ -252,9 +252,7 @@ def to_array(self) -> DataArray:
 T_Group = Union["T_DataArray", _DummyGroup]
 
 
-def _ensure_1d(
-    group: T_Group, obj: T_DataWithCoords
-) -> tuple[
+def _ensure_1d(group: T_Group, obj: T_DataWithCoords) -> tuple[
     T_Group,
     T_DataWithCoords,
     Hashable | None,
@@ -1233,7 +1231,7 @@ def _flox_scan(
         parsed_dim = self._parse_dim(dim)
         obj = self._original_obj
         codes = tuple(g.codes for g in self.groupers)
-        a = 2
+
         g = groupby_scan(
             obj.data,
             *codes,
