diff --git a/xarray/core/groupby.py b/xarray/core/groupby.py
--- a/xarray/core/groupby.py
+++ b/xarray/core/groupby.py
@@ -1219,13 +1219,9 @@ def _flox_scan(
     ) -> T_Xarray:
         from flox import groupby_scan
 
-        obj = self._original_obj
         parsed_dim = self._parse_dim(dim)
-
-        obj = obj.transpose(..., *parsed_dim)
+        obj = self._original_obj.transpose(..., *parsed_dim)
         axis = range(-len(parsed_dim), 0)
-
-        # axis = (axis_,) if isinstance(axis_, int) else axis_
         codes = tuple(g.codes for g in self.groupers)
 
         def wrapper(array, *by, func: str, skipna: bool | None, **kwargs):
