diff --git a/xarray/core/groupby.py b/xarray/core/groupby.py
--- a/xarray/core/groupby.py
+++ b/xarray/core/groupby.py
@@ -1211,8 +1211,8 @@ def _flox_scan(
         dim: Dims,
         *,
         func: str,
-        keep_attrs: bool | None = None,
         skipna: bool | None = None,
+        keep_attrs: bool | None = None,
         **kwargs: Any,
     ) -> DataArray:
         from flox import groupby_scan
