diff --git a/sympy/geometry/line.py b/sympy/geometry/line.py
--- a/sympy/geometry/line.py
+++ b/sympy/geometry/line.py
@@ -236,7 +236,7 @@
 
         angle_between, is_perpendicular, Ray2D.closing_angle
         """
-        if not isinstance(l1, LinearEntity) and not isinstance(l2, LinearEntity):
+        if not isinstance(l1, LinearEntity) or not isinstance(l2, LinearEntity):
             raise TypeError('Must pass only LinearEntity objects')
 
         v1, v2 = l1.direction, l2.direction
