diff --git a/xarray/core/extension_array.py b/xarray/core/extension_array.py
--- a/xarray/core/extension_array.py
+++ b/xarray/core/extension_array.py
@@ -2,7 +2,7 @@
 
 from collections.abc import Callable, Sequence
 from dataclasses import dataclass
-from typing import Generic, cast
+from typing import Any, Generic, cast
 
 import numpy as np
 import pandas as pd
@@ -142,3 +142,10 @@ def __array__(
             return np.asarray(self.array, dtype=dtype, copy=copy)
         else:
             return np.asarray(self.array, dtype=dtype)
+
+    def __getattr__(self, attr: str) -> Any:
+        #  with __deepcopy__ or __copy__, the object is first constructed and then the sub-objects are attached (see https://docs.python.org/3/library/copy.html)
+        # Thus, if we didn't have `super().__getattribute__("array")` this method would call `self.array` (i.e., `getattr(self, "array")`) again while looking for `__setstate__`
+        # (which is apparently the first thing sought in copy.copy from the under-construction copied object),
+        # which would cause a recursion error since `array` is not present on the object when it is being constructed during `__{deep}copy__`.
+        return getattr(super().__getattribute__("array"), attr)