diff --git a/pylint/lint/pylinter.py b/pylint/lint/pylinter.py
--- a/pylint/lint/pylinter.py
+++ b/pylint/lint/pylinter.py
@@ -518,7 +518,7 @@ def register_checker(self, checker: checkers.BaseChecker) -> None:
     def _deregister_checkers(
         self, checker_collection: Collection[tuple[str, checkers.BaseChecker]]
     ) -> None:
-        """Deregistered a collection of checkers with its reports and messages."""
+        """De-registered a collection of checkers with its reports and messages."""
         for checker_name, checker in checker_collection:
             self._checkers[checker_name].remove(checker)
             if checker.reports:
