diff --git a/django/contrib/staticfiles/management/commands/collectstatic.py b/django/contrib/staticfiles/management/commands/collectstatic.py
--- a/django/contrib/staticfiles/management/commands/collectstatic.py
+++ b/django/contrib/staticfiles/management/commands/collectstatic.py
@@ -215,9 +215,10 @@
             modified_count = len(collected["modified"])
             unmodified_count = len(collected["unmodified"])
             post_processed_count = len(collected["post_processed"])
+            skipped_count = len(collected["skipped"])
             return (
                 "\n%(modified_count)s %(identifier)s %(action)s"
-                "%(destination)s%(unmodified)s%(post_processed)s."
+                "%(destination)s%(unmodified)s%(post_processed)s%(skipped)s."
             ) % {
                 "modified_count": modified_count,
                 "identifier": "static file" + ("" if modified_count == 1 else "s"),

 
