diff --git a/sphinx/ext/intersphinx/_resolve.py b/sphinx/ext/intersphinx/_resolve.py
--- a/sphinx/ext/intersphinx/_resolve.py
+++ b/sphinx/ext/intersphinx/_resolve.py
@@ -46,7 +46,7 @@
         # get correct path in case of subdirectories
         uri = (_relative_path(Path(), Path(node['refdoc']).parent) / uri).as_posix()
     if inv_item.project_version:
-        if not
+        if not inv_item.project_version[0].isdigit():
         reftitle = _('(in %s v%s)') % (inv_item.project_name, inv_item.project_version)
     else:
         reftitle = _('(in %s)') % (inv_item.project_name,)