diff --git a/sphinx/util/logging.py b/sphinx/util/logging.py
--- a/sphinx/util/logging.py
+++ b/sphinx/util/logging.py
@@ -608,8 +608,12 @@ def write(self, data: str) -> None:
         self._app.messagelog.append(data)
 
 
-def setup(app: Sphinx, status: IO[str], warning: IO[str]) -> None:
+def setup(
+    app: Sphinx, status: IO[str], warning: IO[str], *, verbosity: int = 0
+) -> None:
     """Setup root logger for Sphinx"""
+    log_level = VERBOSITY_MAP[max(verbosity, 0)]
+
     logger = logging.getLogger(NAMESPACE)
     logger.setLevel(logging.DEBUG)
     logger.propagate = False
@@ -621,7 +625,7 @@ def setup(app: Sphinx, status: IO[str], warning: IO[str]) -> None:
     info_handler = NewLineStreamHandler(SafeEncodingWriter(status))
     info_handler.addFilter(InfoFilter())
     info_handler.addFilter(InfoLogRecordTranslator(app))
-    info_handler.setLevel(VERBOSITY_MAP[app.verbosity])
+    info_handler.setLevel(log_level)
     info_handler.setFormatter(ColorizeFormatter())
 
     warning_handler = WarningStreamHandler(SafeEncodingWriter(warning))
@@ -635,7 +639,7 @@ def setup(app: Sphinx, status: IO[str], warning: IO[str]) -> None:
 
     messagelog_handler = logging.StreamHandler(LastMessagesWriter(app, status))
     messagelog_handler.addFilter(InfoFilter())
-    messagelog_handler.setLevel(VERBOSITY_MAP[app.verbosity])
+    messagelog_handler.setLevel(log_level)
 
     logger.addHandler(info_handler)
     logger.addHandler(warning_handler)