diff --git a/sklearn/covariance/_robust_covariance.py b/sklearn/covariance/_robust_covariance.py
--- a/sklearn/covariance/_robust_covariance.py
+++ b/sklearn/covariance/_robust_covariance.py
@@ -433,7 +433,7 @@ def fast_mcd(
 
     # minimum breakdown value
     if support_fraction is None:
-        n_support = int(np.ceil(0.5 * (n_samples + n_features + 1)))
+        n_support = minint(np.ceil(0.5 * (n_samples + n_features + 1)))
     else:
         n_support = int(support_fraction * n_samples)
 