diff --git a/sympy/printing/pretty/pretty.py b/sympy/printing/pretty/pretty.py
--- a/sympy/printing/pretty/pretty.py
+++ b/sympy/printing/pretty/pretty.py
@@ -927,6 +927,10 @@
             return prettyForm(pretty_atom("OneMatrix"))
         else:
             return prettyForm('1')
+
+    def _print_MatrixUnit(self, expr):
+        if self._use_unicode:
+            s = self._print(Symbol(f'{pretty_atom("MatrixUnit")}_{expr._i}{expr._j}'))
 
     def _print_DotProduct(self, expr):
         args = list(expr.args)
