diff --git a/astropy/units/deprecated.py b/astropy/units/deprecated.py
--- a/astropy/units/deprecated.py
+++ b/astropy/units/deprecated.py
@@ -16,6 +16,11 @@
     >>> deprecated.enable()  # doctest: +SKIP
 
 """
+
+import warnings
+
+from astropy.utils.decorators import deprecated
+from astropy.utils.exceptions import AstropyDeprecationWarning
 
 from . import astrophys, cgs
 from .core import UnitBase, _add_prefixes, add_enabled_units, def_unit
@@ -61,4 +66,4 @@
     This may be used with the ``with`` statement to enable deprecated
     units only temporarily.
     """
-    return add_enabled_units()
+    return add_enabled_units(local_units)
