diff --git a/sklearn/utils/_plotting.py b/sklearn/utils/_plotting.py
--- a/sklearn/utils/_plotting.py
+++ b/sklearn/utils/_plotting.py
@@ -209,13 +209,16 @@
         # Ensure `curve_kwargs` is of correct length
         if isinstance(curve_kwargs, Mapping):
             curve_kwargs = [curve_kwargs] * n_curves
-
-        default_multi_curve_kwargs = {"alpha": 0.5, "linestyle": "--", "color": "blue"}
-        if curve_kwargs is None:
-            if n_curves > 1:
-                curve_kwargs = [default_multi_curve_kwargs] * n_curves
-            else:
-                curve_kwargs = [{}]
+        elif curve_kwargs is None:
+            curve_kwargs = [{}] * n_curves
+
+        if default_curve_kwargs is None:
+            default_curve_kwargs = {}
+        if default_multi_curve_kwargs is None:
+            default_multi_curve_kwargs = {}
+
+        if n_curves > 1:
+            default_curve_kwargs.update(default_multi_curve_kwargs)
 
         labels = []
         if "mean" in legend_metric:
