diff --git a/sklearn/utils/_plotting.py b/sklearn/utils/_plotting.py
--- a/sklearn/utils/_plotting.py
+++ b/sklearn/utils/_plotting.py
@@ -136,6 +136,7 @@ def _validate_curve_kwargs(
         legend_metric_name,
         curve_kwargs,
         default_curve_kwargs=None,
+        default_multi_curve_kwargs={"alpha": 0.5, "linestyle": "--", "color": "blue"},
         **kwargs,
     ):
         """Get validated line kwargs for each curve.
@@ -213,12 +214,11 @@ def _validate_curve_kwargs(
         # Ensure `curve_kwargs` is of correct length
         if isinstance(curve_kwargs, Mapping):
             curve_kwargs = [curve_kwargs] * n_curves
-        if curve_kwargs is None:
+        elif curve_kwargs is None:
             curve_kwargs = [{}] * n_curves
 
         if default_curve_kwargs is None:
             default_curve_kwargs = {}
-        default_multi_curve_kwargs = {"alpha": 0.5, "linestyle": "--", "color": "blue"}
         if n_curves > 1:
             default_curve_kwargs_ = {
                 **default_multi_curve_kwargs,
