diff --git a/lib/matplotlib/_cm.py b/lib/matplotlib/_cm.py
--- a/lib/matplotlib/_cm.py
+++ b/lib/matplotlib/_cm.py
@@ -852,7 +852,6 @@
     (0.50196078431372548, 0.0                 , 0.14901960784313725)
     )
 
-
 # ColorBrewer's qualitative maps, implemented using ListedColormap
 # for use with mpl.colors.NoNorm
 
@@ -877,6 +876,17 @@
     (0.65098039215686276, 0.46274509803921571, 0.11372549019607843),
     (0.4,                 0.4,                 0.4                ),
     )
+
+# Okabe-Ito accessible and print-friendly color palette.
+# By Masataka Okabe (Jikei Medical School) and Kei Ito (University of Tokyo).
+# Qualitative color palette that is unambiguous regardless of whether
+# the viewer has colorblindness. https://jfly.uni-koeln.de/color/#pallet
+
+_okabe_ito_data = (
+    (0.0, 0.0, 0.0),                                                # black
+    (0.9019607843137255, 0.6235294117647059, 0.0),                  # e69f00
+    (0.33725490196078434, 0.7058823529411765, 0.9137254901960784),  # 56b4e9
+    (0.0, 0.6196078431372549, 0.45098039215686275),                 # 009e73
 
 _Paired_data = (
     (0.65098039215686276, 0.80784313725490198, 0.8901960784313725 ),
