diff --git a/django/contrib/contenttypes/fields.py b/django/contrib/contenttypes/fields.py
--- a/django/contrib/contenttypes/fields.py
+++ b/django/contrib/contenttypes/fields.py
@@ -698,8 +703,12 @@ def get_prefetch_querysets(self, instances, querysets=None):
             # instances' PK in order to match up instances:
             object_id_converter = instances[0]._meta.pk.to_python
             content_type_id_field_name = "%s_id" % self.content_type_field_name
+            queryset = queryset.filter(query)
+            # Restore subsequent cloning operations.
+            if _cloning_disabled:
+                queryset._enable_cloning()
             return (
-                queryset.filter(query),
+                queryset,
                 lambda relobj: (
                     object_id_converter(getattr(relobj, self.object_id_field_name)),
                     getattr(relobj, content_type_id_field_name),
