diff --git a/lib/matplotlib/axes/_secondary_axes.py b/lib/matplotlib/axes/_secondary_axes.py
--- a/lib/matplotlib/axes/_secondary_axes.py
+++ b/lib/matplotlib/axes/_secondary_axes.py
@@ -145,6 +145,12 @@ def apply_aspect(self, position=None):
         self._set_lims()
         super().apply_aspect(position)

+
+    @functools.wraps(_AxesBase.set_xticks)
+    def set_xticks(self, *args, **kwargs):
+        if self._orientation == "y":
+            raise TypeError("Cannot set xticks on a secondary y-axis")
+
     @_docstring.copy(Axis.set_ticks)
     def set_ticks(self, ticks, labels=None, *, minor=False, **kwargs):
         ret = self._axis.set_ticks(ticks, labels, minor=minor, **kwargs)
 