diff --git a/astropy/modeling/core.py b/astropy/modeling/core.py
--- a/astropy/modeling/core.py
+++ b/astropy/modeling/core.py
@@ -3331,6 +3331,12 @@
             # passing input with shape (5, 3) might produce a deriv array with
             # shape (6, 5, 3) or (6, 15). We therefore normalize this to always
             # ravel all but the first dimension
+            left_deriv = left_deriv.reshape((left_deriv.shape[0], -1))
+            right_deriv = right_deriv.reshape((right_deriv.shape[0], -1))
+
+            # Convert the arrays back to lists over the first dimension so as to
+            # be able to concatenate them (we don't use .tolist() which would
+            # convert to a list of lists instead of a list of arrays)
 
             # We now have to use various differentiation rules to apply the
             # arithmetic operators to the derivatives.

