diff --git a/xarray/core/groupby.py b/xarray/core/groupby.py
--- a/xarray/core/groupby.py
+++ b/xarray/core/groupby.py
@@ -828,7 +828,7 @@ def groups(self) -> dict[GroupKey, GroupIndex]:
             self._groups = dict(
                 zip(
                     self.encoded.unique_coord.data,
-                    self.encoded.group_indices,
+                    tuple(self.encoded.group_indices),
                     strict=True,
                 )
             )