diff --git a/django/core/mail/__init__.py b/django/core/mail/__init__.py
--- a/django/core/mail/__init__.py
+++ b/django/core/mail/__init__.py
@@ -132,14 +132,12 @@ def _send_server_message(
     if not recipients:
         return
 
-    if not all(isinstance(a, (list, tuple)) and len(a) == 2 for a in recipients):
-        raise ValueError(f"The {setting_name} setting must be a list of 2-tuples.")
 
     mail = EmailMultiAlternatives(
         subject="%s%s" % (settings.EMAIL_SUBJECT_PREFIX, subject),
         body=message,
         from_email=settings.SERVER_EMAIL,
-        to=[a[1] for a in recipients],
+        to=recipients,
         connection=connection,
     )
     if html_message: