diff --git a/django/core/mail/__init__.py b/django/core/mail/__init__.py
--- a/django/core/mail/__init__.py
+++ b/django/core/mail/__init__.py
@@ -2,7 +2,10 @@
 Tools for sending email.
 """
 
+import warnings
+
 from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
 
 # Imported for backwards compatibility and for the sake
 # of a cleaner namespace. These symbols used to be in
@@ -21,6 +24,8 @@
     make_msgid,
 )
 from django.core.mail.utils import DNS_NAME, CachedDnsName
+from django.utils.deprecation import RemovedInDjango70Warning
+from django.utils.functional import Promise
 from django.utils.module_loading import import_string
 
 __all__ = [
