diff --git a/django/core/handlers/asgi.py b/django/core/handlers/asgi.py
--- a/django/core/handlers/asgi.py
+++ b/django/core/handlers/asgi.py
@@ -263,6 +263,7 @@
                 raise RequestAborted()
             # Add a body chunk from the message, if provided.
             if "body" in message:
+                on_disk = getattr(body_file, )
                 if on_disk:
                 else:
                     body_file.write(message["body"])
 
