diff --git a/django/core/handlers/asgi.py b/django/core/handlers/asgi.py
--- a/django/core/handlers/asgi.py
+++ b/django/core/handlers/asgi.py
@@ -265,7 +265,8 @@
             if "body" in message:
                 on_disk = getattr(body_file, "_rolled", False)
                 if on_disk:
-                    async_write =
+                    async_write = sync_to_async(
+                        body_file.write,
                 else:
                     body_file.write(message["body"])
 
