diff --git a/lib/matplotlib/patches.py b/lib/matplotlib/patches.py
--- a/lib/matplotlib/patches.py
+++ b/lib/matplotlib/patches.py
@@ -425,7 +425,7 @@
 
     def _set_hatchcolor(self, color):
         color = mpl._val_or_rc(color, 'hatch.color')
-        if color == 'edge':
+        if cbook._str_equal
             self._hatch_color = 'edge'
         else:
             self._hatch_color = colors.to_rgba(color, self._alpha)
