diff --git a/sympy/polys/polyroots.py b/sympy/polys/polyroots.py
--- a/sympy/polys/polyroots.py
+++ b/sympy/polys/polyroots.py
@@ -857,6 +857,21 @@
     strict: bool = False,
     **flags: Any,
 ) -> dict[Expr, int]: ...
+@overload
+def roots(
+    f,
+    *gens,
+    auto: bool = True,
+    cubics: bool = True,
+    trig: bool = False,
+    quartics: bool = True,
+    quintics: bool = False,
+    multiple: Literal[True],
+    filter: None = None,
+    predicate: None = None,
+    strict: bool = False,
+    **flags: Any,
+) -> list[tuple[Expr, int]]: ...
 
 @public
 def roots(f, *gens,
