diff --git a/sphinx/ext/autodoc/_member_finder.py b/sphinx/ext/autodoc/_member_finder.py
--- a/sphinx/ext/autodoc/_member_finder.py
+++ b/sphinx/ext/autodoc/_member_finder.py
@@ -32,10 +32,11 @@
     from collections.abc import Iterable, Iterator, Mapping, Sequence, Set
     from typing import Any, Literal
 
+    from docutils.statemachine import StringList
+
     from sphinx.config import Config
-    from sphinx.environment import _CurrentDocument
+    from sphinx.environment import BuildEnvironment, _CurrentDocument
     from sphinx.events import EventManager
-    from sphinx.ext.autodoc import Documenter
     from sphinx.ext.autodoc._directive_options import _AutoDocumenterOptions
     from sphinx.ext.autodoc._property_types import _AutodocObjType, _ItemProperties
     from sphinx.ext.autodoc._sentinels import (
@@ -44,7 +45,6 @@
         INSTANCE_ATTR_T,
         SLOTS_ATTR_T,
     )
-    from sphinx.ext.autodoc.directive import DocumenterBridge
     from sphinx.ext.autodoc.importer import _AttrGetter
     from sphinx.registry import SphinxComponentRegistry
 
@@ -98,14 +98,16 @@
     attr_docs: dict[tuple[str, str], list[str]],
     config: Config,
     current_document: _CurrentDocument,
-    directive: DocumenterBridge,
+    env: BuildEnvironment,
     events: EventManager,
     get_attr: _AttrGetter,
     indent: str,
     options: _AutoDocumenterOptions,
     props: _ItemProperties,
     real_modname: str,
+    record_dependencies: set[str],
     registry: SphinxComponentRegistry,
+    result: StringList,
 ) -> None:
     """Generate reST for member documentation.
 
@@ -126,12 +128,11 @@
         attr_docs=attr_docs,
         config=config,
         current_document=current_document,
-        directive=directive,
+        env=env,
         events=events,
         get_attr=get_attr,
         options=options,
         props=props,
-        registry=registry,
     )
 
     # for implicit module members, check __module__ to avoid
@@ -163,20 +164,17 @@
     attr_docs: dict[tuple[str, str], list[str]],
     config: Config,
     current_document: _CurrentDocument,
-    directive: DocumenterBridge,
+    env: BuildEnvironment,
     events: EventManager,
     get_attr: _AttrGetter,
     options: _AutoDocumenterOptions,
     props: _ItemProperties,
-    registry: SphinxComponentRegistry,
-) -> list[tuple[Documenter, bool]]:
+) -> list[tuple[_ItemProperties, bool, str]]:
     """Generate reST for member documentation.
 
     If *want_all* is True, document all members, else those given by
     *self.options.members*.
     """
-    env = directive.env
-
     if props.obj_type not in {'module', 'class', 'exception'}:
         msg = 'must be implemented in subclasses'
         raise NotImplementedError(msg)
@@ -220,7 +218,7 @@
         attr_docs=attr_docs,
     )
     # document non-skipped members
-    member_documenters: list[tuple[Documenter, bool]] = []
+    member_documenters: list[tuple[_ItemProperties, bool, str]] = []
     for member_name, member, is_attr in filtered_members:
         # prefer the documenter with the highest priority
         obj_type = _best_object_type_for_member(
@@ -251,16 +249,11 @@
             env=env,
             events=events,
             get_attr=get_attr,
-            options=directive.genopt,
+            options=options,
         )
         if member_props is None:
             continue
-
-        doccls = registry.documenters[obj_type]
-        documenter = doccls(directive, full_name, indent)
-        documenter.props = member_props
-
-        member_documenters.append((documenter, is_attr))
+        member_documenters.append((member_props, is_attr, indent))
 
     member_order = options.member_order or config.autodoc_member_order
     member_documenters = _sort_members(
