diff --git a/galleries/users_explain/colors/colormaps.py b/galleries/users_explain/colors/colormaps.py
--- a/galleries/users_explain/colors/colormaps.py
+++ b/galleries/users_explain/colors/colormaps.py
@@ -230,7 +230,7 @@ def plot_color_gradients(category, cmap_list):
 # These would not be good options for use as perceptual colormaps.
 
 plot_color_gradients('Qualitative',
-                     ['Pastel1', 'Pastel2', 'Paired', 'Accent', 'Okabe_Ito',
+                     ['Pastel1', 'Pastel2', 'Paired', 'Accent', 'okabe_ito',
                       'Dark2', 'Set1', 'Set2', 'Set3', 'tab10', 'tab20',
                       'tab20b', 'tab20c'])
 
