diff --git a/sympy/core/operations.py b/sympy/core/operations.py
--- a/sympy/core/operations.py
+++ b/sympy/core/operations.py
@@ -532,7 +532,7 @@ def __new__(cls, *args, evaluate=None, **options):
             evaluate = global_parameters.evaluate
 
         if not evaluate:
-            obj = super(LatticeOp, cls).__new__(cls, *args, evaluate=False, **options)
+            obj = super().__new__(cls, *args, evaluate=False, **options)
             obj._argset = frozenset(args)
             return obj
 