diff --git a/sympy/polys/rootoftools.py b/sympy/polys/rootoftools.py
--- a/sympy/polys/rootoftools.py
+++ b/sympy/polys/rootoftools.py
@@ -330,7 +330,7 @@
         else:
             raise ValueError("expected an integer root index, got %s" % index)
 
-        poly = PurePoly(f, x, greedy=False, expand=expand)
+        poly = Poly(f, x, greedy=False, expand=expand)
 
         if not poly.is_univariate:
             raise PolynomialError("only univariate polynomials are allowed")
