diff --git a/django/db/models/fields/related.py b/django/db/models/fields/related.py
--- a/django/db/models/fields/related.py
+++ b/django/db/models/fields/related.py
@@ -1122,25 +1122,12 @@
                         id="fields.E322",
                     )
                 )
-            for db in databases:
-                if not router.allow_migrate_model(db, self.model):
-                    continue
-                connection = connections[db]
-                if not (
-                    "supports_on_delete_db_default"
-                    in self.model._meta.required_db_features
-                    or connection.features.supports_on_delete_db_default
-                ):
-                    errors.append(
-                        checks.Error(
-                            f"{connection.display_name} does not support a "
-                            "DB_SET_DEFAULT.",
-                            hint="Change the on_delete rule to SET_DEFAULT.",
-                            obj=self,
-                            id="fields.E324",
-                        ),
-                    )
-        elif not isinstance(self.remote_field.model, str) and on_delete != DO_NOTHING:
+            errors.extend(
+                self._check_on_delete_db_support(
+                    on_delete, "supports_on_delete_db_default", databases
+                )
+            )
+        if not isinstance(self.remote_field.model, str) and on_delete != DO_NOTHING:
             # Database and Python variants cannot be mixed in a chain of
             # model references.
             is_db_on_delete = isinstance(on_delete, DatabaseOnDelete)
