diff --git a/sympy/matrices/determinant.py b/sympy/matrices/determinant.py
--- a/sympy/matrices/determinant.py
+++ b/sympy/matrices/determinant.py
@@ -760,7 +760,7 @@ def _det_bareiss(
     """
 
     # Recursively implemented Bareiss' algorithm as per Deanna Richelle Leggett's
-    # thesis http://www.math.usm.edu/perry/Research/Thesis_DRL.pdf
+    # thesis https://aquila.usm.edu/cgi/viewcontent.cgi?article=1001&context=masters_theses
     def bareiss(mat: MatrixBase, cumm: Expr = S.One):
         if mat.rows == 0:
             return mat.one
