diff --git a/sphinx/ext/autodoc/__init__.py b/sphinx/ext/autodoc/__init__.py
--- a/sphinx/ext/autodoc/__init__.py
+++ b/sphinx/ext/autodoc/__init__.py
@@ -35,7 +35,6 @@
     ModuleDocumenter,
     PropertyDocumenter,
     TypeAliasDocumenter,
-    autodoc_attrgetter,
 )
 from sphinx.ext.autodoc._event_listeners import between, cut_lines
 from sphinx.ext.autodoc._member_finder import ObjectMember, special_member_re
@@ -94,7 +93,6 @@
     'ObjectMember',
     'py_ext_sig_re',
     'special_member_re',
-    'autodoc_attrgetter',
     'Documenter',
 )
 
