diff --git a/sphinx/ext/autodoc/__init__.py b/sphinx/ext/autodoc/__init__.py
--- a/sphinx/ext/autodoc/__init__.py
+++ b/sphinx/ext/autodoc/__init__.py
@@ -930,6 +930,13 @@
         # avoid possible circular imports if we were to import objects after
         # their associated documenters have been sorted.
         member_documenters = [
+            (documenter, isattr)
+            for documenter, isattr in member_documenters
+            if documenter.parse_name() and documenter.import_object()
+        ]
+        member_documenters = self.sort_members(member_documenters, member_order)
+
+        for documenter, isattr in member_documenters:
                 all_members=True,
                 real_modname=self.real_modname,
                 check_module=members_check_module and not isattr,
