diff --git a/pylint/checkers/typecheck.py b/pylint/checkers/typecheck.py
--- a/pylint/checkers/typecheck.py
+++ b/pylint/checkers/typecheck.py
@@ -150,8 +150,7 @@
     return itertools.chain(values, other_values)


-def _string_distance(seq1: str, seq2: str) -> int:
-    seq2_length = len(seq2)
+def _string_distance(seq1: str, seq2: str, seq1_length: int, seq2_length: int) -> int:

     row = [*list(range(1, seq2_length + 1)), 0]
     for seq1_index, seq1_char in enumerate(seq1):