diff --git a/pylint/checkers/typecheck.py b/pylint/checkers/typecheck.py
--- a/pylint/checkers/typecheck.py
+++ b/pylint/checkers/typecheck.py
@@ -199,7 +199,7 @@
         if min_distance > distance_threshold:
             continue

-        distance = _string_distance(attrname or "", name)
+        distance = _string_distance(attr_str, name, attr_len, name_len)
         if distance <= distance_threshold:
             possible_names.append((name, distance))

