diff --git a/sphinx/util/images.py b/sphinx/util/images.py
--- a/sphinx/util/images.py
+++ b/sphinx/util/images.py
@@ -5,6 +5,7 @@
 import base64
 from pathlib import Path
 from typing import TYPE_CHECKING, NamedTuple, overload
+from urllib.parse import unquote_to_bytes
 
 import imagesize
 
@@ -90,7 +91,6 @@ def get_image_extension(mimetype: str) -> str | None:
 def parse_data_uri(uri: str) -> DataURI | None:
     if not uri.startswith('data:'):
         return None
-    uri = uri[5:]
 
     if ',' not in uri:
         msg = 'malformed data URI'
@@ -100,16 +100,19 @@ def parse_data_uri(uri: str) -> DataURI | None:
     mimetype = 'text/plain'
     charset = 'US-ASCII'
 
+    uri = uri[5:]
     properties, _, data = uri.partition(',')
     for prop in properties.split(';'):
         if prop == 'base64':
             pass  # skip
-        elif prop.startswith('charset='):
+        elif prop.lower().startswith('charset='):
             charset = prop[8:]
         elif prop:
-            mimetype = prop
+            mimetype = prop.lower()
 
-    image_data = base64.b64decode(data)
+    image_data = unquote_to_bytes(data)  # data might be percent-encoded
+    if properties.endswith(';base64'):
+        image_data = base64.decodebytes(image_data)
     return DataURI(mimetype, charset, image_data)
 
 