diff --git a/lib/matplotlib/markers.py b/lib/matplotlib/markers.py
--- a/lib/matplotlib/markers.py
+++ b/lib/matplotlib/markers.py
@@ -308,7 +308,7 @@ def _set_marker(self, marker):
         """
         if isinstance(marker, str) and cbook.is_math_text(marker):
             self._marker_function = self._set_mathtext_path
-        elif isinstance(marker, (int, str)) and marker in self.markers:
+        elif isinstance(marker, (int, str, np.integer)) and marker in self.markers:
             self._marker_function = getattr(self, '_set_' + self.markers[marker])
         elif (isinstance(marker, np.ndarray) and marker.ndim == 2 and
                 marker.shape[1] == 2):
