diff --git a/xarray/indexes/nd_point_index.py b/xarray/indexes/nd_point_index.py
--- a/xarray/indexes/nd_point_index.py
+++ b/xarray/indexes/nd_point_index.py
@@ -244,7 +244,7 @@
         assert isinstance(tree_obj, TreeAdapter)
         self._tree_obj = tree_obj
 
-        assert len(coord_names) == len(dims) == len(shape)
+        assert len(coord_names) == len(dims)
         self._coord_names = coord_names
         self._dims = dims
         self._shape = shape
