diff --git a/xarray/indexes/nd_point_index.py b/xarray/indexes/nd_point_index.py
--- a/xarray/indexes/nd_point_index.py
+++ b/xarray/indexes/nd_point_index.py
@@ -244,7 +244,7 @@ def __init__(
         assert isinstance(tree_obj, TreeAdapter)
         self._tree_obj = tree_obj
 
-        assert len(coord_names) == len(dims) == len(shape)
+        assert len(dims) == len(shape)
         self._coord_names = coord_names
         self._dims = dims
         self._shape = shape
@@ -264,12 +264,6 @@ def from_variables(
 
         var0 = next(iter(variables.values()))
 
-        if len(variables) != len(var0.dims):
-            raise ValueError(
-                f"the number of variables {len(variables)} doesn't match "
-                f"the number of dimensions {len(var0.dims)}"
-            )
-
         opts = dict(options)
 
         tree_adapter_cls: type[T_TreeAdapter] = opts.pop("tree_adapter_cls", None)
