diff --git a/sklearn/utils/_tags.py b/sklearn/utils/_tags.py
--- a/sklearn/utils/_tags.py
+++ b/sklearn/utils/_tags.py
@@ -393,7 +393,9 @@ def get_tags(estimator) -> Tags:
     tag_provider = _find_tags_provider(estimator)
 
     if tag_provider == "__sklearn_tags__":
-        tags = estimator.__sklearn_tags__()
+        # TODO(1.7): turn the warning into an error
+        try:
+            tags = estimator.__sklearn_tags__()
     else:
         # TODO(1.7): Remove this branch of the code
         # Let's go through the MRO and patch each class implementing _more_tags