diff --git a/django/template/library.py b/django/template/library.py
--- a/django/template/library.py
+++ b/django/template/library.py
@@ -123,6 +123,15 @@ def dec(func):
                 ) = getfullargspec(unwrap(func))
             function_name = name or func.__name__
 
+            if takes_context:
+                if params and params[0] == "context":
+                    del params[0]
+                else:
+                    raise TemplateSyntaxError(
+                        f"{function_name!r} is decorated with takes_context=True so it "
+                        "must have a first argument of 'context'"
+                    )
+
             @wraps(func)
             def compile_func(parser, token):
                 bits = token.split_contents()[1:]
@@ -139,7 +148,6 @@ def compile_func(parser, token):
                     defaults,
                     kwonly,
                     kwonly_defaults,
-                    takes_context,
                     function_name,
                 )
                 return SimpleNode(func, takes_context, args, kwargs, target_var)
@@ -182,26 +190,32 @@ def dec(func):
             if end_name is None:
                 end_name = f"end{function_name}"
 
-            @wraps(func)
-            def compile_func(parser, token):
-                tag_params = params.copy()
+            if takes_context:
+                if len(params) >= 2 and params[1] == "content":
+                    del params[1]
+                else:
+                    raise TemplateSyntaxError(
+                        f"{function_name!r} is decorated with takes_context=True so"
+                        " it must have a first argument of 'context' and a second "
+                        "argument of 'content'"
+                    )
 
-                if takes_context:
-                    if len(tag_params) >= 2 and tag_params[1] == "content":
-                        del tag_params[1]
-                    else:
-                        raise TemplateSyntaxError(
-                            f"{function_name!r} is decorated with takes_context=True so"
-                            " it must have a first argument of 'context' and a second "
-                            "argument of 'content'"
-                        )
-                elif tag_params and tag_params[0] == "content":
-                    del tag_params[0]
+                if params and params[0] == "context":
+                    del params[0]
                 else:
                     raise TemplateSyntaxError(
-                        f"'{function_name}' must have a first argument of 'content'"
+                        f"{function_name!r} is decorated with takes_context=True so it "
+                        "must have a first argument of 'context'"
                     )
+            elif params and params[0] == "content":
+                del params[0]
+            else:
+                raise TemplateSyntaxError(
+                    f"{function_name!r} must have a first argument of 'content'"
+                )
 
+            @wraps(func)
+            def compile_func(parser, token):
                 bits = token.split_contents()[1:]
                 target_var = None
                 if len(bits) >= 2 and bits[-2] == "as":
@@ -214,13 +228,12 @@ def compile_func(parser, token):
                 args, kwargs = parse_bits(
                     parser,
                     bits,
-                    tag_params,
+                    params,
                     varargs,
                     varkw,
                     defaults,
                     kwonly,
                     kwonly_defaults,
-                    takes_context,
                     function_name,
                 )
 
@@ -263,6 +276,15 @@ def dec(func):
                 ) = getfullargspec(unwrap(func))
             function_name = name or func.__name__
 
+            if takes_context:
+                if params and params[0] == "context":
+                    params = params[1:]
+                else:
+                    raise TemplateSyntaxError(
+                        f"{function_name!r} is decorated with takes_context=True so it "
+                        "must have a first argument of 'context'"
+                    )
+
             @wraps(func)
             def compile_func(parser, token):
                 bits = token.split_contents()[1:]
@@ -275,7 +297,6 @@ def compile_func(parser, token):
                     defaults,
                     kwonly,
                     kwonly_defaults,
-                    takes_context,
                     function_name,
                 )
                 return InclusionNode(
@@ -394,22 +415,13 @@ def parse_bits(
     defaults,
     kwonly,
     kwonly_defaults,
-    takes_context,
     name,
 ):
     """
     Parse bits for template tag helpers simple_tag and inclusion_tag, in
     particular by detecting syntax errors and by extracting positional and
     keyword arguments.
     """
-    if takes_context:
-        if params and params[0] == "context":
-            params = params[1:]
-        else:
-            raise TemplateSyntaxError(
-                "'%s' is decorated with takes_context=True so it must "
-                "have a first argument of 'context'" % name
-            )
     args = []
     kwargs = {}
     unhandled_params = list(params)
