diff --git a/lib/matplotlib/scale.py b/lib/matplotlib/scale.py
--- a/lib/matplotlib/scale.py
+++ b/lib/matplotlib/scale.py
@@ -77,10 +77,18 @@
         For back-compatibility reasons, scales take an `~matplotlib.axis.Axis`
         object as the first argument.
 
-        The current recommendation for `.ScaleBase` subclasses is to have the
-        *axis* parameter for API compatibility, but not make use of it. This is
-        because we plan to remove this argument to make a scale object usable
-        by multiple `~matplotlib.axis.Axis`\es at the same time.
+        .. deprecated:: 3.11
+
+           The *axis* parameter is now optional, i.e. matplotlib is compatible
+           with `.ScaleBase` subclasses that do not take an *axis* parameter.
+
+           The *axis* parameter is pending-deprecated. It will be deprecated
+           in matplotlib 3.13, and removed in matplotlib 3.15.
+
+           3rd-party scales are recommended to remove the *axis* parameter now
+           if they can afford to restrict compatibility to matplotlib >= 3.11
+           already. Otherwise, they may keep the *axis* parameter and remove it
+           in time for matplotlib 3.13.
         """
 
     def get_transform(self):
@@ -854,6 +862,10 @@
     """
     _scale_mapping[scale_class.name] = scale_class
 
+    # migration code to handle the *axis* parameter
+    has_axis_parameter = "axis" in inspect.signature(scale_class).parameters
+    _scale_has_axis_parameter[scale_class.name] = has_axis_parameter
+
 
 def _get_scale_docs():
     """
