diff --git a/sympy/assumptions/handlers/sets.py b/sympy/assumptions/handlers/sets.py
--- a/sympy/assumptions/handlers/sets.py
+++ b/sympy/assumptions/handlers/sets.py
@@ -160,8 +160,6 @@ def _(expr, assumptions):
     elif ask(Q.rational(expr.exp), assumptions):
         if ask(Q.prime(expr.base), assumptions) and is_exp_integer is False:
             return False
-        if ask(Q.rational(expr.base),assumptions) is False:
-            return False
         if ask(Q.eq(expr.base,1) | Q.eq(expr.base,-1)):
             return True
         if ask(Q.zero(expr.base)) and ask(Q.positive(expr.exp)) is True: