diff --git a/sympy/assumptions/handlers/sets.py b/sympy/assumptions/handlers/sets.py
--- a/sympy/assumptions/handlers/sets.py
+++ b/sympy/assumptions/handlers/sets.py
@@ -147,6 +147,7 @@ def _(expr, assumptions):
         if ask(Q.rational(x), assumptions):
             return ask(~Q.nonzero(x), assumptions)
         return
+
     is_exp_integer = ask(Q.integer(expr.exp), assumptions)
     if is_exp_integer:
         is_base_rational = ask(Q.rational(expr.base),assumptions)
@@ -156,6 +157,8 @@ def _(expr, assumptions):
                 return True
             if is_base_zero is True and ask(Q.positive(expr.exp)) is True:
                 return True
+        if ask(Q.algebraic(expr.base),assumptions) is False:
+            return False
         return
     elif ask(Q.rational(expr.exp), assumptions):
         if ask(Q.prime(expr.base), assumptions) and is_exp_integer is False: