diff --git a/lib/mpl_toolkits/mplot3d/axis3d.py b/lib/mpl_toolkits/mplot3d/axis3d.py
--- a/lib/mpl_toolkits/mplot3d/axis3d.py
+++ b/lib/mpl_toolkits/mplot3d/axis3d.py
@@ -708,7 +708,7 @@
         bb_1, bb_2 = self._get_ticklabel_bboxes(ticks, renderer)
         other = []
 
-        if self.offsetText.get_visible():
+        if self.offsetText.get_visible() and self.offsetText.get_text():
             other.append(self.offsetText.get_window_extent(renderer))
         if self.line.get_visible():
             other.append(self.line.get_window_extent(renderer))
