diff --git a/sympy/assumptions/handlers/order.py b/sympy/assumptions/handlers/order.py
--- a/sympy/assumptions/handlers/order.py
+++ b/sympy/assumptions/handlers/order.py
@@ -3,7 +3,7 @@
 """
 
 from sympy.assumptions import Q, ask
-from sympy.core import Add, Basic, Expr, Mul, Pow
+from sympy.core import Add, Basic, Expr, Mul, Pow, S
 from sympy.core.logic import fuzzy_not, fuzzy_and, fuzzy_or
 from sympy.core.numbers import E, ImaginaryUnit, NaN, I, pi
 from sympy.functions import Abs, acos, acot, asin, atan, exp, factorial, log
@@ -23,6 +23,10 @@
 
 def _NegativePredicate_number(expr, assumptions):
     r, i = expr.as_real_imag()
+
+    if r == S.NaN or i == S.NaN:
+        return None
+
     # If the imaginary part can symbolically be shown to be zero then
     # we just evaluate the real part; otherwise we evaluate the imaginary
     # part to see if it actually evaluates to zero and if it does then