diff --git a/django/tasks/backends/base.py b/django/tasks/backends/base.py
--- a/django/tasks/backends/base.py
+++ b/django/tasks/backends/base.py
@@ -39,16 +39,8 @@ class BaseTaskBackend(metaclass=ABCMeta):
     def __init__(self, alias, params):
         self.alias = alias
         self.queues = set(params.get("QUEUES", [DEFAULT_TASK_QUEUE_NAME]))
-        self.enqueue_on_commit = bool(params.get("ENQUEUE_ON_COMMIT", True))
         self.options = params.get("OPTIONS", {})
 
-    def _get_enqueue_on_commit_for_task(self, task):
-        return (
-            task.enqueue_on_commit
-            if task.enqueue_on_commit is not None
-            else self.enqueue_on_commit
-        )
-
     def validate_task(self, task):
         """
         Determine whether the provided Task can be executed by the backend.
