diff --git a/xarray/backends/zarr.py b/xarray/backends/zarr.py
--- a/xarray/backends/zarr.py
+++ b/xarray/backends/zarr.py
@@ -1438,7 +1438,8 @@
     group : str, optional
         Group path. (a.k.a. `path` in zarr terminology.)
     chunks : int, dict, "auto" or None, optional
-        If provided, used to load the data into dask arrays.
+        Used to load the data into dask arrays. Default behavior is to use
+        ``chunks={}`` if dask is available, otherwise ``chunks=None``.
 
         - ``chunks='auto'`` will use dask ``auto`` chunking taking into account the
           engine preferred chunks.
