diff --git a/sphinx/ext/autodoc/_docstrings.py b/sphinx/ext/autodoc/_docstrings.py
--- a/sphinx/ext/autodoc/_docstrings.py
+++ b/sphinx/ext/autodoc/_docstrings.py
@@ -88,7 +88,7 @@ def _prepare_docstrings(
 ) -> list[list[str]] | None:
     """Add content from docstrings, attribute documentation and user."""
     # add content from attribute documentation
-    if props.obj_type not in {'data', 'attribute'} and props.parts:
+    if props.obj_type not in {'data', 'attribute', 'type'} and props.parts:
         key = ('.'.join(props.parent_names), props.name)
         try:
             # make a copy of docstring for attributes to avoid cache
@@ -244,7 +244,7 @@ def _get_docstring_lines(
             return []
         return [prepare_docstring(docstring, tab_width)]
 
-    if props.obj_type in {'attribute', 'type'}:
+    if props.obj_type == 'attribute':
         # Check the attribute has a docstring-comment
         comment = _get_attribute_comment(
             parent=parent, obj_path=props.parts, attrname=props.parts[-1]
@@ -298,6 +298,20 @@ def _get_docstring_lines(
             return []
         return [prepare_docstring(docstring, tab_width)]
 
+    if props.obj_type == 'type':
+        try:
+            analyzer = ModuleAnalyzer.for_module(props.module_name)
+            analyzer.analyze()
+        except PycodeError:
+            return None
+
+        key = ('', props.name)
+        if key in analyzer.attr_docs:
+            if comment := list(analyzer.attr_docs[key]):
+                return [comment]
+
+        return None
+
     docstring = getdoc(
         obj,
         allow_inherited=inherit_docstrings,
