diff --git a/django/db/backends/postgresql/compiler.py b/django/db/backends/postgresql/compiler.py
--- a/django/db/backends/postgresql/compiler.py
+++ b/django/db/backends/postgresql/compiler.py
@@ -36,11 +36,11 @@
             # Lack of fields denote the usage of the DEFAULT keyword
             # for the insertion of empty rows.
             or any(field is None for field in fields)
-            # Field.get_placeholder takes value as an argument, so the
+            # Field.get_placeholder_sql takes value as an argument, so the
             # resulting placeholder might be dependent on the value.
             # in UNNEST requires a single placeholder to "fit all values" in
             # the array.
-            or any(hasattr(field, "get_placeholder") for field in fields)
+            or any(hasattr(field, "get_placeholder_sql") for field in fields)
             # Fields that don't use standard internal types might not be
             # unnest'able (e.g. array and geometry types are known to be
             # problematic).
