diff --git a/django/db/models/aggregates.py b/django/db/models/aggregates.py
--- a/django/db/models/aggregates.py
+++ b/django/db/models/aggregates.py
@@ -120,11 +120,6 @@ def resolve_expression(
     ):
         # Aggregates are not allowed in UPDATE queries, so ignore for_save
         c = super().resolve_expression(query, allow_joins, reuse, summarize)
-        c.filter = (
-            c.filter.resolve_expression(query, allow_joins, reuse, summarize)
-            if c.filter
-            else None
-        )
         c.order_by = (
             c.order_by.resolve_expression(query, allow_joins, reuse, summarize)
             if c.order_by