diff --git a/astropy/coordinates/sky_coordinate_parsers.py b/astropy/coordinates/sky_coordinate_parsers.py
--- a/astropy/coordinates/sky_coordinate_parsers.py
+++ b/astropy/coordinates/sky_coordinate_parsers.py
@@ -398,7 +398,7 @@ def _parse_coordinate_arg(coords, frame, units):
                 if (
                     reprname == "d_distance"
                     and not hasattr(orig_vel, reprname)
-                    and "unit" in orig_vel.get_name()
+                    and "unit" in orig_vel.name
                 ):
                     continue
                 values.append(getattr(vel, reprname))
@@ -507,7 +507,7 @@ def _parse_coordinate_arg(coords, frame, units):
             if n_coords > n_attr_names:
                 raise ValueError(
                     f"Input coordinates have {n_coords} values but representation"
-                    f" {frame.representation_type.get_name()} only accepts"
+                    f" {frame.representation_type.name} only accepts"
                     f" {n_attr_names}"
                 )
 