diff --git a/astropy/table/index.py b/astropy/table/index.py
--- a/astropy/table/index.py
+++ b/astropy/table/index.py
@@ -694,7 +694,7 @@ def __init__(self, index: Index, index_slice, original=False):
         if isinstance(index_slice, tuple):
             self.start, self._stop, self.step = index_slice
         elif isinstance(index_slice, slice):  # index_slice is an actual slice
-            num_rows = len(index.columns[0])
+            num_rows = len(index)
             self.start, self._stop, self.step = index_slice.indices(num_rows)
         else:
             raise TypeError("index_slice must be tuple or slice")
