diff --git a/doc/api_reference.py b/doc/api_reference.py
--- a/doc/api_reference.py
+++ b/doc/api_reference.py
@@ -744,6 +744,7 @@ def _get_submodule(module_name, submodule_name):
                     "jaccard_score",
                     "log_loss",
                     "matthews_corrcoef",
+                    "metric_at_thresholds",
                     "multilabel_confusion_matrix",
                     "ndcg_score",
                     "precision_recall_curve",
