diff --git a/sympy/physics/control/lti.py b/sympy/physics/control/lti.py
--- a/sympy/physics/control/lti.py
+++ b/sympy/physics/control/lti.py
@@ -5193,18 +5193,12 @@
         return Matrix(ob)

     def unobservable_subspace(self):
-        """
-        Returns the unobservable subspace of the state space model.
-
-        Note: this method raises an error for matrices with symbolic entries
-        because it is not possible determine the unobservable subspace
-        in general.
-
-        Raises
-        ======
-        NotImplementedError
-            If the state space model has symbolic entries, the unobservable
-            subspace cannot be determined in general.
+        """Returns the unobservable subspace of the state space model."""
+        return self.observability_matrix().nullspace()
+
+    def observable_subspace(self):
+        """
+        Returns the observable subspace of the state space model.

         Examples
         ========
@@ -5223,28 +5217,14 @@
         [0]]), Matrix([
         [0],
         [1]])]
-        """
-        M = self.observability_matrix()
-        if len(M.free_symbols) > 0:
-            raise NotImplementedError(
-                "Unbservable subspace cannot be determined for " \
-                "symbolic matrices."
-            )
-        return M.nullspace()
-
-    def observable_subspace(self):
-        """
-        Returns the observable subspace of the state space model.
-
-        Note: this method raises an error for matrices with symbolic entries
-        because it is not possible determine the observable subspace
-        in general.
-
-        Raises
-        ======
-        NotImplementedError
-            If the state space model has symbolic entries, the observable
-            subspace cannot be determined in general.
+
+        """
+        M = Matrix.hstack(*self.unobservable_subspace())
+        return self._calc_orthogonal_complement(M, self._A.shape[0])
+
+    def is_observable(self):
+        """
+        Returns if the state space model is observable.

         Examples
         ========
@@ -5256,38 +5236,7 @@
         >>> C = Matrix([[0, 1]])
         >>> D = Matrix([1])
         >>> ss = StateSpace(A, B, C, D)
-        >>> ob_subspace = ss.observable_subspace()
-        >>> ob_subspace
-        [Matrix([
-        [1],
-        [0]]), Matrix([
-        [0],
-        [1]])]
-
-        """
-        M = Matrix.hstack(*self.unobservable_subspace())
-        if len(M.free_symbols) > 0:
-                raise NotImplementedError(
-                    "Observable subspace cannot be determined for " \
-                    "symbolic matrices."
-                )
-        return self._calc_orthogonal_complement(M, self._A.shape[0])
-
-    def is_observable(self):
-        """
-        Returns conditions for the state space model to be observable.
-
-        Examples
-        ========
-
-        >>> from sympy import Matrix
-        >>> from sympy.physics.control import StateSpace
-        >>> A1 = Matrix([[-1.5, -2], [1, 0]])
-        >>> B1 = Matrix([0.5, 0])
-        >>> C1 = Matrix([[0, 1]])
-        >>> D1 = Matrix([1])
-        >>> ss1 = StateSpace(A1, B1, C1, D1)
-        >>> ss1.is_observable()
+        >>> ss.is_observable()
         True

         """
@@ -5323,7 +5272,12 @@
         for i in range(1, n):
             co = co.row_join(((self._A)**i) * self._B)

-        return co
+        return Matrix(co)
+
+    def uncontrollable_subspace(self):
+        """Returns the uncontrollable subspace of the state space model."""
+        M = Matrix.hstack(*self.controllable_subspace())
+        return self._calc_orthogonal_complement(M, self._A.shape[0])

     def controllable_subspace(self):
         """