diff --git a/sympy/physics/control/lti.py b/sympy/physics/control/lti.py
--- a/sympy/physics/control/lti.py
+++ b/sympy/physics/control/lti.py
@@ -7,7 +7,6 @@
 from sympy.core.expr import Expr
 from sympy.core.function import expand
 from sympy.core.logic import fuzzy_or
-from sympy.core.logic import fuzzy_or
 from sympy.core.relational import Unequality
 from sympy.core.mul import Mul
 from sympy.core.numbers import I, pi, oo