diff --git a/sympy/physics/control/lti.py b/sympy/physics/control/lti.py
--- a/sympy/physics/control/lti.py
+++ b/sympy/physics/control/lti.py
@@ -4870,7 +4870,7 @@ def apply_similarity_transform(self, transform_matrix):
             \bar C=CT\\
             \bar D=D\end{cases}
 
-        Parameters:
+        Parameters
         ==========
 
         transform_matrix : Matrix
@@ -4879,13 +4879,13 @@ def apply_similarity_transform(self, transform_matrix):
             The transformation matrix must be invertible and have the same
             dimensions as the state matrix `A`.
 
-        Returns:
+        Returns
         =======
 
         StateSpace
             The transformed state space model.
 
-        Examples:
+        Examples
         ========
 
         >>> from sympy import Matrix, Rational