diff --git a/xarray/coding/frequencies.py b/xarray/coding/frequencies.py
--- a/xarray/coding/frequencies.py
+++ b/xarray/coding/frequencies.py
@@ -48,6 +48,7 @@
 from xarray.coding.cftime_offsets import _MONTH_ABBREVIATIONS, _legacy_to_new_freq
 from xarray.coding.cftimeindex import CFTimeIndex
 from xarray.core.common import _contains_datetime_like_objects
+from xarray.core.dtypes import _is_numpy_subdtype
 
 _ONE_MICRO = 1
 _ONE_MILLI = _ONE_MICRO * 1000
@@ -88,9 +89,10 @@ def infer_freq(index):
         elif not _contains_datetime_like_objects(Variable("dim", index)):
             raise ValueError("'index' must contain datetime-like objects")
         dtype = np.asarray(index).dtype
-        if dtype == "datetime64[ns]":
+
+        if _is_numpy_subdtype(dtype, "datetime64"):
             index = pd.DatetimeIndex(index.values)
-        elif dtype == "timedelta64[ns]":
+        elif _is_numpy_subdtype(dtype, "timedelta64"):
             index = pd.TimedeltaIndex(index.values)
         else:
             index = CFTimeIndex(index.values)