diff --git a/sympy/integrals/manualintegrate.py b/sympy/integrals/manualintegrate.py
--- a/sympy/integrals/manualintegrate.py
+++ b/sympy/integrals/manualintegrate.py
@@ -1452,6 +1452,9 @@
     guassian_pattern = exp(a * symbol**2 + b * symbol + c)
     trigexp_over_x_pattern = f*exp(a * symbol)/symbol
     trigexp_over_x_match = integrand.match(trigexp_over_x_pattern)
+    if not (any(term.match(guassian_pattern) for term in integrand.atoms(exp))
+            or (trigexp_over_x_match and
+                trigexp_over_x_match[f].has(sin, cos, sinh, cosh))):
         return
 
     # Replace trig and hyperbolic functions with their exponential forms
