diff --git a/sympy/integrals/manualintegrate.py b/sympy/integrals/manualintegrate.py
--- a/sympy/integrals/manualintegrate.py
+++ b/sympy/integrals/manualintegrate.py
@@ -1447,10 +1447,10 @@ def trig_cmplx_exp_rule(integral: IntegralInfo):
     a = Wild('a', exclude=[symbol, 0])
     b = Wild('b', exclude=[symbol])
     c = Wild('c', exclude=[symbol])
-    n = Wild('n', exclude=[symbol], properties=[lambda n: n > 0])
+    # n = Wild('n', exclude=[symbol], properties=[lambda n: n > 0])
     f = WildFunction('f')
     guassian_pattern = exp(a * symbol**2 + b * symbol + c)
-    trigexp_over_x_pattern = f*exp(a * symbol)/symbol**n
+    trigexp_over_x_pattern = f*exp(a * symbol)/symbol
     trigexp_over_x_match = integrand.match(trigexp_over_x_pattern)
     if not (any(term.match(guassian_pattern) for term in integrand.atoms(exp))
             or (trigexp_over_x_match and