diff --git a/django/db/migrations/serializer.py b/django/db/migrations/serializer.py
--- a/django/db/migrations/serializer.py
+++ b/django/db/migrations/serializer.py
@@ -222,6 +222,17 @@ def serialize(self):
         )
 
 
+class GenericAliasSerializer(BaseSerializer):
+    def serialize(self):
+        imports = set()
+        # Avoid iterating self.value, because it returns itself.
+        # https://github.com/python/cpython/issues/103450
+        for item in self.value.__args__:
+            _, item_imports = serializer_factory(item).serialize()
+            imports.update(item_imports)
+        return repr(self.value), imports
+
+
 class IterableSerializer(BaseSerializer):
     def serialize(self):
         imports = set()
@@ -364,6 +375,7 @@ class Serializer:
         decimal.Decimal: DecimalSerializer,
         (functools.partial, functools.partialmethod): FunctoolsPartialSerializer,
         FUNCTION_TYPES: FunctionTypeSerializer,
+        types.GenericAlias: GenericAliasSerializer,
         collections.abc.Iterable: IterableSerializer,
         (COMPILED_REGEX_TYPE, RegexObject): RegexSerializer,
         uuid.UUID: UUIDSerializer,