diff --git a/xarray/indexes/range_index.py b/xarray/indexes/range_index.py
--- a/xarray/indexes/range_index.py
+++ b/xarray/indexes/range_index.py
@@ -79,12 +79,8 @@ def equals(
         )
 
     def slice(self, sl: slice) -> "RangeCoordinateTransform":
-        sl = normalize_slice(sl, self.size)
-
-        # TODO: support reverse transform (i.e., start > stop)?
-        assert sl.start < sl.stop
-
-        new_size = (sl.stop - sl.start) // sl.step
+        new_range = range(self.size)[sl]
+        new_size = len(new_range)
         new_start = self.start + sl.start * self.step
         new_stop = new_start + new_size * sl.step * self.step
