diff --git a/sphinx/ext/autodoc/__init__.py b/sphinx/ext/autodoc/__init__.py
--- a/sphinx/ext/autodoc/__init__.py
+++ b/sphinx/ext/autodoc/__init__.py
@@ -2431,9 +2431,9 @@ def add_directive_header(self, sig: str) -> None:
         if inspect.iscoroutinefunction(obj) or inspect.isasyncgenfunction(obj):
             self.add_line('   :async:', sourcename)
         if (
-            inspect.isclassmethod(obj)
+            inspect.is_classmethod_like(obj)
             or inspect.is_singledispatch_method(obj)
-            and inspect.isclassmethod(obj.func)
+            and inspect.is_classmethod_like(obj.func)
         ):
             self.add_line('   :classmethod:', sourcename)
         if inspect.isstaticmethod(obj, cls=self.parent, name=self.object_name):
