diff --git a/sklearn/preprocessing/_target_encoder.py b/sklearn/preprocessing/_target_encoder.py
--- a/sklearn/preprocessing/_target_encoder.py
+++ b/sklearn/preprocessing/_target_encoder.py
@@ -412,9 +412,9 @@ def fit_transform(self, X, y, groups=None, **params):
             if not hasattr(self.cv, "split") or isinstance(self.cv, str):
                 if not isinstance(self.cv, Iterable) or isinstance(self.cv, str):
                     raise ValueError(
-                        "Expected `cv` as an integer, cross-validation "
-                        "object (from sklearn.model_selection) "
-                        f"or an iterable. Got {self.cv}."
+                        "Expected `cv` as an integer, a cross-validation object "
+                        "(from sklearn.model_selection), or an iterable yielding "
+                        f"(train, test) splits as arrays of indices. Got {self.cv}."
                     )
                 _test_indices = np.array([])
                 for fold in self.cv:
