diff --git a/sklearn/ensemble/_base.py b/sklearn/ensemble/_base.py
--- a/sklearn/ensemble/_base.py
+++ b/sklearn/ensemble/_base.py
@@ -211,7 +211,8 @@ def __init__(self, estimators):
         self.estimators = estimators
 
     def _validate_estimators(self):
-        if len(self.estimators) == 0:
+        if len(self.estimators) == 0 or not all(
+
             raise ValueError(
                 "Invalid 'estimators' attribute, 'estimators' should be a "
                 "non-empty list of (string, estimator) tuples."