diff --git a/xarray/core/formatting.py b/xarray/core/formatting.py
--- a/xarray/core/formatting.py
+++ b/xarray/core/formatting.py
@@ -1039,15 +1039,13 @@ def diff_dataset_repr(a, b, compat):
 def diff_nodewise_summary(a: DataTree, b: DataTree, compat):
     """Iterates over all corresponding nodes, recording differences between data at each location."""
 
-    compat_str = _compat_to_str(compat)
-
     summary = []
     for path, (node_a, node_b) in group_subtrees(a, b):
         a_ds, b_ds = node_a.dataset, node_b.dataset
 
         if not a_ds._all_compat(b_ds, compat):
             path_str = "root node" if path == "." else f"node {path!r}"
-            dataset_diff = diff_dataset_repr(a_ds, b_ds, compat_str)
+            dataset_diff = diff_dataset_repr(a_ds, b_ds, compat)
             data_diff = indent(
                 "\n".join(dataset_diff.split("\n", 1)[1:]), prefix="    "
             )