diff --git a/astropy/io/fits/convenience.py b/astropy/io/fits/convenience.py
--- a/astropy/io/fits/convenience.py
+++ b/astropy/io/fits/convenience.py
@@ -466,7 +466,7 @@ def writeto(
     )
 
 
-def table_to_hdu(table, character_as_bytes=False):
+def table_to_hdu(table, character_as_bytes=False, name=None):
     """
     Convert an `~astropy.table.Table` object to a FITS
     `~astropy.io.fits.BinTableHDU`.
@@ -479,6 +479,8 @@ def table_to_hdu(table, character_as_bytes=False):
         Whether to return bytes for string columns when accessed from the HDU.
         By default this is `False` and (unicode) strings are returned, but for
         large tables this may use up a lot of memory.
+    name : str
+        Name to be populated in ``EXTNAME`` keyword.
 
     Returns
     -------