diff --git a/sympy/assumptions/handlers/sets.py b/sympy/assumptions/handlers/sets.py
--- a/sympy/assumptions/handlers/sets.py
+++ b/sympy/assumptions/handlers/sets.py
@@ -53,7 +53,7 @@ def _(expr, assumptions):
         raise MDNotImplementedError
     return ret
 
-@IntegerPredicate.register_many(Add, Pow)
+@IntegerPredicate.register(Add)
 def _(expr, assumptions):
     """
     * Integer + Integer       -> Integer