diff --git a/lib/matplotlib/_cm.py b/lib/matplotlib/_cm.py
--- a/lib/matplotlib/_cm.py
+++ b/lib/matplotlib/_cm.py
@@ -1365,6 +1365,10 @@ def _gist_yarg(x): return 1 - x
     (0.8509803921568627,  0.8509803921568627,  0.8509803921568627  ),  # d9d9d9
 )
 
+# Colorblind accessible palettes from
+# Matthew A. Petroff, Accessible Color Sequences for Data Visualization
+# https://arxiv.org/abs/2107.02270
+
 
 _petroff10_data = (
     (0.24705882352941178, 0.5647058823529412,  0.8549019607843137),    # 3f90da