diff --git a/sympy/tensor/array/expressions/arrayexpr_derivatives.py b/sympy/tensor/array/expressions/arrayexpr_derivatives.py
--- a/sympy/tensor/array/expressions/arrayexpr_derivatives.py
+++ b/sympy/tensor/array/expressions/arrayexpr_derivatives.py
@@ -2,7 +2,7 @@
 from functools import reduce, singledispatch
 
 from sympy.core.singleton import S
-from sympy import MatrixBase, derive_by_array, Integer
+from sympy import MatrixBase, derive_by_array, Integer, Function
 from sympy.tensor.array import NDimArray
 from sympy.core.expr import Expr
 from sympy.matrices.expressions.hadamard import HadamardProduct
@@ -35,6 +35,12 @@
             return MatrixUnit(x.shape[0], x.shape[1], expr.i, expr.j)
         raise NotImplementedError("algorithm not implemented for this case")
     return ZeroArray(*x.shape)
+
+
+@array_derive.register(Function)
+def _(expr: Function, x: _ArrayExpr):
+    if len(expr.args) != 1:
+        raise NotImplementedError("only 1-parameter functions are supported")
 
 
 @array_derive.register(ArrayTensorProduct)
