diff --git a/sympy/tensor/array/expressions/arrayexpr_derivatives.py b/sympy/tensor/array/expressions/arrayexpr_derivatives.py
--- a/sympy/tensor/array/expressions/arrayexpr_derivatives.py
+++ b/sympy/tensor/array/expressions/arrayexpr_derivatives.py
@@ -2,7 +2,7 @@
 from functools import reduce, singledispatch
 
 from sympy.core.singleton import S
-from sympy import MatrixBase, derive_by_array, Integer, Function
+from sympy import MatrixBase, derive_by_array, Integer, Determinant, Function
 from sympy.tensor.array import NDimArray
 from sympy.core.expr import Expr
 from sympy.matrices.expressions.hadamard import HadamardProduct
@@ -96,6 +96,12 @@
     return ZeroArray(*(x.shape + expr.shape))
 
 
+@array_derive.register(Determinant)
+def _(expr: Determinant, x: Expr):
+    arg = expr.arg
+    arg_inverse = arg.inv()
+
+
 @array_derive.register(Identity)
 def _(expr: Identity, x: _ArrayExpr):
     return ZeroArray(*(x.shape + expr.shape))
