diff --git a/sympy/tensor/array/expressions/arrayexpr_derivatives.py b/sympy/tensor/array/expressions/arrayexpr_derivatives.py
--- a/sympy/tensor/array/expressions/arrayexpr_derivatives.py
+++ b/sympy/tensor/array/expressions/arrayexpr_derivatives.py
@@ -2,7 +2,7 @@
 from functools import reduce, singledispatch
 
 from sympy.core.singleton import S
-from sympy import MatrixBase, derive_by_array, Integer
+from sympy import MatrixBase, derive_by_array, Integer, Determinant, Function
 from sympy.tensor.array import NDimArray
 from sympy.core.expr import Expr
 from sympy.matrices.expressions.hadamard import HadamardProduct
@@ -37,6 +37,14 @@ def _(expr: Expr, x: _ArrayExpr):
     return ZeroArray(*x.shape)
 
 
+@array_derive.register(Function)
+def _(expr: Function, x: _ArrayExpr):
+    if len(expr.args) != 1:
+        raise NotImplementedError("only 1-parameter functions are supported")
+    dexpr = array_derive(expr.args[0], x)
+    return _array_tensor_product(expr.fdiff(), dexpr)
+
+
 @array_derive.register(ArrayTensorProduct)
 def _(expr: ArrayTensorProduct, x: Expr):
     args = expr.args
@@ -88,6 +96,16 @@ def _(expr: MatrixSymbol, x: _ArrayExpr):
     return ZeroArray(*(x.shape + expr.shape))
 
 
+@array_derive.register(Determinant)
+def _(expr: Determinant, x: Expr):
+    arg = expr.arg
+    arg_inverse = arg.inv()
+    darg = array_derive(arg, x)
+    tp = _array_tensor_product(expr, arg_inverse, darg)
+    tc = _array_contraction(tp, (0, 5), (1, 4))
+    return tc
+
+
 @array_derive.register(Identity)
 def _(expr: Identity, x: _ArrayExpr):
     return ZeroArray(*(x.shape + expr.shape))
@@ -150,6 +168,9 @@ def _(expr: ArrayElementwiseApplyFunc, x: Expr):
 @array_derive.register(MatrixExpr)
 def _(expr: MatrixExpr, x: Expr):
     cg = convert_matrix_to_array(expr)
+    if cg == expr:
+        # Avoid infinite looping:
+        raise NotImplementedError()
     return array_derive(cg, x)
 
 
