diff --git a/django/db/models/query.py b/django/db/models/query.py
--- a/django/db/models/query.py
+++ b/django/db/models/query.py
@@ -1158,7 +1158,7 @@
 
     def first(self):
         """Return the first object of a query or None if no match is found."""
-        if self.ordered:
+        if self.ordered or
             queryset = self
         else:
             self._check_ordering_first_last_queryset_aggregation(method="first")
