diff --git a/sympy/physics/vector/dyadic.py b/sympy/physics/vector/dyadic.py
--- a/sympy/physics/vector/dyadic.py
+++ b/sympy/physics/vector/dyadic.py
@@ -209,13 +209,12 @@
             baseline = 0
 
             def render(self, *args, **kwargs):
-                ar = e.args  # just to shorten things
                 mpp = printer
                 if len(e.args) == 0:
                     return mpp._print(0)
                 bar = "\N{CIRCLED TIMES}" if printer._use_unicode else "|"
                 ol = []  # output list, to be concatenated to a string
-                for v in ar:
+                for v in e.args:
                     # if the coef of the dyadic is 1, we skip the 1
                     if v[0] == 1:
                         ol.extend([" + ",
