diff --git a/sympy/physics/vector/dyadic.py b/sympy/physics/vector/dyadic.py
--- a/sympy/physics/vector/dyadic.py
+++ b/sympy/physics/vector/dyadic.py
@@ -1,6 +1,7 @@
 from sympy import sympify, Add, ImmutableMatrix as Matrix
 from sympy.core.evalf import EvalfMixin
 from sympy.printing.defaults import Printable
+from sympy.printing.pretty.stringpict import prettyForm
 
 from mpmath.libmp.libmpf import prec_to_dps
 
@@ -212,7 +213,9 @@
                 mpp = printer
                 if len(e.args) == 0:
                     return mpp._print(0)
-                bar = "\N{CIRCLED TIMES}" if printer._use_unicode else "|"
+
+                outstr = None
+                bar = prettyForm(u"\N{CIRCLED TIMES}" if mpp._use_unicode else "|")
                 ol = []  # output list, to be concatenated to a string
                 for v in e.args:
                     # if the coef of the dyadic is 1, we skip the 1
