diff --git a/sympy/physics/vector/dyadic.py b/sympy/physics/vector/dyadic.py
--- a/sympy/physics/vector/dyadic.py
+++ b/sympy/physics/vector/dyadic.py
@@ -247,8 +247,7 @@ def render(self, *args, **kwargs):
                         outstr = prettyForm(*p_term.left(sign))
                     else:
                         outstr = prettyForm(*outstr.right(sign, p_term))
-
-                return outstr
+                return outstr.__str__()
         return Fake()
 
     def __rsub__(self, other):
