diff --git a/sympy/physics/vector/dyadic.py b/sympy/physics/vector/dyadic.py
--- a/sympy/physics/vector/dyadic.py
+++ b/sympy/physics/vector/dyadic.py
@@ -209,50 +209,44 @@ class Fake:
             baseline = 0
 
             def render(self, *args, **kwargs):
-                ar = e.args  # just to shorten things
                 mpp = printer
-                if len(ar) == 0:
-                    return str(0)
-                bar = "\N{CIRCLED TIMES}" if printer._use_unicode else "|"
-                ol = []  # output list, to be concatenated to a string
-                for v in ar:
-                    # if the coef of the dyadic is 1, we skip the 1
-                    if v[0] == 1:
-                        ol.extend([" + ",
-                                  mpp.doprint(v[1]),
-                                  bar,
-                                  mpp.doprint(v[2])])
-
-                    # if the coef of the dyadic is -1, we skip the 1
-                    elif v[0] == -1:
-                        ol.extend([" - ",
-                                  mpp.doprint(v[1]),
-                                  bar,
-                                  mpp.doprint(v[2])])
-
-                    # If the coefficient of the dyadic is not 1 or -1,
-                    # we might wrap it in parentheses, for readability.
-                    elif v[0] != 0:
-                        if isinstance(v[0], Add):
-                            arg_str = mpp._print(
-                                v[0]).parens()[0]
+                if len(e.args) == 0:
+                    return mpp._print(0)
+
+                outstr = None
+                bar = prettyForm(u"\N{CIRCLED TIMES}" if mpp._use_unicode else "|")
+                for i, v in enumerate(e.args):
+                    p_v1 = mpp._print(v[1])
+                    p_v2 = mpp._print(v[2])
+                    p_dyad = prettyForm(*p_v1.right(bar, p_v2))
+
+                    c = v[0]
+                    if i == 0:
+                        sign = ""
+                    else:
+                        sign = " + "
+
+                    coeff, rest = c.as_coeff_Mul()
+                    if coeff < 0:
+                        if i == 0:
+                            sign = "- "
                         else:
-                            arg_str = mpp.doprint(v[0])
-                        if arg_str.startswith("-"):
-                            arg_str = arg_str[1:]
-                            str_start = " - "
-                        else:
-                            str_start = " + "
-                        ol.extend([str_start, arg_str, " ",
-                                  mpp.doprint(v[1]),
-                                  bar,
-                                  mpp.doprint(v[2])])
-
-                outstr = "".join(ol)
-                if outstr.startswith(" + "):
-                    outstr = outstr[3:]
-                elif outstr.startswith(" "):
-                    outstr = outstr[1:]
+                            sign = " - "
+                        c = -c
+
+                    if c == 1:
+                        p_term = p_dyad
+                    else:
+                        p_c = mpp._print(c)
+                        if isinstance(c, Add):
+                            p_c = prettyForm(*p_c.parens())
+                        p_term = prettyForm(*p_c.right(" ", p_dyad))
+
+                    if outstr is None:
+                        outstr = prettyForm(*p_term.left(sign))
+                    else:
+                        outstr = prettyForm(*outstr.right(sign, p_term))
+
                 return outstr
         return Fake()
 
