diff --git a/sympy/parsing/sympy_parser.py b/sympy/parsing/sympy_parser.py
--- a/sympy/parsing/sympy_parser.py
+++ b/sympy/parsing/sympy_parser.py
@@ -1163,7 +1163,7 @@ def flatten(self, args, func):
                 arg_func = arg.func
                 if isinstance(arg_func, ast.Call):
                     arg_func = arg_func.func
-                if arg_func.id == func:
+                if isinstance(arg_func, ast.Name) and arg_func.id == func:
                     result.extend(self.flatten(arg.args, func))
                 else:
                     result.append(arg)
@@ -1185,16 +1185,6 @@ def visit_BinOp(self, node):
                     keywords=[ast.keyword(arg='evaluate', value=ast.Constant(value=False))]
                 )
             elif isinstance(node.op, ast.Div):
-                # If numerator is 1, directly return Pow(denominator, -1)
-                if isinstance(left, ast.Call) and \
-                     (isinstance(left.func, ast.Name) and left.func.id == 'Integer') and \
-                     (isinstance(left.args[0], ast.Constant) and left.args[0].value == 1):
-                    return ast.Call(
-                    func=ast.Name(id='Pow', ctx=ast.Load()),
-                    args=[right, ast.UnaryOp(op=ast.USub(), operand=ast.Constant(1))],
-                    keywords=[ast.keyword(arg='evaluate', value=ast.Constant(value=False))]
-                    )
-
                 if isinstance(node.left, ast.UnaryOp):
                     left, right = right, left
                     rev = True
@@ -1212,6 +1202,15 @@ def visit_BinOp(self, node):
 
                 if rev:  # undo reversal
                     left, right = right, left
+                # Check if numerator is 1, if yes, directly returns Pow(denom, -1)
+                temp_node = ast.parse("lambda n, d: d if n == 1 else (Mul(*n.args, d, evaluate=False) \
+                                      if isinstance(n, Mul) else Mul(n, d, evaluate=False))").body[0].value
+
+                return ast.Call(
+                    func=temp_node,
+                    args=[left, right],
+                    keywords=[]
+                )
 
             new_node = ast.Call(
                 func=ast.Name(id=sympy_class, ctx=ast.Load()),
