diff --git a/sklearn/tree/_classes.py b/sklearn/tree/_classes.py
--- a/sklearn/tree/_classes.py
+++ b/sklearn/tree/_classes.py
@@ -1126,8 +1126,6 @@ class DecisionTreeRegressor(RegressorMixin, BaseDecisionTree):
         absolute error, which minimizes the L1 loss using the median of each terminal
         node, and "poisson" which uses reduction in Poisson deviance to find splits,
         also using the mean of each terminal node.
-        Training using "absolute_error" is significantly slower
-        than when using "squared_error".
 
         .. versionadded:: 0.18
            Mean Absolute Error (MAE) criterion.
@@ -1777,8 +1775,6 @@ class ExtraTreeRegressor(DecisionTreeRegressor):
         absolute error, which minimizes the L1 loss using the median of each terminal
         node, and "poisson" which uses reduction in Poisson deviance to find splits,
         also using the mean of each terminal node.
-        Training using "absolute_error" is significantly slower
-        than when using "squared_error".
 
         .. versionadded:: 0.18
            Mean Absolute Error (MAE) criterion.
