diff --git a/sklearn/tree/_classes.py b/sklearn/tree/_classes.py
--- a/sklearn/tree/_classes.py
+++ b/sklearn/tree/_classes.py
@@ -1124,7 +1124,8 @@ class DecisionTreeRegressor(RegressorMixin, BaseDecisionTree):
         variance reduction as feature selection criterion and minimizes the L2
         loss using the mean of each terminal node, "absolute_error" for the mean
         absolute error, which minimizes the L1 loss using the median of each terminal
-        node, and "poisson" which uses reduction in Poisson deviance to find splits, also using the mean of each terminal node.
+        node, and "poisson" which uses reduction in Poisson deviance to find splits,
+        also using the mean of each terminal node.
         Training using "absolute_error" is significantly slower
         than when using "squared_error".
 
@@ -1774,7 +1775,8 @@ class ExtraTreeRegressor(DecisionTreeRegressor):
         variance reduction as feature selection criterion and minimizes the L2
         loss using the mean of each terminal node, "absolute_error" for the mean
         absolute error, which minimizes the L1 loss using the median of each terminal
-        node, and "poisson" which uses reduction in Poisson deviance to find splits, also using the mean of each terminal node.
+        node, and "poisson" which uses reduction in Poisson deviance to find splits,
+        also using the mean of each terminal node.
         Training using "absolute_error" is significantly slower
         than when using "squared_error".
 
