diff --git a/sympy/combinatorics/named_groups.py b/sympy/combinatorics/named_groups.py
--- a/sympy/combinatorics/named_groups.py
+++ b/sympy/combinatorics/named_groups.py
@@ -93,6 +93,8 @@
     # small cases are special
     if n == 1:
         G = PermutationGroup([Permutation([0])])
+    elif n == 2:
+        G = PermutationGroup([Permutation([0,1])])
 
     a = list(range(n))
     a[0], a[1], a[2] = a[1], a[2], a[0]
