diff --git a/sympy/functions/special/error_functions.py b/sympy/functions/special/error_functions.py
--- a/sympy/functions/special/error_functions.py
+++ b/sympy/functions/special/error_functions.py
@@ -1749,7 +1749,7 @@
             raise ArgumentIndexError(self, argindex)
 
     def _eval_evalf(self, prec):
-        return self.rewrite(li).evalf(prec)
+        return self.rewrite(li)._eval_evalf(prec)
 
     def _eval_rewrite_as_li(self, z, **kwargs):
         return li(z) - li(2)
