diff --git a/sympy/printing/repr.py b/sympy/printing/repr.py
--- a/sympy/printing/repr.py
+++ b/sympy/printing/repr.py
@@ -9,6 +9,8 @@
 from typing import Any
 
 from sympy.core.function import AppliedUndef
+from sympy.core.mul import Mul
+from sympy.core.add import Add
 from mpmath.libmp import repr_dps, to_str as mlib_to_str
 
 from .printer import Printer, print_function
@@ -47,7 +49,8 @@ def emptyPrinter(self, expr):
             return str(expr)
 
     def _print_Add(self, expr, order=None):
-        args = map(self._print, expr.args)
+        args = Add.make_args(expr)
+        args = map(self._print, args)
         clsname = type(expr).__name__
         return clsname + "(%s)" % ", ".join(args)
 
@@ -191,7 +194,8 @@ def _print_NaN(self, expr):
         return "nan"
 
     def _print_Mul(self, expr, order=None):
-        args = map(self._print, expr.args)
+        args = Mul.make_args(expr)
+        args = map(self._print, args)
         clsname = type(expr).__name__
         return clsname + "(%s)" % ", ".join(args)
 