diff --git a/examples/release_highlights/plot_release_highlights_1_3_0.py b/examples/release_highlights/plot_release_highlights_1_3_0.py
--- a/examples/release_highlights/plot_release_highlights_1_3_0.py
+++ b/examples/release_highlights/plot_release_highlights_1_3_0.py
@@ -58,7 +58,7 @@
 X, true_labels = load_digits(return_X_y=True)
 print(f"number of digits: {len(np.unique(true_labels))}")
 
-hdbscan = HDBSCAN(min_cluster_size=15).fit(X)
+hdbscan = HDBSCAN(min_cluster_size=15, copy=True).fit(X)
 non_noisy_labels = hdbscan.labels_[hdbscan.labels_ != -1]
 print(f"number of clusters found: {len(np.unique(non_noisy_labels))}")
 
