diff --git a/sphinx/io.py b/sphinx/io.py
--- a/sphinx/io.py
+++ b/sphinx/io.py
@@ -23,9 +23,7 @@
     from docutils.parsers import Parser
     from docutils.transforms import Transform
 
-    from sphinx.application import Sphinx
     from sphinx.environment import BuildEnvironment
-    from sphinx.registry import SphinxComponentRegistry
 
 
 logger = logging.getLogger(__name__)
@@ -127,7 +125,7 @@
         destination=NullOutput(),
     )
     # Propagate exceptions by default when used programmatically:
-    defaults = {'traceback': True, **app.env.settings}
+    defaults = {'traceback': True, **env.settings}
     # Set default settings
     pub.get_settings(**defaults)
     return pub
