diff --git a/xarray/core/options.py b/xarray/core/options.py
--- a/xarray/core/options.py
+++ b/xarray/core/options.py
@@ -183,7 +183,7 @@ class set_options:
 
     Parameters
     ----------
-    arithmetic_broadcast: bool, default: True
+    arithmetic_broadcast : bool, default: True
         Whether to perform automatic broadcasting in binary operations.
     arithmetic_compat: {"identical", "equals", "broadcast_equals", "no_conflicts", "override", "minimal"}, default: "minimal"
         How to compare non-index coordinates of the same name for potential
