diff --git a/sklearn/inspection/_plot/decision_boundary.py b/sklearn/inspection/_plot/decision_boundary.py
--- a/sklearn/inspection/_plot/decision_boundary.py
+++ b/sklearn/inspection/_plot/decision_boundary.py
@@ -246,12 +246,15 @@
                 elif cmap == "tab20" and n_responses <= 20:
                     colors = plt.get_cmap("tab20", 20).colors[:n_responses]
                 else:
-                    colors = plt.get_cmap(cmap, n_responses).colors
-            elif isinstance(self.multiclass_colors, str):
-                colors = colors = plt.get_cmap(
-                    self.multiclass_colors, n_responses
-                ).colors
+                    cmap = plt.get_cmap(cmap, n_responses)
+                    if not hasattr(cmap, "colors"):
+                        # For LinearSegmentedColormap
+                        colors = cmap(np.linspace(0, 1, n_responses))
+                    else:
+                        colors = cmap.colors
+            elif isinstance(self.multiclass_colors, list):
+                colors = [mpl.colors.to_rgba(color) for color in self.multiclass_colors]
             else:
-                colors = [mpl.colors.to_rgba(color) for color in self.multiclass_colors]
+                raise ValueError("'multiclass_colors' must be a list or a str.")
 
             self.multiclass_colors_ = colors
             multiclass_cmaps = [
