diff --git a/sphinx/util/docutils.py b/sphinx/util/docutils.py
--- a/sphinx/util/docutils.py
+++ b/sphinx/util/docutils.py
@@ -5,7 +5,7 @@
 import os
 import re
 import warnings
-from contextlib import contextmanager
+from contextlib import contextmanager, nullcontext
 from copy import copy
 from pathlib import Path
 from typing import TYPE_CHECKING
@@ -22,7 +22,8 @@
 
 from sphinx.errors import SphinxError
 from sphinx.locale import __
-from sphinx.util import logging
+from sphinx.transforms import SphinxTransformer
+from sphinx.util import logging, rst
 from sphinx.util.parsing import nested_parse_to_nodes
 
 logger = logging.getLogger(__name__)
