diff --git a/sphinx/ext/autodoc/_generate.py b/sphinx/ext/autodoc/_generate.py
--- a/sphinx/ext/autodoc/_generate.py
+++ b/sphinx/ext/autodoc/_generate.py
@@ -16,11 +16,11 @@
 from sphinx.util.typing import restify, stringify_annotation
 
 if TYPE_CHECKING:
-    from collections.abc import Iterator
+    from collections.abc import Iterator, Mapping, MutableSet
     from typing import Literal
 
     from sphinx.config import Config
-    from sphinx.environment import BuildEnvironment, _CurrentDocument
+    from sphinx.environment import _CurrentDocument
     from sphinx.events import EventManager
     from sphinx.ext.autodoc._directive_options import _AutoDocumenterOptions
     from sphinx.ext.autodoc._property_types import _ItemProperties
@@ -38,13 +38,14 @@
     *,
     config: Config,
     current_document: _CurrentDocument,
-    env: BuildEnvironment,
     events: EventManager,
     get_attr: _AttrGetter,
     indent: str,
     options: _AutoDocumenterOptions,
     props: _ItemProperties,
     record_dependencies: set[str],
+    ref_context: Mapping[str, str | None],
+    reread_always: MutableSet[str],
     result: StringList,
 ) -> None:
     """Generate reST for the object given by *props*, and possibly for its members.
@@ -136,7 +137,6 @@
         attr_docs=analyzer.attr_docs if analyzer is not None else {},
         config=config,
         current_document=current_document,
-        env=env,
         events=events,
         get_attr=get_attr,
         indent=indent,
@@ -144,6 +144,8 @@
         props=props,
         real_modname=real_modname,
         record_dependencies=record_dependencies,
+        ref_context=ref_context,
+        reread_always=reread_always,
         result=result,
     )
 
