diff --git a/lib/matplotlib/_mathtext.py b/lib/matplotlib/_mathtext.py
--- a/lib/matplotlib/_mathtext.py
+++ b/lib/matplotlib/_mathtext.py
@@ -2639,7 +2639,7 @@
 
         x: List
         if super is None:
-            # node757
+            # Align subscript without superscript (node757).
             # Note: One of super or sub must be a Node if we're in this function, but
             # mypy can't know this, since it can't interpret pyparsing expressions,
             # hence the cast.
@@ -2648,29 +2648,37 @@
             if self.is_dropsub(last_char):
                 shift_down = lc_baseline + consts.subdrop * x_height
             else:
-                shift_down = consts.sub1 * x_height
+                shift_down = max(shift_down, consts.sub1 * x_height,
+                                 x.height - x_height * 4 / 5)
             x.shift_amount = shift_down
         else:
+            # Align superscript (node758).
             x = Hlist([Kern(superkern), super])
             x.shrink()
             if self.is_dropsub(last_char):
                 shift_up = lc_height - consts.subdrop * x_height
             else:
-                shift_up = consts.sup1 * x_height
+                shift_up = max(shift_up, consts.sup1 * x_height, x.depth + x_height / 4)
             if sub is None:
                 x.shift_amount = -shift_up
-            else:  # Both sub and superscript
+            else:
+                # Align subscript with superscript (node759).
                 y = Hlist([Kern(subkern), sub])
                 y.shrink()
                 if self.is_dropsub(last_char):
                     shift_down = lc_baseline + consts.subdrop * x_height
                 else:
-                    shift_down = consts.sub2 * x_height
-                # If sub and superscript collide, move super up
-                clr = (2 * rule_thickness -
+                    shift_down = max(shift_down, consts.sub2 * x_height)
+                # If the subscript and superscript are too close to each other,
+                # move the subscript down.
+                clr = (4 * rule_thickness -
                        ((shift_up - x.depth) - (y.height - shift_down)))
                 if clr > 0.:
-                    shift_up += clr
+                    shift_down += clr
+                    clr = x_height * 4 / 5 - shift_up + x.depth
+                    if clr > 0:
+                        shift_up += clr
+                        shift_down -= clr
                 x = Vlist([
                     x,
                     Kern((shift_up - x.depth) - (y.height - shift_down)),
