diff --git a/sympy/polys/ring_series.py b/sympy/polys/ring_series.py
--- a/sympy/polys/ring_series.py
+++ b/sympy/polys/ring_series.py
@@ -506,11 +506,13 @@ def _series_inversion1(p, x, prec):
     if _has_constant_term(p - c, x):
         raise ValueError("p cannot contain a constant term depending on "
                          "parameters")
+    if not R.domain.is_unit(c):
+        raise ValueError(f"Constant term {c} must be a unit in {R.domain}")
+
     one = R(1)
     if R.domain is EX:
         one = 1
     if c != one:
-        # TODO add check that it is a unit
         p1 = R(1)/c
     else:
         p1 = R(1)