diff --git a/sympy/polys/ring_series.py b/sympy/polys/ring_series.py
--- a/sympy/polys/ring_series.py
+++ b/sympy/polys/ring_series.py
@@ -506,7 +506,7 @@ def _series_inversion1(p, x, prec):
     if _has_constant_term(p - c, x):
         raise ValueError("p cannot contain a constant term depending on "
                          "parameters")
-    if hasattr(R.domain, 'is_unit') and not R.domain.is_unit(c):
+    if not R.domain.is_unit(c):
         raise ValueError(f"Constant term {c} must be a unit in {R.domain}")
 
     one = R(1)