diff --git a/astropy/units/format/ogip.py b/astropy/units/format/ogip.py
--- a/astropy/units/format/ogip.py
+++ b/astropy/units/format/ogip.py
@@ -48,10 +48,9 @@ class OGIP(Base, _ParsingFormatMixin):
         "WHITESPACE",
         "POWER",
         "STAR",
-        "SIGN",
-        "UFLOAT",
+        "FLOAT",
         "LIT10",
-        "UINT",
+        "INT",
         "UNKNOWN",
         "FUNCNAME",
         "UNIT",
@@ -106,21 +105,16 @@ def _lexer(cls) -> Lexer:
 
         # NOTE THE ORDERING OF THESE RULES IS IMPORTANT!!
         # Regular expression rules for simple tokens
-        def t_UFLOAT(t):
-            r"(((\d+\.?\d*)|(\.\d+))([eE][+-]?\d+))|(((\d+\.\d*)|(\.\d+))([eE][+-]?\d+)?)"
+        def t_FLOAT(t):
+            r"[+-]?((((\d+\.?\d*)|(\.\d+))([eE][+-]?\d+))|(((\d+\.\d*)|(\.\d+))([eE][+-]?\d+)?))"
             t.value = float(t.value)
             return t
 
-        def t_UINT(t):
-            r"\d+"
+        def t_INT(t):
+            r"[+-]?\d+"
             t.value = int(t.value)
             return t
 
-        def t_SIGN(t):
-            r"[+-](?=\d)"
-            t.value = 1 if t.value == "+" else -1
-            return t
-
         def t_LIT10(t):
             r"10"
             return 10
@@ -254,9 +248,8 @@ def p_scale_factor(p):
             """
             scale_factor : LIT10 POWER numeric_power
                          | LIT10
-                         | signed_float
-                         | signed_float POWER numeric_power
-                         | signed_int POWER numeric_power
+                         | number
+                         | number POWER numeric_power
             """
             if len(p) == 4:
                 p[0] = 10 ** p[3]
@@ -280,11 +273,9 @@ def p_product(p):
 
         def p_numeric_power(p):
             """
-            numeric_power : UINT
-                          | signed_float
-                          | OPEN_PAREN signed_int CLOSE_PAREN
-                          | OPEN_PAREN signed_float CLOSE_PAREN
-                          | OPEN_PAREN signed_float DIVISION UINT CLOSE_PAREN
+            numeric_power : number
+                          | OPEN_PAREN number CLOSE_PAREN
+                          | OPEN_PAREN INT DIVISION INT CLOSE_PAREN
             """
             if len(p) == 6:
                 p[0] = Fraction(int(p[2]), int(p[4]))
@@ -300,28 +291,12 @@ def p_numeric_power(p):
                         )
                     )
 
-        def p_sign(p):
-            """
-            sign : SIGN
-                 |
+        def p_number(p):
             """
-            if len(p) == 2:
-                p[0] = p[1]
-            else:
-                p[0] = 1.0
-
-        def p_signed_int(p):
+            number : INT
+                   | FLOAT
             """
-            signed_int : SIGN UINT
-            """
-            p[0] = p[1] * p[2]
-
-        def p_signed_float(p):
-            """
-            signed_float : sign UINT
-                         | sign UFLOAT
-            """
-            p[0] = p[1] * p[2]
+            p[0] = p[1]
 
         def p_error(p):
             raise ValueError()
