diff --git a/sympy/matrices/expressions/determinant.py b/sympy/matrices/expressions/determinant.py
--- a/sympy/matrices/expressions/determinant.py
+++ b/sympy/matrices/expressions/determinant.py
@@ -52,6 +52,10 @@ def doit(self, **hints):
 
         return self
 
+    def _eval_derivative(self, x):
+        # Derivative currently implements `hasattr(..., "_eval_derivative")` to proceed:
+        return None
+
 
 def det(matexpr):
     """ Matrix Determinant
diff --git a/sympy/matrices/expressions/special.py b/sympy/matrices/expressions/special.py
--- a/sympy/matrices/expressions/special.py
+++ b/sympy/matrices/expressions/special.py
@@ -328,7 +328,7 @@ def _entry(self, i, j, **kwargs):
 
 class MatrixUnit(MatrixExpr):
     """
-    Matrix with only one nonzero entry with value 1.
+    Matrix with only one nonzero entry with value 1. Also called single-entry matrix.
 
     https://en.wikipedia.org/wiki/Matrix_unit
 
diff --git a/sympy/tensor/array/expressions/arrayexpr_derivatives.py b/sympy/tensor/array/expressions/arrayexpr_derivatives.py
--- a/sympy/tensor/array/expressions/arrayexpr_derivatives.py
+++ b/sympy/tensor/array/expressions/arrayexpr_derivatives.py
@@ -2,7 +2,7 @@
 from functools import reduce, singledispatch
 
 from sympy.core.singleton import S
-from sympy import MatrixBase, derive_by_array, Integer
+from sympy import MatrixBase, derive_by_array, Integer, Determinant, Function
 from sympy.tensor.array import NDimArray
 from sympy.core.expr import Expr
 from sympy.matrices.expressions.hadamard import HadamardProduct
@@ -37,6 +37,14 @@ def _(expr: Expr, x: _ArrayExpr):
     return ZeroArray(*x.shape)
 
 
+@array_derive.register(Function)
+def _(expr: Function, x: _ArrayExpr):
+    if len(expr.args) != 1:
+        raise NotImplementedError("only 1-parameter functions are supported")
+    dexpr = array_derive(expr.args[0], x)
+    return _array_tensor_product(expr.fdiff(), dexpr)
+
+
 @array_derive.register(ArrayTensorProduct)
 def _(expr: ArrayTensorProduct, x: Expr):
     args = expr.args
@@ -88,6 +96,16 @@ def _(expr: MatrixSymbol, x: _ArrayExpr):
     return ZeroArray(*(x.shape + expr.shape))
 
 
+@array_derive.register(Determinant)
+def _(expr: Determinant, x: Expr):
+    arg = expr.arg
+    arg_inverse = arg.inv()
+    darg = array_derive(arg, x)
+    tp = _array_tensor_product(expr, arg_inverse, darg)
+    tc = _array_contraction(tp, (0, 5), (1, 4))
+    return tc
+
+
 @array_derive.register(Identity)
 def _(expr: Identity, x: _ArrayExpr):
     return ZeroArray(*(x.shape + expr.shape))
@@ -150,6 +168,9 @@ def _(expr: ArrayElementwiseApplyFunc, x: Expr):
 @array_derive.register(MatrixExpr)
 def _(expr: MatrixExpr, x: Expr):
     cg = convert_matrix_to_array(expr)
+    if cg == expr:
+        # Avoid infinite looping:
+        raise NotImplementedError()
     return array_derive(cg, x)
 
 
diff --git a/sympy/tensor/array/expressions/from_matrix_to_array.py b/sympy/tensor/array/expressions/from_matrix_to_array.py
--- a/sympy/tensor/array/expressions/from_matrix_to_array.py
+++ b/sympy/tensor/array/expressions/from_matrix_to_array.py
@@ -69,8 +69,10 @@ def convert_matrix_to_array(expr: Basic) -> Basic:
     elif isinstance(expr, MatPow):
         base = convert_matrix_to_array(expr.base)
         if expr.exp.is_Integer != True:
-            b = symbols("b", cls=Dummy)
-            return ArrayElementwiseApplyFunc(Lambda(b, b**expr.exp), convert_matrix_to_array(base))
+            if get_shape(expr) == (1, 1):
+                b = symbols("b", cls=Dummy)
+                return ArrayElementwiseApplyFunc(Lambda(b, b ** expr.exp), convert_matrix_to_array(base))
+            return expr
         elif (expr.exp > 0) == True:
             return convert_matrix_to_array(MatMul.fromiter(base for i in range(expr.exp)))
         else: