diff --git a/sphinx/ext/autodoc/directive.py b/sphinx/ext/autodoc/directive.py
--- a/sphinx/ext/autodoc/directive.py
+++ b/sphinx/ext/autodoc/directive.py
@@ -10,6 +10,7 @@
     _AutoDocumenterOptions,
     _process_documenter_options,
 )
+from sphinx.ext.autodoc._documenters import _AutodocAttrGetter
 from sphinx.util import logging
 from sphinx.util.docutils import SphinxDirective, switch_source_input
 from sphinx.util.parsing import nested_parse_to_nodes
@@ -25,6 +26,7 @@
     from sphinx.environment import BuildEnvironment
     from sphinx.ext.autodoc import Documenter
     from sphinx.ext.autodoc._directive_options import Options
+    from sphinx.ext.autodoc.importer import _AttrGetter
 
 logger = logging.getLogger(__name__)
 
@@ -50,6 +52,7 @@
         options: _AutoDocumenterOptions,
         lineno: int,
         state: Any,
+        get_attr: _AttrGetter,
     ) -> None:
         self.env = env
         self._reporter = reporter
@@ -58,6 +61,7 @@
         self.record_dependencies: set[str] = set()
         self.result = StringList()
         self.state = state
+        self.get_attr = get_attr
 
 
 def process_documenter_options(
