diff --git a/django/views/debug.py b/django/views/debug.py
--- a/django/views/debug.py
+++ b/django/views/debug.py
@@ -415,6 +415,8 @@
             c["exception_type"] = self.exc_type.__name__
         if self.exc_value:
             c["exception_value"] = getattr(
+                self.exc_value, "raw_error_message", self.exc_value
+            )
             if exc_notes := getattr(self.exc_value, "__notes__", None):
                 c["exception_notes"] = "\n" + "\n".join(exc_notes)
         if frames:
