diff --git a/astropy/samp/hub_script.py b/astropy/samp/hub_script.py
--- a/astropy/samp/hub_script.py
+++ b/astropy/samp/hub_script.py
@@ -3,6 +3,7 @@
 
 import argparse
 import copy
+import logging
 import sys
 import time
 
@@ -106,16 +107,20 @@ def hub_script(timeout=0):
         "specify the output files where redirect the logging messages.",
     )
 
+    # valid logging levels as strings, reverse ordered by level.
+    # the default 'INFO' is assumed to exist (...)
+    log_levels = sorted(d := logging.getLevelNamesMapping(), key=d.get, reverse=True)
     log_group.add_argument(
         "-L",
         "--log-level",
         dest="loglevel",
         metavar="LEVEL",
-        help="set the Hub instance log level (OFF, ERROR, WARNING, INFO, DEBUG).",
+        help=f"set the Hub instance log level ({', '.join(log_levels)}).",
         type=str,
-        choices=["OFF", "ERROR", "WARNING", "INFO", "DEBUG"],
+        choices=log_levels,
         default="INFO",
     )
+    del log_levels
 
     log_group.add_argument(
         "-O",
@@ -170,8 +175,7 @@ def hub_script(timeout=0):
     options = parser.parse_args()
 
     try:
-        if options.loglevel in ("OFF", "ERROR", "WARNING", "DEBUG", "INFO"):
-            log.setLevel(options.loglevel)
+        log.setLevel(options.loglevel)
 
         if options.logout != "":
             context = log.log_to_file(options.logout)