diff --git a/django/db/models/fields/related_descriptors.py b/django/db/models/fields/related_descriptors.py
--- a/django/db/models/fields/related_descriptors.py
+++ b/django/db/models/fields/related_descriptors.py
@@ -1270,6 +1270,7 @@
                     self.target_field_name,
                     *objs,
                     through_defaults=through_defaults,
+                    using=db,
                 )
                 # If this is a symmetrical m2m relation to self, add the mirror
                 # entry in the m2m table.
@@ -1279,6 +1280,7 @@
                         self.source_field_name,
                         *objs,
                         through_defaults=through_defaults,
+                        using=db,
                     )
 
         def add(self, *objs, through_defaults=None):
@@ -1520,7 +1522,12 @@
             )
 
         def _add_items(
-            self, source_field_name, target_field_name, *objs, through_defaults=None
+            self,
+            source_field_name,
+            target_field_name,
+            *objs,
+            through_defaults=None,
+            using=None,
         ):
             # source_field_name: the PK fieldname in join table for the source
             # object target_field_name: the PK fieldname in join table for the
@@ -1531,7 +1538,7 @@
 
             through_defaults = dict(resolve_callables(through_defaults or {}))
             target_ids = self._get_target_ids(target_field_name, objs)
-            db = router.db_for_write(self.through, instance=self.instance)
+            db = using or router.db_for_write(self.through, instance=self.instance)
             can_ignore_conflicts, must_send_signals, can_fast_add = self._get_add_plan(
                 db, source_field_name
             )
