diff --git a/django/db/models/fields/related_descriptors.py b/django/db/models/fields/related_descriptors.py
--- a/django/db/models/fields/related_descriptors.py
+++ b/django/db/models/fields/related_descriptors.py
@@ -1315,8 +1315,8 @@
 
         aremove.alters_data = True
 
-        def clear(self):
-            db = router.db_for_write(self.through, instance=self.instance)
+        def _clear_base(self, using=None):
+            db = using or router.db_for_write(self.through, instance=self.instance)
             with transaction.atomic(using=db, savepoint=False):
                 signals.m2m_changed.send(
                     sender=self.through,
@@ -1327,7 +1327,6 @@
                     pk_set=None,
                     using=db,
                 )
-                self._remove_prefetched_objects()
                 filters = self._build_remove_filters(super().get_queryset().using(db))
                 self.through._default_manager.using(db).filter(filters).delete()
 
@@ -1340,6 +1339,11 @@
                     pk_set=None,
                     using=db,
                 )
+
+        def clear(self):
+            self._remove_prefetched_objects()
+            db = router.db_for_write(self.through, instance=self.instance)
+            self._clear_base(using=db)
 
         clear.alters_data = True
 
