diff --git a/sympy/polys/partfrac.py b/sympy/polys/partfrac.py
--- a/sympy/polys/partfrac.py
+++ b/sympy/polys/partfrac.py
@@ -52,7 +52,7 @@
     You can choose Bronstein's algorithm by setting ``full=True``:
 
     >>> apart(y/(x**2 + x + 1), x, full=True)
-    RootSum(_w**2 + _w + 1, Lambda(_a, (-2*_a*y/3 - y/3)/(-_a + x)))
+    RootSum(w**2 + w + 1, Lambda(w, (-2*w*y/3 - y/3)/(-w + x)))
 
     Calling ``doit()`` yields a human-readable result:
 
@@ -286,7 +286,7 @@
     1)])
 
     >>> assemble_partfrac_list(pfd)
-    RootSum(_w**2 + _w + t, Lambda(_a, (-2*_a*t/(4*t - 1) - t/(4*t - 1))/(-_a + x)))
+    RootSum(t + w**2 + w, Lambda(w, (-2*t*w/(4*t - 1) - t/(4*t - 1))/(-w + x)))
 
     This example is taken from Bronstein's original paper:
 
@@ -454,7 +454,7 @@
 
     >>> pfda = assemble_partfrac_list(pfd)
     >>> pfda
-    RootSum(_w**2 - 2, Lambda(_a, _a/(-_a + x)))/2
+    RootSum(w**2 - 2, Lambda(w, w/(-w + x)))/2
 
     >>> pfda.doit()
     -sqrt(2)/(2*(x + sqrt(2))) + sqrt(2)/(2*(x - sqrt(2)))
