diff --git a/lib/matplotlib/widgets.py b/lib/matplotlib/widgets.py
--- a/lib/matplotlib/widgets.py
+++ b/lib/matplotlib/widgets.py
@@ -14,6 +14,7 @@
 import enum
 import functools
 import itertools
+import weakref
 from numbers import Integral, Number
 
 from cycler import cycler
@@ -120,16 +121,6 @@ def __init__(self, ax):
         self._cids = []
         self._blit_background_id = None
 
-    def __del__(self):
-        blit_background_id = getattr(self, '_blit_background_id', None)
-        # __del__ may be called on a partially initialized object, e.g.,
-        # when __init__ raises. Therefore, we handle missing attributes
-        # gracefully.
-        if blit_background_id is not None:
-            canvas = getattr(self, 'canvas', None)
-            if canvas is not None:
-                canvas._release_blit_background_id(blit_background_id)
-
     canvas = property(
         lambda self: getattr(self.ax.get_figure(root=True), 'canvas', None)
     )
@@ -170,7 +161,9 @@ def _save_blit_background(self, background):
         good enough for all existing widgets.
         """
         if self._blit_background_id is None:
-            self._blit_background_id = self.canvas._get_blit_background_id()
+            bbid = self.canvas._get_blit_background_id()
+            weakref.finalize(self, self.canvas._release_blit_background_id, bbid)
+            self._blit_background_id = bbid
         self.canvas._blit_backgrounds[self._blit_background_id] = background
 
     def _load_blit_background(self):
