diff --git a/sklearn/utils/multiclass.py b/sklearn/utils/multiclass.py
--- a/sklearn/utils/multiclass.py
+++ b/sklearn/utils/multiclass.py
@@ -120,7 +120,9 @@ def unique_labels(*ys, ys_types=None):
     )
     # Check that we don't mix string type with number type
     if len(set(isinstance(label, str) for label in ys_labels)) > 1:
-        msg_details = "Got " + " ".join([f"{xp.unique(y)}" for y in ys])
+        msg_details = (
+            "Got " + " and ".join([f"{xp.unique_values(y)}" for y in ys]) + "."
+        )
         raise ValueError(f"Mix of label input types (string and number); {msg_details}")
 
     return xp.asarray(sorted(ys_labels))
