diff --git a/sklearn/utils/multiclass.py b/sklearn/utils/multiclass.py
--- a/sklearn/utils/multiclass.py
+++ b/sklearn/utils/multiclass.py
@@ -80,9 +80,10 @@ def unique_labels(*ys, ys_types=None):
         raise ValueError("No argument has been passed.")
     # Check that we don't mix label format
 
-    ys_types = set(type_of_target(x) for x in ys)
-    if ys_types == {"binary", "multiclass"}:
-        ys_types = {"multiclass"}
+    if ys_types is None:
+        ys_types = set(type_of_target(x) for x in ys)
+        if ys_types == {"binary", "multiclass"}:
+            ys_types = {"multiclass"}
 
     if len(ys_types) > 1:
         raise ValueError("Mix type of y not allowed, got types %s" % ys_types)
