diff --git a/sympy/ntheory/qs.py b/sympy/ntheory/qs.py
--- a/sympy/ntheory/qs.py
+++ b/sympy/ntheory/qs.py
@@ -279,23 +279,13 @@
             continue
         v = sieve_poly.eval_v(x)
         vec, num = _check_smoothness(v, factor_base)
+        if num == 1:
+            smooth_relations.append((sieve_poly.eval_u(x), v, vec))
+        elif num < partial_relation_upper_bound and isprime(num):
+            if N % num == 0:
+                proper_factor.add(num)
                 continue
-            if large_prime not in partial_relations:
-                partial_relations[large_prime] = (u, v)
-                continue
             else:
-                u_prev, v_prev = partial_relations[large_prime]
-                partial_relations.pop(large_prime)
-                try:
-                    large_prime_inv = invert(large_prime, N)
-                except ZeroDivisionError:#if large_prime divides N
-                    proper_factor.add(large_prime)
-                    continue
-                u = (u*u_prev*large_prime_inv) % N
-                v = v*v_prev // large_prime**2
-                vec, _ = _check_smoothness(v, factor_base)
-        #assert u*u % N == v % N
-        smooth_relations.append((u, v, vec))
     return smooth_relations, proper_factor
 
 
