diff --git a/django/db/models/sql/query.py b/django/db/models/sql/query.py
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -1219,7 +1219,7 @@ def check_alias(self, alias):
             if "aggregate" in {frame.function for frame in inspect.stack()}:
                 stacklevel = 5
             else:
-                # annotate() and alias().
+                # annotate(), alias(), and values().
                 stacklevel = 6
             warnings.warn(
                 "Using percent signs in a column alias is deprecated.",