diff --git a/sympy/matrices/expressions/blockmatrix.py b/sympy/matrices/expressions/blockmatrix.py
--- a/sympy/matrices/expressions/blockmatrix.py
+++ b/sympy/matrices/expressions/blockmatrix.py
@@ -1,7 +1,6 @@
 from sympy.assumptions.ask import (Q, ask)
 from sympy.core import Basic, Add, Mul, S
 from sympy.core.sympify import _sympify
-from sympy.functions import adjoint
 from sympy.functions.elementary.complexes import re, im
 from sympy.strategies import typed, exhaust, condition, do_one, unpack
 from sympy.strategies.traverse import bottom_up
@@ -187,13 +186,9 @@ def _eval_transpose(self):
         return BlockMatrix(M)
 
     def _eval_adjoint(self):
-        # Adjoint all the individual matrices
-        matrices = [adjoint(matrix) for matrix in self.blocks]
-        # Make a copy
-        M = Matrix(self.blockshape[0], self.blockshape[1], matrices)
-        # Transpose the block structure
-        M = M.transpose()
-        return BlockMatrix(M)
+        return BlockMatrix(
+            Matrix(self.blockshape[0], self.blockshape[1], self.blocks).adjoint()
+        )
 
     def _eval_trace(self):
         if self.rowblocksizes == self.colblocksizes: