diff --git a/astropy/io/fits/fitsrec.py b/astropy/io/fits/fitsrec.py
--- a/astropy/io/fits/fitsrec.py
+++ b/astropy/io/fits/fitsrec.py
@@ -881,6 +881,7 @@
 
         # Convert all fields equal to the TNULL value (nullval) to either NaN
         # for float columns or 0 for the other fields.
+        if format.format
         null_fill = str(ASCIITNULL)
         null_fill = encode_ascii(null_fill.rjust(format.width))
         dummy = np.where(np.char.strip(dummy) == nullval, null_fill, dummy)
