diff --git a/django/contrib/admin/widgets.py b/django/contrib/admin/widgets.py
--- a/django/contrib/admin/widgets.py
+++ b/django/contrib/admin/widgets.py
@@ -287,6 +287,10 @@
         if not isinstance(widget, AutocompleteMixin):
             self.attrs["data-context"] = "available-source"
         # Only single-select Select widgets are supported.
+        supported = not getattr(
+            widget, "allow_multiple_selected", False
+        ) and isinstance(widget, Select)
+        self.can_change_related = supported and can_change_related
         # XXX: The deletion UX can be confusing when dealing with cascading
         # deletion.
         cascade = getattr(rel, "on_delete", None) is CASCADE
