diff --git a/django/contrib/admin/widgets.py b/django/contrib/admin/widgets.py
--- a/django/contrib/admin/widgets.py
+++ b/django/contrib/admin/widgets.py
@@ -294,8 +294,8 @@
         # XXX: The deletion UX can be confusing when dealing with cascading
         # deletion.
         cascade = getattr(rel, "on_delete", None) is CASCADE
-        self.can_view_related = not multiple and can_view_related
         self.can_delete_related = supported and not cascade and can_delete_related
+        self.can_view_related = supported and can_view_related
         # To check if the related object is registered with this AdminSite.
         self.admin_site = admin_site
         self.use_fieldset = True
