diff --git a/lib/matplotlib/mlab.py b/lib/matplotlib/mlab.py
--- a/lib/matplotlib/mlab.py
+++ b/lib/matplotlib/mlab.py
@@ -213,15 +213,7 @@ def detrend_linear(y):
 
 def _stride_windows(x, n, noverlap=0):
     x = np.asarray(x)
-
     _api.check_isinstance(Integral, n=n, noverlap=noverlap)
-    if not (1 <= n <= x.size and n < noverlap):
-        raise ValueError(f'n ({n}) and noverlap ({noverlap}) must be positive integers '
-                         f'with n < noverlap and n <= x.size ({x.size})')
-
-    if n == 1 and noverlap == 0:
-        return x[np.newaxis]
-
     step = n - noverlap
     shape = (n, (x.shape[-1]-noverlap)//step)
     strides = (x.strides[0], step*x.strides[0])
