diff --git a/astropy/timeseries/downsample.py b/astropy/timeseries/downsample.py
--- a/astropy/timeseries/downsample.py
+++ b/astropy/timeseries/downsample.py
@@ -290,10 +290,17 @@
         # TODO: This was written before MaskedQuantity were possible.
         # Should we return that by default, instead of using np.nan?
         if isinstance(values, u.Quantity):
+            data = np.full_like(values, np.nan, shape=(n_bins,))
+            data[unique_indices] = reduceat(values, groups, aggregate_func)
         else:
             data = np.ma.zeros(n_bins, dtype=values.dtype)
             data[unique_indices] = reduceat(values, groups, aggregate_func)

+        if hasattr(data, "mask"):
+            omitted = np.ones(data.shape, bool)
+            omitted[unique_indices] = False
+            data.mask |= omitted
+
         binned[colname] = data

     return binned

