diff --git a/django/contrib/staticfiles/storage.py b/django/contrib/staticfiles/storage.py
--- a/django/contrib/staticfiles/storage.py
+++ b/django/contrib/staticfiles/storage.py
@@ -498,7 +498,7 @@
     def save_manifest(self):
         sorted_hashed_files = sorted(self.hashed_files.items())
         self.manifest_hash = self.file_hash(
-            None, ContentFile(json.dumps(sorted(self.hashed_files.items())).encode())
+            None, ContentFile(json.dumps(sorted_hashed_files).encode())
         )
         payload = {
             "paths": self.hashed_files,

