diff --git a/sympy/polys/ring_series.py b/sympy/polys/ring_series.py
--- a/sympy/polys/ring_series.py
+++ b/sympy/polys/ring_series.py
@@ -51,7 +51,8 @@
 from sympy.core import PoleError, Function, Expr
 from sympy.core.numbers import Rational
 from sympy.core.intfunc import igcd
-from sympy.functions import sin, cos, tan, atan, exp, atanh, tanh, log, ceiling
+from sympy.functions import (sin, cos, tan, atan, exp, atanh, tanh, log, ceiling,
+                             sinh, cosh)
 from sympy.utilities.misc import as_int
 from mpmath.libmp.libintmath import giant_steps
 import math
@@ -1858,7 +1859,10 @@ def rs_compose_add(p1, p2):
         'exp': 'rs_exp',
         'tan': 'rs_tan',
         'log': 'rs_log',
-        'atan': 'rs_atan'
+        'atan': 'rs_atan',
+        'sinh': 'rs_sinh',
+        'cosh': 'rs_cosh',
+        'tanh': 'rs_tanh'
         }
 
 def rs_min_pow(expr, series_rs, a):