diff --git a/galleries/examples/color/color_sequences.py b/galleries/examples/color/color_sequences.py
--- a/galleries/examples/color/color_sequences.py
+++ b/galleries/examples/color/color_sequences.py
@@ -38,8 +38,8 @@ def plot_color_sequences(names, ax):
 
 built_in_color_sequences = [
     'tab10', 'tab20', 'tab20b', 'tab20c', 'Pastel1', 'Pastel2', 'Paired',
-    'Accent', 'Dark2', 'Set1', 'Set2', 'Set3', 'petroff6', 'petroff8',
-    'petroff10']
+    'Accent', 'Accessible', 'Dark2', 'Set1', 'Set2', 'Set3', 'petroff6',
+    'petroff8', 'petroff10']
 
 
 fig, ax = plt.subplots(figsize=(6.4, 9.6), layout='constrained')
