diff --git a/sphinx/writers/latex.py b/sphinx/writers/latex.py
--- a/sphinx/writers/latex.py
+++ b/sphinx/writers/latex.py
@@ -132,11 +132,9 @@ def __init__(self, node: Element) -> None:
             self.colsep = None
         self.colwidths: list[int] = []
         self.has_problematic = False
-        self.has_oldproblematic = False
         self.has_verbatim = False
         # cf https://github.com/sphinx-doc/sphinx/issues/13646#issuecomment-2958309632
         self.is_nested = False
-        self.entry_needs_linetrimming = 0
         self.caption: list[str] = []
         self.stubs: list[int] = []
 
@@ -223,12 +221,6 @@ def get_colspec(self) -> str:
         elif self.get_table_type() == 'tabulary':
             # sphinx.sty sets T to be J by default.
             return '{' + _colsep + (('T' + _colsep) * self.colcount) + '}' + CR
-        elif self.has_oldproblematic:
-            return (
-                r'{%s*{%d}{\X{1}{%d}%s}}'
-                % (_colsep, self.colcount, self.colcount, _colsep)
-                + CR
-            )
         else:
             return '{' + _colsep + (('l' + _colsep) * self.colcount) + '}' + CR
 
@@ -804,8 +796,6 @@ def visit_desc(self, node: Element) -> None:
         else:
             self.body.append(BLANKLINE)
             self.body.append(r'\begin{fulllineitems}' + CR)
-        if self.table:
-            self.table.has_problematic = True
 
     def depart_desc(self, node: Element) -> None:
         if self.in_desc_signature:
@@ -1107,8 +1097,6 @@ def visit_seealso(self, node: Element) -> None:
             r'\begin{sphinxseealso}{%s:}' % admonitionlabels['seealso'] + CR
         )
         self.no_latex_floats += 1
-        if self.table:
-            self.table.has_problematic = True
 
     def depart_seealso(self, node: Element) -> None:
         self.body.append(BLANKLINE)
@@ -1367,19 +1355,25 @@ def visit_entry(self, node: Element) -> None:
                 r'\sphinxmultirow{%d}{%d}{%%' % (cell.height, cell.cell_id) + CR
             )
             context = '}%' + CR + context
-        if cell.width > 1 or cell.height > 1:
-            self.body.append(
-                r'\begin{varwidth}[t]{\sphinxcolwidth{%d}{%d}}'
-                % (cell.width, self.table.colcount)
-                + CR
-            )
-            context = (
-                r'\par' + CR + r'\vskip-\baselineskip'
-                r'\vbox{\hbox{\strut}}\end{varwidth}%' + CR + context
-            )
-            self.table.entry_needs_linetrimming = 1
-        if len(list(node.findall(nodes.paragraph))) >= 2:
-            self.table.has_oldproblematic = True
+        # 8.3.0 wraps ALL cells contents in "varwidth".  This fixes a
+        # number of issues and allows more usage of tabulary.
+        #
+        # "varwidth" usage allows a *tight fit* to multiple paragraphs,
+        # line blocks, bullet lists, enumerated lists; it is less
+        # successful at finding a tight fit for object descriptions or
+        # admonitions: the table will then probably occupy full-width, and
+        # columns containing such cells will auto-divide the total width
+        # equally.
+        #
+        # "\sphinxcolwidth" has an appropriate definition in
+        # sphinxlatextables.sty which in particular takes into account
+        # tabulary "two-pass" system.
+        self.body.append(
+            r'\begin{varwidth}[t]{\sphinxcolwidth{%d}{%d}}'
+            % (cell.width, self.table.colcount)
+            + CR
+        )
+        context = r'\sphinxbeforeendvarwidth' + CR + r'\end{varwidth}%' + CR + context
         if (
             isinstance(node.parent.parent, nodes.thead)
             or (cell.col in self.table.stubs)
@@ -1392,20 +1386,17 @@ def visit_entry(self, node: Element) -> None:
                 pass
             else:
                 self.body.append(r'\sphinxstyletheadfamily ')
-        if self.table.entry_needs_linetrimming:
-            self.pushbody([])
+        self.pushbody([])
         self.context.append(context)
 
     def depart_entry(self, node: Element) -> None:
         assert self.table is not None
-        if self.table.entry_needs_linetrimming:
-            self.table.entry_needs_linetrimming = 0
-            body = self.popbody()
+        body = self.popbody()
 
-            # Remove empty lines from top of merged cell
-            while body and body[0] == CR:
-                body.pop(0)
-            self.body.extend(body)
+        # Remove empty lines from top of merged cell
+        while body and body[0] == CR:
+            body.pop(0)
+        self.body.extend(body)
 
         self.body.append(self.context.pop())
 
@@ -1446,8 +1437,6 @@ def visit_acks(self, node: Element) -> None:
     def visit_bullet_list(self, node: Element) -> None:
         if not self.compact_list:
             self.body.append(r'\begin{itemize}' + CR)
-        if self.table:
-            self.table.has_problematic = True
 
     def depart_bullet_list(self, node: Element) -> None:
         if not self.compact_list:
@@ -1485,8 +1474,6 @@ def get_nested_level(node: Element) -> int:
         )
         if 'start' in node:
             self.body.append(r'\setcounter{%s}{%d}' % (enum, node['start'] - 1) + CR)
-        if self.table:
-            self.table.has_problematic = True
 
     def depart_enumerated_list(self, node: Element) -> None:
         self.body.append(r'\end{enumerate}' + CR)
@@ -1501,8 +1488,6 @@ def depart_list_item(self, node: Element) -> None:
 
     def visit_definition_list(self, node: Element) -> None:
         self.body.append(r'\begin{description}' + CR)
-        if self.table:
-            self.table.has_problematic = True
 
     def depart_definition_list(self, node: Element) -> None:
         self.body.append(r'\end{description}' + CR)
@@ -1542,8 +1527,6 @@ def depart_definition(self, node: Element) -> None:
 
     def visit_field_list(self, node: Element) -> None:
         self.body.append(r'\begin{quote}\begin{description}' + CR)
-        if self.table:
-            self.table.has_problematic = True
 
     def depart_field_list(self, node: Element) -> None:
         self.body.append(r'\end{description}\end{quote}' + CR)
@@ -1585,8 +1568,6 @@ def depart_paragraph(self, node: Element) -> None:
 
     def visit_centered(self, node: Element) -> None:
         self.body.append(CR + r'\begin{center}')
-        if self.table:
-            self.table.has_problematic = True
 
     def depart_centered(self, node: Element) -> None:
         self.body.append(CR + r'\end{center}')
@@ -1601,8 +1582,6 @@ def visit_hlist(self, node: Element) -> None:
             r'\begin{itemize}\setlength{\itemsep}{0pt}'
             r'\setlength{\parskip}{0pt}' + CR
         )
-        if self.table:
-            self.table.has_problematic = True
 
     def depart_hlist(self, node: Element) -> None:
         self.compact_list -= 1
@@ -1798,8 +1777,6 @@ def depart_legend(self, node: Element) -> None:
     def visit_admonition(self, node: Element) -> None:
         self.body.append(CR + r'\begin{sphinxadmonition}{note}')
         self.no_latex_floats += 1
-        if self.table:
-            self.table.has_problematic = True
 
     def depart_admonition(self, node: Element) -> None:
         self.body.append(r'\end{sphinxadmonition}' + CR)
@@ -1811,8 +1788,6 @@ def _visit_named_admonition(self, node: Element) -> None:
             CR + r'\begin{sphinxadmonition}{%s}{%s:}' % (node.tagname, label)
         )
         self.no_latex_floats += 1
-        if self.table:
-            self.table.has_problematic = True
 
     def _depart_named_admonition(self, node: Element) -> None:
         self.body.append(r'\end{sphinxadmonition}' + CR)
@@ -2312,8 +2287,6 @@ def visit_line_block(self, node: Element) -> None:
             self.body.append(r'\begin{DUlineblock}{\DUlineblockindent}' + CR)
         else:
             self.body.append(CR + r'\begin{DUlineblock}{0em}' + CR)
-        if self.table:
-            self.table.has_problematic = True
 
     def depart_line_block(self, node: Element) -> None:
         self.body.append(r'\end{DUlineblock}' + CR)
@@ -2329,8 +2302,6 @@ def visit_block_quote(self, node: Element) -> None:
                 done = 1
         if not done:
             self.body.append(r'\begin{quote}' + CR)
-            if self.table:
-                self.table.has_problematic = True
 
     def depart_block_quote(self, node: Element) -> None:
         done = 0
@@ -2370,8 +2341,6 @@ def depart_option_group(self, node: Element) -> None:
 
     def visit_option_list(self, node: Element) -> None:
         self.body.append(r'\begin{optionlist}{3cm}' + CR)
-        if self.table:
-            self.table.has_problematic = True
 
     def depart_option_list(self, node: Element) -> None:
         self.body.append(r'\end{optionlist}' + CR)