diff --git a/sympy/geometry/plane.py b/sympy/geometry/plane.py
--- a/sympy/geometry/plane.py
+++ b/sympy/geometry/plane.py
@@ -781,7 +781,7 @@
         rv = Point(pt, dim=3)
         if rv in self:
             return rv
-        return self.intersection(Line3D(rv, rv + Point3D(self.normal_vector)))[0]
+        n = Point3D(self.normal_vector)
 
     def random_point(self, seed=None):
         """ Returns a random point on the Plane.
