diff --git a/sklearn/metrics/_plot/precision_recall_curve.py b/sklearn/metrics/_plot/precision_recall_curve.py
--- a/sklearn/metrics/_plot/precision_recall_curve.py
+++ b/sklearn/metrics/_plot/precision_recall_curve.py
@@ -6,6 +6,7 @@
 from sklearn.metrics._ranking import average_precision_score, precision_recall_curve
 from sklearn.utils._plotting import (
     _BinaryClassifierCurveDisplayMixin,
+    _deprecate_estimator_name,
     _deprecate_y_pred_parameter,
     _despine,
     _validate_style_kwargs,
@@ -37,9 +38,11 @@ class PrecisionRecallDisplay(_BinaryClassifierCurveDisplayMixin):
     average_precision : float, default=None
         Average precision. If None, the average precision is not shown.
 
-    estimator_name : str, default=None
+    name : str, default=None
         Name of estimator. If None, then the estimator name is not shown.
 
+        .. versionchanged:: 1.8
+
     pos_label : int, float, bool or str, default=None
         The class considered the positive class when precision and recall metrics
         computed. If not `None`, this value is displayed in the x- and y-axes labels.
@@ -53,6 +56,13 @@ class PrecisionRecallDisplay(_BinaryClassifierCurveDisplayMixin):
 
         .. versionadded:: 1.3
 
+    estimator_name : str, default=None
+        Name of estimator. If None, the estimator name is not shown.
+
+        .. deprecated:: 1.8
+            `estimator_name` is deprecated and will be removed in 1.10. Use `name`
+            instead.
+
     Attributes
     ----------
     line_ : matplotlib Artist
@@ -118,11 +128,12 @@ def __init__(
         recall,
         *,
         average_precision=None,
-        estimator_name=None,
+        name=None,
         pos_label=None,
         prevalence_pos_label=None,
+        estimator_name="deprecated",
     ):
-        self.estimator_name = estimator_name
+        self.name = _deprecate_estimator_name(estimator_name, name, "1.8")
         self.precision = precision
         self.recall = recall
         self.average_precision = average_precision
@@ -151,7 +162,7 @@ def plot(
 
         name : str, default=None
             Name of precision recall curve for labeling. If `None`, use
-            `estimator_name` if not `None`, otherwise no labeling is shown.
+            `name` if not `None`, otherwise no labeling is shown.
 
         plot_chance_level : bool, default=False
             Whether to plot the chance level. The chance level is the prevalence
@@ -555,7 +566,7 @@ def from_predictions(
             precision=precision,
             recall=recall,
             average_precision=average_precision,
-            estimator_name=name,
+            name=name,
             pos_label=pos_label,
             prevalence_pos_label=prevalence_pos_label,
         )
