diff --git a/sklearn/cluster/_kmeans.py b/sklearn/cluster/_kmeans.py
--- a/sklearn/cluster/_kmeans.py
+++ b/sklearn/cluster/_kmeans.py
@@ -935,10 +935,13 @@
 
     def _validate_center_shape(self, X, centers, n_centroids=None):
         """Check if the shape of the centers is correct."""
-        if centers.shape[0] != self.n_clusters:
+        if n_centroids is None:
+            n_centroids = self.n_clusters
+
+        if centers.shape[0] != n_centroids:
             raise ValueError(
                 f"The shape of the initial centers {centers.shape} does not "
-                f"match the number of clusters {self.n_clusters}."
+                f"match the number of clusters {n_centroids}."
             )
         if centers.shape[1] != X.shape[1]:
             raise ValueError(
