diff --git a/sklearn/cluster/_kmeans.py b/sklearn/cluster/_kmeans.py
--- a/sklearn/cluster/_kmeans.py
+++ b/sklearn/cluster/_kmeans.py
@@ -943,7 +943,6 @@ def _validate_center_shape(self, X, centers, n_centroids=None):
                 f"The shape of the initial centers {centers.shape} does not "
                 f"match the number of clusters {n_centroids}."
             )
-
         if centers.shape[1] != X.shape[1]:
             raise ValueError(
                 f"The shape of the initial centers {centers.shape} does not "
@@ -1041,7 +1040,7 @@ def _init_centroids(
         elif callable(init):
             centers = init(X, n_clusters, random_state=random_state)
             centers = check_array(centers, dtype=X.dtype, copy=False, order="C")
-            self._validate_center_shape(X, centers, n_centers=n_clusters)
+            self._validate_center_shape(X, centers, n_centroids=n_clusters)
 
         if sp.issparse(centers):
             centers = centers.toarray()
