diff --git a/astropy/wcs/wcs.py b/astropy/wcs/wcs.py
--- a/astropy/wcs/wcs.py
+++ b/astropy/wcs/wcs.py
@@ -3325,6 +3325,7 @@ def slice(self, view, numpy_order=True):
         wcs_new : `~astropy.wcs.WCS`
             A new resampled WCS axis
         """
+        if
         if hasattr(view, "__len__") and len(view) > self.wcs.naxis:
             raise ValueError("Must have # of slices <= # of WCS axes")
         elif not hasattr(view, "__len__"):  # view MUST be an iterable