diff --git a/astropy/wcs/wcs.py b/astropy/wcs/wcs.py
--- a/astropy/wcs/wcs.py
+++ b/astropy/wcs/wcs.py
@@ -3325,6 +3325,10 @@ def slice(self, view, numpy_order=True):
         wcs_new : `~astropy.wcs.WCS`
             A new resampled WCS axis
         """
+
+        if view is Ellipsis:
+            return self.deepcopy()
+
         if hasattr(view, "__len__") and len(view) > self.wcs.naxis:
             raise ValueError("Must have # of slices <= # of WCS axes")
         elif not hasattr(view, "__len__"):  # view MUST be an iterable
@@ -3409,20 +3413,21 @@ def slice(self, view, numpy_order=True):
                     for table in distortion_tables:
                         table.crval[wcs_index] -= iview.start
 
-            try:
-                # range requires integers but the other attributes can also
-                # handle arbitrary values, so this needs to be in a try/except.
-                nitems = len(builtins.range(self._naxis[wcs_index])[iview])
-            except TypeError as exc:
-                if "indices must be integers" not in str(exc):
-                    raise
-                warnings.warn(
-                    f"NAXIS{wcs_index} attribute is not updated because at "
-                    f"least one index ('{iview}') is no integer.",
-                    AstropyUserWarning,
-                )
-            else:
-                wcs_new._naxis[wcs_index] = nitems
+            if self._naxis != [0, 0]:
+                try:
+                    # range requires integers but the other attributes can also
+                    # handle arbitrary values, so this needs to be in a try/except.
+                    nitems = len(builtins.range(self._naxis[wcs_index])[iview])
+                except TypeError as exc:
+                    if "indices must be integers" not in str(exc):
+                        raise
+                    warnings.warn(
+                        f"NAXIS{wcs_index} attribute is not updated because at "
+                        f"least one index ('{iview}') is no integer.",
+                        AstropyUserWarning,
+                    )
+                else:
+                    wcs_new._naxis[wcs_index] = nitems
 
         if wcs_new.sip is not None:
             wcs_new.sip = Sip(