diff --git a/sklearn/tree/_export.py b/sklearn/tree/_export.py
--- a/sklearn/tree/_export.py
+++ b/sklearn/tree/_export.py
@@ -334,9 +334,7 @@
 
         # Write impurity
         if self.impurity:
-            if isinstance(criterion, _criterion.FriedmanMSE):
-                criterion = "friedman_mse"
-            elif isinstance(criterion, _criterion.MSE) or criterion == "squared_error":
+            if isinstance(criterion, _criterion.MSE) or criterion == "squared_error":
                 criterion = "squared_error"
             elif not isinstance(criterion, str):
                 criterion = "impurity"
