diff --git a/astropy/units/quantity_helper/function_helpers.py b/astropy/units/quantity_helper/function_helpers.py
--- a/astropy/units/quantity_helper/function_helpers.py
+++ b/astropy/units/quantity_helper/function_helpers.py
@@ -514,68 +514,115 @@ def _block(arrays, max_depth, result_ndim, depth=0):
 
 UNIT_FROM_LIKE_ARG = object()
 
-if NUMPY_LT_2_0:
 
-    @function_helper
-    def arange(*args, start=None, stop=None, step=None, dtype=None):
-        return arange_impl(*args, start=start, stop=stop, step=step, dtype=dtype)
-else:
+if not NUMPY_LT_2_0:
 
     @function_helper
-    def arange(*args, start=None, stop=None, step=None, dtype=None, device=None):
+    def arange(
+        start_or_stop,
+        /,
+        stop=None,
+        step=1,
+        *,
+        dtype=None,
+        device=None,
+    ):
         return arange_impl(
-            *args, start=start, stop=stop, step=step, dtype=dtype, device=device
+            start_or_stop, stop=stop, step=step, dtype=dtype, device=device
         )
 
+else:
 
-def arange_impl(*args, start=None, stop=None, step=None, dtype=None, **kwargs):
-    # NumPy is supposed to validate the input parameters before this dispatched
-    # function is reached. Nevertheless, we'll sprinkle a few rundundant
-    # sanity checks in the form of `assert` statements.
-    # As they are not part of the business logic, it is fine if they are
-    # compiled-away (e.g. the Python interpreter runs with -O)
-    assert len(args) <= 4
-
-    # bind positional arguments to their meaningful names
-    # following the (complex) logic of np.arange
-    match args:
-        case (pos1,):
-            assert stop is None or start is None
-            if stop is None:
-                stop = pos1
-            elif start is None:
-                start = pos1
-        case start, stop, *rest:
-            if start is not None and stop is None:
-                start, stop = stop, start
-            match rest:
-                # rebind step and dtype if possible
-                case (step,):
-                    pass
-                case step, dtype:
-                    pass
-
-    # as the only required argument, we want stop to set the unit of the output
-    # so it's important that it comes first in the qty_kwargs
-    qty_kwargs = {
-        k: v
-        for k, v in (("stop", stop), ("start", start), ("step", step))
-        if v is not None
-    }
+    @function_helper
+    def arange(start_or_stop, /, stop=None, step=1, *, dtype=None):
+        return arange_impl(start_or_stop, stop=stop, step=step, dtype=dtype)
+
+
+def unwrap_arange_args(*, start_or_stop, stop_, step_):
+    # handle the perilous task of disentangling original arguments
+    # This isn't trivial because start_or_stop may actually bind to two
+    # different inputs, as the name suggests.
+    # We (ab)use structural pattern matching here to bind output variables
+    # (start, stop, step), so no additional logic is actually needed after
+    # a match is found.
+    match (start_or_stop, stop_, step_):
+        case (stop, None as start, step):
+            pass
+        case (start, stop, step):
+            pass
+
+    # purely defensive programming
+    assert stop is not None, "Please report this."
+    return start, stop, step
+
+
+def wrap_arange_args(*, start, stop, step, expected_out_unit):
+    # do the reverse operation than unwrap_arange_args
+    # this is needed because start_or_stop *must* be passed as positional
+
+    # purely defensive programming
+    assert stop is not None, "Please report this."
+
+    match start, stop:
+        case (None, _):
+            qty_args = (stop,)
+        case _:
+            qty_args = (start, stop)
+
+    step_val = step.to_value(expected_out_unit) if hasattr(step, "unit") else step
+
+    kwargs = {} if step == 1 else {"step": step_val}
+
+    # reverse positional arguments so `stop` always comes first
+    # this is done to ensure that the arrays are first converted to the
+    # expected unit, which we guarantee should be stop's
+    args_rev, out_unit = _quantities2arrays(*qty_args[::-1])
+    if expected_out_unit is not UNIT_FROM_LIKE_ARG:
+        assert out_unit == expected_out_unit
+    if hasattr(stop, "unit"):
+        assert out_unit == stop.unit
+
+    # reverse args again to restore initial order
+    args = args_rev[::-1]
+
+    if "step" in kwargs:
+        kwargs["step"] = step_val
+    return args, kwargs
+
+
+def arange_impl(start_or_stop, /, *, stop, step, dtype, device=None):
+    # Because this wrapper requires exceptional amounts of additional logic
+    # to unwrap/wrap its complicated signature, we'll sprinkle a few
+    # sanity checks in the form of `assert` statements, which should help making
+    # heads or tails of what's happening in the event of an unexpected exception.
+    #
+    # also note that we intentionally choose to match numpy.arange's signature
+    # at typecheck time, as opposed to its actual runtime signature, which is
+    # even richer (as of numpy 2.4). For instance, this means we don't support
+    # `start` being passed as keyword, or `dtype` being passed as positional.
+    # This is done to improve the overall stability and maintainability of this
+    # complicated wrapper function.
+    start, stop, step = unwrap_arange_args(
+        start_or_stop=start_or_stop, stop_=stop, step_=step
+    )
     out_unit = getattr(stop, "unit", UNIT_FROM_LIKE_ARG)
-    if out_unit is UNIT_FROM_LIKE_ARG:
-        if hasattr(start, "unit") or hasattr(step, "unit"):
-            raise TypeError(
-                "stop without a unit cannot be combined with start or step with a unit."
-            )
-        kwargs.update(qty_kwargs)
-    else:
-        # Convert possible start, step to stop units.
-        new_values, _ = _quantities2arrays(*qty_kwargs.values())
-        kwargs.update(zip(qty_kwargs.keys(), new_values))
+
+    if out_unit is UNIT_FROM_LIKE_ARG and (
+        hasattr(start, "unit") or hasattr(step, "unit")
+    ):
+        raise TypeError(
+            "stop without a unit cannot be combined with start or step with a unit."
+        )
+
+    args, kwargs = wrap_arange_args(
+        start=start, stop=stop, step=step, expected_out_unit=out_unit
+    )
 
     kwargs["dtype"] = dtype
-    return (), kwargs, out_unit, None
+    if not NUMPY_LT_2_0:
+        kwargs["device"] = device
+
+    return args, kwargs, out_unit, None
 
 
 if NUMPY_LT_2_0: