diff --git a/sympy/utilities/iterables.py b/sympy/utilities/iterables.py
--- a/sympy/utilities/iterables.py
+++ b/sympy/utilities/iterables.py
@@ -1,6 +1,6 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, overload
 
 from collections import Counter, defaultdict, OrderedDict
 from itertools import (
@@ -21,7 +21,7 @@
 
 
 if TYPE_CHECKING:
-    from typing import TypeVar, Iterable, Callable, Sequence
+    from typing import TypeVar, Iterable, Callable, Literal, Sequence
     T = TypeVar('T')
 
 
