diff --git a/sphinx/builders/__init__.py b/sphinx/builders/__init__.py
--- a/sphinx/builders/__init__.py
+++ b/sphinx/builders/__init__.py
@@ -269,7 +269,7 @@ def cat2relpath(cat: CatalogInfo, srcdir: Path = self.srcdir) -> str:
             __('writing output... '),
             'darkgreen',
             len(catalogs),
-            self._app.verbosity,
+            self.config.verbosity,
             stringify_func=cat2relpath,
         ):
             catalog.write_mo(
@@ -587,7 +587,7 @@ def _read_serial(self, docnames: list[str]) -> None:
             __('reading sources... '),
             'purple',
             len(docnames),
-            self._app.verbosity,
+            self.config.verbosity,
         ):
             # remove all inventory entries for that file
             self.events.emit('env-purge-doc', self.env, docname)
@@ -604,7 +604,7 @@ def _read_parallel(self, docnames: list[str], nproc: int) -> None:
             __('reading sources... '),
             'purple',
             len(chunks),
-            self._app.verbosity,
+            self.config.verbosity,
         )
 
         # clear all outdated docs at once
@@ -776,7 +776,7 @@ def _write_serial(self, docnames: Sequence[str]) -> None:
                 __('writing output... '),
                 'darkgreen',
                 len(docnames),
-                self._app.verbosity,
+                self.config.verbosity,
             ):
                 _write_docname(docname, env=self.env, builder=self, tags=self.tags)
 
@@ -800,7 +800,7 @@ def write_process(docs: list[tuple[str, nodes.document]]) -> None:
             __('writing output... '),
             'darkgreen',
             len(chunks),
-            self._app.verbosity,
+            self.config.verbosity,
         )
 
         def on_chunk_done(args: list[tuple[str, nodes.document]], result: None) -> None:
