diff --git a/astropy/visualization/wcsaxes/wcsapi.py b/astropy/visualization/wcsaxes/wcsapi.py
--- a/astropy/visualization/wcsaxes/wcsapi.py
+++ b/astropy/visualization/wcsaxes/wcsapi.py
@@ -107,6 +107,9 @@
     for k, v in mapping.items():
         k = k.removeprefix("custom:")
         if k in CUSTOM_UCD_COORD_META_MAPPING:
+            if not overwrite:
+                raise ValueError(f"UCD metadata mapping {k} already exists.")
+
         CUSTOM_UCD_COORD_META_MAPPING[k] = v
 
     yield

