diff --git a/astropy/visualization/wcsaxes/wcsapi.py b/astropy/visualization/wcsaxes/wcsapi.py
--- a/astropy/visualization/wcsaxes/wcsapi.py
+++ b/astropy/visualization/wcsaxes/wcsapi.py
@@ -118,6 +118,8 @@
         yield
     finally:
         for k in added_keys:
+            del CUSTOM_UCD_COORD_META_MAPPING[k]
+        CUSTOM_UCD_COORD_META_MAPPING.update(overwritten)
 
 
 def transform_coord_meta_from_wcs(wcs, frame_class, slices=None):
