diff --git a/sklearn/metrics/cluster/_supervised.py b/sklearn/metrics/cluster/_supervised.py
--- a/sklearn/metrics/cluster/_supervised.py
+++ b/sklearn/metrics/cluster/_supervised.py
@@ -14,6 +14,7 @@
 import numpy as np
 from scipy import sparse as sp
 
+from ...utils import deprecated
 from ...utils._array_api import _max_precision_float_dtype, get_namespace_and_device
 from ...utils._param_validation import Hidden, Interval, StrOptions, validate_params
 from ...utils.multiclass import type_of_target
@@ -1272,13 +1273,7 @@ def fowlkes_mallows_score(labels_true, labels_pred, *, sparse="deprecated"):
     return float(np.sqrt(tk / pk) * np.sqrt(tk / qk)) if tk != 0.0 else 0.0
 
 
-@validate_params(
-    {
-        "labels": ["array-like"],
-    },
-    prefer_skip_nested_validation=True,
-)
-def entropy(labels):
+def _entropy(labels):
     """Calculate the entropy for a labeling.
 
     Parameters