diff --git a/astropy/wcs/wcsapi/wrappers/sliced_wcs.py b/astropy/wcs/wcsapi/wrappers/sliced_wcs.py
--- a/astropy/wcs/wcsapi/wrappers/sliced_wcs.py
+++ b/astropy/wcs/wcsapi/wrappers/sliced_wcs.py
@@ -268,7 +268,7 @@ def world_to_pixel_values(self, *world_arrays):
                 world_arrays_new.append(sliced_out_world_coords[iworld])
 
         world_arrays_new = np.broadcast_arrays(*world_arrays_new)
-        pixel_arrays = list(self._wcs.world_to_pixel_values(*world_arrays_new))
+        pixel_arrays = self._wcs.world_to_pixel_values(*world_arrays_new)
 
         for ipixel in range(self._wcs.pixel_n_dim):
             if (
 