diff --git a/xarray/core/datatree.py b/xarray/core/datatree.py
--- a/xarray/core/datatree.py
+++ b/xarray/core/datatree.py
@@ -1237,7 +1237,7 @@ def from_dict(
         -------
         DataTree
 
-        See also
+        See Also
         --------
         Dataset
 
@@ -1428,7 +1428,7 @@ def to_dict(self, relative: bool = False) -> dict[str, Dataset]:
         -------
         dict[str, Dataset]
 
-        See also
+        See Also
         --------
         DataTree.subtree_with_keys
         """
@@ -1805,7 +1805,7 @@ def map_over_datasets(
         subtrees : DataTree, tuple of DataTrees
             One or more subtrees containing results from applying ``func`` to the data at each node.
 
-        See also
+        See Also
         --------
         map_over_datasets
         """
@@ -2052,8 +2052,8 @@ def to_netcdf(
             * ``dask.delayed.Delayed`` if compute is False
             * ``None`` otherwise
 
-        Note
-        ----
+        Notes
+        -----
             Due to file format specifications the on-disk root group name
             is always ``"/"`` overriding any given ``DataTree`` root node name.
         """
@@ -2147,8 +2147,8 @@ def to_zarr(
         kwargs :
             Additional keyword arguments to be passed to ``xarray.Dataset.to_zarr``
 
-        Note
-        ----
+        Notes
+        -----
             Due to file format specifications the on-disk root group name
             is always ``"/"`` overriding any given ``DataTree`` root node name.
         """
