diff --git a/xarray/core/datatree.py b/xarray/core/datatree.py
--- a/xarray/core/datatree.py
+++ b/xarray/core/datatree.py
@@ -2223,14 +2223,21 @@ def reduce(
         result = {}
         for path, node in self.subtree_with_keys:
             reduce_dims = [d for d in node._node_dims if d in dims]
-            node_result = node.dataset.reduce(
-                func,
-                reduce_dims,
-                keep_attrs=keep_attrs,
-                keepdims=keepdims,
-                numeric_only=numeric_only,
-                **kwargs,
-            )
+
+            # Prefer Dataset.func(...) over Dataset.reduce(func, ...),
+            # because Dataset.func(...) may do further func-specific processing:
+            f = getattr(node.dataset, func.__name__, None)
+            if f:
+                node_result = f(reduce_dims, keep_attrs=keep_attrs, **kwargs)
+            else:
+                node_result = node.dataset.reduce(
+                    func,
+                    reduce_dims,
+                    keep_attrs=keep_attrs,
+                    keepdims=keepdims,
+                    numeric_only=numeric_only,
+                    **kwargs,
+                )
             result[path] = node_result
         return type(self).from_dict(result, name=self.name)
 
