diff --git a/astropy/units/core.py b/astropy/units/core.py
--- a/astropy/units/core.py
+++ b/astropy/units/core.py
@@ -1839,12 +1839,13 @@ def _inject(self, namespace: MutableMapping[str, object] | None = None) -> None:
 
         # Loop through all of the names first, to ensure all of them
         # are new, then add them all as a single "transaction" below.
-        for name in (unicodedata.normalize("NFKC", name) for name in self._names):
+        for name in self._names:
+            nfkc_name = unicodedata.normalize("NFKC", name)
             if name in namespace and self != namespace[name]:
                 raise ValueError(
                     f"Object with NFKC normalized name {name!r} already exists in "
                     f"given namespace ({namespace[name]!r})."
                 )
 
         for name in self._names:
             namespace[name] = self