diff --git a/xarray/util/generate_aggregations.py b/xarray/util/generate_aggregations.py
--- a/xarray/util/generate_aggregations.py
+++ b/xarray/util/generate_aggregations.py
@@ -568,7 +568,7 @@ def generate_code(self, method, has_keep_attrs):
 
         if method.aggregation_type == "scan" and self.datastructure.name == "Dataset":
             # Scans retain dimensions, datasets drops them somehow:
-            out_finalized = "out.assign_coords(self._obj.coords)"
+            out_finalized = "out.assign_coords(self.coords)"
         else:
             out_finalized = "out"
 
