diff --git a/xarray/util/generate_aggregations.py b/xarray/util/generate_aggregations.py
--- a/xarray/util/generate_aggregations.py
+++ b/xarray/util/generate_aggregations.py
@@ -4,8 +4,10 @@
 
 Usage:
     python xarray/util/generate_aggregations.py
-    pytest --doctest-modules xarray/{core,namedarray}/_aggregations.py --accept || true
-    pytest --doctest-modules xarray/{core,namedarray}/_aggregations.py
+    pytest --doctest-modules xarray/core/_aggregations.py --accept
+    pytest --doctest-modules xarray/core/_aggregations.py
+    pytest --doctest-modules xarray/namedarray/_aggregations.py --accept
+    pytest --doctest-modules xarray/namedarray/_aggregations.py
 
 This requires [pytest-accept](https://github.com/max-sixty/pytest-accept).
 The second run of pytest is deliberate, since the first will return an error
@@ -15,7 +17,7 @@
 
 import textwrap
 from dataclasses import dataclass, field
-from typing import NamedTuple
+from typing import Literal, NamedTuple
 
 MODULE_PREAMBLE = '''\
 """Mixin classes with reduction operations."""
@@ -24,7 +26,7 @@
 
 from __future__ import annotations
 
-from collections.abc import Callable, Sequence
+from collections.abc import Callable, Mapping, Sequence
 from typing import TYPE_CHECKING, Any
 
 from xarray.core import duck_array_ops
@@ -33,6 +35,7 @@
 from xarray.core.utils import contains_only_chunked_or_numpy, module_available
 
 if TYPE_CHECKING:
+    from xarray.core.coordinates import DatasetCoordinates
     from xarray.core.dataarray import DataArray
     from xarray.core.dataset import Dataset
 
@@ -69,6 +72,34 @@ def reduce(
     ) -> Self:
         raise NotImplementedError()"""
 
+DATASET_PREAMBLE = """
+
+class {obj}{cls}Aggregations:
+    __slots__ = ()
+
+    def reduce(
+        self,
+        func: Callable[..., Any],
+        dim: Dims = None,
+        *,
+        axis: int | Sequence[int] | None = None,
+        keep_attrs: bool | None = None,
+        keepdims: bool = False,
+        **kwargs: Any,
+    ) -> Self:
+        raise NotImplementedError()
+
+    @property
+    def coords(self) -> DatasetCoordinates:
+        raise NotImplementedError()
+
+    def assign_coords(
+        self,
+        coords: Mapping | None = None,
+        **coords_kwargs: Any,
+    ) -> Self:
+        raise NotImplementedError()"""
+
 NAMED_ARRAY_AGGREGATIONS_PREAMBLE = """
 
 class {obj}{cls}Aggregations:
@@ -107,6 +138,17 @@ def _flox_reduce(
         self,
         dim: Dims,
         **kwargs: Any,
+    ) -> {obj}:
+        raise NotImplementedError()
+
+    def _flox_scan(
+        self,
+        dim: Dims,
+        *,
+        func: str,
+        skipna: bool | None = None,
+        keep_attrs: bool | None = None,
+        **kwargs: Any,
     ) -> {obj}:
         raise NotImplementedError()"""
 
@@ -131,6 +173,17 @@ def _flox_reduce(
         self,
         dim: Dims,
         **kwargs: Any,
+    ) -> {obj}:
+        raise NotImplementedError()
+
+    def _flox_scan(
+        self,
+        dim: Dims,
+        *,
+        func: str,
+        skipna: bool | None = None,
+        keep_attrs: bool | None = None,
+        **kwargs: Any,
     ) -> {obj}:
         raise NotImplementedError()"""
 
@@ -284,6 +337,7 @@ def __init__(
         see_also_methods=(),
         min_flox_version=None,
         additional_notes="",
+        aggregation_type: Literal["reduce", "scan"] = "reduce",
     ):
         self.name = name
         self.extra_kwargs = extra_kwargs
@@ -292,6 +346,7 @@ def __init__(
         self.see_also_methods = see_also_methods
         self.min_flox_version = min_flox_version
         self.additional_notes = additional_notes
+        self.aggregation_type = aggregation_type
         if bool_reduce:
             self.array_method = f"array_{name}"
             self.np_example_array = (
@@ -444,7 +499,7 @@ def generate_code(self, method, has_keep_attrs):
 
         # median isn't enabled yet, because it would break if a single group was present in multiple
         # chunks. The non-flox code path will just rechunk every group to a single chunk and execute the median
-        method_is_not_flox_supported = method.name in ("median", "cumsum", "cumprod")
+        method_is_not_flox_supported = method.name in ("median", "cumprod")
         if method_is_not_flox_supported:
             indent = 12
         else:
@@ -455,14 +510,21 @@ def generate_code(self, method, has_keep_attrs):
         else:
             extra_kwargs = ""
 
+        if method.aggregation_type == "scan":
+            # Scans retain dimensions.
+            out_finalized = "out.assign_coords(self._obj.coords)"
+        else:
+            out_finalized = "out"
+
         if method_is_not_flox_supported:
             return f"""\
-        return self.reduce(
+        out = self.reduce(
             duck_array_ops.{method.array_method},
             dim=dim,{extra_kwargs}
             keep_attrs=keep_attrs,
             **kwargs,
-        )"""
+        )
+        return {out_finalized}"""
 
         min_version_check = f"""
             and module_available("flox", minversion="{method.min_flox_version}")"""
@@ -476,20 +538,21 @@ def generate_code(self, method, has_keep_attrs):
             + f"""
             and contains_only_chunked_or_numpy(self._obj)
         ):
-            return self._flox_reduce(
+            return self._flox_{method.aggregation_type}(
                 func="{method.name}",
                 dim=dim,{extra_kwargs}
                 # fill_value=fill_value,
                 keep_attrs=keep_attrs,
                 **kwargs,
             )
         else:
-            return self.reduce(
+            out = self.reduce(
                 duck_array_ops.{method.array_method},
                 dim=dim,{extra_kwargs}
                 keep_attrs=keep_attrs,
                 **kwargs,
-            )"""
+            )
+            return {out_finalized}"""
         )
 
 
@@ -507,12 +570,20 @@ def generate_code(self, method, has_keep_attrs):
         keep_attrs = (
             "\n" + 12 * " " + "keep_attrs=keep_attrs," if has_keep_attrs else ""
         )
+
+        if method.aggregation_type == "scan" and self.datastructure.name == "Dataset":
+            # Scans retain dimensions, datasets drops them somehow:
+            out_finalized = "out.assign_coords(self.coords)"
+        else:
+            out_finalized = "out"
+
         return f"""\
-        return self.reduce(
+        out = self.reduce(
             duck_array_ops.{method.array_method},
             dim=dim,{extra_kwargs}{keep_attrs}
             **kwargs,
-        )"""
+        )
+        return {out_finalized}"""
 
 
 AGGREGATION_METHODS = (
@@ -530,20 +601,23 @@ def generate_code(self, method, has_keep_attrs):
     Method(
         "median", extra_kwargs=(skipna,), numeric_only=True, min_flox_version="0.9.2"
     ),
-    # Cumulatives:
+    # Scans:
     Method(
         "cumsum",
         extra_kwargs=(skipna,),
         numeric_only=True,
         see_also_methods=("cumulative",),
         additional_notes=_CUM_NOTES,
+        min_flox_version="0.10.5",
+        aggregation_type="scan",
     ),
     Method(
         "cumprod",
         extra_kwargs=(skipna,),
         numeric_only=True,
         see_also_methods=("cumulative",),
         additional_notes=_CUM_NOTES,
+        aggregation_type="scan",
     ),
 )
 
@@ -611,7 +685,7 @@ def generate_code(self, method, has_keep_attrs):
     docref="agg",
     docref_description="reduction or aggregation operations",
     example_call_preamble="",
-    definition_preamble=AGGREGATIONS_PREAMBLE,
+    definition_preamble=DATASET_PREAMBLE,
 )
 DATAARRAY_GENERATOR = GenericAggregationGenerator(
     cls="",
@@ -700,8 +774,11 @@ def write_methods(filepath, generators, preamble):
     from pathlib import Path
 
     p = Path(os.getcwd())
+
+    filepath = p.parent / "xarray" / "xarray" / "core" / "_aggregations.py"
+    # filepath = p.parent / "core" / "_aggregations.py"  # Run from script location
     write_methods(
-        filepath=p.parent / "xarray" / "xarray" / "core" / "_aggregations.py",
+        filepath=filepath,
         generators=[
             DATATREE_GENERATOR,
             DATASET_GENERATOR,
@@ -713,9 +790,12 @@ def write_methods(filepath, generators, preamble):
         ],
         preamble=MODULE_PREAMBLE,
     )
+
+    # NamedArray:
+    filepath = p.parent / "xarray" / "xarray" / "namedarray" / "_aggregations.py"
+    # filepath = p.parent / "namedarray" / "_aggregations.py"  # Run from script location
     write_methods(
-        filepath=p.parent / "xarray" / "xarray" / "namedarray" / "_aggregations.py",
+        filepath=filepath,
         generators=[NAMED_ARRAY_GENERATOR],
         preamble=NAMED_ARRAY_MODULE_PREAMBLE,
     )
-    # filepath = p.parent / "core" / "_aggregations.py"  # Run from script location



