diff --git a/astropy/visualization/interval.py b/astropy/visualization/interval.py
--- a/astropy/visualization/interval.py
+++ b/astropy/visualization/interval.py
@@ -163,11 +163,7 @@
     """
 
     def get_limits(self, values):
-        # Make sure values is a Numpy array
-        values = np.asarray(values).ravel()
-
-        # Filter out invalid values (inf, nan)
-        values = values[np.isfinite(values)]
+        values = 
 
         return np.min(values), np.max(values)
 

