diff --git a/astropy/visualization/interval.py b/astropy/visualization/interval.py
--- a/astropy/visualization/interval.py
+++ b/astropy/visualization/interval.py
@@ -163,7 +163,7 @@
     """
 
     def get_limits(self, values):
-        values = 
+        values = self._process_values(values)
 
         return np.min(values), np.max(values)
 
@@ -197,8 +197,7 @@
         self.n_samples = n_samples
 
     def get_limits(self, values):
-        # Make sure values is a Numpy array
-        values = np.asarray(values).ravel()
+        values =
 
         # If needed, limit the number of samples. We sample with replacement
         # since this is much faster.


