diff --git a/astropy/visualization/interval.py b/astropy/visualization/interval.py
--- a/astropy/visualization/interval.py
+++ b/astropy/visualization/interval.py
@@ -9,8 +9,6 @@
 
 import numpy as np
 
-import astropy.units as u
-
 from .transform import BaseTransform
 
 __all__ = [
@@ -110,14 +108,6 @@ def __call__(self, values, clip=True, out=None):
         result : ndarray
             The transformed values.
         """
-        # Drop units if present
-        if isinstance(values, u.Quantity):
-            values = values.value
-        if isinstance(values, np.ma.MaskedArray) and isinstance(
-            values.data, u.Quantity
-        ):
-            values = np.ma.MaskedArray(values.data.value, values.mask)
-
         vmin, vmax = self.get_limits(values)
 
         if out is None: