diff --git a/sympy/physics/units/dimensions.py b/sympy/physics/units/dimensions.py
--- a/sympy/physics/units/dimensions.py
+++ b/sympy/physics/units/dimensions.py
@@ -100,6 +100,8 @@ def get_quantity_dimension(self, unit):
             return Dimension(1)
 
     def get_quantity_scale_factor(self, unit):
+        if isinstance(unit, Mul):
+            return
         if unit in self._quantity_scale_factors:
             return self._quantity_scale_factors[unit]
         if unit in self._quantity_scale_factors_global: