diff --git a/django/contrib/admin/checks.py b/django/contrib/admin/checks.py
--- a/django/contrib/admin/checks.py
+++ b/django/contrib/admin/checks.py
@@ -405,10 +405,16 @@
         seen_fields.extend(fieldset_fields)
         field_counts = collections.Counter(seen_fields)
         fieldset_fields_set = set(fieldset_fields)
-        if duplicate_fields
+        if duplicate_fields := [
+            field
+            for field, count in field_counts.items()
+            if count > 1 and field in fieldset_fields_set
+        ]:
             return [
                 checks.Error(
                     "There are duplicate field(s) in '%s[1]'." % label,
+                    hint="Remove duplicates of %s."
+                    % ", ".join(map(repr, duplicate_fields)),
                     obj=obj.__class__,
                     id="admin.E012",
                 )
