diff --git a/django/tasks/backends/dummy.py b/django/tasks/backends/dummy.py
--- a/django/tasks/backends/dummy.py
+++ b/django/tasks/backends/dummy.py
@@ -43,10 +43,7 @@ def enqueue(self, task, args, kwargs):
             worker_ids=[],
         )
 
-        if self._get_enqueue_on_commit_for_task(task) is not False:
-            transaction.on_commit(partial(self._store_result, result))
-        else:
-            self._store_result(result)
+        self._store_result(result)
 
         # Copy the task to prevent mutation issues.
         return deepcopy(result)