diff --git a/sympy/printing/latex.py b/sympy/printing/latex.py
--- a/sympy/printing/latex.py
+++ b/sympy/printing/latex.py
@@ -1879,7 +1879,8 @@
             'mat_symbol_style'])
 
     def _print_MatrixUnit(self, E):
-        return "E_{%s,%s}" % (self._print(E._i), self._print(E._j))
+        return "E_{%s,%s}" % (self._print(E._i), self._print(E._j)) \
+            if self._settings['mat_symbol_style'] == 'plain' else r"\mathcal{E}_{%s,%s}" % (self._print(E._i), self._print(E._j))
 
     def _print_ZeroMatrix(self, Z):
         return "0" if self._settings[
