diff --git a/sympy/combinatorics/fp_groups.py b/sympy/combinatorics/fp_groups.py
--- a/sympy/combinatorics/fp_groups.py
+++ b/sympy/combinatorics/fp_groups.py
@@ -16,6 +16,7 @@
 from sympy.utilities import public
 from sympy.utilities.magic import pollute
 
+from collections import defaultdict
 from itertools import product
 
 
@@ -635,6 +636,19 @@
                                 break
                         if not folded:
                             break
+
+            def _check_membership(graph, base, word):
+                current = base
+                for label in word.letter_form:
+                    targets = graph[current].get(label)
+                    if not targets:
+                        return False
+                    current = next(iter(targets))
+                return current == base
+
+            graph = defaultdict(lambda: defaultdict(set))
+            base_vertex = 0
+            next_vertex_id = 1
 
             if self._min_words is None:
                 # make _min_words - a list of subwords such that
