diff --git a/sympy/assumptions/refine.py b/sympy/assumptions/refine.py
--- a/sympy/assumptions/refine.py
+++ b/sympy/assumptions/refine.py
@@ -403,6 +403,33 @@ def refine_matrixelement(expr, assumptions):
             return expr
         return MatrixElement(matrix, j, i)
 
+
+def refine_sin_cos(expr, assumptions):
+    from sympy.functions.elementary.trigonometric import sin, cos
+    arg = expr.args[0]
+    k_terms = []
+    rem_terms = []
+    terms = arg.args if arg.is_Add else (arg,)
+    for term in terms:
+        pi_coeff = term.coeff(S.Pi)
+        if pi_coeff:
+            coeff = 2 * pi_coeff
+            if ask(Q.integer(coeff), assumptions):
+                k_terms.append(coeff)
+            else:
+                rem_terms.append(term)
+        else:
+            rem_terms.append(term)
+    k = Add(*k_terms, evaluate=False)
+    rem = Add(*rem_terms, evaluate=False)
+    if isinstance(expr, sin):
+        k = Add(k, -1, evaluate=False)
+    if ask(Q.odd(k), assumptions):
+        return ((-1)**((k + 1) / 2)) * sin(rem)
+    elif ask(Q.even(k), assumptions):
+        return ((-1)**(k / 2)) * cos(rem)
+    return expr
+
 handlers_dict: dict[str, Callable[[Basic, Boolean | bool], Expr]] = {
     'Abs': refine_abs,
     'Pow': refine_Pow,
@@ -411,5 +438,7 @@ def refine_matrixelement(expr, assumptions):
     'im': refine_im,
     'arg': refine_arg,
     'sign': refine_sign,
-    'MatrixElement': refine_matrixelement
+    'MatrixElement': refine_matrixelement,
+    'cos': refine_sin_cos,
+    'sin': refine_sin_cos,
 }



