diff --git a/sympy/concrete/products.py b/sympy/concrete/products.py
--- a/sympy/concrete/products.py
+++ b/sympy/concrete/products.py
@@ -285,11 +285,6 @@ def doit(self, **hints):
         else:
             return powsimp(f)
 
-    def _eval_adjoint(self):
-        if self.is_commutative:
-            return self.func(self.function.adjoint(), *self.limits)
-        return None
-
     def _eval_conjugate(self):
         return self.func(self.function.conjugate(), *self.limits)
 