diff --git a/sympy/series/sequences.py b/sympy/series/sequences.py
--- a/sympy/series/sequences.py
+++ b/sympy/series/sequences.py
@@ -644,6 +644,19 @@
         if shift1 is None or shift2 is None:
             return None
 
+        v1 = self.variables[0]
+        v2 = other.variables[0]
+        if v1 != v2:
+            per2 = [p.subs(v2, v1) for p in per2]
+
+        new_per = []
+        for x in range(per_length):
+            idx1 = (shift1 + dir1*x) % lper1
+            idx2 = (shift2 + dir2*x) % lper2
+            new_per.append(op(per1[idx1], per2[idx2]))
+
+        return SeqPer(new_per, (v1, start, stop))
+
     def coeff_mul(self, coeff):
         """See docstring of SeqBase.coeff_mul"""
         coeff = sympify(coeff)
