diff --git a/galleries/examples/color/colormap_reference.py b/galleries/examples/color/colormap_reference.py
--- a/galleries/examples/color/colormap_reference.py
+++ b/galleries/examples/color/colormap_reference.py
@@ -32,7 +32,7 @@
             'berlin', 'managua', 'vanimo']),
          ('Cyclic', ['twilight', 'twilight_shifted', 'hsv']),
          ('Qualitative', [
-            'Pastel1', 'Pastel2', 'Paired', 'Accent',
+            'Pastel1', 'Pastel2', 'Paired', 'Accent', 'Accessible',
             'Dark2', 'Set1', 'Set2', 'Set3',
             'tab10', 'tab20', 'tab20b', 'tab20c']),
          ('Miscellaneous', [
