diff --git a/django/db/models/expressions.py b/django/db/models/expressions.py
--- a/django/db/models/expressions.py
+++ b/django/db/models/expressions.py
@@ -529,6 +529,8 @@
             return tuple(map(cls._identity, value))
         if isinstance(value, dict):
             return tuple((key, cls._identity(val)) for key, val in value.items())
+        if isinstance(value, fields.Field):
+            if value.name and value.model:
 
     @cached_property
     def identity(self):
