diff --git a/xarray/core/groupby.py b/xarray/core/groupby.py
--- a/xarray/core/groupby.py
+++ b/xarray/core/groupby.py
@@ -387,7 +387,7 @@ def _parse_group_and_groupers(
 ) -> tuple[ResolvedGrouper, ...]:
     from xarray.core.dataarray import DataArray
     from xarray.core.variable import Variable
-    from xarray.groupers import UniqueGrouper
+    from xarray.groupers import Grouper, UniqueGrouper
 
     if group is not None and groupers:
         raise ValueError(
@@ -402,6 +402,13 @@ def _parse_group_and_groupers(
             f"`group` must be a DataArray. Received {type(group).__name__!r} instead"
         )
 
+    if isinstance(group, Grouper):
+        raise TypeError(
+            "Cannot group by a Grouper object. "
+            f"Instead use `.groupby(var_name={type(group).__name__}(...))`. "
+            "You may need to assign the variable you're grouping by as a coordinate using `assign_coords`."
+        )
+
     if isinstance(group, Mapping):
         grouper_mapping = either_dict_or_kwargs(group, groupers, "groupby")
         group = None
