diff --git a/lib/matplotlib/figure.py b/lib/matplotlib/figure.py
--- a/lib/matplotlib/figure.py
+++ b/lib/matplotlib/figure.py
@@ -3003,12 +3003,10 @@ def _set_base_canvas(self):
         This is used upon initialization of the Figure, but also
         to reset the canvas when decoupling from pyplot.
         """
-        # check if we have changed the DPI due to hi-dpi screens
-        orig_dpi = getattr(self, '_original_dpi', self._dpi)
         FigureCanvasBase(self)  # Set self.canvas as a side-effect
-        # put it back to what it was
-        if orig_dpi != self._dpi:
-            self.dpi = orig_dpi
+        # undo any high-dpi scaling
+        if self._dpi != self._original_dpi:
+            self.dpi = self._original_dpi
 
     def set_canvas(self, canvas):
         """
