diff --git a/lib/matplotlib/spines.py b/lib/matplotlib/spines.py
--- a/lib/matplotlib/spines.py
+++ b/lib/matplotlib/spines.py
@@ -232,12 +232,7 @@ def _clear(self):
         """
         self._position = None  # clear position
 
-    def _adjust_location(self):
-        """Automatically set spine bounds to the view interval."""
-
-        if self.spine_type == 'circle':
-            return
-
+    def _get_bounds_or_viewLim(self):
         if self._bounds is not None:
             low, high = self._bounds
         elif self.spine_type in ('left', 'right'):
@@ -246,6 +241,15 @@ def _adjust_location(self):
             low, high = self.axes.viewLim.intervalx
         else:
             raise ValueError(f'unknown spine spine_type: {self.spine_type}')
+        return low, high
+
+    def _adjust_location(self):
+        """Automatically set spine bounds to the view interval."""
+
+        if self.spine_type == 'circle':
+            return
+
+        low, high = self._get_bounds_or_viewLim()
 
         if self._patch_type == 'arc':
             if self.spine_type in ('bottom', 'top'):
@@ -424,7 +428,7 @@ def set_bounds(self, low=None, high=None):
                 'set_bounds() method incompatible with circular spines')
         if high is None and np.iterable(low):
             low, high = low
-        old_low, old_high = self.get_bounds() or (None, None)
+        old_low, old_high = self._get_bounds_or_viewLim()
         if low is None:
             low = old_low
         if high is None: