diff --git a/sympy/physics/secondquant.py b/sympy/physics/secondquant.py
--- a/sympy/physics/secondquant.py
+++ b/sympy/physics/secondquant.py
@@ -19,6 +19,7 @@
 from sympy.core.sorting import default_sort_key
 from sympy.core.symbol import Dummy, Symbol
 from sympy.core.sympify import sympify
+from sympy.functions.elementary.complexes import conjugate
 from sympy.functions.elementary.miscellaneous import sqrt
 from sympy.functions.special.tensor_functions import KroneckerDelta
 from sympy.matrices.dense import zeros
@@ -144,8 +145,8 @@ def eval(cls, arg):
                 return Add(*tuple(map(Dagger, arg.args)))
             if arg.is_Mul:
                 return Mul(*tuple(map(Dagger, reversed(arg.args))))
-            if arg.is_Number:
-                return arg
+            if arg.is_Number or arg.is_scalar:
+                return conjugate(arg)
             if arg.is_Pow:
                 return Pow(Dagger(arg.args[0]), arg.args[1])
             if arg == I: