diff --git a/astropy/io/fits/fitsrec.py b/astropy/io/fits/fitsrec.py
--- a/astropy/io/fits/fitsrec.py
+++ b/astropy/io/fits/fitsrec.py
@@ -1382,7 +1382,7 @@
     out_dtype = np.dtype((f"S{inarray.dtype.itemsize // 4}", inarray.dtype.shape))
     if out is not None:
         out = out.view(out_dtype)
-    if
+    if inarray.size == 0:
 
     op_dtypes = [inarray.dtype, out_dtype]
     op_flags = [["readonly"], ["writeonly", "allocate"]]
