diff --git a/xarray/core/groupby.py b/xarray/core/groupby.py
--- a/xarray/core/groupby.py
+++ b/xarray/core/groupby.py
@@ -1402,7 +1402,11 @@ def _first_or_last(
             and contains_only_chunked_or_numpy(self._obj)
         ):
             result = self._flox_reduce(
-                dim=None, func=op, skipna=skipna, keep_attrs=keep_attrs
+                dim=None,
+                func=op,
+                skipna=skipna,
+                keep_attrs=keep_attrs,
+                fill_value=flox.xrdtypes.NA,
             )
         else:
             result = self.reduce(