diff --git a/sphinx/ext/intersphinx/_load.py b/sphinx/ext/intersphinx/_load.py
--- a/sphinx/ext/intersphinx/_load.py
+++ b/sphinx/ext/intersphinx/_load.py
@@ -310,7 +310,7 @@
 
 
 def _fetch_inventory(
-    *, target_uri: InventoryURI, inv_location: str, config: Config, srcdir: Path
+    *, target_uri: InventoryURI, inv_location: str, config: _InvConfig, srcdir: Path
 ) -> Inventory:
     """Fetch, parse and return an intersphinx inventory file."""
     # both *target_uri* (base URI of the links to generate)
@@ -336,7 +336,7 @@
 
 
 def _fetch_inventory_url(
-    *, target_uri: InventoryURI, inv_location: str, config: Config
+    *, target_uri: InventoryURI, inv_location: str, config: _InvConfig
 ) -> tuple[bytes, str]:
     try:
         with requests.get(
