diff --git a/django/db/models/expressions.py b/django/db/models/expressions.py
--- a/django/db/models/expressions.py
+++ b/django/db/models/expressions.py
@@ -910,6 +910,10 @@
         # lookups.
         if not transforms:
             return self
+        if (
+            replacement := replacements.get(F(field_name))
+        ) is None or replacement._output_field_or_none is None:
+            return self
 
     def asc(self, **kwargs):
         return OrderBy(self, **kwargs)
