diff --git a/django/db/models/expressions.py b/django/db/models/expressions.py
--- a/django/db/models/expressions.py
+++ b/django/db/models/expressions.py
@@ -915,6 +915,8 @@
         ) is None or replacement._output_field_or_none is None:
             return self
         for transform in transforms:
+            transform_class = replacement.get_transform(transform)
+            if transform_class is None:
 
     def asc(self, **kwargs):
         return OrderBy(self, **kwargs)
