diff --git a/astropy/io/misc/ecsv.py b/astropy/io/misc/ecsv.py
--- a/astropy/io/misc/ecsv.py
+++ b/astropy/io/misc/ecsv.py
@@ -394,7 +394,6 @@ def get_data_kwargs(
         header: ECSVHeader,
         null_values: list[str],
     ) -> dict[str, Any]:
-        kw = {}
         fill_values = get_null_values_per_column(
             header.cols, header.table_meta, null_values
         )
@@ -408,14 +407,17 @@ def get_data_kwargs(
                 for nan in ("nan", "NaN"):
                     null_values[col_name].append(nan)
 
-        kw["na_values"] = null_values
-        kw["keep_default_na"] = False
-        kw["comment"] = "#"
-        # Would prefer `kw["skiprows"] = header.n_header` here, but there is a bug in
-        # pandas.read_csv where skiprows does not behave expectedly.
-        # https://github.com/pandas-dev/pandas/issues/62739
-        kw["dtype"] = converters
-        return kw
+        kw = {
+            "na_values": null_values,
+            "keep_default_na": False,
+            "comment": "#",
+            "dtype": converters,
+        }
+        # Would prefer setting `"skiprows": header.n_header` above (as in the original
+        # implementation prior to #18756) instead of "comment": "#". However there is a
+        # bug in pandas.read_csv where skiprows does not work when the line includes a
+        # quote character, see https://github.com/pandas-dev/pandas/issues/62739.
+        return kw  # noqa: RET504
 
 
 def is_numpy_dtype(np_type: str) -> bool:
@@ -1230,7 +1232,7 @@ def read_ecsv(
     return table  # noqa: RET504
 
 
-def write_ecsv(tbl, output, **kwargs):
+def write_ecsv(tbl, output, engine="io.ascii", **kwargs):
     """Thin wrapper around the ``io.ascii`` ECSV writer to write ECSV files.
 
     Parameters
@@ -1239,14 +1241,16 @@ def write_ecsv(tbl, output, **kwargs):
         The table to write to ECSV format.
     output : str or os.PathLike or file-like object
         The output file path or file-like object to write the ECSV data to.
+    engine : str, optional
+        The engine to use for writing the CSV data. Default is "io.ascii", which uses
+        astropy to write the CSV data. Currently this is the only option.
     **kwargs : dict, optional
         Additional keyword arguments passed to the ECSV writer. These can include
         options like ``delimiter``, ``encoding``, and others supported by the
         `astropy.io.ascii.Ecsv` writer.
     """
-    # Calling writer like `t.write(out, format="ecsv", engine="pyarrow")` needs to drop
-    # the engine kwarg.
-    kwargs.pop("engine", None)
+    if engine != "io.ascii":
+        raise ValueError("'io.ascii' is the only available `engine` for writing")
     tbl.write(output, format="ascii.ecsv", **kwargs)
 
 
