diff --git a/astropy/io/misc/ecsv.py b/astropy/io/misc/ecsv.py
--- a/astropy/io/misc/ecsv.py
+++ b/astropy/io/misc/ecsv.py
@@ -365,6 +365,7 @@ def get_data_kwargs(
         # encoding. Converters are required, e.g. for a string column that looks like
         # floats. Would be nice to fix this, but in mean time use Python CSV reader.
         kw["fast_reader"] = False
+        kw["strip_column_names"] = False
         return kw
 
 
