diff --git a/sphinx/domains/javascript.py b/sphinx/domains/javascript.py
--- a/sphinx/domains/javascript.py
+++ b/sphinx/domains/javascript.py
@@ -137,8 +137,9 @@ def handle_signature(self, sig: str, signode: desc_signature) -> tuple[str, str]
                 _pseudo_parse_arglist(
                     signode,
                     arglist,
-                    multi_line_parameter_list,
-                    trailing_comma,
+                    multi_line_parameter_list=multi_line_parameter_list,
+                    trailing_comma=trailing_comma,
+                    env=self.env,
                 )
         return fullname, prefix
 
