diff --git a/sympy/solvers/ode/single.py b/sympy/solvers/ode/single.py
--- a/sympy/solvers/ode/single.py
+++ b/sympy/solvers/ode/single.py
@@ -2112,11 +2112,11 @@ class NthLinearConstantCoeffHomogeneous(SingleODESolver):
     >>> dsolve(f(x).diff(x, 5) + 10*f(x).diff(x) - 2*f(x), f(x),
     ... hint='nth_linear_constant_coeff_homogeneous')
     ... # doctest: +NORMALIZE_WHITESPACE
-    Eq(f(x), C5*exp(x*CRootOf(_x**5 + 10*_x - 2, 0))
-    + (C1*sin(x*im(CRootOf(_x**5 + 10*_x - 2, 1)))
-    + C2*cos(x*im(CRootOf(_x**5 + 10*_x - 2, 1))))*exp(x*re(CRootOf(_x**5 + 10*_x - 2, 1)))
-    + (C3*sin(x*im(CRootOf(_x**5 + 10*_x - 2, 3)))
-    + C4*cos(x*im(CRootOf(_x**5 + 10*_x - 2, 3))))*exp(x*re(CRootOf(_x**5 + 10*_x - 2, 3))))
+    Eq(f(x), C5*exp(x*CRootOf(x**5 + 10*x - 2, 0))
+    + (C1*sin(x*im(CRootOf(x**5 + 10*x - 2, 1)))
+    + C2*cos(x*im(CRootOf(x**5 + 10*x - 2, 1))))*exp(x*re(CRootOf(x**5 + 10*x - 2, 1)))
+    + (C3*sin(x*im(CRootOf(x**5 + 10*x - 2, 3)))
+    + C4*cos(x*im(CRootOf(x**5 + 10*x - 2, 3))))*exp(x*re(CRootOf(x**5 + 10*x - 2, 3))))
 
     Note that because this method does not involve integration, there is no
     ``nth_linear_constant_coeff_homogeneous_Integral`` hint.
