diff --git a/sphinx/ext/autodoc/type_comment.py b/sphinx/ext/autodoc/type_comment.py
--- a/sphinx/ext/autodoc/type_comment.py
+++ b/sphinx/ext/autodoc/type_comment.py
@@ -127,13 +127,8 @@
         return None
 
 
-def update_annotations_using_type_comments(
-    app: Sphinx, obj: Any, bound_method: bool
-) -> None:
+def _update_annotations_using_type_comments(obj: Any, bound_method: bool) -> None:
     """Update annotations info of *obj* using type_comments."""
-    if not app.config.autodoc_use_type_comments:
-        return
-
     try:
         type_sig = get_type_comment(obj, bound_method)
         if type_sig:
