diff --git a/sphinx/ext/autodoc/type_comment.py b/sphinx/ext/autodoc/type_comment.py
--- a/sphinx/ext/autodoc/type_comment.py
+++ b/sphinx/ext/autodoc/type_comment.py
@@ -6,7 +6,6 @@
 from inspect import Parameter, Signature, getsource
 from typing import TYPE_CHECKING, cast
 
-import sphinx
 from sphinx.locale import __
 from sphinx.pycode.ast import unparse as ast_unparse
 from sphinx.util import inspect, logging
@@ -15,8 +14,6 @@
     from collections.abc import Sequence
     from typing import Any
 
-    from sphinx.application import Sphinx
-    from sphinx.util.typing import ExtensionMetadata
 
 logger = logging.getLogger(__name__)
 
@@ -147,17 +144,3 @@
         )
     except NotImplementedError as exc:  # failed to ast.unparse()
         logger.warning(__('Failed to parse type_comment for %r: %s'), obj, exc)
-
-
-def setup(app: Sphinx) -> ExtensionMetadata:
-    app.add_config_value(
-        'autodoc_use_type_comments', True, 'env', types=frozenset({bool})
-    )
-    app.connect(
-        'autodoc-before-process-signature', update_annotations_using_type_comments
-    )
-
-    return {
-        'version': sphinx.__display_version__,
-        'parallel_read_safe': True,
-    }
