diff --git a/sympy/geometry/entity.py b/sympy/geometry/entity.py
--- a/sympy/geometry/entity.py
+++ b/sympy/geometry/entity.py
@@ -138,7 +138,7 @@ def _eval_subs(self, old, new):
             else:
                 old = Point(old)
                 new = Point(new)
-            return  self._subs(old, new)
+            return self.xreplace({old: new})
 
     def _repr_svg_(self):
         """SVG representation of a GeometryEntity suitable for IPython"""
