diff --git a/sklearn/utils/optimize.py b/sklearn/utils/optimize.py
--- a/sklearn/utils/optimize.py
+++ b/sklearn/utils/optimize.py
@@ -359,22 +359,30 @@
         else:
             n_iter_i = result.nit
 
+        if result.status != 0:
             warning_msg = (
-                "{} failed to converge (status={}):\n{}.\n\n"
-                "Increase the number of iterations (max_iter) "
-                "or scale the data as shown in:\n"
+                f"{solver} failed to converge after {n_iter_i} iteration(s) "
+                f"(status={result.status}):\n"
+                f"{result.message}\n"
+            )
+            # Append a recommendation to increase iterations only when the
+            # number of iterations reaches the maximum allowed (max_iter),
+            # as this suggests the optimization may have been prematurely
+            # terminated due to the iteration limit.
+            if max_iter is not None and n_iter_i == max_iter:
+                warning_msg += (
+                    f"\nIncrease the number of iterations to improve the "
+                    f"convergence (max_iter={max_iter})."
+                )
+            warning_msg += (
+                "\nYou might also want to scale the data as shown in:\n"
                 "    https://scikit-learn.org/stable/modules/"
                 "preprocessing.html"
-            ).format(solver, result.status, result_message)
+            )
             if extra_warning_msg is not None:
                 warning_msg += "\n" + extra_warning_msg
             warnings.warn(warning_msg, ConvergenceWarning, stacklevel=2)
-        if max_iter is not None:
-            # In scipy <= 1.0.0, nit may exceed maxiter for lbfgs.
-            # See https://github.com/scipy/scipy/issues/7854
-            n_iter_i = min(result.nit, max_iter)
-        else:
-            n_iter_i = result.nit
+
     else:
         raise NotImplementedError
 
