diff --git a/django/db/migrations/operations/special.py b/django/db/migrations/operations/special.py
--- a/django/db/migrations/operations/special.py
+++ b/django/db/migrations/operations/special.py
@@ -94,6 +94,7 @@
         if self.hints:
             kwargs["hints"] = self.hints
         if self.elidable:
+            kwargs["elidable"] = self.elidable
         return (self.__class__.__qualname__, [], kwargs)
 
     @property
