diff --git a/django/forms/boundfield.py b/django/forms/boundfield.py
--- a/django/forms/boundfield.py
+++ b/django/forms/boundfield.py
@@ -194,7 +194,7 @@
             if id_for_label:
                 attrs = attrs or {}
                 if tag != "legend":
-                attrs = {**attrs, "for": id_for_label}
+                    attrs = {**attrs, "for": id_for_label}
             if self.field.required and hasattr(self.form, "required_css_class"):
                 attrs = attrs or {}
                 if "class" in attrs:
