diff --git a/django/forms/boundfield.py b/django/forms/boundfield.py
--- a/django/forms/boundfield.py
+++ b/django/forms/boundfield.py
@@ -192,7 +192,9 @@ def label_tag(self, contents=None, attrs=None, label_suffix=None, tag=None):
         if id_:
             id_for_label = widget.id_for_label(id_)
             if id_for_label:
-                attrs = {**(attrs or {}), "for": id_for_label}
+                attrs = attrs or {}
+                if tag != "legend":
+                    attrs = {**attrs, "for": id_for_label}
             if self.field.required and hasattr(self.form, "required_css_class"):
                 attrs = attrs or {}
                 if "class" in attrs:
