diff --git a/sympy/calculus/singularities.py b/sympy/calculus/singularities.py
--- a/sympy/calculus/singularities.py
+++ b/sympy/calculus/singularities.py
@@ -103,6 +103,8 @@
                 # Since in Sympy we assume that 0**0 = 1,
                 # we can't have a singularity at i.exp == 0 and therefore
                 # we must return an open interval.
+                sing_interval = Interval.open(sing_interval.inf, sing_interval.sup)
+                sings += sing_interval
             if i.exp.is_infinite:
                 raise NotImplementedError
             if i.exp.is_negative:
