diff --git a/xarray/core/groupby.py b/xarray/core/groupby.py
--- a/xarray/core/groupby.py
+++ b/xarray/core/groupby.py
@@ -177,7 +177,9 @@ def _inverse_permutation_indices(positions, N: int | None = None) -> np.ndarray
             return None
         positions = [np.arange(sl.start, sl.stop, sl.step) for sl in positions]
 
-    newpositions = nputils.inverse_permutation(np.concatenate(positions), N)
+    newpositions = nputils.inverse_permutation(
+        np.concatenate(tuple(p for p in positions if p)), N
+    )
     return newpositions[newpositions != -1]
 
 
