diff --git a/django/db/models/fields/json.py b/django/db/models/fields/json.py
--- a/django/db/models/fields/json.py
+++ b/django/db/models/fields/json.py
@@ -12,6 +12,7 @@
     PostgresOperatorLookup,
     Transform,
 )
+from django.utils.deconstruct import deconstructible
 from django.utils.deprecation import RemovedInDjango70Warning, django_file_prefixes
 from django.utils.translation import gettext_lazy as _
 
@@ -150,6 +151,7 @@ def formfield(self, **kwargs):
         )
 
 
+@deconstructible(path="django.db.models.JSONNull")
 class JSONNull(expressions.Value):
     """Represent JSON `null` primitive."""
 