diff --git a/sklearn/metrics/_plot/roc_curve.py b/sklearn/metrics/_plot/roc_curve.py
--- a/sklearn/metrics/_plot/roc_curve.py
+++ b/sklearn/metrics/_plot/roc_curve.py
@@ -2,8 +2,6 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
 
-import warnings
-
 import numpy as np
 
 from ...utils import _safe_indexing
@@ -12,6 +10,7 @@
     _check_param_lengths,
     _convert_to_list_leaving_none,
     _deprecate_estimator_name,
+    _deprecate_y_pred_parameter,
     _despine,
     _validate_style_kwargs,
 )
@@ -576,24 +575,7 @@ def from_predictions(
         <...>
         >>> plt.show()
         """
-        # TODO(1.9): remove after the end of the deprecation period of `y_pred`
-        if y_score is not None and not (
-            isinstance(y_pred, str) and y_pred == "deprecated"
-        ):
-            raise ValueError(
-                "`y_pred` and `y_score` cannot be both specified. Please use `y_score`"
-                " only as `y_pred` is deprecated in 1.7 and will be removed in 1.9."
-            )
-        if not (isinstance(y_pred, str) and y_pred == "deprecated"):
-            warnings.warn(
-                (
-                    "y_pred is deprecated in 1.7 and will be removed in 1.9. "
-                    "Please use `y_score` instead."
-                ),
-                FutureWarning,
-            )
-            y_score = y_pred
-
+        y_score = _deprecate_y_pred_parameter(y_score, y_pred, "1.7")
         pos_label_validated, name = cls._validate_from_predictions_params(
             y_true, y_score, sample_weight=sample_weight, pos_label=pos_label, name=name
         )
