diff --git a/astropy/io/votable/tree.py b/astropy/io/votable/tree.py
--- a/astropy/io/votable/tree.py
+++ b/astropy/io/votable/tree.py
@@ -181,10 +181,10 @@ def lookup_by_attr(self, ref, before=None):
             if element is before:
                 if getattr(element, attr, None) == ref:
                     vo_raise(
+                        KeyError,
                         f"{element_name} references itself",
                         element._config,
                         element._pos,
-                        KeyError,
                     )
                 break
             if getattr(element, attr, None) == ref:
@@ -224,10 +224,10 @@ def lookup_by_id_or_name(self, ref, before=None):
             if element is before:
                 if ref in (element.ID, element.name):
                     vo_raise(
+                        KeyError,
                         f"{element_name} references itself",
                         element._config,
                         element._pos,
-                        KeyError,
                     )
                 break
             if ref in (element.ID, element.name):
@@ -2250,7 +2250,7 @@ def get_ref(self):
         for field in self._table._votable.iter_fields_and_params():
             if isinstance(field, Field) and field.ID == self.ref:
                 return field
-        vo_raise(f"No field named '{self.ref}'", self._config, self._pos, KeyError)
+        vo_raise(KeyError, f"No field named '{self.ref}'", self._config, self._pos)
 
 
 class ParamRef(SimpleElement, _UtypeProperty, _UcdProperty):
@@ -2315,7 +2315,7 @@ def get_ref(self):
         for param in self._table._votable.iter_fields_and_params():
             if isinstance(param, Param) and param.ID == self.ref:
                 return param
-        vo_raise(f"No params named '{self.ref}'", self._config, self._pos, KeyError)
+        vo_raise(KeyError, f"No params named '{self.ref}'", self._config, self._pos)
 
 
 class Group(
@@ -2620,10 +2620,10 @@ def format(self, format):
         format = format.lower()
         if format == "fits":
             vo_raise(
+                NotImplementedError,
                 "fits format can not be written out, only read.",
                 self._config,
                 self._pos,
-                NotImplementedError,
             )
         if format == "binary2":
             if not self._config.get("version_1_3_or_later"):
@@ -3134,10 +3134,10 @@ def _get_binary_data_stream(self, iterator, config):
             vo_prot = ("http", "https", "ftp", "file")
             if not href.startswith(vo_prot):
                 vo_raise(
+                    NotImplementedError,
                     f"The vo package only supports remote data through {vo_prot}",
                     self._config,
                     self._pos,
-                    NotImplementedError,
                 )
             fd = urllib.request.urlopen(href)
             if encoding is not None:
@@ -3147,10 +3147,10 @@ def _get_binary_data_stream(self, iterator, config):
                     fd = codecs.EncodedFile(fd, "base64")
                 else:
                     vo_raise(
+                        NotImplementedError,
                         f"Unknown encoding type '{encoding}'",
                         self._config,
                         self._pos,
-                        NotImplementedError,
                     )
             read = fd.read
 
@@ -3248,10 +3248,10 @@ def _parse_fits(self, iterator, extnum, config):
 
         if not href.startswith(("http", "ftp", "file")):
             vo_raise(
+                NotImplementedError,
                 "The vo package only supports remote data through http, ftp or file",
                 self._config,
                 self._pos,
-                NotImplementedError,
             )
 
         fd = urllib.request.urlopen(href)
@@ -3262,10 +3262,10 @@ def _parse_fits(self, iterator, extnum, config):
                 fd = codecs.EncodedFile(fd, "base64")
             else:
                 vo_raise(
+                    NotImplementedError,
                     f"Unknown encoding type '{encoding}'",
                     self._config,
                     self._pos,
-                    NotImplementedError,
                 )
         hdulist = fits.open(fd)
 
@@ -3302,10 +3302,10 @@ def _parse_parquet(self, iterator, config):
 
         if not href.startswith(("http", "ftp", "file")):
             vo_raise(
+                NotImplementedError,
                 "The vo package only supports remote data through http, ftp or file",
                 self._config,
                 self._pos,
-                NotImplementedError,
             )
 
         try:
@@ -3330,10 +3330,10 @@ def _parse_parquet(self, iterator, config):
                     fd = codecs.EncodedFile(fd, "base64")
                 else:
                     vo_raise(
+                        NotImplementedError,
                         f"Unknown encoding type '{encoding}'",
                         self._config,
                         self._pos,
-                        NotImplementedError,
                     )
 
             array = Table.read(fd, format="parquet")