diff --git a/django/forms/models.py b/django/forms/models.py
--- a/django/forms/models.py
+++ b/django/forms/models.py
@@ -1562,7 +1562,12 @@ def to_python(self, value):
             if isinstance(value, self.queryset.model):
                 value = getattr(value, key)
             value = self.queryset.get(**{key: value})
-        except (ValueError, TypeError, self.queryset.model.DoesNotExist):
+        except (
+            ValueError,
+            TypeError,
+            self.queryset.model.DoesNotExist,
+            ValidationError,
+        ):
             raise ValidationError(
                 self.error_messages["invalid_choice"],
                 code="invalid_choice",
