diff --git a/django/forms/models.py b/django/forms/models.py
--- a/django/forms/models.py
+++ b/django/forms/models.py
@@ -1645,7 +1645,7 @@
             self.validate_no_null_characters(pk)
             try:
                 self.queryset.filter(**{key: pk})
-            except (ValueError, TypeError):
+            except (ValueError, TypeError, ValidationError):
                 raise ValidationError(
                     self.error_messages["invalid_pk_value"],
                     code="invalid_pk_value",
