diff --git a/astropy/visualization/stretch.py b/astropy/visualization/stretch.py
--- a/astropy/visualization/stretch.py
+++ b/astropy/visualization/stretch.py
@@ -281,14 +281,13 @@ class PowerDistStretch(BaseStretch):
     Parameters
     ----------
     a : float, optional
-        The ``a`` parameter used in the above formula.  ``a`` must be
-        greater than or equal to 0, but cannot be set to 1.  Default is
-        1000.
+        The ``a`` parameter used in the above formula. ``a`` must be
+        greater than 0, but cannot be set to 1. Default is 1000.
     """
 
     def __init__(self, a=1000.0):
-        if a < 0 or a == 1:  # singularity
-            raise ValueError("a must be >= 0, but cannot be set to 1")
+        if a <= 0 or a == 1:  # singularity
+            raise ValueError("a must be > 0, but cannot be set to 1")
         super().__init__()
         self.a = a
 
@@ -318,14 +317,13 @@ class InvertedPowerDistStretch(BaseStretch):
     Parameters
     ----------
     a : float, optional
-        The ``a`` parameter used in the above formula.  ``a`` must be
-        greater than or equal to 0, but cannot be set to 1.  Default is
-        1000.
+        The ``a`` parameter used in the above formula. ``a`` must be
+        greater than 0, but cannot be set to 1. Default is 1000.
     """
 
     def __init__(self, a=1000.0):
-        if a < 0 or a == 1:  # singularity
-            raise ValueError("a must be >= 0, but cannot be set to 1")
+        if a <= 0 or a == 1:  # singularity
+            raise ValueError("a must be > 0, but cannot be set to 1")
         super().__init__()
         self.a = a
 