diff --git a/sympy/physics/quantum/qexpr.py b/sympy/physics/quantum/qexpr.py
--- a/sympy/physics/quantum/qexpr.py
+++ b/sympy/physics/quantum/qexpr.py
@@ -97,10 +97,6 @@ class QExpr(Expr):
     # The separator used in printing the label.
     _label_separator = ''
 
-    # @property
-    # def free_symbols(self):
-    #     return {self}
-
     def __new__(cls, *args, **kwargs):
         """Construct a new quantum object.
 