diff --git a/xarray/structure/merge.py b/xarray/structure/merge.py
--- a/xarray/structure/merge.py
+++ b/xarray/structure/merge.py
@@ -276,6 +276,7 @@ def merge_collected(
             if index is not None:
                 merged_indexes[name] = index
         else:
+            attrs: dict[Any, Any] = {}
             indexed_elements = [
                 (variable, index)
                 for variable, index in elements_list
