diff --git a/xarray/backends/zarr.py b/xarray/backends/zarr.py
--- a/xarray/backends/zarr.py
+++ b/xarray/backends/zarr.py
@@ -933,7 +933,7 @@
             existing_keys = {}
             existing_variable_names = {}
         else:
-            existing_keys = tuple(self.zarr_group.array_keys())
+            existing_keys = self.array_keys()
             existing_variable_names = {
                 vn for vn in variables if _encode_variable_name(vn) in existing_keys
             }
@@ -1111,7 +1111,7 @@
             dimensions.
         """
 
-        existing_keys = tuple(self.zarr_group.array_keys())
+        existing_keys = self.array_keys()
         is_zarr_v3_format = _zarr_v3() and self.zarr_group.metadata.zarr_format == 3
 
         for vn, v in variables.items():
@@ -1159,7 +1159,6 @@
                     zarr_array.resize(new_shape)
 
                 zarr_shape = zarr_array.shape
-
             region = tuple(write_region[dim] for dim in dims)
 
             # We need to do this for both new and existing variables to ensure we're not
@@ -1301,7 +1300,7 @@
 
     def _validate_encoding(self, encoding) -> None:
         if encoding and self._mode in ["a", "a-", "r+"]:
-            existing_var_names = set(self.zarr_group.array_keys())
+            existing_var_names = self.array_keys()
             for var_name in existing_var_names:
                 if var_name in encoding:
                     raise ValueError(
