diff --git a/sklearn/metrics/_plot/det_curve.py b/sklearn/metrics/_plot/det_curve.py
--- a/sklearn/metrics/_plot/det_curve.py
+++ b/sklearn/metrics/_plot/det_curve.py
@@ -4,7 +4,10 @@
 import numpy as np
 import scipy as sp

-from ...utils._plotting import _BinaryClassifierCurveDisplayMixin
+from ...utils._plotting import (
+    _BinaryClassifierCurveDisplayMixin,
+    _deprecate_y_pred_parameter,
+)
 from .._ranking import det_curve


@@ -67,8 +70,8 @@ class DetCurveDisplay(_BinaryClassifierCurveDisplayMixin):
     >>> X_train, X_test, y_train, y_test = train_test_split(
     ...     X, y, test_size=0.4, random_state=0)
     >>> clf = SVC(random_state=0).fit(X_train, y_train)
-    >>> y_pred = clf.decision_function(X_test)
-    >>> fpr, fnr, _ = det_curve(y_test, y_pred)
+    >>> y_score = clf.decision_function(X_test)
+    >>> fpr, fnr, _ = det_curve(y_test, y_score)
     >>> display = DetCurveDisplay(
     ...     fpr=fpr, fnr=fnr, estimator_name="SVC"
     ... )
@@ -178,7 +181,7 @@ def from_estimator(
         <...>
         >>> plt.show()
         """
-        y_pred, pos_label, name = cls._validate_and_get_response_values(
+        y_score, pos_label, name = cls._validate_and_get_response_values(
             estimator,
             X,
             y,
@@ -189,7 +192,7 @@ def from_estimator(
 
         return cls.from_predictions(
             y_true=y,
-            y_pred=y_pred,
+            y_score=y_score,
             sample_weight=sample_weight,
             drop_intermediate=drop_intermediate,
             name=name,
@@ -202,13 +205,14 @@ def from_estimator(
     def from_predictions(
         cls,
         y_true,
-        y_pred,
+        y_score=None,
         *,
         sample_weight=None,
         drop_intermediate=True,
         pos_label=None,
         name=None,
         ax=None,
+        y_pred="deprecated",
         **kwargs,
     ):
         """Plot the DET curve given the true and predicted labels.
@@ -225,11 +229,14 @@ def from_predictions(
         y_true : array-like of shape (n_samples,)
             True labels.
 
-        y_pred : array-like of shape (n_samples,)
+        y_score : array-like of shape (n_samples,)
             Target scores, can either be probability estimates of the positive
             class, confidence values, or non-thresholded measure of decisions
             (as returned by `decision_function` on some classifiers).
 
+            .. versionadded:: 1.8
+                `y_pred` has been renamed to `y_score`.
+
         sample_weight : array-like of shape (n_samples,), default=None
             Sample weights.
 
@@ -253,6 +260,15 @@ def from_predictions(
             Axes object to plot on. If `None`, a new figure and axes is
             created.
 
+        y_pred : array-like of shape (n_samples,)
+            Target scores, can either be probability estimates of the positive
+            class, confidence values, or non-thresholded measure of decisions
+            (as returned by “decision_function” on some classifiers).
+
+            .. deprecated:: 1.8
+                `y_pred` is deprecated and will be removed in 1.10. Use
+                `y_score` instead.
+
         **kwargs : dict
             Additional keywords arguments passed to matplotlib `plot` function.
 
