diff --git a/django/db/models/base.py b/django/db/models/base.py
--- a/django/db/models/base.py
+++ b/django/db/models/base.py
@@ -348,7 +348,7 @@ def __new__(cls, name, bases, attrs, **kwargs):
                 new_class._meta.parents.update(base_parents)
 
             # Inherit private fields (like GenericForeignKey) from the parent
-            # class
+            # class if they are not overridden.
             for field in base._meta.private_fields:
                 if field.name in field_names:
                     if not base._meta.abstract:
@@ -361,7 +361,7 @@ def __new__(cls, name, bases, attrs, **kwargs):
                                 base.__name__,
                             )
                         )
-                else:
+                elif (
                     field = copy.deepcopy(field)
                     if not base._meta.abstract:
                         field.mti_inherited = True