diff --git a/sklearn/utils/validation.py b/sklearn/utils/validation.py
--- a/sklearn/utils/validation.py
+++ b/sklearn/utils/validation.py
@@ -1427,7 +1427,7 @@ def _check_y(y, multi_output=False, y_numeric=False, estimator=None):
     return y
 
 
-def column_or_1d(y, *, dtype=None, warn=False, device=None):
+def column_or_1d(y, *, dtype=None, input_name="y", warn=False, device=None):
     """Ravel column or 1d numpy array, else raises an error.
 
     Parameters
@@ -1440,6 +1440,11 @@ def column_or_1d(y, *, dtype=None, warn=False, device=None):
 
         .. versionadded:: 1.2
 
+    input_name : str, default="y"
+        The data name used to construct the error message.
+
+        .. versionadded:: 1.8
+
     warn : bool, default=False
        To control display of warnings.
 
@@ -1470,7 +1475,7 @@ def column_or_1d(y, *, dtype=None, warn=False, device=None):
         y,
         ensure_2d=False,
         dtype=dtype,
-        input_name="y",
+        input_name=input_name,
         ensure_all_finite=False,
         ensure_min_samples=0,
     )