diff --git a/django/core/mail/message.py b/django/core/mail/message.py
--- a/django/core/mail/message.py
+++ b/django/core/mail/message.py
@@ -191,7 +191,7 @@ def __setitem__(self, name, val):
         MIMEMultipart.__setitem__(self, name, val)


-EmailAlternative = namedtuple("Alternative", ["content", "mimetype"])
 EmailAttachment = namedtuple("Attachment", ["filename", "content", "mimetype"])
+EmailAlternative = namedtuple("EmailAlternative", ["content", "mimetype"])

 
 class EmailMessage:
