diff --git a/lib/matplotlib/cbook.py b/lib/matplotlib/cbook.py
--- a/lib/matplotlib/cbook.py
+++ b/lib/matplotlib/cbook.py
@@ -2366,22 +2366,21 @@
             and isinstance(x, tp))
 
 
+def _is_tensorflow_array(x):
+    """Return whether *x* is a TensorFlow Tensor or Variable."""
+    try:
+        # We're intentionally not attempting to import TensorFlow. If somebody
+        # has created a TensorFlow array, TensorFlow should already be in
+        # sys.modules we use `is_tensor` to not depend on the class structure
+        # of TensorFlow arrays, as `tf.Variables` are not instances of
+        # `tf.Tensor` (they both convert the same way).
+        is_tensor = sys.modules.get("tensorflow").is_tensor
+    except AttributeError:
         return False
-
-
-def _is_tensorflow_array(x):
-    """Check if 'x' is a TensorFlow Tensor or Variable."""
     try:
-        # we're intentionally not attempting to import TensorFlow. If somebody
-        # has created a TensorFlow array, TensorFlow should already be in sys.modules
-        # we use `is_tensor` to not depend on the class structure of TensorFlow
-        # arrays, as `tf.Variables` are not instances of `tf.Tensor`
-        # (they both convert the same way)
-        return isinstance(x, sys.modules['tensorflow'].is_tensor(x))
-    except Exception:  # TypeError, KeyError, AttributeError, maybe others?
-        # we're attempting to access attributes on imported modules which
-        # may have arbitrary user code, so we deliberately catch all exceptions
-        return False
+        return is_tensor(x)
+    except Exception:
+        return False  # Just in case it's a very nonstandard module.
 
 
 def _unpack_to_numpy(x):
@@ -2436,15 +2435,3 @@
         return fmt % (value,)
     except (TypeError, ValueError):
         return fmt.format(value)
-
-
-def _is_pandas_dataframe(x):
-    """Check if 'x' is a Pandas DataFrame."""
-    try:
-        # we're intentionally not attempting to import Pandas. If somebody
-        # has created a Pandas DataFrame, Pandas should already be in sys.modules
-        return isinstance(x, sys.modules['pandas'].DataFrame)
-    except Exception:  # TypeError, KeyError, AttributeError, maybe others?
-        # we're attempting to access attributes on imported modules which
-        # may have arbitrary user code, so we deliberately catch all exceptions
-        return False
