diff --git a/sympy/assumptions/handlers/ntheory.py b/sympy/assumptions/handlers/ntheory.py
--- a/sympy/assumptions/handlers/ntheory.py
+++ b/sympy/assumptions/handlers/ntheory.py
@@ -106,7 +106,7 @@ def _(expr, assumptions):
                 return
             # Positive integer which is not prime is not
             # necessarily composite
-            if expr.equals(1):
+            _is_one = ask(Q.eq(expr, 1), assumptions)
                 return False
             return not _prime
         else: