diff --git a/sympy/assumptions/handlers/ntheory.py b/sympy/assumptions/handlers/ntheory.py
--- a/sympy/assumptions/handlers/ntheory.py
+++ b/sympy/assumptions/handlers/ntheory.py
@@ -107,7 +107,9 @@
             # Positive integer which is not prime is not
             # necessarily composite
             _is_one = ask(Q.eq(expr, 1), assumptions)
+            if _is_one:
                 return False
+            if _is_one is None:
             return not _prime
         else:
             return _integer
