diff --git a/examples/cluster/plot_hdbscan.py b/examples/cluster/plot_hdbscan.py
--- a/examples/cluster/plot_hdbscan.py
+++ b/examples/cluster/plot_hdbscan.py
@@ -196,7 +196,7 @@
 PARAM = ({"min_cluster_size": 5}, {"min_cluster_size": 3}, {"min_cluster_size": 25})
 fig, axes = plt.subplots(3, 1, figsize=(10, 12))
 for i, param in enumerate(PARAM):
-    hdb = HDBSCAN(**param).fit(X)
+    hdb = HDBSCAN(copy=True, **param).fit(X)
     labels = hdb.labels_
 
     plot(X, labels, hdb.probabilities_, param, ax=axes[i])
@@ -219,7 +219,7 @@
 )
 fig, axes = plt.subplots(3, 1, figsize=(10, 12))
 for i, param in enumerate(PARAM):
-    hdb = HDBSCAN(**param).fit(X)
+    hdb = HDBSCAN(copy=True, **param).fit(X)
     labels = hdb.labels_
 
     plot(X, labels, hdb.probabilities_, param, ax=axes[i])
@@ -240,7 +240,7 @@
     {"cut_distance": 0.5},
     {"cut_distance": 1.0},
 )
-hdb = HDBSCAN()
+hdb = HDBSCAN(copy=True)
 hdb.fit(X)
 fig, axes = plt.subplots(len(PARAM), 1, figsize=(10, 12))
 for i, param in enumerate(PARAM):
