diff --git a/django/db/models/sql/datastructures.py b/django/db/models/sql/datastructures.py
--- a/django/db/models/sql/datastructures.py
+++ b/django/db/models/sql/datastructures.py
@@ -120,7 +120,9 @@
             )
         on_clause_sql = " AND ".join(join_conditions)
         alias_str = (
-            "" if self.table_alias == self.table_name else (" %s" % self.table_alias)
+            ""
+            if self.table_alias == self.table_name
+            else (" %s" % qn(self.table_alias))
         )
         sql = "%s %s%s ON (%s)" % (
             self.join_type,
@@ -195,7 +197,9 @@
     def as_sql(self, compiler, connection):
         qn = compiler.quote_name
         alias_str = (
-            "" if self.table_alias == self.table_name else (" %s" % self.table_alias)
+            ""
+            if self.table_alias == self.table_name
+            else (" %s" % qn(self.table_alias))
         )
         base_sql = qn(self.table_name)
         return base_sql + alias_str, []
