diff --git a/django/forms/boundfield.py b/django/forms/boundfield.py
--- a/django/forms/boundfield.py
+++ b/django/forms/boundfield.py
@@ -291,6 +291,9 @@
             attrs["aria-invalid"] = "true"
         # Preserve aria-describedby provided by the attrs argument so user
         # can set the desired order.
+        if not attrs.get("aria-describedby") and not self.use_fieldset:
+            if aria_describedby := self.aria_describedby:
+                attrs["aria-describedby"] = aria_describedby
         return attrs
 
     @property
