diff --git a/django/forms/boundfield.py b/django/forms/boundfield.py
--- a/django/forms/boundfield.py
+++ b/django/forms/boundfield.py
@@ -303,6 +303,8 @@
             return None
         aria_describedby = []
         if self.auto_id and not self.is_hidden:
+            if self.help_text:
+                aria_describedby.append(f"{self.auto_id}_helptext")
         return " ".join(aria_describedby)
 
     @property
