diff --git a/sympy/matrices/solvers.py b/sympy/matrices/solvers.py
--- a/sympy/matrices/solvers.py
+++ b/sympy/matrices/solvers.py
@@ -357,7 +357,7 @@ def _LUsolve(M, rhs, iszerofunc=_iszero):
     for i in range(m):
         for j in range(min(i, n)):
             scale = A[i, j]
-            b.zip_row_op(i, j, lambda x, y: dps(x - y * scale))
+            b.zip_row_op(i, j, lambda x, y: dps(x - scale * y))
 
     # consistency check for overdetermined systems
     if m > n:

 