diff --git a/sympy/matrices/solvers.py b/sympy/matrices/solvers.py
--- a/sympy/matrices/solvers.py
+++ b/sympy/matrices/solvers.py
@@ -372,7 +372,7 @@
     for i in range(n - 1, -1, -1):
         for j in range(i + 1, n):
             scale = A[i, j]
-            b.zip_row_op(i, j, lambda x, y: dps(x - y * scale))
+            b.zip_row_op(i, j, lambda x, y: dps(x - scale * y))
 
         scale = A[i, i]
         b.row_op(i, lambda x, _: dps(x / scale))
 
