diff --git a/lib/matplotlib/axes/_axes.py b/lib/matplotlib/axes/_axes.py
--- a/lib/matplotlib/axes/_axes.py
+++ b/lib/matplotlib/axes/_axes.py
@@ -2888,6 +2888,8 @@ def bar_label(self, container, labels=None, *, fmt="%g", label_type="edge",
             Distance of label from the end of the bar, in points.
             If an array-like is provided, the padding values are applied
             to each label individually. Must have the same length as container.
+            
+            .. versionadded:: 3.11
 
         **kwargs
             Any remaining keyword arguments are passed through to