diff --git a/galleries/examples/images_contours_and_fields/plot_streamplot.py b/galleries/examples/images_contours_and_fields/plot_streamplot.py
--- a/galleries/examples/images_contours_and_fields/plot_streamplot.py
+++ b/galleries/examples/images_contours_and_fields/plot_streamplot.py
@@ -14,6 +14,8 @@
 * Unbroken streamlines even when exceeding the limit of lines within a single
   grid cell.
 """
+import time
+
 import matplotlib.pyplot as plt
 import numpy as np
 
@@ -74,6 +76,42 @@
 axs[7].streamplot(X, Y, U, V, broken_streamlines=False)
 axs[7].set_title('Streamplot with unbroken streamlines')
 
+plt.tight_layout()
+# plt.show()
+
+# %%
+# Streamline computation
+# ----------------------
+#
+# The streamlines are computed by integrating along the provided vector field
+# from the seed points, which are either automatically generated or manually
+# specified. The accuracy and smoothness of the streamlines can be adjusted using
+# the ``integration_max_step_scale`` and ``integration_max_error_scale`` optional
+# parameters. See the `~.axes.Axes.streamplot` function documentation for more
+# details.
+#
+# This example shows how adjusting the maximum allowed step size and error for
+# the integrator changes the appearance of the streamline. The differences can
+# be subtle, but can be observed particularly where the streamlines have
+# high curvature (as shown in the zoomed in region).
+
+# Linear potential flow over a lifting cylinder
+n = 50
+x, y = np.meshgrid(np.linspace(-2, 2, n), np.linspace(-3, 3, n))
+th = np.arctan2(y, x)
+r = np.sqrt(x**2 + y**2)
+vr = -np.cos(th) / r**2
+vt = -np.sin(th) / r**2 - 1 / r
+vx = vr * np.cos(th) - vt * np.sin(th) + 1.0
+vy = vr * np.sin(th) + vt * np.cos(th)
+
+# Seed points
+n_seed = 50
+seed_pts = np.column_stack((np.full(n_seed, -1.75), np.linspace(-2, 2, n_seed)))
+
+_, axs = plt.subplots(3, 1, figsize=(6, 14))
+th_circ = np.linspace(0, 2 * np.pi, 100)
+for ax, max_val in zip(axs, [0.05, 1, 5]):
 plt.tight_layout()
 plt.show()
 # %%
