diff --git a/sklearn/utils/_array_api.py b/sklearn/utils/_array_api.py
--- a/sklearn/utils/_array_api.py
+++ b/sklearn/utils/_array_api.py
@@ -135,9 +135,9 @@ def yield_mixed_namespace_input_permutations():
 
     We do not test for all possible permutations of namespace/device from
     `yield_namespace_device_dtype_combinations` (excluding dtype variations, this is
-    C(8,2)=56), to avoid slow testing and maintenance burden.
+    P(8,2)=56), to avoid slow testing and maintenance burden.
 
-    The included selection covers ensures that the following conversions are tested:
+    The included selection ensures that the following conversions are tested:
 
     * non-NumPy to NumPy (including GPU to CPU)
     * NumPy to non-NumPy (including CPU to GPU)
