diff --git a/sklearn/utils/_array_api.py b/sklearn/utils/_array_api.py
--- a/sklearn/utils/_array_api.py
+++ b/sklearn/utils/_array_api.py
@@ -627,7 +627,6 @@ def move_to(*arrays, xp, device):
                     # more details about the introduction of the `copy` and `device`
                     # kwargs in the from_dlpack method and their expected
                     # meaning by namespaces implementing the array API spec.
-                    # `RuntimeError`: occurs when converting from GPU to NumPy
                     # TODO: try removing this once DLPack v1 more widely supported
                     # TODO: ValueError not needed once min NumPy >=2.4.0:
                     # https://github.com/numpy/numpy/issues/30341
@@ -637,7 +636,6 @@ def move_to(*arrays, xp, device):
                     NotImplementedError,
                     BufferError,
                     ValueError,
-                    RuntimeError,
                 ):
                     # Converting to numpy is tricky, handle this via dedicated function
                     if _is_numpy_namespace(xp):
