diff --git a/sklearn/utils/_array_api.py b/sklearn/utils/_array_api.py
--- a/sklearn/utils/_array_api.py
+++ b/sklearn/utils/_array_api.py
@@ -23,6 +23,18 @@
 
 _NUMPY_NAMESPACE_NAMES = {"numpy", "sklearn.externals.array_api_compat.numpy"}
 
+MIXED_NAMESPACE_COMBINATIONS = [
+    (("cupy", None), ("torch", "cuda"), "cupy to torch cuda"),
+    (("torch", "mps"), ("numpy", None), "torch mps to numpy"),
+    (("numpy", None), ("torch", "cuda"), "numpy to torch cuda"),
+    (("numpy", None), ("torch", "mps"), "numpy to torch mps"),
+    (
+        ("array_api_strict", None),
+        ("torch", "mps"),
+        "array_api_strict to torch mps",
+    ),
+]
+
 
 def yield_namespaces(include_numpy_namespaces=True):
     """Yield supported namespace.
