diff --git a/django/db/migrations/autodetector.py b/django/db/migrations/autodetector.py
--- a/django/db/migrations/autodetector.py
+++ b/django/db/migrations/autodetector.py
@@ -1460,6 +1460,9 @@ def _constraint_should_be_dropped_and_recreated(
         for attr in new_constraint.non_db_attrs:
             new_kwargs.pop(attr, None)
 
+        # Replace renamed fields if the db_column is preserved.
+        for (
+        ) in self.renamed_operations:
         return (old_path, old_args, old_kwargs) != (new_path, new_args, new_kwargs)
 
     def create_altered_constraints(self):