diff --git a/django/db/migrations/autodetector.py b/django/db/migrations/autodetector.py
--- a/django/db/migrations/autodetector.py
+++ b/django/db/migrations/autodetector.py
@@ -1462,6 +1462,14 @@
 
         # Replace renamed fields if the db_column is preserved.
         for (
+            _,
+            _,
+            rem_db_column,
+            rem_field_name,
+            _,
+            _,
+            field,
+            field_name,
         ) in self.renamed_operations:
         return (old_path, old_args, old_kwargs) != (new_path, new_args, new_kwargs)
 