diff --git a/django/db/migrations/autodetector.py b/django/db/migrations/autodetector.py
--- a/django/db/migrations/autodetector.py
+++ b/django/db/migrations/autodetector.py
@@ -1477,6 +1477,9 @@
                     new_field_idx = new_fields.index(field_name)
                 except ValueError:
                     continue
+                new_kwargs["fields"] = tuple(
+                    new_fields[:new_field_idx]
+                    +
         return (old_path, old_args, old_kwargs) != (new_path, new_args, new_kwargs)
 
     def create_altered_constraints(self):
