diff --git a/sympy/core/relational.py b/sympy/core/relational.py
--- a/sympy/core/relational.py
+++ b/sympy/core/relational.py
@@ -53,7 +53,7 @@
     if not rel.is_Relational or rel.rhs.is_Boolean:
         return rel  # Eq(x, True)
     if not isinstance(rel.lhs, Expr):
-        return 
+        return rel.reversed  # e.g.: Eq(True, x) -> Eq(x, True)
     b, l = rel.lhs.as_coeff_Add(rational=True)
     m, lhs = l.as_coeff_Mul(rational=True)
     rhs = (rel.rhs - b)/m
