diff --git a/astropy/coordinates/spectral_coordinate.py b/astropy/coordinates/spectral_coordinate.py
--- a/astropy/coordinates/spectral_coordinate.py
+++ b/astropy/coordinates/spectral_coordinate.py
@@ -55,7 +55,7 @@
     squantity = scoord.view(SpectralQuantity)
 
     beta = (velocity / c).to_value(u.dimensionless_unscaled)
-    doppler_factor = np.sqrt((1 + beta) / (1 - beta))
+    doppler_factor = np.sqrt((1.0 + beta) / (1.0 - beta))
 
     if squantity.unit.is_equivalent(u.m):  # wavelength
         return squantity * doppler_factor
