diff --git a/astropy/coordinates/spectral_coordinate.py b/astropy/coordinates/spectral_coordinate.py
--- a/astropy/coordinates/spectral_coordinate.py
+++ b/astropy/coordinates/spectral_coordinate.py
@@ -57,16 +57,13 @@
     beta = (velocity / c).to_value(u.dimensionless_unscaled)
     doppler_factor = np.sqrt((1.0 + beta) / (1.0 - beta))
 
-    if squantity.unit.is_equivalent(u.m):  # wavelength
-        return squantity * doppler_factor
-    elif (
-        squantity.unit.is_equivalent(u.Hz)
-        or squantity.unit.is_equivalent(u.eV)
-        or squantity.unit.is_equivalent(1 / u.m)
-    ):
-        return squantity / doppler_factor
-    elif squantity.unit.is_equivalent(KMS):  # velocity
-        return (squantity.to(u.Hz) / doppler_factor).to(squantity.unit)
+    match squantity.unit.physical_type:
+        case u.physical.length:
+            return squantity * doppler_factor
+        case u.physical.frequency | u.physical.energy | u.physical.wavenumber:
+            return squantity / doppler_factor
+        case u.physical.velocity:
+            return (squantity.to(u.Hz) / doppler_factor).to(squantity.unit)
     else:  # pragma: no cover
         raise RuntimeError(
             f"Unexpected units in velocity shift: {squantity.unit}. This should not"
