diff --git a/astropy/wcs/wcsapi/fitswcs.py b/astropy/wcs/wcsapi/fitswcs.py
--- a/astropy/wcs/wcsapi/fitswcs.py
+++ b/astropy/wcs/wcsapi/fitswcs.py
@@ -690,6 +690,16 @@
                                     AstropyDeprecationWarning,
                                 )
 
+                        if ctype == "VELO":
+                            kwargs["doppler_convention"] = "relativistic"
+                            kwargs["doppler_rest"] = restfrq
+                        elif ctype == "VRAD":
+                            kwargs["doppler_convention"] = "radio"
+                            kwargs["doppler_rest"] = restfrq
+                        elif ctype == "VOPT":
+                            kwargs["doppler_convention"] = "optical"
+                            kwargs["doppler_rest"] = restwav
+
                 def spectralcoord_from_value(value):
                     if isinstance(value, SpectralCoord):
                         return value
