diff --git a/astropy/wcs/wcsapi/fitswcs.py b/astropy/wcs/wcsapi/fitswcs.py
--- a/astropy/wcs/wcsapi/fitswcs.py
+++ b/astropy/wcs/wcsapi/fitswcs.py
@@ -15,7 +15,7 @@
     update_differentials_to_match,
 )
 from astropy.units import allclose as quantity_allclose
-from astropy.utils.exceptions import AstropyUserWarning
+from astropy.utils.exceptions import AstropyDeprecationWarning, AstropyUserWarning
 
 from .high_level_api import HighLevelWCSMixin
 from .low_level_api import BaseLowLevelWCS
@@ -664,34 +664,41 @@ def beta_from_spectralcoord(spectralcoord):
                 # RESTFRQ is defined in the original FITS header, wcs.restwav
                 # is 0.
 
-                restfrq = self.wcs.restfrq
-                restwav = self.wcs.restwav
-
-                if restfrq > 0 or restwav > 0:
-                    if restwav == 0:
-                        restfrq = u.Quantity(restfrq, u.Hz)
-                        restwav = restfrq.to(u.m, u.spectral())
-                    elif restfrq == 0:
-                        restwav = u.Quantity(restwav, u.m)
-                        restfrq = restwav.to(u.Hz, u.spectral())
-                    else:
-                        restfrq = u.Quantity(restfrq, u.Hz)
-                        restwav = u.Quantity(restwav, u.m)
-                        restfrq_derived = restwav.to(u.Hz, u.spectral())
-                        if not quantity_allclose(restfrq, restfrq_derived, rtol=1e-4):
-                            raise ValueError(
-                                f"restfrq={restfrq} and restwav={restwav}={restfrq_derived} are not consistent to 10^-4 or better precision"
-                            )
-
-                    if ctype == "VELO":
-                        kwargs["doppler_convention"] = "relativistic"
-                        kwargs["doppler_rest"] = restfrq
-                    elif ctype == "VRAD":
-                        kwargs["doppler_convention"] = "radio"
-                        kwargs["doppler_rest"] = restfrq
-                    elif ctype == "VOPT":
-                        kwargs["doppler_convention"] = "optical"
-                        kwargs["doppler_rest"] = restwav
+                if ctype in ("VELO", "VRAD", "VOPT"):
+                    restfrq = self.wcs.restfrq
+                    restwav = self.wcs.restwav
+
+                    if restfrq > 0 or restwav > 0:
+                        if restwav == 0:
+                            restfrq = u.Quantity(restfrq, u.Hz)
+                            restwav = restfrq.to(u.m, u.spectral())
+                        elif restfrq == 0:
+                            restwav = u.Quantity(restwav, u.m)
+                            restfrq = restwav.to(u.Hz, u.spectral())
+                        else:
+                            restfrq = u.Quantity(restfrq, u.Hz)
+                            restwav = u.Quantity(restwav, u.m)
+                            restfrq_derived = restwav.to(u.Hz, u.spectral())
+                            if not quantity_allclose(
+                                restfrq, restfrq_derived, rtol=1e-4
+                            ):
+                                used = "restwav" if ctype == "VOPT" else "restfrq"
+                                warnings.warn(
+                                    f"restfrq={restfrq} and restwav={restwav}={restfrq_derived} "
+                                    f"are not consistent to rtol=1e-4, choosing {used}. In future, "
+                                    f"this will raise an exception.",
+                                    AstropyDeprecationWarning,
+                                )
+
+                        if ctype == "VELO":
+                            kwargs["doppler_convention"] = "relativistic"
+                            kwargs["doppler_rest"] = restfrq
+                        elif ctype == "VRAD":
+                            kwargs["doppler_convention"] = "radio"
+                            kwargs["doppler_rest"] = restfrq
+                        elif ctype == "VOPT":
+                            kwargs["doppler_convention"] = "optical"
+                            kwargs["doppler_rest"] = restwav
 
                 def spectralcoord_from_value(value):
                     if isinstance(value, SpectralCoord):