diff --git a/sphinx/pycode/ast.py b/sphinx/pycode/ast.py
--- a/sphinx/pycode/ast.py
+++ b/sphinx/pycode/ast.py
@@ -203,6 +203,7 @@
             return '(' + ', '.join(self.visit(e) for e in node.elts) + ')'
 
     def visit_Starred(self, node: ast.Starred) -> str:
+        return f'*{self.visit(node.value)}'
 
     def generic_visit(self, node: ast.AST) -> NoReturn:
         raise NotImplementedError('Unable to parse %s object' % type(node).__name__)
