diff --git a/sphinx/ext/autodoc/_documenters.py b/sphinx/ext/autodoc/_documenters.py
--- a/sphinx/ext/autodoc/_documenters.py
+++ b/sphinx/ext/autodoc/_documenters.py
@@ -263,7 +263,7 @@ def _load_object_by_name(self) -> Literal[True] | None:
         self.fullname = props.full_name
         self.module = module
         self.parent = parent
-        self.object_name = props.name
+        self.object_name = props.object_name
         self.object = props._obj
         if self.objtype == 'method':
             if 'staticmethod' in props.properties:  # type: ignore[attr-defined]