diff --git a/sympy/combinatorics/free_groups.py b/sympy/combinatorics/free_groups.py
--- a/sympy/combinatorics/free_groups.py
+++ b/sympy/combinatorics/free_groups.py
@@ -363,24 +363,19 @@
         return _reduce_array_form(array_form)
 
     def __new__(cls, init=()):
-        return cls._new(init)
+        if isinstance(init, FreeGroupElement):
+            return cls._new(tuple(init))
+        return cls._new(_reduce_array_form(init))
 
     @classmethod
-    def _new(cls, init=(), *, reduced=False, boundary_index=None):
-        if not init:
-            return tuple.__new__(cls, ())
-        if boundary_index is not None:
-            init = _reduce_array_form(init, boundary_index=boundary_index)
-            return tuple.__new__(cls, init)
-        if reduced:
-            return tuple.__new__(cls, tuple(init))
-        return tuple.__new__(cls, _reduce_array_form(init))
+    def _new(cls, array_form=()):
+        return tuple.__new__(cls, array_form)
 
     @classmethod
-    def new(cls, init):
-        if isinstance(init, FreeGroupElement):
-            return cls._new(init, reduced=True)
-        return cls._new(init)
+    def _new_reduce_at_boundary(cls, array_form, boundary_index):
+        array_form = _reduce_array_form(array_form,
+                                        boundary_index=boundary_index)
+        return cls._new(array_form)
 
     _hash = None
 
@@ -391,7 +386,7 @@
         return _hash
 
     def copy(self):
-        return self.new(self)
+        return self._new(tuple(self))
 
     @property
     def is_identity(self):
@@ -558,7 +553,7 @@
         if other.is_identity:
             return self
         r = self.array_form + other.array_form
-        return group.dtype._new(r, boundary_index=len(self.array_form) - 1)
+        return group.dtype._new_reduce_at_boundary(r, len(self.array_form) - 1)
 
     def __truediv__(self, other):
         group = self.group
@@ -594,7 +589,7 @@
         """
         group = self.group
         r = tuple((i, -j) for i, j in self.array_form[::-1])
-        return group.dtype._new(r, reduced=True)
+        return group.dtype._new(r)
 
     def order(self):
         """Find the order of a ``FreeGroupElement``.
@@ -930,7 +925,7 @@
         else:
             letter_form = self.letter_form[from_i: to_j]
             array_form = self._array_form_from_letter_form(letter_form)
-            return group.dtype._new(array_form, reduced=True)
+            return group.dtype._new(array_form)
 
     def subword_index(self, word, start = 0):
         '''
@@ -1031,7 +1026,7 @@
         period2 = int(to_j/l) - 1
         word += letter_form*period2 + letter_form[:diff-l+from_i-l*period2]
         array_form = self._array_form_from_letter_form(word)
-        return group.dtype._new(array_form, reduced=True)
+        return group.dtype._new(array_form)
 
     def cyclic_conjugates(self):
         """Returns a words which are cyclic to the word `self`.
