diff --git a/astropy/io/fits/convenience.py b/astropy/io/fits/convenience.py
--- a/astropy/io/fits/convenience.py
+++ b/astropy/io/fits/convenience.py
@@ -453,10 +453,10 @@ def writeto(
 
     Notes
     -----
-    gzip, zip and bzip2 compression algorithms are natively supported.
+    gzip, zip, bzip2 and lzma compression algorithms are natively supported.
     Compression mode is determined from the filename extension
-    ('.gz', '.zip' or '.bz2' respectively).  It is also possible to pass a
-    compressed file object, e.g. `gzip.GzipFile`.
+    ('.gz', '.zip', '.bz2' or '.xz' respectively).  It is also possible to pass
+    a compressed file object, e.g. `gzip.GzipFile`.
     """
     hdu = _makehdu(data, header)
     if hdu.is_image and not isinstance(hdu, PrimaryHDU):