diff --git a/sklearn/metrics/_ranking.py b/sklearn/metrics/_ranking.py
--- a/sklearn/metrics/_ranking.py
+++ b/sklearn/metrics/_ranking.py
@@ -2256,7 +2256,7 @@ def metric_at_thresholds(
     sample_weight=None,
     metric_params=None,
 ):
-    """Compute `metric_func` per threshold for :term:`binary` data.
+    r"""Compute `metric_func` per threshold for :term:`binary` data.
 
     Read more in the :ref:`User Guide <threshold_tunning>`.
 
@@ -2285,10 +2285,10 @@ def metric_at_thresholds(
 
     Returns
     -------
-    metric_values : ndarray of shape (n_thresholds,) or (n_thresholds, *n_outputs)
+    metric_values : ndarray of shape (n_thresholds,) or (n_thresholds, \*n_outputs)
         The scores associated with each threshold. If `metric_func` returns a
         collection (e.g., a tuple of floats), the output would be a 2D array
-        of shape (n_thresholds, *n_outputs).
+        of shape (n_thresholds, \*n_outputs).
 
     thresholds : ndarray of shape (n_thresholds,)
         The thresholds used to compute the scores.
