diff --git a/sklearn/metrics/_ranking.py b/sklearn/metrics/_ranking.py
--- a/sklearn/metrics/_ranking.py
+++ b/sklearn/metrics/_ranking.py
@@ -2297,7 +2297,7 @@ def decision_threshold_curve(
     >>> y_true = np.array([0, 0, 1, 1])
     >>> y_score = np.array([0.1, 0.4, 0.35, 0.8])
     >>> scores, thresholds = decision_threshold_curve(
-    ...     accuracy_score, y_true, y_score)
+    ...     y_true, y_score, accuracy_score)
     >>> thresholds
     array([0.8 , 0.4 , 0.35, 0.1 ])
     >>> scores
