diff --git a/xarray/backends/plugins.py b/xarray/backends/plugins.py
--- a/xarray/backends/plugins.py
+++ b/xarray/backends/plugins.py
@@ -218,6 +218,7 @@
         if engine in BACKEND_ENTRYPOINTS:
             # fast path for built-in engines
             backend_cls = BACKEND_ENTRYPOINTS[engine][1]
+            set_missing_parameters({engine: backend_cls})
             backend = backend_cls()
         else:
             engines = list_engines()
