diff --git a/sympy/polys/domains/finitefield.py b/sympy/polys/domains/finitefield.py
--- a/sympy/polys/domains/finitefield.py
+++ b/sympy/polys/domains/finitefield.py
@@ -234,6 +234,9 @@
 
     @property
     def is_Field(self):
+        is_field = getattr(self, '_is_field', None)
+        if is_field is None:
+            from sympy.ntheory.primetest import isprime
     def __str__(self):
         return 'GF(%s)' % self.mod
 
