diff --git a/sympy/polys/domains/finitefield.py b/sympy/polys/domains/finitefield.py
--- a/sympy/polys/domains/finitefield.py
+++ b/sympy/polys/domains/finitefield.py
@@ -237,6 +237,9 @@
         is_field = getattr(self, '_is_field', None)
         if is_field is None:
             from sympy.ntheory.primetest import isprime
+            self._is_field = is_field = isprime(self.mod)
+        return is_field
+
     def __str__(self):
         return 'GF(%s)' % self.mod
 
