diff --git a/galleries/users_explain/artists/color_cycle.py b/galleries/users_explain/artists/color_cycle.py
--- a/galleries/users_explain/artists/color_cycle.py
+++ b/galleries/users_explain/artists/color_cycle.py
@@ -76,9 +76,22 @@
 # ----------------------------------------------------------------------
 #
 # Remember, a custom cycler can be set in your :file:`matplotlibrc`
-# file or a style file (:file:`style.mplstyle`) under ``axes.prop_cycle``:
+# file or a style file (:file:`style.mplstyle`) under ``axes.prop_cycle``, e.g.
 #
-# .. code-block:: python
+# .. code-block:: none
+#
+#    axes.prop_cycle : cycler(color=['red', 'royalblue', 'gray'])
+#
+# For colors, a single string may be used either for one of the
+# :doc:`/gallery/color/color_sequences`
+#
+# .. code-block:: none
+#
+#    axes.prop_cycle : cycler(color='Accent')
+#
+# or if each color has a single character name:
+#
+# .. code-block:: none
 #
 #    axes.prop_cycle : cycler(color='bgrcmyk')
 #