diff --git a/astropy/io/misc/ecsv.py b/astropy/io/misc/ecsv.py
--- a/astropy/io/misc/ecsv.py
+++ b/astropy/io/misc/ecsv.py
@@ -64,14 +64,14 @@
 import re
 import warnings
 from collections.abc import Iterable
-from contextlib import ExitStack
 from dataclasses import dataclass
-from pathlib import Path
 from typing import TYPE_CHECKING, Any, Final, NamedTuple
 
 import numpy as np
 import numpy.typing as npt
 
+from astropy.utils.data import get_readable_fileobj
+
 if TYPE_CHECKING:
     from astropy.table import SerializedColumn, Table
 
@@ -441,8 +441,8 @@ def get_header_lines(
     ----------
     input_file : str | os.PathLike | io.BytesIO
         The input file path or file-like object to read. If a file path is
-        provided, the function automatically handles compressed files with
-        extensions `.gz` or `.bz2`.
+        provided, the function automatically handles compressed file formats
+        that are supported by `~astropy.utils.data.get_readable_fileobj`.
     encoding : str, optional
         The encoding used to decode the file content. Default is "utf-8".
 
@@ -463,25 +463,8 @@ def get_header_lines(
     n_empty = 0
     n_comment = 0
 
-    with ExitStack() as stack:
-        # Get a file-like object as tmp_input_file
-        if isinstance(input_file, (str, os.PathLike)):
-            ext = Path(input_file).suffix
-            if ext == ".gz":
-                import gzip
-
-                opener = gzip.open
-            elif ext == ".bz2":
-                import bz2
-
-                opener = bz2.open
-            else:
-                opener = open
-            tmp_input_file = stack.enter_context(opener(input_file, "rb"))
-        else:
-            tmp_input_file = input_file
-
-        for idx, line in enumerate(tmp_input_file):
+    with get_readable_fileobj(input_file, encoding="binary") as f:
+        for idx, line in enumerate(f):
             line_strip = line.strip()
             if line_strip.startswith(header_prefix):
                 lines.append(line_strip[2:].decode(encoding))
diff --git a/astropy/io/misc/pyarrow/csv.py b/astropy/io/misc/pyarrow/csv.py
--- a/astropy/io/misc/pyarrow/csv.py
+++ b/astropy/io/misc/pyarrow/csv.py
@@ -6,13 +6,13 @@
 import datetime
 import io
 import os
-from contextlib import ExitStack
 from typing import TYPE_CHECKING, BinaryIO, Literal, Union
 
 import numpy as np
 from numpy.typing import DTypeLike, NDArray
 
 from astropy.utils.compat.optional_deps import HAS_PYARROW
+from astropy.utils.data import get_readable_fileobj
 
 if TYPE_CHECKING:
     import pyarrow
@@ -159,12 +159,13 @@ def read_csv(
         include_names, dtypes, null_values, timestamp_parsers
     )
 
-    table_pa = csv.read_csv(
-        input_file,
-        parse_options=parse_options,
-        read_options=read_options,
-        convert_options=convert_options,
-    )
+    with get_readable_fileobj(input_file, encoding="binary") as f:
+        table_pa = csv.read_csv(
+            f,
+            parse_options=parse_options,
+            read_options=read_options,
+            convert_options=convert_options,
+        )
     return convert_pa_table_to_astropy_table(table_pa)
 
 
@@ -388,29 +389,26 @@ def strip_comment_lines(
     """
     comment_encode = comment.encode(encoding)
 
-    with ExitStack() as stack:
-        if isinstance(input_file, (str, os.PathLike)):
-            input_file = stack.enter_context(open(input_file, "rb"))
-
+    with get_readable_fileobj(input_file, encoding="binary") as f:
         if header_start in (None, 0) and data_start is None:
             idx_last_comment = -1
-            for idx, line in enumerate(input_file):
+            for idx, line in enumerate(f):
                 if line.lstrip().startswith(comment_encode):
                     if idx - idx_last_comment == 1:
                         idx_last_comment = idx
                     else:
                         # Gap between comment lines, need to reset input file handle and
                         # break out to the logic below.
-                        input_file.seek(0)
+                        f.seek(0)
                         break
             else:
-                input_file.seek(0)
+                f.seek(0)
                 return None, idx_last_comment + 1
 
         # If we get here, we need to read the whole file and remove comment lines and
         # write into an output BytesIO file.
         output = io.BytesIO()
-        for idx, line in enumerate(input_file):
+        for idx, line in enumerate(f):
             if not line.lstrip().startswith(comment_encode):
                 output.write(line)
 