diff --git a/astropy/table/soco.py b/astropy/table/soco.py
--- a/astropy/table/soco.py
+++ b/astropy/table/soco.py
@@ -86,7 +86,7 @@ def add(self, key: tuple, row: int) -> None:
         """
         Add a key, value pair.
         """
-        if self._unique and (key in self._nodes):
+        if self.unique and (key in self._nodes):
             message = f"duplicate {key!r} in unique index"
             raise ValueError(message)
         self._nodes.add(Node(key, row))
@@ -186,3 +186,10 @@ def __repr__(self):
             nodes = self._nodes
         nodes_str = ", ".join(str(node) for node in nodes)
         return f"<{self.__class__.__name__} nodes={nodes_str}>"
+
+    @property
+    def unique(self):
+        return self._unique
+
+    def __len__(self) -> int:
+        return len(self._nodes)



