diff --git a/sympy/tensor/array/expressions/from_array_to_matrix.py b/sympy/tensor/array/expressions/from_array_to_matrix.py
--- a/sympy/tensor/array/expressions/from_array_to_matrix.py
+++ b/sympy/tensor/array/expressions/from_array_to_matrix.py
@@ -377,10 +377,15 @@ def _(expr: ArrayTensorProduct):
             continue
         if not isinstance(arg, (MatrixExpr, MatrixBase)):
             rarg, rem = _remove_trivial_dims(arg)
-            removed.extend(rem)
-            newargs.append(rarg)
-            continue
-        elif getattr(arg, "is_Identity", False) and arg.shape == (1, 1):
+            rem_dims = [current_range[j] for j in rem]
+            removed.extend(rem_dims)
+            if not isinstance(rarg, MatrixExpr) or 1 not in rarg.shape:
+                newargs.append(rarg)
+                continue
+            else:
+                current_range = [j for j in current_range if j not in rem_dims]
+                arg = rarg
+        if getattr(arg, "is_Identity", False) and arg.shape == (1, 1):
             if arg.shape == (1, 1):
                 # Ignore identity matrices of shape (1, 1) - they are equivalent to scalar 1.
                 removed.extend(current_range)
