diff --git a/astropy/io/misc/pyarrow/csv.py b/astropy/io/misc/pyarrow/csv.py
--- a/astropy/io/misc/pyarrow/csv.py
+++ b/astropy/io/misc/pyarrow/csv.py
@@ -159,12 +159,13 @@ def read_csv(
         include_names, dtypes, null_values, timestamp_parsers
     )
 
-    table_pa = csv.read_csv(
-        input_file,
-        parse_options=parse_options,
-        read_options=read_options,
-        convert_options=convert_options,
-    )
+    with get_readable_fileobj(input_file, encoding="binary") as f:
+        table_pa = csv.read_csv(
+            f,
+            parse_options=parse_options,
+            read_options=read_options,
+            convert_options=convert_options,
+        )
     return convert_pa_table_to_astropy_table(table_pa)
 
 
