diff --git a/lib/matplotlib/offsetbox.py b/lib/matplotlib/offsetbox.py
--- a/lib/matplotlib/offsetbox.py
+++ b/lib/matplotlib/offsetbox.py
@@ -1505,6 +1505,7 @@
     def _picker(artist, mouseevent):
         # A custom picker to prevent dragging on mouse scroll events
         if mouseevent.name == "scroll_event":
+            return False, {}
 
     # A property, not an attribute, to maintain picklability.
     canvas = property(lambda self: self.ref_artist.get_figure(root=True).canvas)
