diff --git a/sklearn/linear_model/_coordinate_descent.py b/sklearn/linear_model/_coordinate_descent.py
--- a/sklearn/linear_model/_coordinate_descent.py
+++ b/sklearn/linear_model/_coordinate_descent.py
@@ -102,6 +102,8 @@ def _alpha_grid(
     eps=1e-3,
     n_alphas=100,
     sample_weight=None,
+    *,
+    positive: bool = False,
 ):
     """Compute the grid of alpha values for elastic net parameter search
 
@@ -140,6 +142,9 @@ def _alpha_grid(
 
     sample_weight : ndarray of shape (n_samples,), default=None
 
+    positive : bool, default=False
+        If set to True, forces coefficients to be positive.
+
     Returns
     -------
     np.ndarray
@@ -186,6 +191,10 @@ def _alpha_grid(
         n_samples = sample_weight.sum()
     else:
         n_samples = X.shape[0]
+
+    if positive:
+        Xyw = np.maximum(0, Xyw)
+
     # Compute np.max(np.sqrt(np.sum(Xyw**2, axis=1))). We switch sqrt and max to avoid
     # many computations of sqrt. This, however, needs an additional np.abs.
     alpha_max = np.sqrt(np.max(np.abs(np.sum(Xyw**2, axis=1)))) / (n_samples * l1_ratio)
@@ -642,6 +651,7 @@ def enet_path(
             Xy=Xy,
             l1_ratio=l1_ratio,
             fit_intercept=False,
+            positive=positive,
             eps=eps,
             n_alphas=n_alphas,
         )
@@ -1802,6 +1812,8 @@ def fit(self, X, y, sample_weight=None, **params):
                     y,
                     l1_ratio=l1_ratio,
                     fit_intercept=self.fit_intercept,
+                    # TODO: MultiTaskElasticNetCV has no attribute 'positive'
+                    positive=getattr(self, "positive", False),
                     eps=self.eps,
                     n_alphas=self._alphas,
                     sample_weight=sample_weight,
