diff --git a/sympy/codegen/numpy_nodes.py b/sympy/codegen/numpy_nodes.py
--- a/sympy/codegen/numpy_nodes.py
+++ b/sympy/codegen/numpy_nodes.py
@@ -2,8 +2,10 @@
 from sympy.core.power import Pow
 from sympy.core.singleton import S
 from sympy.core.sorting import default_sort_key
+from sympy.core.sympify import sympify
 from sympy.functions.elementary.exponential import exp, log
 from sympy.functions.elementary.miscellaneous import Max, Min
+from .ast import Token, none
 
 
 def _logaddexp(x1, x2, *, evaluate=True):
@@ -111,7 +113,7 @@ def _eval_simplify(self, *args, **kwargs):
             return logaddexp2(a, b)
 
 
-class amin(Function):
+class amin(Token):
     """ Minimum value along an axis.
 
     Helper class for use with e.g. numpy.amin
@@ -122,10 +124,12 @@ class amin(Function):
 
     https://numpy.org/doc/stable/reference/generated/numpy.amin.html
     """
-    nargs = 1
+    __slots__ = _fields = ('array', 'axis')
+    defaults = {'axis': none}
+    _construct_axis = staticmethod(sympify)
 
 
-class amax(Function):
+class amax(Token):
     """ Maximum value along an axis.
 
     Helper class for use with e.g. numpy.amax
@@ -136,7 +140,9 @@ class amax(Function):
 
     https://numpy.org/doc/stable/reference/generated/numpy.amax.html
     """
-    nargs = 1
+    __slots__ = _fields = ('array', 'axis')
+    defaults = {'axis': none}
+    _construct_axis = staticmethod(sympify)
 
 
 class maximum(Function):
diff --git a/sympy/printing/numpy.py b/sympy/printing/numpy.py
--- a/sympy/printing/numpy.py
+++ b/sympy/printing/numpy.py
@@ -230,8 +230,7 @@ def _print_Min(self, expr):
         return '{}({}.asarray([{}]), axis=0)'.format(self._module_format(self._module + '.amin'), self._module_format(self._module), ','.join(self._print(i) for i in expr.args))
 
     def _print_amin(self, expr):
-        inner, = expr.args
-        return '{}({})'.format(self._module_format(self._module + '.amin'), self._print(inner))
+        return '{}({}, axis={})'.format(self._module_format(self._module + '.amin'), self._print(expr.array), self._print(expr.axis))
 
     def _print_minimum(self, expr):
         op = self._module_format(self._module + '.minimum')
@@ -241,8 +240,7 @@ def _print_Max(self, expr):
         return '{}({}.asarray([{}]), axis=0)'.format(self._module_format(self._module + '.amax'), self._module_format(self._module), ','.join(self._print(i) for i in expr.args))
 
     def _print_amax(self, expr):
-        inner, = expr.args
-        return '{}({})'.format(self._module_format(self._module + '.amax'), self._print(inner))
+        return '{}({}, axis={})'.format(self._module_format(self._module + '.amax'), self._print(expr.array), self._print(expr.axis))
 
     def _print_maximum(self, expr):
         op = self._module_format(self._module + '.maximum')