diff --git a/sympy/vector/scalar.py b/sympy/vector/scalar.py
--- a/sympy/vector/scalar.py
+++ b/sympy/vector/scalar.py
@@ -1,6 +1,5 @@
 from sympy.core import AtomicExpr, Symbol, S
 from sympy.core.sympify import _sympify
-from sympy.printing.pretty.stringpict import prettyForm
 from sympy.printing.precedence import PRECEDENCE
 from sympy.core.kind import NumberKind
 
@@ -56,12 +55,6 @@
             return S.One
         return S.Zero
 
-    def _latex(self, printer=None):
-        return self._latex_form
-
-    def _pretty(self, printer=None):
-        return prettyForm(self._pretty_form)
-
     precedence = PRECEDENCE['Atom']
 
     @property
