diff --git a/django/db/models/expressions.py b/django/db/models/expressions.py
--- a/django/db/models/expressions.py
+++ b/django/db/models/expressions.py
@@ -1626,7 +1626,9 @@
         c = super().resolve_expression(query, allow_joins, reuse, summarize, for_save)
         if for_save and c.condition is not None:
             # Resolve condition with for_save=False, since it's used as a filter.
-            c.condition =
+            c.condition = self.condition.resolve_expression(
+                query, allow_joins, reuse, summarize, for_save=False
+            )
     def get_source_fields(self):
         # We're only interested in the fields of the result expressions.
         return [self.result._output_field_or_none]
