diff --git a/xarray/coding/calendar_ops.py b/xarray/coding/calendar_ops.py
--- a/xarray/coding/calendar_ops.py
+++ b/xarray/coding/calendar_ops.py
@@ -213,7 +213,7 @@ def convert_calendar(
         out[dim] = new_times
 
         # Remove NaN that where put on invalid dates in target calendar
-        out = out.sel(time=out[dim].notnull())
+        out = out.sel({:out[dim].notnull()})
 
         if use_cftime:
             # Reassign times to ensure time index of output is a CFTimeIndex