diff --git a/sympy/codegen/numpy_nodes.py b/sympy/codegen/numpy_nodes.py
--- a/sympy/codegen/numpy_nodes.py
+++ b/sympy/codegen/numpy_nodes.py
@@ -2,8 +2,10 @@
 from sympy.core.power import Pow
 from sympy.core.singleton import S
 from sympy.core.sorting import default_sort_key
+from sympy.core.sympify import sympify
 from sympy.functions.elementary.exponential import exp, log
 from sympy.functions.elementary.miscellaneous import Max, Min
+from .ast import Token, none
 
 
 def _logaddexp(x1, x2, *, evaluate=True):
