diff --git a/xarray/groupers.py b/xarray/groupers.py
--- a/xarray/groupers.py
+++ b/xarray/groupers.py
@@ -398,6 +398,8 @@
         if not isinstance(self.bins, pd.IntervalIndex):
             dummy, _ = self._cut(np.array([0]).astype(group.dtype))
             full_index = dummy.categories
+        else:
+            full_index = pd.Index(self.labels)
 
         if not by_is_chunked:
             uniques = np.sort(pd.unique(codes.data.ravel()))
