diff --git a/django/db/models/base.py b/django/db/models/base.py
--- a/django/db/models/base.py
+++ b/django/db/models/base.py
@@ -1176,7 +1176,8 @@
             for field in insert_fields:
                 value = (
                     getattr(self, field.attname)
-                    if raw else field.pre_save(self, add=True)
+                    if raw
+                    else field.pre_save(self, add=True)
                 )
                 if hasattr(value, "resolve_expression"):
                     if field not in returning_fields:
