diff --git a/sympy/vector/coordsysrect.py b/sympy/vector/coordsysrect.py
--- a/sympy/vector/coordsysrect.py
+++ b/sympy/vector/coordsysrect.py
@@ -248,6 +248,31 @@
         obj._parent_rotation_matrix = rotation_matrix
         obj._origin = origin
 
+        # compute the transformation matrix from this coordinate system
+        # to the parent
+        if parent is not None:
+            u = (x1, x2, x3)
+            X = Matrix(obj._transformation_lambda(*u))
+            h = obj._lame_coefficients
+            J = X.jacobian(u)
+
+            # detects coordinate systems like spherical/cylindrical,
+            # where the Jacobian depends on the position
+            is_curvilinear = J.has(BaseScalar)
+            if is_curvilinear:
+                # normalized jacobian
+                J = Matrix([(J.col(i) / h[i]).T for i in range(3)]).T
+
+            T_to_parent = J
+        else:
+            is_curvilinear = False
+            T_to_parent = eye(3)
+
+        obj._transformation_matrix_to_parent = T_to_parent
+        obj._is_cartesian = (obj._lame_coefficients == (S.One, S.One, S.One))
+        obj._is_curvilinear = is_curvilinear
+
+
         # Return the instance
         return obj
 
