diff --git a/sympy/vector/coordsysrect.py b/sympy/vector/coordsysrect.py
--- a/sympy/vector/coordsysrect.py
+++ b/sympy/vector/coordsysrect.py
@@ -16,6 +16,7 @@
 from sympy.matrices.dense import eye
 from sympy.matrices.immutable import ImmutableDenseMatrix
 from sympy.simplify.simplify import simplify
+from sympy.simplify.fu import TR5
 from sympy.simplify.trigsimp import trigsimp
 import sympy.vector
 from sympy.vector.orienters import (Orienter, AxisOrienter, BodyOrienter,
@@ -510,25 +511,74 @@
 
         A SymPy Matrix is returned.
 
+        Note that rotation_matrix only consider pure rotations between
+        Cartesian systems. It doesn't account for changes in base vectors,
+        like curvilinear to Cartesian. Use change_of_basis_matrix_from if
+        those are important.
+
         Parameters
         ==========
 
         other : CoordSys3D
             The system which the DCM is generated to.
 
+        See Also
+        ========
+
+        CoordSys3D.change_of_basis_matrix_from
+
         Examples
         ========
 
         >>> from sympy.vector import CoordSys3D
         >>> from sympy import symbols
+
+        Let's consider two Cartesian systems connected by rotations:
+
         >>> q1 = symbols('q1')
         >>> N = CoordSys3D('N')
         >>> A = N.orient_new_axis('A', q1, N.i)
-        >>> N.rotation_matrix(A)
+
+        The rotation matrix from A to N is:
+
+        >>> R_fromA_toN = N.rotation_matrix(A)
+        >>> R_fromA_toN
         Matrix([
         [1,       0,        0],
         [0, cos(q1), -sin(q1)],
         [0, sin(q1),  cos(q1)]])
+
+        Because the two systems are Cartesians and connected by pure rotation,
+        the rotation matrix is equal to the change-of-basis matrix:
+
+        >>> R_fromA_toN == N.change_of_basis_matrix_from(A)
+        True
+
+        This is generally not true. Specifically if a non-Cartesian coordinate
+        system (like curvilinear system, or a scaled system, or a
+        reflected system, etc.) is in the path of two connected systems,
+        then the two matrices are different, because the one computed by
+        change_of_basis_matrix_from considers both the change in base vectors
+        and the orientation between the systems, whereas the one computed by
+        rotation_matrix only consider the orientation between Cartesian
+        systems. For example:
+
+        >>> B = A.create_new("B", transformation=lambda x,y,z: (2*x, z, y))
+        >>> C = B.orient_new_axis("C", q1, B.i)
+        >>> R_fromC_toN = N.rotation_matrix(C).simplify()
+        >>> R_fromC_toN
+        Matrix([
+        [1,         0,          0],
+        [0, cos(2*q1), -sin(2*q1)],
+        [0, sin(2*q1),  cos(2*q1)]])
+        >>> T_fromC_toN = N.change_of_basis_matrix_from(C).simplify()
+        >>> T_fromC_toN
+        Matrix([
+        [2, 0, 0],
+        [0, 0, 1],
+        [0, 1, 0]])
+
+        Hence, the use of change_of_basis_matrix_from is recommended.
 
         """
         from sympy.vector.functions import _path
@@ -656,6 +706,8 @@
         for i in range(rootindex + 1, len(path)):
             result *= path[i]._transformation_matrix_to_parent
 
+        # attempts to cancel out opposite rotations
+        result = TR5(result)
         return result
 
     @cacheit
