diff --git a/sympy/vector/coordsysrect.py b/sympy/vector/coordsysrect.py
--- a/sympy/vector/coordsysrect.py
+++ b/sympy/vector/coordsysrect.py
@@ -2,9 +2,10 @@
 
 from sympy.core.basic import Basic
 from sympy.core.cache import cacheit
-from sympy.core import S, Dummy, Lambda
+from sympy.core import S, Dummy, Lambda, factor_terms
 from sympy.core.symbol import Str
 from sympy.core.symbol import symbols
+from sympy.functions.elementary.trigonometric import TrigonometricFunction
 from sympy.matrices.immutable import ImmutableDenseMatrix as Matrix
 from sympy.matrices.matrixbase import MatrixBase
 from sympy.solvers import solve
@@ -273,7 +274,6 @@
         obj._is_cartesian = (obj._lame_coefficients == (S.One, S.One, S.One))
         obj._is_curvilinear = is_curvilinear
 
-
         # Return the instance
         return obj
 
@@ -791,8 +791,21 @@
             sx, sy, sz = system.base_scalars()
             current = {sx: xs, sy: ys, sz: zs}
 
+        def post_process(expr):
+            # attempts to cancel out opposite rotations
+            def pattern(t):
+                return (
+                    t.is_Mul
+                    and any(isinstance(a, TrigonometricFunction) for a in t.args)
+                    and any(a.is_Add and a.has(TrigonometricFunction) for a in t.args)
+                )
+
+            if not expr.find(pattern):
+                return expr
+            return TR5(factor_terms(expr.expand()))
+
+        current = {k: post_process(v) for k, v in current.items()}
         return current
-
 
     def locate_new(self, name, position, vector_names=None,
                    variable_names=None):
