diff --git a/sympy/vector/coordsysrect.py b/sympy/vector/coordsysrect.py
--- a/sympy/vector/coordsysrect.py
+++ b/sympy/vector/coordsysrect.py
@@ -513,8 +513,8 @@ def rotation_matrix(self, other):
 
         Note that rotation_matrix only consider pure rotations between
         Cartesian systems. It doesn't account for changes in base vectors,
-        like curvilinear to Cartesian. Use transformation_matrix if those
-        are important.
+        like curvilinear to Cartesian. Use transformation_matrix_from if
+        those are important.
 
         Parameters
         ==========
@@ -525,7 +525,7 @@ def rotation_matrix(self, other):
         See Also
         ========
 
-        CoordSys3D.transformation_matrix
+        CoordSys3D.transformation_matrix_from
 
         Examples
         ========
@@ -551,14 +551,17 @@ def rotation_matrix(self, other):
         Because the two systems are Cartesians and connected by pure rotation,
         the rotation matrix is equal to the transformation matrix:
 
-        >>> R_fromA_toN == N.transformation_matrix(A)
+        >>> R_fromA_toN == N.transformation_matrix_from(A)
         True
 
         This is generally not true. Specifically if a non-Cartesian coordinate
         system (like curvilinear system, or a scaled system, or a
         reflected system, etc.) is in the path of two connected systems,
-        then the matrix computed by transformation_matrix is correct,
-        whereas the one computed by rotation_matrix is wrong. For example:
+        then the two matrices are different, because the one computed by
+        transformation_matrix_from considers both the change in base vectors
+        and the orientation between the systems, whereas the one computed by
+        rotation_matrix only consider the orientation between Cartesian
+        systems. For example:
 
         >>> B = A.create_new("B", transformation=lambda x,y,z: (2*x, z, y))
         >>> C = B.orient_new_axis("C", q1, B.i)
@@ -568,14 +571,14 @@ def rotation_matrix(self, other):
         [1,         0,          0],
         [0, cos(2*q1), -sin(2*q1)],
         [0, sin(2*q1),  cos(2*q1)]])
-        >>> T_fromC_toN = N.transformation_matrix(C).simplify()
+        >>> T_fromC_toN = N.transformation_matrix_from(C).simplify()
         >>> T_fromC_toN
         Matrix([
         [2, 0, 0],
         [0, 0, 1],
         [0, 1, 0]])
 
-        Hence, the use of transformation_matrix is recommended.
+        Hence, the use of transformation_matrix_from is recommended.
 
         """
         from sympy.vector.functions import _path
@@ -598,15 +601,15 @@ def rotation_matrix(self, other):
             result *= path[i]._parent_rotation_matrix.T
         return result
 
-    def transformation_matrix(self, other):
+    def transformation_matrix_from(self, other):
         """
         Returns the transformation matrix of this coordinate system with
         respect to another system, which takes into account rotation
         and scaling.
 
         If v_a is a vector defined in system 'A' (in matrix format)
         and v_b is the same vector defined in system 'B', then
-        v_a = A.transformation_matrix(B) * v_b.
+        v_a = A.transformation_matrix_from(B) * v_b.
 
         A SymPy Matrix is returned.
 
@@ -635,7 +638,7 @@ def transformation_matrix(self, other):
 
         The transformation matrix from A to N is:
 
-        >>> T_fromA_toN = N.transformation_matrix(A)
+        >>> T_fromA_toN = N.transformation_matrix_from(A)
         >>> T_fromA_toN
         Matrix([
         [1,       0,        0],
@@ -665,7 +668,7 @@ def transformation_matrix(self, other):
         >>> Cart = CoordSys3D("Cart")
         >>> S = Cart.create_new("S", transformation="spherical")
         >>> C = Cart.create_new("C", transformation="cylindrical")
-        >>> Cart.transformation_matrix(S)
+        >>> Cart.transformation_matrix_from(S)
         Matrix([
         [sin(S.theta)*cos(S.phi), cos(S.phi)*cos(S.theta), -sin(S.phi)],
         [sin(S.phi)*sin(S.theta), sin(S.phi)*cos(S.theta),  cos(S.phi)],
@@ -676,7 +679,7 @@ def transformation_matrix(self, other):
 
         >>> r_s, theta_s, phi_s = S.base_scalars()
         >>> r_c, theta_c, z_c = C.base_scalars()
-        >>> C.transformation_matrix(S).subs(phi_s, theta_c).simplify()
+        >>> C.transformation_matrix_from(S).subs(phi_s, theta_c).simplify()
         Matrix([
         [sin(S.theta),  cos(S.theta), 0],
         [           0,             0, 1],
