diff --git a/sklearn/linear_model/_logistic.py b/sklearn/linear_model/_logistic.py
--- a/sklearn/linear_model/_logistic.py
+++ b/sklearn/linear_model/_logistic.py
@@ -728,7 +728,7 @@ def calc_score(log_reg):
         # We need to pass the classes as "labels" argument to scorers that support
         # it, e.g. scoring = "neg_brier_score", because y_test may not contain all
         # class labels.
-        # There now 2 routes:
+        # There are 2 routes:
         #  - metadata routing is enabled: A try except clause is possible with
         #    adding labels to score_params.
         # - metadata routing is disabled: We have to reconstruct the scorer and
