diff --git a/sympy/algebras/quaternion.py b/sympy/algebras/quaternion.py
--- a/sympy/algebras/quaternion.py
+++ b/sympy/algebras/quaternion.py
@@ -1039,9 +1039,6 @@
         if q_norm.is_zero:
             raise ValueError("Cannot compute logarithm for a zero quaternion")
         a = ln(q_norm)
-        b = q.b * acos(q.a / q_norm) / vector_norm
-        c = q.c * acos(q.a / q_norm) / vector_norm
-        d = q.d * acos(q.a / q_norm) / vector_norm
 
         return Quaternion(a, b, c, d)
 
