diff --git a/xarray/core/groupby.py b/xarray/core/groupby.py
--- a/xarray/core/groupby.py
+++ b/xarray/core/groupby.py
@@ -1368,22 +1368,9 @@
             return self._obj
         if keep_attrs is None:
             keep_attrs = _get_keep_attrs(default=True)
-        if (
-            module_available("flox", minversion="0.9.16")
-            and OPTIONS["use_flox"]
-            and contains_only_chunked_or_numpy(self._obj)
-        ):
-            result, *_ = self._flox_reduce(
-                dim=None, func=op, skipna=skipna, keep_attrs=keep_attrs
-            )
-        else:
-            result = self.reduce(
-                getattr(duck_array_ops, op),
-                dim=[self._group_dim],
-                skipna=skipna,
-                keep_attrs=keep_attrs,
-            )
-        return result
+        return self.reduce(
+            op, dim=[self._group_dim], skipna=skipna, keep_attrs=keep_attrs
+        )
 
     def first(self, skipna: bool | None = None, keep_attrs: bool | None = None):
         """Return the first element of each group along the group dimension"""
