diff --git a/galleries/examples/specialty_plots/skewt.py b/galleries/examples/specialty_plots/skewt.py
--- a/galleries/examples/specialty_plots/skewt.py
+++ b/galleries/examples/specialty_plots/skewt.py
@@ -34,9 +34,9 @@ def draw(self, renderer):
             for artist in [self.gridline, self.tick1line, self.tick2line,
                            self.label1, self.label2]:
                 stack.callback(artist.set_visible, artist.get_visible())
-            needs_lower = transforms.interval_contains(
+            needs_lower = transforms._interval_contains(
                 self.axes.lower_xlim, self.get_loc())
-            needs_upper = transforms.interval_contains(
+            needs_upper = transforms._interval_contains(
                 self.axes.upper_xlim, self.get_loc())
             self.tick1line.set_visible(
                 self.tick1line.get_visible() and needs_lower)
