diff --git a/sympy/ntheory/elliptic_curve.py b/sympy/ntheory/elliptic_curve.py
--- a/sympy/ntheory/elliptic_curve.py
+++ b/sympy/ntheory/elliptic_curve.py
@@ -353,7 +353,7 @@ def __rmul__(self, n):
 
     def __neg__(self):
         if self.z == 0:
-            return self
+            return self.point_at_infinity(self._curve)
         return EllipticCurvePoint(self.x, -self.y - self._curve._a1*self.x - self._curve._a3, self.z, self._curve)
 
     def __repr__(self):
