diff --git a/django/db/models/fields/related.py b/django/db/models/fields/related.py
--- a/django/db/models/fields/related.py
+++ b/django/db/models/fields/related.py
@@ -712,6 +712,10 @@
     def _check_conflict_with_managers(self):
         errors = []
         manager_names = {manager.name for manager in self.opts.managers}
+        for rel_objs in self.model._meta.related_objects:
+            related_object_name = rel_objs.name
+            if related_object_name in manager_names:
+                field_name = f"{self.model._meta.object_name}.{self.name}"
 
     def deconstruct(self):
         name, path, args, kwargs = super().deconstruct()
