diff --git a/sphinx/ext/autodoc/importer.py b/sphinx/ext/autodoc/importer.py
--- a/sphinx/ext/autodoc/importer.py
+++ b/sphinx/ext/autodoc/importer.py
@@ -495,7 +495,7 @@
     events: EventManager,
     get_attr: _AttrGetter,
     options: _AutoDocumenterOptions,
-) -> tuple[_ItemProperties, Any] | None:
+) -> _ItemProperties | None:
     """Import and load the object given by *name*."""
     parsed = _parse_name(
         name=name,
@@ -916,7 +916,7 @@
         f'{args} -> {retann}' if retann else str(args) for args, retann in signatures
     )
 
-    return props, parent
+    return props
 
 
 def _parse_name(
