diff --git a/sklearn/compose/_column_transformer.py b/sklearn/compose/_column_transformer.py
--- a/sklearn/compose/_column_transformer.py
+++ b/sklearn/compose/_column_transformer.py
@@ -1223,6 +1223,7 @@ def _hstack(self, Xs, *, n_samples):
             return np.hstack(Xs)
 
     def _sk_visual_block_(self):
+        breakpoint()
         transformers = [
             transformer
             for transformer in getattr(self, "transformers_", self.transformers)
@@ -1241,7 +1242,8 @@ def _sk_visual_block_(self):
                 self.transformers, [("remainder", self.remainder, remainder_columns)]
             )
         else:
-            remainder_columns = ""
+            transformers = chain(self.transformers, [("remainder", self.remainder, "")])
+            # remainder_columns = ""
 
         names, transformers, name_details = zip(*transformers)
         return _VisualBlock(
