diff --git a/astropy/wcs/wcsapi/fitswcs.py b/astropy/wcs/wcsapi/fitswcs.py
--- a/astropy/wcs/wcsapi/fitswcs.py
+++ b/astropy/wcs/wcsapi/fitswcs.py
@@ -592,18 +592,24 @@ def redshift_from_spectralcoord(spectralcoord):
                         or spectralcoord.observer is None
                         or spectralcoord.target is None
                     ):
-                        if observer is None:
-                            msg = "No observer defined on WCS"
-                        elif spectralcoord.observer is None:
-                            msg = "No observer defined on SpectralCoord"
-                        else:
-                            msg = "No target defined on SpectralCoord"
-                        warnings.warn(
-                            f"{msg}, SpectralCoord "
-                            "will be converted without any velocity "
-                            "frame change",
-                            AstropyUserWarning,
-                        )
+                        # Note that if observer and spectralcoord.observer are
+                        # both None, we don't emit any warning since this is
+                        # likely to be fine and is a very common use case. It
+                        # is more important to emit a warning if one has
+                        # observer information and the other does not.
+                        if not (observer is None and spectralcoord.observer is None):
+                            if observer is None:
+                                msg = "No observer defined on WCS"
+                            elif spectralcoord.observer is None:
+                                msg = "No observer defined on SpectralCoord"
+                            else:
+                                msg = "No target defined on SpectralCoord"
+                            warnings.warn(
+                                f"{msg}, SpectralCoord "
+                                "will be converted without any velocity "
+                                "frame change",
+                                AstropyUserWarning,
+                            )
                         return spectralcoord.to_value(u.m) / self.wcs.restwav - 1.0
                     else:
                         return (
@@ -640,18 +646,24 @@ def beta_from_spectralcoord(spectralcoord):
                         or spectralcoord.observer is None
                         or spectralcoord.target is None
                     ):
-                        if observer is None:
-                            msg = "No observer defined on WCS"
-                        elif spectralcoord.observer is None:
-                            msg = "No observer defined on SpectralCoord"
-                        else:
-                            msg = "No target defined on SpectralCoord"
-                        warnings.warn(
-                            f"{msg}, SpectralCoord "
-                            "will be converted without any velocity "
-                            "frame change",
-                            AstropyUserWarning,
-                        )
+                        # Note that if observer and spectralcoord.observer are
+                        # both None, we don't emit any warning since this is
+                        # likely to be fine and is a very common use case. It
+                        # is more important to emit a warning if one has
+                        # observer information and the other does not.
+                        if not (observer is None and spectralcoord.observer is None):
+                            if observer is None:
+                                msg = "No observer defined on WCS"
+                            elif spectralcoord.observer is None:
+                                msg = "No observer defined on SpectralCoord"
+                            else:
+                                msg = "No target defined on SpectralCoord"
+                            warnings.warn(
+                                f"{msg}, SpectralCoord "
+                                "will be converted without any velocity "
+                                "frame change",
+                                AstropyUserWarning,
+                            )
                         return spectralcoord.to_value(u.m / u.s, doppler_equiv) / C_SI
                     else:
                         return (
@@ -723,18 +735,24 @@ def value_from_spectralcoord(spectralcoord):
                         or spectralcoord.observer is None
                         or spectralcoord.target is None
                     ):
-                        if observer is None:
-                            msg = "No observer defined on WCS"
-                        elif spectralcoord.observer is None:
-                            msg = "No observer defined on SpectralCoord"
-                        else:
-                            msg = "No target defined on SpectralCoord"
-                        warnings.warn(
-                            f"{msg}, SpectralCoord "
-                            "will be converted without any velocity "
-                            "frame change",
-                            AstropyUserWarning,
-                        )
+                        # Note that if observer and spectralcoord.observer are
+                        # both None, we don't emit any warning since this is
+                        # likely to be fine and is a very common use case. It
+                        # is more important to emit a warning if one has
+                        # observer information and the other does not.
+                        if not (observer is None and spectralcoord.observer is None):
+                            if observer is None:
+                                msg = "No observer defined on WCS"
+                            elif spectralcoord.observer is None:
+                                msg = "No observer defined on SpectralCoord"
+                            else:
+                                msg = "No target defined on SpectralCoord"
+                            warnings.warn(
+                                f"{msg}, SpectralCoord "
+                                "will be converted without any velocity "
+                                "frame change",
+                                AstropyUserWarning,
+                            )
                         return spectralcoord.to_value(**kwargs)
                     else:
                         return spectralcoord.with_observer_stationary_relative_to(