diff --git a/lib/matplotlib/path.py b/lib/matplotlib/path.py
--- a/lib/matplotlib/path.py
+++ b/lib/matplotlib/path.py
@@ -290,7 +290,21 @@
         p._readonly = False
         return p
 
-    deepcopy = __deepcopy__
+    def deepcopy(self, memo=None):
+        """
+        Return a deep copy of the `Path`.  The `Path` will not be readonly,
+        even if the source `Path` is.
+
+        Parameters
+        ----------
+        memo : dict, optional
+            A dictionary to use for memoizing, passed to `copy.deepcopy`.
+
+        Returns
+        -------
+        Path
+            A deep copy of the `Path`, but not readonly.
+        """
 
     @classmethod
     def make_compound_path_from_polys(cls, XY):
