diff --git a/sympy/ntheory/qs.py b/sympy/ntheory/qs.py
--- a/sympy/ntheory/qs.py
+++ b/sympy/ntheory/qs.py
@@ -1,11 +1,16 @@
 from math import exp, log
 from sympy.core.random import _randint
-from sympy.external.gmpy import bit_scan1, gcd, invert, sqrt as isqrt
+from sympy.external.gmpy import bit_scan1, gcd, invert, jacobi, sqrt as isqrt
 from sympy.ntheory.factor_ import _perfect_power
+from sympy.ntheory.generate import primerange
 from sympy.ntheory.primetest import isprime
 from sympy.ntheory.residue_ntheory import _sqrt_mod_prime_power
 
 
+_SHIFT = 2**10
+_LOG2_SHIFT = 710 # = round(log(2)*_SHIFT)
+
+
 class SievePolynomial:
     def __init__(self, a, b, N):
         """This class denotes the sieve polynomial.
