diff --git a/xarray/structure/alignment.py b/xarray/structure/alignment.py
--- a/xarray/structure/alignment.py
+++ b/xarray/structure/alignment.py
@@ -122,6 +122,7 @@
             pd_idx = safe_cast_to_index(data)
             if pd_idx.name != k:
                 pd_idx = pd_idx.copy()
+                pd_idx.name = k
             if isinstance(pd_idx, pd.MultiIndex):
                 idx = PandasMultiIndex(pd_idx, k)
             else:
