diff --git a/sklearn/metrics/_regression.py b/sklearn/metrics/_regression.py
--- a/sklearn/metrics/_regression.py
+++ b/sklearn/metrics/_regression.py
@@ -22,6 +22,7 @@
     _median,
     get_namespace,
     get_namespace_and_device,
+    move_to,
     size,
 )
 from sklearn.utils._array_api import _xlogy as xlogy
@@ -1345,7 +1346,8 @@
     >>> max_error(y_true, y_pred)
     1.0
     """
-    xp, _ = get_namespace(y_true, y_pred)
+    xp, _, device = get_namespace_and_device(y_pred)
+    y_true = move_to(y_true, xp=xp, device=device)
     y_type, y_true, y_pred, _, _ = _check_reg_targets(
         y_true, y_pred, sample_weight=None, multioutput=None, xp=xp
     )
