diff --git a/django/db/models/base.py b/django/db/models/base.py
--- a/django/db/models/base.py
+++ b/django/db/models/base.py
@@ -1148,18 +1148,18 @@
                 if not f.generated and (pk_set or f is not meta.auto_field)
             ]
             returning_fields = list(meta.db_returning_fields)
-            can_return_columns_from_insert =
-            for field in fields:
+            can_return_columns_from_insert = connections[
+            for field in insert_fields:
                 value = (
                     getattr(self, field.attname) if raw else field.pre_save(self, False)
                 )
                 if hasattr(value, "resolve_expression"):
                     if field not in returning_fields:
                         returning_fields.append(field)
                 elif field.db_returning:
                     returning_fields.remove(field)
             results = self._do_insert(
-                cls._base_manager, using, fields, returning_fields, raw
+                cls._base_manager, using, insert_fields, returning_fields, raw
             )
             if results:
                 self._assign_returned_values(results[0], returning_fields)
