diff --git a/django/db/models/base.py b/django/db/models/base.py
--- a/django/db/models/base.py
+++ b/django/db/models/base.py
@@ -1161,7 +1161,8 @@
                 elif (
                     field.db_returning
                     and not can_return_columns_from_insert
-                    and
+                    and not (pk_set and field is meta.auto_field)
+                ):
                     returning_fields.remove(field)
             results = self._do_insert(
                 cls._base_manager, using, insert_fields, returning_fields, raw
