diff --git a/sklearn/linear_model/_ridge.py b/sklearn/linear_model/_ridge.py
--- a/sklearn/linear_model/_ridge.py
+++ b/sklearn/linear_model/_ridge.py
@@ -60,7 +60,6 @@
     _routing_enabled,
     process_routing,
 )
-from sklearn.utils.sparsefuncs import mean_variance_axis
 from sklearn.utils.validation import (
     _check_sample_weight,
     check_is_fitted,
@@ -1837,105 +1836,92 @@
             D = D[(slice(None),) + (None,) * (len(B.shape) - 1)]
         return D * B
 
-    def _compute_gram(self, X, sqrt_sw):
-        """Computes the Gram matrix XX^T with possible centering.
+    def _compute_gram(self, X, X_mean, sqrt_sw):
+        """Computes the Gram matrix X X' with possible centering.
 
         Parameters
         ----------
-        X : {ndarray, sparse matrix} of shape (n_samples, n_features)
+        X : {ndarray, sparse matrix, sparse array} of shape (n_samples, n_features)
             The preprocessed design matrix.
 
+        X_mean : ndarray of shape (n_feature,)
+            The weighted mean of X for each feature.
+
         sqrt_sw : ndarray of shape (n_samples,)
-            square roots of sample weights
+            Square roots of sample weights.
 
         Returns
         -------
         gram : ndarray of shape (n_samples, n_samples)
             The Gram matrix.
-        X_mean : ndarray of shape (n_feature,)
-            The weighted mean of ``X`` for each feature.
 
         Notes
         -----
+        When self.fit_intercept is False no centering is done.
+
         When X is dense the centering has been done in preprocessing
-        so the mean is 0 and we just compute XX^T.
-
-        When X is sparse it has not been centered in preprocessing, but it has
-        been scaled by sqrt(sample weights).
-
-        When self.fit_intercept is False no centering is done.
-
-        The centered X is never actually computed because centering would break
-        the sparsity of X.
+        so the mean is 0 and we just compute X X'.
+
+        When X is sparse it has not been centered in preprocessing, but
+        it has been scaled by sqrt_sw. The centered X is never actually
+        computed because centering would break the sparsity of X.
         """
-        xp, _ = get_namespace(X)
         center = self.fit_intercept and sparse.issparse(X)
         if not center:
             # in this case centering has been done in preprocessing
             # or we are not fitting an intercept.
-            X_mean = xp.zeros(X.shape[1], dtype=X.dtype)
-            return safe_sparse_dot(X, X.T, dense_output=True), X_mean
-        # X is sparse
-        n_samples = X.shape[0]
-        sample_weight_matrix = sparse.dia_matrix(
-            (sqrt_sw, 0), shape=(n_samples, n_samples)
+            return safe_sparse_dot(X, X.T, dense_output=True)
+        # X is sparse and fit_intercept is True
+        # centered matrix = X - sqrt_sw X_mean'
+        X_Xm = safe_sparse_dot(X, X_mean, dense_output=True)
+        return (
+            safe_sparse_dot(X, X.T, dense_output=True)
+            - X_Xm[:, None] * sqrt_sw[None, :]
+            - sqrt_sw[:, None] * X_Xm[None, :]
+            + (X_mean @ X_mean) * sqrt_sw[:, None] * sqrt_sw[None, :]
         )
-        X_weighted = sample_weight_matrix.dot(X)
-        X_mean, _ = mean_variance_axis(X_weighted, axis=0)
-        X_mean *= n_samples / sqrt_sw.dot(sqrt_sw)
-        X_mX = sqrt_sw[:, None] * safe_sparse_dot(X_mean, X.T, dense_output=True)
-        X_mX_m = np.outer(sqrt_sw, sqrt_sw) * np.dot(X_mean, X_mean)
-        return (
-            safe_sparse_dot(X, X.T, dense_output=True) + X_mX_m - X_mX - X_mX.T,
-            X_mean,
-        )
-
-    def _compute_covariance(self, X, sqrt_sw):
-        """Computes covariance matrix X^TX with possible centering.
+
+    def _compute_covariance(self, X, X_mean, sqrt_sw):
+        """Computes covariance matrix X' X with possible centering.
 
         Parameters
         ----------
-        X : sparse matrix of shape (n_samples, n_features)
+        X : {ndarray, sparse matrix, sparse array} of shape (n_samples, n_features)
             The preprocessed design matrix.
 
+        X_mean : ndarray of shape (n_feature,)
+            The weighted mean of X for each feature.
+
         sqrt_sw : ndarray of shape (n_samples,)
-            square roots of sample weights
+            Square roots of sample weights.
 
         Returns
         -------
         covariance : ndarray of shape (n_features, n_features)
             The covariance matrix.
-        X_mean : ndarray of shape (n_feature,)
-            The weighted mean of ``X`` for each feature.
 
         Notes
         -----
-        Since X is sparse it has not been centered in preprocessing, but it has
-        been scaled by sqrt(sample weights).
-
         When self.fit_intercept is False no centering is done.
 
-        The centered X is never actually computed because centering would break
-        the sparsity of X.
+        When X is dense the centering has been done in preprocessing
+        so the mean is 0 and we just compute X' X.
+
+        When X is sparse it has not been centered in preprocessing, but
+        it has been scaled by sqrt_sw. The centered X is never actually
+        computed because centering would break the sparsity of X.
         """
-        if not self.fit_intercept:
+        center = self.fit_intercept and sparse.issparse(X)
+        if not center:
             # in this case centering has been done in preprocessing
             # or we are not fitting an intercept.
-            X_mean = np.zeros(X.shape[1], dtype=X.dtype)
-            return safe_sparse_dot(X.T, X, dense_output=True), X_mean
-        # this function only gets called for sparse X
-        n_samples = X.shape[0]
-        sample_weight_matrix = sparse.dia_matrix(
-            (sqrt_sw, 0), shape=(n_samples, n_samples)
-        )
-        X_weighted = sample_weight_matrix.dot(X)
-        X_mean, _ = mean_variance_axis(X_weighted, axis=0)
-        X_mean = X_mean * n_samples / sqrt_sw.dot(sqrt_sw)
-        weight_sum = sqrt_sw.dot(sqrt_sw)
+            return safe_sparse_dot(X.T, X, dense_output=True)
+        # X is sparse and fit_intercept is True
+        # centered matrix = X - sqrt_sw X_mean'
+        sw_sum = sqrt_sw @ sqrt_sw
         return (
             safe_sparse_dot(X.T, X, dense_output=True)
-            - weight_sum * np.outer(X_mean, X_mean),
-            X_mean,
+            - sw_sum * X_mean[:, None] * X_mean[None, :]
         )
 
     def _sparse_multidot_diag(self, X, A, X_mean, sqrt_sw):
