diff --git a/sklearn/linear_model/_ridge.py b/sklearn/linear_model/_ridge.py
--- a/sklearn/linear_model/_ridge.py
+++ b/sklearn/linear_model/_ridge.py
@@ -2099,65 +2099,72 @@
         coef = Hinv_XT_y
         return looe, coef
 
-    def _svd_decompose_design_matrix(self, X, y, sqrt_sw):
-        xp, _, device_ = get_namespace_and_device(X)
-        # X already centered
-        X_mean = xp.zeros(X.shape[1], dtype=X.dtype, device=device_)
-        if self.fit_intercept:
-            # to emulate fit_intercept=True situation, add a column
-            # containing the square roots of the sample weights
-            # by centering, the other columns are orthogonal to that one
-            intercept_column = sqrt_sw[:, None]
-            X = xp.concat((X, intercept_column), axis=1)
+    def _svd_decompose_design_matrix(self, X, X_mean, y, sqrt_sw):
+        """Reduced SVD decomposition of X"""
+        xp, _ = get_namespace(X)
         # reduced svd
         U, singvals, VT = xp.linalg.svd(X, full_matrices=False)
         UT_y = U.T @ y
+        UT_sqrt_sw = U.T @ sqrt_sw
         V = VT.T
-        return X_mean, singvals, U, V, UT_y
-
-    def _solve_svd_design_matrix(self, alpha, y, sqrt_sw, X_mean, singvals, U, V, UT_y):
-        """Compute dual coefficients and diagonal of G^-1.
-
-        Used when we have an SVD decomposition of X
-        (n_samples > n_features and X is dense).
+        return singvals, U, V, UT_y, UT_sqrt_sw
+
+    def _solve_svd_design_matrix_long(
+        self, alpha, y, sqrt_sw, singvals, U, V, UT_y, UT_sqrt_sw
+    ):
+        """Compute looe and coef when we have an SVD decomposition of X.
+
+        Long X case (n_features < n_samples).
         """
-        xp, is_array_api = get_namespace(U)
+        M = alpha / (singvals**2 + alpha) - 1
+        alpha_c = U @ self._diag_dot(M, UT_y) + y
+        alpha_d = self._decomp_diag(M, U) + 1
         if self.fit_intercept:
-            # detect intercept column
-            normalized_sw = sqrt_sw / xp.linalg.vector_norm(sqrt_sw)
-            intercept_dim = int(_find_smallest_angle(normalized_sw, U))
-
-        n, r = U.shape
-        p, r = V.shape
-        if p < n:
-            assert p == r == len(singvals)
-            w = alpha / (singvals**2 + alpha) - 1
-            if self.fit_intercept:
-                # cancel the regularization for the intercept
-                w[intercept_dim] = -1
-            alpha_c = U @ self._diag_dot(w, UT_y) + y
-            alpha_d = self._decomp_diag(w, U) + 1
-        else:
-            assert n == r == len(singvals)
-            g = alpha / (singvals**2 + alpha)
-            if self.fit_intercept:
-                # cancel the regularization for the intercept
-                g[intercept_dim] = 0
-            alpha_c = U @ self._diag_dot(g, UT_y)
-            alpha_d = self._decomp_diag(g, U)
-
-        if len(y.shape) != 1:
+            sw_sum = sqrt_sw @ sqrt_sw
+            alpha_Ginv_sqrt_sw = U @ self._diag_dot(M, UT_sqrt_sw) + sqrt_sw
+            alpha_d -= alpha_Ginv_sqrt_sw * sqrt_sw / sw_sum
+        if y.ndim == 2:
             # handle case where y is 2-d
             alpha_d = alpha_d[:, None]
-
-        # coefficient and leave-one-out-errors
         looe = alpha_c / alpha_d
-        h = singvals / (singvals**2 + alpha)
-        coef = V @ self._diag_dot(h, UT_y)
+        coef = V @ self._diag_dot(singvals / (singvals**2 + alpha), UT_y)
+        return looe, coef
+
+    def _solve_svd_design_matrix_wide(
+        self, alpha, y, sqrt_sw, singvals, U, V, UT_y, UT_sqrt_sw
+    ):
+        """Compute looe and coef when we have an SVD decomposition of X.
+
+        Wide X case (n_samples < n_features).
+        """
+        alpha_D = alpha / (singvals**2 + alpha)
+        alpha_c = U @ self._diag_dot(alpha_D, UT_y)
+        alpha_d = self._decomp_diag(alpha_D, U)
         if self.fit_intercept:
-            # remove intercept dim
-            coef = np.delete(coef, intercept_dim)
+            sw_sum = sqrt_sw @ sqrt_sw
+            alpha_Ginv_sqrt_sw = U @ self._diag_dot(alpha_D, UT_sqrt_sw)
+            alpha_d -= alpha_Ginv_sqrt_sw * sqrt_sw / sw_sum
+        if y.ndim == 2:
+            # handle case where y is 2-d
+            alpha_d = alpha_d[:, None]
+        looe = alpha_c / alpha_d
+        coef = V @ self._diag_dot(singvals / (singvals**2 + alpha), UT_y)
         return looe, coef
+
+    def _solve_svd_design_matrix(
+        self, alpha, y, sqrt_sw, singvals, U, V, UT_y, UT_sqrt_sw
+    ):
+        """Compute looe and coef when we have an SVD decomposition of X."""
+        n_samples = U.shape[0]
+        n_features = V.shape[0]
+        if n_samples <= n_features:
+            return self._solve_svd_design_matrix_wide(
+                alpha, y, sqrt_sw, singvals, U, V, UT_y, UT_sqrt_sw
+            )
+        else:
+            return self._solve_svd_design_matrix_long(
+                alpha, y, sqrt_sw, singvals, U, V, UT_y, UT_sqrt_sw
+            )
 
     def fit(self, X, y, sample_weight=None, score_params=None):
         """Fit Ridge regression model with gcv.
