diff --git a/sklearn/linear_model/_ridge.py b/sklearn/linear_model/_ridge.py
--- a/sklearn/linear_model/_ridge.py
+++ b/sklearn/linear_model/_ridge.py
@@ -1618,13 +1618,24 @@
 
 
 def _check_gcv_mode(X, gcv_mode):
-    if gcv_mode in ["eigen", "svd"]:
-        return gcv_mode
-    # if X has more rows than columns, use decomposition of X^T.X,
-    # otherwise X.X^T
-    if X.shape[0] > X.shape[1]:
-        return "svd"
-    return "eigen"
+    # svd only implemented for dense X
+    if gcv_mode == "svd":
+        if sparse.issparse(X):
+            # TODO(1.11) raise ValueError
+            msg = (
+                "The 'svd' mode is not supported for sparse X, we fallback to "
+                "`gcv_mode='eigen'`. Passing `gcv_mode='svd'` on sparse X will raise "
+                "an error in 1.11, use the default or pass `gcv_mode='eigen'` to "
+                "suppress this warning."
+            )
+            warnings.warn(msg, FutureWarning)
+        else:
+            return "svd"
+
+    # All other cases ("auto", "eigen")
+    # fallbacks to gram (n <= p) or cov (p < n)
+    n, p = X.shape
+    return "gram" if n <= p else "cov"
 
 
 def _find_smallest_angle(query, vectors):
@@ -2196,13 +2207,15 @@
         """
         xp, is_array_api, device_ = get_namespace_and_device(X)
         y, sample_weight = move_to(y, sample_weight, xp=xp, device=device_)
-        if is_array_api or hasattr(getattr(X, "dtype", None), "kind"):
-            original_dtype = X.dtype
+        if (is_array_api and xp.isdtype(X.dtype, "real floating")) or getattr(
+            getattr(X, "dtype", None), "kind", None
+        ) == "f":
+            original_floating_dtype = X.dtype
         else:
             # for X that does not have a simple dtype (e.g. pandas dataframe)
             # the attributes will be stored in the dtype chosen by
             # `validate_data``, i.e. np.float64
-            original_dtype = None
+            original_floating_dtype = None
         # Using float32 can be numerically unstable for this estimator. So if
         # the array API namespace and device allow, convert the input values
         # to float64 whenever possible before converting the results back to
@@ -2240,25 +2253,25 @@
 
         gcv_mode = _check_gcv_mode(X, self.gcv_mode)
 
-        if gcv_mode == "eigen":
+        n_samples, n_features = X.shape
+        if gcv_mode == "gram":
             decompose = self._eigen_decompose_gram
             solve = self._solve_eigen_gram
+        elif gcv_mode == "cov":
+            decompose = self._eigen_decompose_covariance
+            solve = self._solve_eigen_covariance
         elif gcv_mode == "svd":
-            if sparse.issparse(X):
-                decompose = self._eigen_decompose_covariance
-                solve = self._solve_eigen_covariance
-            else:
-                decompose = self._svd_decompose_design_matrix
-                solve = self._solve_svd_design_matrix
-
-        n_samples, n_features = X.shape
+            decompose = self._svd_decompose_design_matrix
+            solve = self._solve_svd_design_matrix
+        else:
+            raise ValueError(f"Unknown {gcv_mode=}")
 
         if sqrt_sw is None:
             sqrt_sw = xp.ones(n_samples, dtype=X.dtype, device=device_)
 
-        X_mean, *decomposition = decompose(X, y, sqrt_sw)
-
-        n_y = 1 if len(y.shape) == 1 else y.shape[1]
+        decomposition = decompose(X, X_offset, y, sqrt_sw)
+
+        n_y = 1 if y.ndim == 1 else y.shape[1]
         if (
             isinstance(self.alphas, numbers.Number)
             or getattr(self.alphas, "ndim", None) == 0
@@ -2270,15 +2283,13 @@
 
         if self.store_cv_results:
             self.cv_results_ = xp.empty(
-                (n_samples * n_y, n_alphas), dtype=original_dtype, device=device_
+                (n_samples * n_y, n_alphas), dtype=X.dtype, device=device_
             )
 
         best_coef, best_score, best_alpha = None, None, None
 
         for i, alpha in enumerate(alphas):
-            looe, coef = solve(float(alpha), y, sqrt_sw, X_mean, *decomposition)
-            assert len(looe) == n_samples
-            assert len(coef) == n_features, "coef wrong size"
+            looe, coef = solve(float(alpha), y, sqrt_sw, *decomposition)
             if self.scoring is None:
                 squared_errors = looe**2
                 alpha_score = self._score_without_scorer(squared_errors=squared_errors)
@@ -2321,7 +2332,7 @@
                 # update
                 if self.alpha_per_target and n_y > 1:
                     to_update = alpha_score > best_score
-                    best_coef[to_update] = coef[to_update]
+                    best_coef[:, to_update] = coef[:, to_update]
                     best_score[to_update] = alpha_score[to_update]
                     best_alpha[to_update] = alpha
                 elif alpha_score > best_score:
@@ -2330,26 +2341,31 @@
         self.alpha_ = best_alpha
         self.best_score_ = best_score
         self.coef_ = best_coef
+        if y.ndim == 2:
+            self.coef_ = self.coef_.T
         if y.ndim == 1 or y.shape[1] == 1:
             self.coef_ = _ravel(self.coef_)
 
-        if sparse.issparse(X):
-            X_offset = X_mean * X_scale
-        else:
-            X_offset += X_mean * X_scale
         self._set_intercept(X_offset, y_offset, X_scale)
 
         if self.store_cv_results:
-            if len(y.shape) == 1:
+            if y.ndim == 1:
                 cv_results_shape = n_samples, n_alphas
             else:
                 cv_results_shape = n_samples, n_y, n_alphas
             self.cv_results_ = xp.reshape(self.cv_results_, shape=cv_results_shape)
 
-        if original_dtype is not None:
+        if original_floating_dtype:
             if type(self.intercept_) is not float:
-                self.intercept_ = xp.astype(self.intercept_, original_dtype, copy=False)
-            self.coef_ = xp.astype(self.coef_, original_dtype, copy=False)
+                self.intercept_ = xp.astype(
+                    self.intercept_, original_floating_dtype, copy=False
+                )
+            self.coef_ = xp.astype(self.coef_, original_floating_dtype, copy=False)
+            if self.store_cv_results:
+                self.cv_results_ = xp.astype(
+                    self.cv_results_, original_floating_dtype, copy=False
+                )
+
         return self
 
     def _score_without_scorer(self, squared_errors):
@@ -2691,13 +2707,14 @@
         Flag indicating which strategy to use when performing
         Leave-One-Out Cross-Validation. Options are::
 
-            'auto' : use 'svd' if n_samples > n_features, otherwise use 'eigen'
-            'svd' : force use of singular value decomposition of X when X is
-                dense, eigenvalue decomposition of X^T.X when X is sparse.
-            'eigen' : force computation via eigendecomposition of X.X^T
+            'auto' : same as 'eigen'
+            'svd' : use singular value decomposition of X when X is dense,
+                fallback to 'eigen' when X is sparse
+            'eigen' : use eigendecomposition of X X' when n_samples <= n_features
+                or X' X when n_features < n_samples
 
         The 'auto' mode is the default and is intended to pick the cheaper
-        option of the two depending on the shape of the training data.
+        option depending on the shape and sparsity of the training data.
 
     store_cv_results : bool, default=False
         Flag indicating if the cross-validation values corresponding to
