diff --git a/sklearn/linear_model/_ridge.py b/sklearn/linear_model/_ridge.py
--- a/sklearn/linear_model/_ridge.py
+++ b/sklearn/linear_model/_ridge.py
@@ -43,6 +43,7 @@
 )
 from sklearn.utils._array_api import (
     _is_numpy_namespace,
+    _max_precision_float_dtype,
     _ravel,
     device,
     get_namespace,
@@ -2201,19 +2202,27 @@ def fit(self, X, y, sample_weight=None, score_params=None):
         """
         xp, is_array_api, device_ = get_namespace_and_device(X)
         y, sample_weight = move_to(y, sample_weight, xp=xp, device=device_)
+        # FIXME original_dtype can be int64 or int32
+        # we would like original_dtype to be float64 or float32
         if is_array_api or hasattr(getattr(X, "dtype", None), "kind"):
             original_dtype = X.dtype
         else:
             # for X that does not have a simple dtype (e.g. pandas dataframe)
             # the attributes will be stored in the dtype chosen by
             # `validate_data``, i.e. np.float64
             original_dtype = None
+        # Using float32 can be numerically unstable for this estimator. So if
+        # the array API namespace and device allow, convert the input values
+        # to float64 whenever possible before converting the results back to
+        # float32.
+        dtype = _max_precision_float_dtype(xp, device=device_)
+
         X, y = validate_data(
             self,
             X,
             y,
             accept_sparse=["csr", "csc", "coo"],
-            dtype=original_dtype,
+            dtype=dtype,
             multi_output=True,
             y_numeric=True,
         )
@@ -2346,6 +2355,12 @@ def fit(self, X, y, sample_weight=None, score_params=None):
                 cv_results_shape = n_samples, n_y, n_alphas
             self.cv_results_ = xp.reshape(self.cv_results_, shape=cv_results_shape)
 
+        # FIXME original_dtype can be int64 or int32
+        if original_dtype == xp.float32:
+            if type(self.intercept_) is not float:
+                self.intercept_ = xp.astype(self.intercept_, original_dtype, copy=False)
+            self.coef_ = xp.astype(self.coef_, original_dtype, copy=False)
+
         return self
 
     def _score_without_scorer(self, squared_errors):
