diff --git a/sympy/integrals/manualintegrate.py b/sympy/integrals/manualintegrate.py
--- a/sympy/integrals/manualintegrate.py
+++ b/sympy/integrals/manualintegrate.py
@@ -1643,6 +1643,17 @@
             coeffs = numer_poly.all_coeffs()
             step = SqrtQuadraticDenomRule(integrand, x, a, b, c, coeffs)
         return step
+
+    def sqrt_quadratic_reduction_rule(integrand: Expr, n: int):
+        # Implementation of Gradshteyn & Ryzhik 2.263.3
+        k = (-n - 1) // 2
+        delta = 4*a*c - b**2
+        R = c*x**2 + b*x + a
+
+        term_denom = (2*k - 1) * delta * (R**(S(2*k - 1)/2))
+        constant_term = f*2*(2*c*x+b) / term_denom
+        coeff = (8*c*(k-1))/((2*k-1) * delta)
+        expr = f * R**(S(1)/2 - k)
 
     if n > 0:  # rewrite poly * sqrt(s)**(2*k-1) to poly*s**k / sqrt(s)
         numer_poly = f_poly * (a+b*x+c*x**2)**((n+1)/2)
