diff --git a/django/contrib/admin/templatetags/base.py b/django/contrib/admin/templatetags/base.py
--- a/django/contrib/admin/templatetags/base.py
+++ b/django/contrib/admin/templatetags/base.py
@@ -1,5 +1,6 @@
 from inspect import getfullargspec
 
+from django.template.exceptions import TemplateSyntaxError
 from django.template.library import InclusionNode, parse_bits
 from django.utils.inspect import lazy_annotations
 
@@ -10,12 +11,21 @@ class InclusionAdminNode(InclusionNode):
     or globally.
     """
 
-    def __init__(self, parser, token, func, template_name, takes_context=True):
+    def __init__(self, name, parser, token, func, template_name, takes_context=True):
         self.template_name = template_name
         with lazy_annotations():
             params, varargs, varkw, defaults, kwonly, kwonly_defaults, _ = (
                 getfullargspec(func)
             )
+        if takes_context:
+            if params and params[0] == "context":
+                del params[0]
+            else:
+                function_name = func.__name__
+                raise TemplateSyntaxError(
+                    f"{name!r} sets takes_context=True so {function_name!r} "
+                    "must have a first argument of 'context'"
+                )
         bits = token.split_contents()
         args, kwargs = parse_bits(
             parser,
@@ -26,7 +36,6 @@ def __init__(self, parser, token, func, template_name, takes_context=True):
             defaults,
             kwonly,
             kwonly_defaults,
-            takes_context,
             bits[0],
         )
         super().__init__(func, takes_context, args, kwargs, filename=None)
