diff --git a/sympy/core/expr.py b/sympy/core/expr.py
--- a/sympy/core/expr.py
+++ b/sympy/core/expr.py
@@ -1918,7 +1918,7 @@ def as_independent(self, *deps, **hint) -> tuple[Expr, Expr]:
 
         func = self.func
         want: type[Add] | type[Mul]
-        if hint.get('as_Add', isinstance(self, Add) ):
+        if hint.get('as_Add', self.is_Add):
             want = Add
         else:
             want = Mul
@@ -1952,15 +1952,15 @@ def has(e):
             else:
                 return (self, S.One)
         else:
-            if issubclass(func, Add):
+            if func.is_Add:
                 args = list(self.args)
             else:
                 args, nc = self.args_cnc()
 
         d = sift(args, has)
         depend = d[True]
         indep = d[False]
-        if issubclass(func, Add):  # all terms were treated as commutative
+        if func.is_Add:  # all terms were treated as commutative
             return (func(*indep), _unevaluated_Add(*depend))
         else:  # handle noncommutative by stopping at first dependent term
             for i, n in enumerate(nc):