diff --git a/xarray/structure/merge.py b/xarray/structure/merge.py
--- a/xarray/structure/merge.py
+++ b/xarray/structure/merge.py
@@ -292,6 +292,8 @@
                     # does not affect the original input variable.
                     merged_vars[name] = variable.copy(deep=False)
                     merged_vars[name].attrs = attrs
+                else:
+                    merged_vars[name] = variable
                 merged_indexes[name] = index
             else:
                 variables = [variable for variable, _ in elements_list]
