diff --git a/astropy/io/fits/card.py b/astropy/io/fits/card.py
--- a/astropy/io/fits/card.py
+++ b/astropy/io/fits/card.py
@@ -1055,7 +1055,7 @@
         # We have to be careful that the first line may be able to hold less
         # of the value, if it is a HIERARCH keyword.
         headstr = self._format_keyword() + VALUE_INDICATOR
-        first_value_length = value_length + KEYWORD_LENGTH + 1 - len(headstr)
+        first_value_length = value_length + KEYWORD_LENGTH + 2 - len(headstr)
 
         # do the value string
         value = self._value.replace("'", "''")
