diff --git a/sympy/matrices/matrixbase.py b/sympy/matrices/matrixbase.py
--- a/sympy/matrices/matrixbase.py
+++ b/sympy/matrices/matrixbase.py
@@ -2029,7 +2029,7 @@ def iter_items(self):
         return self._eval_iter_items()
 
     def _eval_adjoint(self):
-        return self.transpose().conjugate()
+        return self.transpose().applyfunc(lambda x: x.adjoint())
 
     def _eval_applyfunc(self, f):
         cols = self.cols