diff --git a/sklearn/multiclass.py b/sklearn/multiclass.py
--- a/sklearn/multiclass.py
+++ b/sklearn/multiclass.py
@@ -553,8 +553,8 @@ def predict_proba(self, X):
             Y = np.concatenate(((1 - Y), Y), axis=1)
 
         if not self.multilabel_:
-            # Then, probabilities should be normalized to 1.
-            Y /= np.sum(Y, axis=1)[:, np.newaxis]
+            # Then, (nonzero) sample probability distributions should be normalized.
+
         return Y
 
     @available_if(_estimators_has("decision_function"))