diff --git a/django/utils/csp.py b/django/utils/csp.py
--- a/django/utils/csp.py
+++ b/django/utils/csp.py
@@ -71,15 +71,16 @@ class LazyNonce(SimpleLazyObject):
     """
 
     def __init__(self):
-        super().__init__(self._generate)
-
-    def _generate(self):
-        return secrets.token_urlsafe(16)
+        super().__init__(generate_nonce)
 
     def __bool__(self):
         return self._wrapped is not empty
 
 
+def generate_nonce():
+    return secrets.token_urlsafe(16)
+
+
 def build_policy(config, nonce=None):
     policy = []
 
diff --git a/django/utils/functional.py b/django/utils/functional.py
--- a/django/utils/functional.py
+++ b/django/utils/functional.py
@@ -407,7 +407,11 @@ def _setup(self):
     # without evaluating the wrapped object.
     def __repr__(self):
         if self._wrapped is empty:
-            repr_attr = self._setupfunc
+            # Avoid recursion if setupfunc is a bound method of this instance.
+            if getattr(self._setupfunc, "__self__", None) is self:
+                repr_attr = f"<bound method {self._setupfunc.__name__}>"
+            else:
+                repr_attr = self._setupfunc
         else:
             repr_attr = self._wrapped
         return "<%s: %r>" % (type(self).__name__, repr_attr)