diff --git a/sphinx/domains/python/_object.py b/sphinx/domains/python/_object.py
--- a/sphinx/domains/python/_object.py
+++ b/sphinx/domains/python/_object.py
@@ -41,7 +41,7 @@
 py_sig_re = re.compile(
     r"""^ ([\w.]*\.)?            # class name(s)
           (\w+)  \s*             # thing name
-          (?: \[\s*(.*)\s*])?    # optional: type parameters list
+          (?: \[\s*(.*?)\s*])?   # optional: type parameters list
           (?: \(\s*(.*)\s*\)     # optional: arguments
            (?:\s* -> \s* (.*))?  #           return annotation
           )? $                   # and nothing more
