diff --git a/astropy/utils/decorators.py b/astropy/utils/decorators.py
--- a/astropy/utils/decorators.py
+++ b/astropy/utils/decorators.py
@@ -711,7 +711,7 @@ def decorated_function(*args, **kwargs):
                     "Pass them as keywords to suppress this warning. "
                     f"(deprecated since {since_details})"
                 )
-                warnings.warn(msg, FutureWarning, stacklevel=2)
+                warnings.warn(msg, AstropyDeprecationWarning, stacklevel=2)
             return raw_function(*args, **kwargs)
 
         return decorated_function