diff --git a/sympy/core/power.py b/sympy/core/power.py
--- a/sympy/core/power.py
+++ b/sympy/core/power.py
@@ -1212,8 +1212,11 @@ def as_real_imag(self, deep=True, **hints):
 
     def _eval_derivative(self, s):
         from sympy.functions.elementary.exponential import log
+        if not self.base.is_commutative:
+            return None
         dbase = self.base.diff(s)
         dexp = self.exp.diff(s)
+
         return self * (dexp * log(self.base) + dbase * self.exp/self.base)
 
     def _eval_evalf(self, prec):
