diff --git a/django/db/backends/base/creation.py b/django/db/backends/base/creation.py
--- a/django/db/backends/base/creation.py
+++ b/django/db/backends/base/creation.py
@@ -362,13 +362,7 @@
         from unittest import expectedFailure, skip
 
         for test_name in self.connection.features.django_test_expected_failures:
-            test_case_name, _, test_method_name = test_name.rpartition(".")
-            test_app = test_name.split(".")[0]
-            # Importing a test app that isn't installed raises RuntimeError.
-            if test_app in settings.INSTALLED_APPS:
-                test_case = import_string(test_case_name)
-                test_method = getattr(test_case, test_method_name)
-                setattr(test_case, test_method_name, expectedFailure(test_method))
+            self._mark_test(test_name)
         for reason, tests in self.connection.features.django_test_skips.items():
             for test_name in tests:
                 test_case_name, _, test_method_name = test_name.rpartition(".")
