diff --git a/django/db/backends/base/creation.py b/django/db/backends/base/creation.py
--- a/django/db/backends/base/creation.py
+++ b/django/db/backends/base/creation.py
@@ -358,21 +358,11 @@
         Mark tests in Django's test suite which are expected failures on this
         database and test which should be skipped on this database.
         """
-        # Only load unittest if we're actually testing.
-        from unittest import expectedFailure, skip
-
         for test_name in self.connection.features.django_test_expected_failures:
             self._mark_test(test_name)
         for reason, tests in self.connection.features.django_test_skips.items():
             for test_name in tests:
-                test_case_name, _, test_method_name = test_name.rpartition(".")
-                test_app = test_name.split(".")[0]
-                # Importing a test app that isn't installed raises
-                # RuntimeError.
-                if test_app in settings.INSTALLED_APPS:
-                    test_case = import_string(test_case_name)
-                    test_method = getattr(test_case, test_method_name)
-                    setattr(test_case, test_method_name, skip(reason)(test_method))
+                self._mark_test(test_name, reason)
 
     def _mark_test(self, test_name, skip_reason=None):
         # Only load unittest during testing.
