diff --git a/django/db/backends/base/creation.py b/django/db/backends/base/creation.py
--- a/django/db/backends/base/creation.py
+++ b/django/db/backends/base/creation.py
@@ -372,8 +372,21 @@
         test_app = test_name.split(".")[0]
         # Importing a test app that isn't installed raises RuntimeError.
         if test_app in settings.INSTALLED_APPS:
-            test_frame = import_string(module_or_class_name)
-            test_to_mark = getattr(test_frame, name_to_mark)
+            try:
+                test_frame = import_string(module_or_class_name)
+            except ImportError:
+                # import_string() can raise ImportError if a submodule's parent
+                # module hasn't already been imported during test discovery.
+                # This can happen in at least two cases:
+                # 1. When running a subset of tests in a module, the test
+                #    runner won't import tests in that module's other
+                #    submodules.
+                # 2. When the parallel test runner spawns workers with an empty
+                #    import cache.
+                test_to_mark = import_string(test_name)
+                test_frame = sys.modules.get(test_to_mark.__module__)
+            else:
+                test_to_mark = getattr(test_frame, name_to_mark)
             if skip_reason:
                 setattr(test_frame, name_to_mark, skip(skip_reason)(test_to_mark))
             else:
