diff --git a/astropy/io/votable/converters.py b/astropy/io/votable/converters.py
--- a/astropy/io/votable/converters.py
+++ b/astropy/io/votable/converters.py
@@ -406,7 +406,6 @@ def _binoutput_var(self, value, mask):
 
         if self.arraysize != "*" and len(value) > self.arraysize:
             vo_warn(W46, ("char", self.arraysize), None, None)
-            value = value[: self.arraysize]
 
         return self._write_length(len(value)) + value
 