diff --git a/sklearn/base.py b/sklearn/base.py
--- a/sklearn/base.py
+++ b/sklearn/base.py
@@ -400,7 +400,7 @@ def _more_tags(self):
         warnings.warn(
             "The `_more_tags` method is deprecated in 1.6 and will be removed in "
             "1.7. Please implement the `__sklearn_tags__` method.",
-            category=FutureWarning,
+            category=DeprecationWarning,
         )
         return _to_old_tags(default_tags(self))
 
