diff --git a/lib/matplotlib/_mathtext.py b/lib/matplotlib/_mathtext.py
--- a/lib/matplotlib/_mathtext.py
+++ b/lib/matplotlib/_mathtext.py
@@ -2478,12 +2478,10 @@
                 shift = hlist.height + vgap + nucleus.depth
             vlt = Vlist(vlist)
             vlt.shift_amount = shift
-            result = Hlist([
-                vlt,
-                *([self._make_space(self._space_widths[r'\,'])]
-                  if self._needs_space_after_subsuper else []),
-            ])
+            optional_spacing = ([self._make_space(self._space_widths[r'\,'])]
+                                if self._needs_space_after_subsuper else [])
             self._needs_space_after_subsuper = False
+            result = Hlist([vlt, *optional_spacing])
             return [result]
 
         # We remove kerning on the last character for consistency (otherwise
@@ -2575,12 +2573,10 @@
 
         # Do we need to add a space after the nucleus?
         # To find out, check the flag set by operatorname
-        spaced_nucleus: list[Node] = [nucleus, x]
-        if self._needs_space_after_subsuper:
-            spaced_nucleus += [self._make_space(self._space_widths[r'\,'])]
-            self._needs_space_after_subsuper = False
-
-        result = Hlist(spaced_nucleus)
+        optional_spacing = ([self._make_space(self._space_widths[r'\,'])]
+                            if self._needs_space_after_subsuper else [])
+        self._needs_space_after_subsuper = False
+        result = Hlist([nucleus, x, *optional_spacing])
         return [result]
 
     def _genfrac(self, ldelim: str, rdelim: str, rule: float | None, style: _MathStyle,
