diff --git a/xarray/core/_aggregations.py b/xarray/core/_aggregations.py
--- a/xarray/core/_aggregations.py
+++ b/xarray/core/_aggregations.py
@@ -1363,6 +1363,13 @@ def reduce(
     ) -> Self:
         raise NotImplementedError()
 
+    def assign_coords(
+        self,
+        coords: Mapping | None = None,
+        **coords_kwargs: Any,
+    ) -> Self:
+        raise NotImplementedError()
+
     def count(
         self,
         dim: Dims = None,
