diff --git a/astropy/io/fits/scripts/fitscheck.py b/astropy/io/fits/scripts/fitscheck.py
--- a/astropy/io/fits/scripts/fitscheck.py
+++ b/astropy/io/fits/scripts/fitscheck.py
@@ -165,33 +165,38 @@ def verify_checksums(filename):
     """
     Prints a message if any HDU in `filename` has a bad checksum or datasum.
     """
+    checksum_errors = 0
+
+    # _verify_checksum_datasum issues warnings when checksums are wrong so we
+    # catch those to avoid duplicates, since we are using logging here to
+    # report problems.
     with warnings.catch_warnings(record=True) as wlist:
         warnings.simplefilter("always")
         with fits.open(filename, checksum=OPTIONS.checksum_kind) as hdulist:
             for i, hdu in enumerate(hdulist):
                 # looping on HDUs is needed to read them and verify the
                 # checksums
                 if not OPTIONS.ignore_missing:
                     if not hdu._checksum:
                         log.warning(
                             f"MISSING {filename!r} .. Checksum not found in HDU #{i}"
                         )
                         return 1
                     if not hdu._datasum:
                         log.warning(
                             f"MISSING {filename!r} .. Datasum not found in HDU #{i}"
                         )
                         return 1

     for w in wlist:
         if str(w.message).startswith(
             ("Checksum verification failed", "Datasum verification failed")
         ):
             log.warning("BAD %r %s", filename, str(w.message))
             return 1

     log.info(f"OK {filename!r}")
     return 0
 
 
 def verify_compliance(filename):