diff --git a/sklearn/utils/estimator_checks.py b/sklearn/utils/estimator_checks.py
--- a/sklearn/utils/estimator_checks.py
+++ b/sklearn/utils/estimator_checks.py
@@ -4000,7 +4000,7 @@
     # Make sure that the dtype of X stays unchanged: for instance estimator
     # that expect categorical inputs typically expected integer-based encoded
     # categories.
-
+    X -= X.mean().astype(X.dtype)

     if tags.input_tags.positive_only:
         with raises(ValueError, match="Negative values in data"):