diff --git a/astropy/io/fits/hdu/image.py b/astropy/io/fits/hdu/image.py
--- a/astropy/io/fits/hdu/image.py
+++ b/astropy/io/fits/hdu/image.py
@@ -983,6 +983,7 @@
     def dtype(self):
         # Implementing `.dtype` enables `astropy.nddata.Cutout2D` to accept
         # `ImageHDU.section` in place of `.data`.
+        if not self.hdu._do_not_scale_image_data and :
     def __getitem__(self, key):
         """Returns a slice of HDU data specified by `key`.
 
