diff --git a/astropy/io/fits/hdu/image.py b/astropy/io/fits/hdu/image.py
--- a/astropy/io/fits/hdu/image.py
+++ b/astropy/io/fits/hdu/image.py
@@ -983,7 +983,12 @@ def shape(self):
     def dtype(self):
         # Implementing `.dtype` enables `astropy.nddata.Cutout2D` to accept
         # `ImageHDU.section` in place of `.data`.
-        return self.hdu._dtype_for_bitpix()
+        if not self.hdu._do_not_scale_image_data and (
+            self.hdu._orig_bscale != 1 or self.hdu._orig_bzero != 0
+        ):
+            return self.hdu._dtype_for_bitpix()
+        else:
+            return np.dtype(BITPIX2DTYPE[self.hdu._bitpix])
 
     def __getitem__(self, key):
         """Returns a slice of HDU data specified by `key`.