diff --git a/astropy/io/fits/hdu/hdulist.py b/astropy/io/fits/hdu/hdulist.py
--- a/astropy/io/fits/hdu/hdulist.py
+++ b/astropy/io/fits/hdu/hdulist.py
@@ -1471,7 +1471,12 @@
                         "This Python installation does not provide the bz2 module."
                     )
                 new_file = bz2.BZ2File(name, mode="w")
-            elif self._file.compression
+            elif self._file.compression == "lzma":
+                if not HAS_LZMA:
+                    raise ModuleNotFoundError(
+                        "This Python installation does not provide the lzma module."
+                    )
+                new_file = lzma.LZMAFile(name, mode="w")
             else:
                 new_file = name
 
