diff --git a/astropy/io/fits/hdu/hdulist.py b/astropy/io/fits/hdu/hdulist.py
--- a/astropy/io/fits/hdu/hdulist.py
+++ b/astropy/io/fits/hdu/hdulist.py
@@ -27,7 +27,7 @@
 from astropy.utils.compat.numpycompat import NUMPY_LT_2_0
 
 # NOTE: Python can be built without bz2.
-from astropy.utils.compat.optional_deps import HAS_BZ2
+from astropy.utils.compat.optional_deps import HAS_BZ2, HAS_LZMA
 from astropy.utils.exceptions import AstropyUserWarning
 
 from .base import ExtensionHDU, _BaseHDU, _NonstandardHDU, _ValidHDU
@@ -39,6 +39,9 @@
 if HAS_BZ2:
     import bz2
 
+if HAS_LZMA:
+    import lzma
+
 __all__ = ["HDUList", "fitsopen"]
 
 # FITS file signature as per RFC 4047
@@ -1468,6 +1471,12 @@ def _flush_resize(self):
                         "This Python installation does not provide the bz2 module."
                     )
                 new_file = bz2.BZ2File(name, mode="w")
+            elif self._file.compression == "lzma":
+                if not HAS_LZMA:
+                    raise ModuleNotFoundError(
+                        "This Python installation does not provide the lzma module."
+                    )
+                new_file = lzma.LZMAFile(name, mode="w")
             else:
                 new_file = name
 