diff --git a/astropy/io/misc/yaml.py b/astropy/io/misc/yaml.py
--- a/astropy/io/misc/yaml.py
+++ b/astropy/io/misc/yaml.py
@@ -55,6 +55,7 @@
 """
 
 import base64
+import itertools
 
 import numpy as np
 import yaml
@@ -335,16 +336,16 @@
     for cls in itertools.chain(
         coords.representation.REPRESENTATION_CLASSES.values(),
         coords.representation.DIFFERENTIAL_CLASSES.values(),
-
+        coords.frame_transform_graph.frame_set,
     )
 ]
 for cls in cls_coords:
     name = cls.__name__
     # Add representations/differentials defined in astropy.
     if name in coords.representation.__all__:
         tag = "!astropy.coordinates." + name
         AstropyDumper.add_multi_representer(cls, _quantity_representer(tag))
         AstropyLoader.add_constructor(tag, _quantity_constructor(cls))
 
 
 def load(stream):
