diff --git a/astropy/io/misc/yaml.py b/astropy/io/misc/yaml.py
--- a/astropy/io/misc/yaml.py
+++ b/astropy/io/misc/yaml.py
@@ -338,14 +338,13 @@
         coords.representation.DIFFERENTIAL_CLASSES.values(),
         coords.frame_transform_graph.frame_set,
     )
+    if cls.__name__ in dir(coords)
 ]
 for cls in cls_coords:
     name = cls.__name__
-    # Add representations/differentials defined in astropy.
-    if name in coords.representation.__all__:
         tag = "!astropy.coordinates." + name
         AstropyDumper.add_multi_representer(cls, _quantity_representer(tag))
         AstropyLoader.add_constructor(tag, _quantity_constructor(cls))
 
 
 def load(stream):
