diff --git a/django/db/models/constraints.py b/django/db/models/constraints.py
--- a/django/db/models/constraints.py
+++ b/django/db/models/constraints.py
@@ -53,7 +53,8 @@
         for field_name, *__ in model._get_expr_references(expression):
             if field_name == "pk":
                 field = model._meta.pk
-            field = get_field(field_name)
+            else:
+                field = get_field(field_name)
             if field.generated and cls._expression_refs_exclude(
                 model, field.expression, exclude
             ):
