diff --git a/django/db/models/constraints.py b/django/db/models/constraints.py
--- a/django/db/models/constraints.py
+++ b/django/db/models/constraints.py
@@ -55,6 +55,8 @@
                 field = model._meta.pk
             else:
                 field = get_field(field_name)
+            if field_name in exclude or field.name in exclude:
+                return True
             if field.generated and cls._expression_refs_exclude(
                 model, field.expression, exclude
             ):
