diff --git a/sklearn/linear_model/_bayes.py b/sklearn/linear_model/_bayes.py
--- a/sklearn/linear_model/_bayes.py
+++ b/sklearn/linear_model/_bayes.py
@@ -17,7 +17,7 @@
 from ..utils._param_validation import Interval
 from ..utils.extmath import fast_logdet
 from ..utils.validation import _check_sample_weight, validate_data
-from ._base import LinearModel, _preprocess_data, _rescale_data
+from ._base import LinearModel, _preprocess_data
 
 ###############################################################################
 # BayesianRidge regression
@@ -254,17 +254,15 @@ def fit(self, X, y, sample_weight=None):
             y_mean = np.average(y, weights=sample_weight)
             y_var = np.average((y - y_mean) ** 2, weights=sample_weight)
 
-        X, y, X_offset_, y_offset_, X_scale_ = _preprocess_data(
+        X, y, X_offset_, y_offset_, X_scale_, _ = _preprocess_data(
             X,
             y,
             fit_intercept=self.fit_intercept,
             copy=self.copy_X,
             sample_weight=sample_weight,
-        )
-
-        if sample_weight is not None:
             # Sample weight can be implemented via a simple rescaling.
-            X, y, _ = _rescale_data(X, y, sample_weight)
+            rescale_with_sw=True,
+        )
 
         self.X_offset_ = X_offset_
         self.X_scale_ = X_scale_
@@ -671,7 +669,7 @@ def fit(self, X, y):
         n_samples, n_features = X.shape
         coef_ = np.zeros(n_features, dtype=dtype)
 
-        X, y, X_offset_, y_offset_, X_scale_ = _preprocess_data(
+        X, y, X_offset_, y_offset_, X_scale_, _ = _preprocess_data(
             X, y, fit_intercept=self.fit_intercept, copy=self.copy_X
         )
 
