diff --git a/sklearn/base.py b/sklearn/base.py
--- a/sklearn/base.py
+++ b/sklearn/base.py
@@ -200,6 +200,9 @@
     def __dir__(self):
         # Filters conditional methods that should be hidden based
         # on the `available_if` decorator
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", category=FutureWarning)
+            return 
 
     _html_repr = estimator_html_repr
 
