diff --git a/sphinx/application.py b/sphinx/application.py
--- a/sphinx/application.py
+++ b/sphinx/application.py
@@ -239,7 +239,7 @@ def __init__(
         self._fail_on_warnings = bool(warningiserror)
         self.pdb = pdb
         self._exception_on_warning = exception_on_warning
-        logging.setup(self, self._status, self._warning)
+        logging.setup(self, self._status, self._warning, verbosity=verbosity)
 
         self.events = EventManager(self)
 
