diff --git a/sympy/printing/fortran.py b/sympy/printing/fortran.py
--- a/sympy/printing/fortran.py
+++ b/sympy/printing/fortran.py
@@ -772,6 +772,17 @@ def _print_ArrayConstructor(self, ac):
         fmtstr = "[%s]" if self._settings["standard"] >= 2003 else '(/%s/)'
         return fmtstr % ', '.join((self._print(arg) for arg in ac.elements))
 
+    def _print_KeywordFunctionCall(self, expr):
+        args = [self._print(arg) for arg in expr.function_args]
+
+        for key, value in expr.keyword_args.items():
+            args.append(f"{key}={self._print(value)}")
+
+        return '{name}({args})'.format(
+            name=self._print(expr.name),
+            args=', '.join(args)
+        )
+
     def _print_ArrayElement(self, elem):
         return '{symbol}({idxs})'.format(
             symbol=self._print(elem.name),