diff --git a/sklearn/utils/_repr_html/params.py b/sklearn/utils/_repr_html/params.py
--- a/sklearn/utils/_repr_html/params.py
+++ b/sklearn/utils/_repr_html/params.py
@@ -90,9 +90,12 @@ def _params_html_repr(params):
     PARAM_AVAILABLE_DOC_LINK_TEMPLATE = """
 
         <a class="param-doc-link"
+            style="anchor-name: --doc-link-{param_name};"
             rel="noreferrer" target="_blank" href="{link}">
             {param_name}
-            <span class="param-doc-description">{param_description}</span>
+            <span class="param-doc-description"
+            style="position-anchor: --doc-link-{param_name};">
+            {param_description}</span>
         </a>
 
     """
