diff --git a/sklearn/metrics/__init__.py b/sklearn/metrics/__init__.py
--- a/sklearn/metrics/__init__.py
+++ b/sklearn/metrics/__init__.py
@@ -91,7 +91,7 @@
     silhouette_score,
     v_measure_score,
 )
-from sklearn.metrics.decision_threshold import metric_threshold_curve
+from sklearn.metrics.decision_threshold import decision_threshold_curve
 from sklearn.metrics.pairwise import (
     euclidean_distances,
     nan_euclidean_distances,
@@ -134,6 +134,7 @@
     "d2_tweedie_score",
     "davies_bouldin_score",
     "dcg_score",
+    "decision_threshold_curve",
     "det_curve",
     "euclidean_distances",
     "explained_variance_score",
@@ -162,7 +163,6 @@
     "mean_squared_log_error",
     "mean_tweedie_deviance",
     "median_absolute_error",
-    "metric_threshold_curve",
     "multilabel_confusion_matrix",
     "mutual_info_score",
     "nan_euclidean_distances",
