diff --git a/sympy/combinatorics/named_groups.py b/sympy/combinatorics/named_groups.py
--- a/sympy/combinatorics/named_groups.py
+++ b/sympy/combinatorics/named_groups.py
@@ -91,25 +91,28 @@ def AlternatingGroup(n):
 
     """
     # small cases are special
-    if n in (1, 2):
-        return PermutationGroup([Permutation([0])])
+    if n == 1:
+        G = PermutationGroup([Permutation([0])])
+    elif n == 2:
+        G = PermutationGroup([Permutation([0,1])])
 
-    a = list(range(n))
-    a[0], a[1], a[2] = a[1], a[2], a[0]
-    gen1 = a
-    if n % 2:
-        a = list(range(1, n))
-        a.append(0)
-        gen2 = a
     else:
-        a = list(range(2, n))
-        a.append(1)
-        a.insert(0, 0)
-        gen2 = a
-    gens = [gen1, gen2]
-    if gen1 == gen2:
-        gens = gens[:1]
-    G = PermutationGroup([_af_new(a) for a in gens], dups=False)
+        a = list(range(n))
+        a[0], a[1], a[2] = a[1], a[2], a[0]
+        gen1 = a
+        if n % 2:
+            a = list(range(1, n))
+            a.append(0)
+            gen2 = a
+        else:
+            a = list(range(2, n))
+            a.append(1)
+            a.insert(0, 0)
+            gen2 = a
+        gens = [gen1, gen2]
+        if gen1 == gen2:
+            gens = gens[:1]
+        G = PermutationGroup([_af_new(a) for a in gens], dups=False)
 
     set_alternating_group_properties(G, n, n)
     G._is_alt = True
@@ -118,6 +121,8 @@ def AlternatingGroup(n):
 
 def set_alternating_group_properties(G, n, degree):
     """Set known properties of an alternating group. """
+    if n < 3:
+        G._order = 1
     if n < 4:
         G._is_abelian = True
         G._is_nilpotent = True
@@ -129,7 +134,10 @@ def set_alternating_group_properties(G, n, degree):
     else:
         G._is_solvable = False
     G._degree = degree
-    G._is_transitive = True
+    if n == 2:
+        G._is_transitive = False
+    else:
+        G._is_transitive = True
     G._is_dihedral = False
 
 