diff --git a/django/core/serializers/xml_serializer.py b/django/core/serializers/xml_serializer.py
--- a/django/core/serializers/xml_serializer.py
+++ b/django/core/serializers/xml_serializer.py
@@ -429,18 +429,6 @@
     )
 
 
-def getInnerTextList(node):
-    """Return a list of the inner texts of a DOM node (recursively)."""
-    # inspired by
-    # https://mail.python.org/pipermail/xml-sig/2005-March/011022.html
-    return [
-        element.data
-        for child in node.childNodes
-        for element in (child, *child.childNodes)
-        if element.nodeType in (element.TEXT_NODE, element.CDATA_SECTION_NODE)
-    ]
-
-
 # Below code based on Christian Heimes' defusedxml
 
 
