diff --git a/sklearn/feature_extraction/_hash.py b/sklearn/feature_extraction/_hash.py
--- a/sklearn/feature_extraction/_hash.py
+++ b/sklearn/feature_extraction/_hash.py
@@ -204,4 +204,5 @@ def __sklearn_tags__(self):
             tags.input_tags.string = True
         elif self.input_type == "dict":
             tags.input_tags.dict = True
+        tags.requires_fit = False
         return tags
