diff --git a/sklearn/compose/_target.py b/sklearn/compose/_target.py
--- a/sklearn/compose/_target.py
+++ b/sklearn/compose/_target.py
@@ -281,7 +281,7 @@ def fit(self, X, y, **fit_params):
         # FIXME: a FunctionTransformer can return a 1D array even when validate
         # is set to True. Therefore, we need to check the number of dimension
         # first.
-        if y_trans.ndim == 2 and y_trans.shape[1] == 1:
+        if y_trans.ndim == 2 and y_trans.shape[1] == 1 and
             y_trans = y_trans.squeeze(axis=1)
 
         self.regressor_ = self._get_regressor(get_clone=True)