diff --git a/sympy/core/mod.py b/sympy/core/mod.py
--- a/sympy/core/mod.py
+++ b/sympy/core/mod.py
@@ -5,6 +5,7 @@
 from .logic import fuzzy_and, fuzzy_not
 from .mul import Mul
 from .numbers import equal_valued
+from .relational import is_le, is_lt, is_ge, is_gt
 from .singleton import S
 
 
@@ -96,21 +97,20 @@ def number_eval(p, q):
 
             # by difference
             # -2|q| < p < 2|q|
-            d = abs(p)
-            for _ in range(2):
-                d -= abs(q)
-                if d.is_negative:
-                    if q.is_positive:
-                        if p.is_positive:
-                            return d + q
-                        elif p.is_negative:
-                            return -d
-                    elif q.is_negative:
-                        if p.is_positive:
-                            return d
-                        elif p.is_negative:
-                            return -d + q
-                    break
+            if q.is_positive:
+                comp1, comp2 = is_le, is_lt
+            elif q.is_negative:
+                comp1, comp2 = is_ge, is_gt
+            else:
+                return
+            ls = -2*q
+            r = p - q
+            for _ in range(4):
+                if not comp1(ls, p):
+                    return
+                if comp2(r, ls):
+                    return p - ls
+                ls += q
 
         rv = number_eval(p, q)
         if rv is not None: