diff --git a/xarray/core/formatting_html.py b/xarray/core/formatting_html.py
--- a/xarray/core/formatting_html.py
+++ b/xarray/core/formatting_html.py
@@ -211,24 +211,17 @@
     max_items_collapse,
     expand_option_name,
     enabled=True,
-    max_option_name: Literal["display_max_children"] | None = None,
+    **kwargs,
 ) -> str:
     n_items = len(mapping)
-    expanded = _get_boolean_with_default(
+    expanded = max_items_collapse is None or _get_boolean_with_default(
         expand_option_name, n_items < max_items_collapse
     )
     collapsed = not expanded
 
-    inline_details = ""
-    if max_option_name and max_option_name in OPTIONS:
-        max_items = int(OPTIONS[max_option_name])
-        if n_items > max_items:
-            inline_details = f"({max_items}/{n_items})"
-
     return collapsible_section(
-        name,
-        inline_details=inline_details,
-        details=details_func(mapping),
+        f"{name}:",
+        details=details_func(mapping, **kwargs),
         n_items=n_items,
         enabled=enabled,
         collapsed=collapsed,
@@ -239,7 +232,7 @@
     dim_list = format_dims(obj.sizes, obj.xindexes.dims)
 
     return collapsible_section(
-        "Dimensions", inline_details=dim_list, enabled=False, collapsed=True
+        "Dimensions:", inline_details=dim_list, enabled=False, collapsed=True
     )
 
 
