diff --git a/xarray/core/formatting_html.py b/xarray/core/formatting_html.py
--- a/xarray/core/formatting_html.py
+++ b/xarray/core/formatting_html.py
@@ -466,6 +466,65 @@
     total = node_count + child_count
     cache[id(node)] = total
     return total
+
+@dataclass
+class _DataTreeDisplay:
+    node: DataTree
+    sections: list[str]
+    item_count: int
+    collapsed: bool
+    disabled: bool
+
+
+def _build_datatree_displays(tree: DataTree) -> dict[str, _DataTreeDisplay]:
+    displayed_line_count = 0
+    html_line_count = 0
+    displays: dict[str, _DataTreeDisplay] = {}
+    item_count_cache: dict[int, int] = {}
+    root = True
+    collapsed = False
+    disabled = False
+
+    html_limit = OPTIONS["display_max_html_elements"]
+    uncollapsed_limit = OPTIONS["display_max_items"]
+
+    too_many_items_section = collapsible_section(
+        "<em>Too many items to display (display_max_html_elements exceeded)</em>",
+        enabled=False,
+        collapsed=True,
+        span_grid=True,
+    )
+
+    for node in tree.subtree:  # breadth-first
+        parent = node.parent
+        if parent is not None:
+            parent_display = displays.get(parent.path, None)
+            if parent_display is not None and parent_display.disabled:
+                break  # no need to build display
+
+        item_count = _tree_item_count(node, item_count_cache)
+
+        sections, node_line_count = _datatree_node_sections(node, root)
+        new_displayed_count = displayed_line_count + node_line_count
+        new_html_count = html_line_count + node_line_count
+
+        disabled = not root and (disabled or new_html_count > html_limit)
+        if disabled:
+            sections = [too_many_items_section]
+            collapsed = True
+        else:
+            html_line_count = new_html_count
+
+        collapsed = not root and (collapsed or new_displayed_count > uncollapsed_limit)
+        if not collapsed:
+            displayed_line_count = new_displayed_count
+
+        displays[node.path] = _DataTreeDisplay(
+            node, sections, item_count, collapsed, disabled
+        )
+        root = False
+
+    return displays
 
 def datatree_child_repr(node: DataTree, end: bool = False) -> str:
     # Wrap DataTree HTML representation with a tee to the left of it.
