diff --git a/xarray/core/formatting_html.py b/xarray/core/formatting_html.py
--- a/xarray/core/formatting_html.py
+++ b/xarray/core/formatting_html.py
@@ -8,7 +8,7 @@
 from html import escape
 from importlib.resources import files
 from math import ceil
-from typing import TYPE_CHECKING, Literal
+from typing import TYPE_CHECKING
 
 from xarray.core.formatting import (
     filter_nondefault_indexes,
