diff --git a/django/contrib/admin/views/main.py b/django/contrib/admin/views/main.py
--- a/django/contrib/admin/views/main.py
+++ b/django/contrib/admin/views/main.py
@@ -64,7 +64,6 @@ def __init__(self, *args, **kwargs):
 
 class ChangeList:
     search_form_class = ChangeListSearchForm
-    formset = None
 
     def __init__(
         self,
@@ -102,7 +101,7 @@ def __init__(
         self.preserved_filters = model_admin.get_preserved_filters(request)
         self.sortable_by = sortable_by
         self.search_help_text = search_help_text
-
+        self.formset = None
         # Get search parameters from the query string.
         _search_form = self.search_form_class(request.GET)
         if not _search_form.is_valid():
