diff --git a/sympy/physics/units/dimensions.py b/sympy/physics/units/dimensions.py
--- a/sympy/physics/units/dimensions.py
+++ b/sympy/physics/units/dimensions.py
@@ -15,6 +15,8 @@
 import collections
 from functools import reduce
 
+from sympy import Mul
+
 from sympy.core.basic import Basic
 from sympy.core.containers import (Dict, Tuple)
 from sympy.core.singleton import S
@@ -100,6 +102,10 @@ def get_quantity_dimension(self, unit):
             return Dimension(1)
 
     def get_quantity_scale_factor(self, unit):
+        if isinstance(unit, Mul):
+            return Mul.fromiter(self.get_quantity_scale_factor(arg) for arg in unit.args)
+        if isinstance(unit, Pow):
+            return self.get_quantity_scale_factor(unit.base)**unit.exp
         if unit in self._quantity_scale_factors:
             return self._quantity_scale_factors[unit]
         if unit in self._quantity_scale_factors_global: