diff --git a/sphinx/ext/autodoc/directive.py b/sphinx/ext/autodoc/directive.py
--- a/sphinx/ext/autodoc/directive.py
+++ b/sphinx/ext/autodoc/directive.py
@@ -171,9 +171,9 @@ def run(self) -> list[Node]:
             options=documenter_options,
             props=props,
             record_dependencies=record_dependencies,
-            result=result,
             ref_context=ref_context,
             reread_always=reread_always,
+            result=result,
         )
         if not result:
             return []
