diff --git a/astropy/units/format/__init__.py b/astropy/units/format/__init__.py
--- a/astropy/units/format/__init__.py
+++ b/astropy/units/format/__init__.py
@@ -50,20 +50,15 @@ def __getattr__(name):
     raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
 
 
-def _known_formats() -> str:
-    in_out = [
-        name
+def known_formats() -> str:
+    return "Valid formatter names are: " + ", ".join(map(repr, Base.registry))
+
+
+def known_parsers() -> str:
+    return "Valid parser names are: " + ", ".join(
+        repr(name)
         for name, cls in Base.registry.items()
         if cls.parse.__func__ is not Base.parse.__func__
-    ]
-    out_only = [
-        name
-        for name, cls in Base.registry.items()
-        if cls.parse.__func__ is Base.parse.__func__
-    ]
-    return (
-        f"Valid formatter names are: {in_out} for input and output, "
-        f"and {out_only} for output only."
     )
 
