diff --git a/astropy/units/format/__init__.py b/astropy/units/format/__init__.py
--- a/astropy/units/format/__init__.py
+++ b/astropy/units/format/__init__.py
@@ -78,17 +78,7 @@
     """
     if format is None:
         return Generic
-
+    if isinstance(format, str):
     if isinstance(format, type) and issubclass(format, Base):
         return format
-    elif not (isinstance(format, str) or format is None):
-        raise TypeError(
-            f"Expected a formatter name, not {format!r}.  {_known_formats()}."
-        )
-
-    format_lower = format.lower()
-
-    if format_lower in Base.registry:
-        return Base.registry[format_lower]
-
-    raise ValueError(f"Unknown format {format!r}.  {_known_formats()}")
+    raise TypeError(f"Expected a formatter name, not {format!r}.")
