diff --git a/sympy/polys/matrices/sdm.py b/sympy/polys/matrices/sdm.py
--- a/sympy/polys/matrices/sdm.py
+++ b/sympy/polys/matrices/sdm.py
@@ -849,8 +849,14 @@
 
         K = A.domain
         zero = K.zero
-        fzero = lambda e: zero
-        Csdm = binop_dict(A, B, mul, fzero, fzero)
+        if K.is_EXRAW:
+            fmul_zero_a = lambda e: e * zero
+            fmul_zero_b = lambda e: zero * e
+            Csdm = binop_dict(A, B, mul, fmul_zero_a, fmul_zero_b)
+        else:
+            fzero = lambda e: zero
+            Csdm = binop_dict(A, B, mul, fzero, fzero)
+
         return A.new(Csdm, A.shape, A.domain)
 
     def add(A, B):
