diff --git a/sphinx/ext/autosummary/generate.py b/sphinx/ext/autosummary/generate.py
--- a/sphinx/ext/autosummary/generate.py
+++ b/sphinx/ext/autosummary/generate.py
@@ -68,6 +68,7 @@
     from sphinx.application import Sphinx
     from sphinx.events import EventManager
     from sphinx.ext.autodoc import Documenter
+    from sphinx.ext.autodoc._property_types import _AutodocObjType
 
 logger = logging.getLogger(__name__)
 
@@ -223,20 +224,18 @@
         *,
         config: Config,
         events: EventManager,
-        registry: SphinxComponentRegistry,
     ) -> None:
         self.config = config
         self.events = events
-        self.registry = registry
         self.object = obj
 
     def get_object_type(self, name: str, value: Any) -> str:
         return _get_documenter(value, self.object)
 
-    def is_skipped(self, name: str, value: Any, objtype: str) -> bool:
+    def is_skipped(self, name: str, value: Any, obj_type: _AutodocObjType) -> bool:
         try:
             return self.events.emit_firstresult(
-                'autodoc-skip-member', objtype, name, value, False, {}
+                'autodoc-skip-member', obj_type, name, value, False, {}
             )
         except Exception as exc:
             logger.warning(
