diff --git a/sphinx/ext/autosummary/generate.py b/sphinx/ext/autosummary/generate.py
--- a/sphinx/ext/autosummary/generate.py
+++ b/sphinx/ext/autosummary/generate.py
@@ -322,16 +322,14 @@
     *,
     config: Config,
     events: EventManager,
-    registry: SphinxComponentRegistry,
 ) -> str:
     obj_type = _get_documenter(obj, parent)
-    doc = registry.documenters[obj_type]
 
     ns: dict[str, Any] = {}
     ns.update(context)
 
-    if doc.objtype == 'module':
-        scanner = ModuleScanner(obj, config=config, events=events, registry=registry)
+    if obj_type == 'module':
+        scanner = ModuleScanner(obj, config=config, events=events)
         ns['members'] = scanner.scan(imported_members)
 
         respect_module_all = not config.autosummary_ignore_module_all
@@ -340,30 +338,27 @@
         )
 
         ns['functions'], ns['all_functions'] = _get_members(
-            doc,
+            obj_type,
             obj,
             {'function'},
             config=config,
             events=events,
-            registry=registry,
             imported=imported_members,
         )
         ns['classes'], ns['all_classes'] = _get_members(
-            doc,
+            obj_type,
             obj,
             {'class'},
             config=config,
             events=events,
-            registry=registry,
             imported=imported_members,
         )
         ns['exceptions'], ns['all_exceptions'] = _get_members(
-            doc,
+            obj_type,
             obj,
             {'exception'},
             config=config,
             events=events,
-            registry=registry,
             imported=imported_members,
         )
         ns['attributes'], ns['all_attributes'] = _get_module_attrs(name, ns['members'])
@@ -386,12 +381,11 @@
             # Otherwise, use get_modules method normally
             if respect_module_all and '__all__' in dir(obj):
                 imported_modules, all_imported_modules = _get_members(
-                    doc,
+                    obj_type,
                     obj,
                     {'module'},
                     config=config,
                     events=events,
-                    registry=registry,
                     imported=True,
                 )
                 skip += all_imported_modules
