diff --git a/django/core/validators.py b/django/core/validators.py
--- a/django/core/validators.py
+++ b/django/core/validators.py
@@ -76,14 +76,13 @@ class DomainNameValidator(RegexValidator):
     # Max length for domain name labels is 63 characters per RFC 1034 sec. 3.1.
     domain_re = r"(?:\.(?!-)[a-z" + ul + r"0-9-]{1,63}(?<!-))*"
     # Top-level domain.
-    tld_re = (
+    tld_no_fqdn_re = (
         r"\."  # dot
         r"(?!-)"  # can't start with a dash
         r"(?:[a-z" + ul + "-]{2,63}"  # domain label
         r"|xn--[a-z0-9]{1,59})"  # or punycode label
         r"(?<!-)"  # can't end with a dash
-        r"\.?"  # may have a trailing dot
     )
     ascii_only_hostname_re = r"[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?"
     ascii_only_domain_re = r"(?:\.(?!-)[a-zA-Z0-9-]{1,63}(?<!-))*"
     ascii_only_tld_re = (
