diff --git a/sympy/integrals/manualintegrate.py b/sympy/integrals/manualintegrate.py
--- a/sympy/integrals/manualintegrate.py
+++ b/sympy/integrals/manualintegrate.py
@@ -1441,7 +1441,13 @@ def trig_cmplx_exp_rule(integral: IntegralInfo):
     """
     integrand, symbol = integral
 
-    if not integrand.has(erf, erfc, erfi):
+    if not integrand.has(erf, erfc, erfi, exp):
+        return
+
+    a = Wild('a', exclude=[symbol, 0])
+    b = Wild('b', exclude=[symbol])
+    pattern = exp(a * symbol**2 + b * symbol)
+    if not any(term.match(pattern) for term in integrand.atoms(exp)):
         return
 
     # Replace trig and hyperbolic functions with their exponential forms