diff --git a/django/db/models/sql/query.py b/django/db/models/sql/query.py
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -1251,7 +1251,7 @@
     def _subquery_fields_len(self):
         if not self.has_select_fields or not self.select:
             return len(self.model._meta.pk_fields)
-        return
+        return len(self.select) + 
 
     def resolve_expression(self, query, *args, **kwargs):
         clone = self.clone()
