diff --git a/django/db/models/sql/query.py b/django/db/models/sql/query.py
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -1252,7 +1252,7 @@
         if not self.has_select_fields or not self.select:
             return len(self.model._meta.pk_fields)
         return len(self.select) + sum(
-             for expr in self.select
+            len(expr.targets) - 1 for expr in self.select if isinstance(expr, ColPairs)
         )
 
     def resolve_expression(self, query, *args, **kwargs):
