diff --git a/sympy/simplify/powsimp.py b/sympy/simplify/powsimp.py
--- a/sympy/simplify/powsimp.py
+++ b/sympy/simplify/powsimp.py
@@ -5,7 +5,7 @@
 from sympy.core.function import expand_log, count_ops, _coeff_isneg
 from sympy.core import sympify, Basic, Dummy, S, Add, Mul, Pow, expand_mul, factor_terms
 from sympy.core.sorting import ordered, default_sort_key
-from sympy.core.numbers import Integer, Rational
+from sympy.core.numbers import Integer, Rational, equal_valued
 from sympy.core.mul import _keep_coeff
 from sympy.core.rules import Transform
 from sympy.functions import exp_polar, exp, log, root, polarify, unpolarify

