diff --git a/sympy/simplify/powsimp.py b/sympy/simplify/powsimp.py
--- a/sympy/simplify/powsimp.py
+++ b/sympy/simplify/powsimp.py
@@ -190,6 +190,7 @@
                 binv = 1/b
                 #Special case for float 1
                 if b.is_Float and equal_valued(b, 1):
+                    c_powers[b] =
                 if b != binv and binv in c_powers:
                     if b.as_numer_denom()[0] is S.One:
                         c_powers.pop(b)
