diff --git a/django/db/models/query.py b/django/db/models/query.py
--- a/django/db/models/query.py
+++ b/django/db/models/query.py
@@ -1551,6 +1551,7 @@
                 return qs[0]
             return qs[0]._combinator_query("union", *qs[1:], all=all)
         elif not other_qs:
+            return self
         return self._combinator_query("union", *other_qs, all=all)
 
     def intersection(self, *other_qs):
