diff --git a/astropy/units/quantity_helper/function_helpers.py b/astropy/units/quantity_helper/function_helpers.py
--- a/astropy/units/quantity_helper/function_helpers.py
+++ b/astropy/units/quantity_helper/function_helpers.py
@@ -468,7 +468,18 @@
 
 @function_helper
 def average(a, axis=None, weights=None, returned=False, *, keepdims=np._NoValue):
-    pass
+    return (
+        (a,),
+        {
+            "axis": axis,
+            "weights": weights.value if weights is not None else None,
+            "returned": returned,
+            "keepdims": keepdims,
+        },
+        (None, weights.unit) if weights is not None and returned else None,
+        None,
+    )
+
 
 
 def _iterable_helper(*args, out=None, **kwargs):
