diff --git a/astropy/units/quantity_helper/function_helpers.py b/astropy/units/quantity_helper/function_helpers.py
--- a/astropy/units/quantity_helper/function_helpers.py
+++ b/astropy/units/quantity_helper/function_helpers.py
@@ -1699,7 +1699,7 @@ def average(a, axis=None, weights=None, returned=False, *, keepdims=np._NoValue)
     w_value, w_unit = (
         (None, dimensionless_unscaled)
         if weights is None
-        else (weights.value, weights.unit) 
+        else (weights.value, weights.unit)
     )
     return (
         (a_value,),
