diff --git a/astropy/units/quantity_helper/function_helpers.py b/astropy/units/quantity_helper/function_helpers.py
--- a/astropy/units/quantity_helper/function_helpers.py
+++ b/astropy/units/quantity_helper/function_helpers.py
@@ -467,6 +467,31 @@ def _quantities2arrays(*args, unit_from_first=False):
     return arrays, q.unit
 
 
+@function_helper
+def average(a, axis=None, weights=None, returned=False, *, keepdims=np._NoValue):
+    # This override should no longer be needed once
+    # https://github.com/numpy/numpy/pull/30522 is merged (likely for
+    # "not NUMPY_LT_2_4_1").
+
+    a_value, a_unit = (_a := _as_quantity(a)).value, _a.unit
+    w_value, w_unit = (
+        (None, dimensionless_unscaled)
+        if weights is None
+        else ((_w := _as_quantity(weights)).value, _w.unit)
+    )
+    return (
+        (a_value,),
+        {
+            "axis": axis,
+            "weights": w_value,
+            "returned": returned,
+            "keepdims": keepdims,
+        },
+        ((a_unit, w_unit) if returned else a_unit),
+        None,
+    )
+
+
 def _iterable_helper(*args, out=None, **kwargs):
     """Convert arguments to Quantity, and treat possible 'out'."""
     if out is not None:
@@ -1685,30 +1710,3 @@ def merge_arrays(
         unit,
         None,
     )
-
-
-@function_helper
-def average(a, axis=None, weights=None, returned=False, *, keepdims=np._NoValue):
-    # This override should no longer be needed once
-    # https://github.com/numpy/numpy/pull/30522 is merged (likely for
-    # "not NUMPY_LT_2_4_1").
-
-    a = _as_quantity(a)
-    a_value, a_unit = a.value, a.unit
-    weights = _as_quantity(weights)
-    w_value, w_unit = (
-        (None, dimensionless_unscaled)
-        if weights is None
-        else (weights.value, weights.unit)
-    )
-    return (
-        (a_value,),
-        {
-            "axis": axis,
-            "weights": w_value,
-            "returned": returned,
-            "keepdims": keepdims,
-        },
-        ((a_unit, w_unit) if returned else a_unit),
-        None,
-    )



