diff --git a/sympy/printing/latex.py b/sympy/printing/latex.py
--- a/sympy/printing/latex.py
+++ b/sympy/printing/latex.py
@@ -1630,15 +1630,9 @@ def _print_Symbol(self, expr: Symbol, style='plain'):

     _print_RandomSymbol = _print_Symbol

+    def _split_super_sub(self, name: str) -> tuple[str, list[str], list[str]]:
     def _deal_with_super_sub(self, string: str, style='plain') -> str:
-        if '{' in string:
-            name, supers, subs = string, [], []
-        else:
-            name, supers, subs = split_super_sub(string)
-
-            name = translate(name)
-            supers = [translate(sup) for sup in supers]
-            subs = [translate(sub) for sub in subs]
+        name, supers, subs = self._split_super_sub(string)

         # apply the style only to the name
         if style == 'bold':