diff --git a/sympy/core/expr.py b/sympy/core/expr.py
--- a/sympy/core/expr.py
+++ b/sympy/core/expr.py
@@ -3669,8 +3669,18 @@ def _expand_hint(expr, hint, deep=True, **hints):
         return (expr, hit)
 
     @cacheit
-    def expand(self, deep=True, modulus=None, power_base=True, power_exp=True,
-            mul=True, log=True, multinomial=True, basic=True, **hints):
+    def expand(
+        self,
+        deep=True,
+        modulus=None,
+        power_base=True,
+        power_exp=True,
+        mul=True,
+        log=True,
+        multinomial=True,
+        basic=True,
+        **hints,
+    ) -> Expr:
         """
         Expand an expression using hints.
 
diff --git a/sympy/matrices/eigen.py b/sympy/matrices/eigen.py
--- a/sympy/matrices/eigen.py
+++ b/sympy/matrices/eigen.py
@@ -238,52 +238,16 @@ def _eigenvals(M,
 
 
 def _eigenvals_list(
-        M: MatrixBase,
-        error_when_incomplete: bool = True,
-        simplify: Callable[[Expr], Expr] | bool = False,
-        **flags
-    ) -> list[Expr]:
-    iblocks = M.strongly_connected_components()
-    all_eigs = []
-    is_dom = M._rep.domain in (ZZ, QQ) # type: ignore
-    for b in iblocks:
-
-        # Fast path for a 1x1 block:
-        if is_dom and len(b) == 1:
-            index = b[0]
-            val = M[index, index]
-            all_eigs.append(val)
-            continue
-
-        block = M[b, b]
-
-        if isinstance(simplify, FunctionType):
-            charpoly = block.charpoly(simplify=simplify)
-        else:
-            charpoly = block.charpoly()
-
-        factors = charpoly.factor_list()[1]
-
-        for factor, multiplicity in factors:
-            eigs = roots(factor, multiple=True, **flags)
-
-            degree = int(factor.degree())
-            if len(eigs) != degree:
-                try:
-                    eigs = factor.all_roots(multiple=True)
-                except NotImplementedError:
-                    if error_when_incomplete:
-                        raise MatrixError(eigenvals_error_message)
-                    else:
-                        eigs = []
-
-            all_eigs += eigs * multiplicity
-
-    if not simplify:
-        return all_eigs
-    if not isinstance(simplify, FunctionType):
-        simplify = _simplify
-    return [simplify(value) for value in all_eigs]
+    M: MatrixBase,
+    error_when_incomplete: bool = True,
+    simplify: Callable[[Expr], Expr] | bool = False,
+    **flags,
+) -> list[Expr]:
+    vals_dict = _eigenvals_dict(M, error_when_incomplete, simplify, **flags)
+    vals = []
+    for val, mult in vals_dict.items():
+        vals.extend([val] * mult)
+    return vals
 
 
 def _eigenvals_dict(
@@ -292,8 +256,11 @@ def _eigenvals_dict(
         simplify: Callable[[Expr], Expr] | bool = False,
         **flags: Any,
     ) -> dict[Expr, int]:
+
     iblocks = M.strongly_connected_components()
     all_eigs: dict[Expr, int] = {}
+    expanded_eigs: dict[Expr, Expr] = {}
+
     # XXX: Only RepMatrix has _rep ...
     is_dom = M._rep.domain in (ZZ, QQ) # type: ignore
     for b in iblocks:
@@ -328,6 +295,16 @@ def _eigenvals_dict(
                         eigs = {}
 
             for k, v in eigs.items():
+                # Try a bit to canonicalize the eigenvalue expressions to get
+                # the multiplicity correct. This is not robust enough in general
+                # if different subroutines in roots can return different forms
+                # for the same root.
+                k_expanded = k.expand()
+                if k_expanded in expanded_eigs:
+                    k = expanded_eigs[k_expanded]
+                else:
+                    expanded_eigs[k_expanded] = k
+
                 v_total = v * multiplicity
                 if k in all_eigs:
                     all_eigs[k] += v_total
diff --git a/sympy/matrices/utilities.py b/sympy/matrices/utilities.py
--- a/sympy/matrices/utilities.py
+++ b/sympy/matrices/utilities.py
@@ -78,7 +78,7 @@ def _is_zero_after_expand_mul(x):
     return expand_mul(x) == 0
 
 
-def _simplify(expr):
+def _simplify(expr: Expr) -> Expr:
     """ Wrapper to avoid circular imports. """
     from sympy.simplify.simplify import simplify
     return simplify(expr)
diff --git a/sympy/polys/polyroots.py b/sympy/polys/polyroots.py
--- a/sympy/polys/polyroots.py
+++ b/sympy/polys/polyroots.py
@@ -1,5 +1,8 @@
 """Algorithms for computing symbolic roots of polynomials. """
 
+from __future__ import annotations
+
+from typing import overload, Any, Literal
 
 import math
 from functools import reduce
@@ -839,6 +842,37 @@ def func(k, coeff):
     return coeff, poly
 
 
+@overload
+def roots(
+    f,
+    *gens,
+    auto: bool = True,
+    cubics: bool = True,
+    trig: bool = False,
+    quartics: bool = True,
+    quintics: bool = False,
+    multiple: Literal[False] = False,
+    filter: None = None,
+    predicate: None = None,
+    strict: bool = False,
+    **flags: Any,
+) -> dict[Expr, int]: ...
+@overload
+def roots(
+    f,
+    *gens,
+    auto: bool = True,
+    cubics: bool = True,
+    trig: bool = False,
+    quartics: bool = True,
+    quintics: bool = False,
+    multiple: Literal[True],
+    filter: None = None,
+    predicate: None = None,
+    strict: bool = False,
+    **flags: Any,
+) -> list[tuple[Expr, int]]: ...
+
 @public
 def roots(f, *gens,
         auto=True,
diff --git a/sympy/polys/polytools.py b/sympy/polys/polytools.py
--- a/sympy/polys/polytools.py
+++ b/sympy/polys/polytools.py
@@ -3668,7 +3668,18 @@ def real_roots(f, multiple=True, radicals=True):
         else:
             return group(reals, multiple=False)
 
-    def all_roots(f, multiple=True, radicals=True):
+    @overload
+    def all_roots(
+        f, multiple: Literal[True] = True, radicals: bool = True
+    ) -> list[Expr]: ...
+    @overload
+    def all_roots(
+        f, multiple: Literal[False], *, radicals: bool = True
+    ) -> list[tuple[Expr, int]]: ...
+
+    def all_roots(
+        f, multiple: bool = True, radicals: bool = True
+    ) -> list[Expr] | list[tuple[Expr, int]]:
         """
         Return a list of real and complex roots with multiplicities.
 
diff --git a/sympy/polys/polyutils.py b/sympy/polys/polyutils.py
--- a/sympy/polys/polyutils.py
+++ b/sympy/polys/polyutils.py
@@ -173,6 +173,8 @@ def _sort_factors(factors, **args):
     def order_key(factor):
         if isinstance(factor, _GF_types):
             return int(factor)
+        elif isinstance(factor, Expr):
+            return factor.sort_key()
         elif isinstance(factor, list):
             return [order_key(f) for f in factor]
         else:
diff --git a/sympy/utilities/iterables.py b/sympy/utilities/iterables.py
--- a/sympy/utilities/iterables.py
+++ b/sympy/utilities/iterables.py
@@ -1,6 +1,6 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, overload
 
 from collections import Counter, defaultdict, OrderedDict
 from itertools import (
@@ -21,7 +21,7 @@
 
 
 if TYPE_CHECKING:
-    from typing import TypeVar, Iterable, Callable, Sequence
+    from typing import TypeVar, Iterable, Callable, Literal, Sequence
     T = TypeVar('T')
 
 
@@ -197,7 +197,14 @@ def reshape(seq, how):
     return type(seq)(rv)
 
 
-def group(seq, multiple=True):
+@overload
+def group(seq: Iterable[T], multiple: Literal[True] = True) -> list[list[T]]: ...
+@overload
+def group(seq: Iterable[T], multiple: Literal[False]) -> list[tuple[T, int]]: ...
+
+def group(
+    seq: Iterable[T], multiple: bool = True
+) -> list[list[T]] | list[tuple[T, int]]:
     """
     Splits a sequence into a list of lists of equal, adjacent elements.
 