diff --git a/sympy/calculus/accumulationbounds.py b/sympy/calculus/accumulationbounds.py
--- a/sympy/calculus/accumulationbounds.py
+++ b/sympy/calculus/accumulationbounds.py
@@ -693,7 +693,7 @@
                 "Input must be AccumulationBounds or FiniteSet object")
 
         if self.min <= other.min and self.max >= other.min:
-            return AccumBounds(self.min, Max(self.max, other.max))
+            return AccumBounds(Min(self.min, other.min), Max(self.max, other.max))
 
         if other.min <= self.min and other.max >= self.min:
             return AccumBounds(other.min, Max(self.max, other.max))
