diff --git a/sympy/calculus/accumulationbounds.py b/sympy/calculus/accumulationbounds.py
--- a/sympy/calculus/accumulationbounds.py
+++ b/sympy/calculus/accumulationbounds.py
@@ -696,7 +696,7 @@
             return AccumBounds(Min(self.min, other.min), Max(self.max, other.max))
 
         if other.min <= self.min and other.max >= self.min:
-            return AccumBounds(other.min, Max(self.max, other.max))
+            return AccumBounds(Min(self.min, other.min), Max(self.max, other.max))
 
 
 @dispatch(AccumulationBounds, AccumulationBounds) # type: ignore # noqa:F811
