diff --git a/sklearn/model_selection/_classification_threshold.py b/sklearn/model_selection/_classification_threshold.py
--- a/sklearn/model_selection/_classification_threshold.py
+++ b/sklearn/model_selection/_classification_threshold.py
@@ -504,7 +504,7 @@
     into a class label. The tuning is done by optimizing a binary metric,
     potentially constrained by another metric.
 
-    Read more in the :ref:`User Guide <TunedThresholdClassifierCV>`.
+    Read more in the :ref:`User Guide <threshold_tunning>`.
 
     .. versionadded:: 1.5
 
