diff --git a/django/contrib/admin/options.py b/django/contrib/admin/options.py
--- a/django/contrib/admin/options.py
+++ b/django/contrib/admin/options.py
@@ -946,7 +946,6 @@ def log_addition(self, request, obj, message):
             queryset=[obj],
             action_flag=ADDITION,
             change_message=message,
-            single_object=True,
         )
 
     def log_change(self, request, obj, message):