diff --git a/django/db/models/fields/tuple_lookups.py b/django/db/models/fields/tuple_lookups.py
--- a/django/db/models/fields/tuple_lookups.py
+++ b/django/db/models/fields/tuple_lookups.py
@@ -329,6 +329,10 @@ def process_rhs(self, compiler, connection):
         lhs = self.lhs
 
         for vals in rhs:
+            # Remove any tuple containing None from the list as NULL is never
+            # equal to anything.
+            if any(val is None for val in vals):
+                continue
             result.append(
                 Tuple(
                     *[
@@ -338,6 +342,9 @@ def process_rhs(self, compiler, connection):
                 )
             )
 
+        if not result:
+            raise EmptyResultSet
+
         return compiler.compile(Tuple(*result))
 
     def get_fallback_sql(self, compiler, connection):
@@ -353,9 +360,15 @@ def get_fallback_sql(self, compiler, connection):
         lhs = self.lhs
 
         for vals in rhs:
+            # Remove any tuple containing None from the list as NULL is never
+            # equal to anything.
+            if any(val is None for val in vals):
+                continue
             lookups = [Exact(col, val) for col, val in zip(lhs, vals)]
             root.children.append(WhereNode(lookups, connector=AND))
 
+        if not root.children:
+            raise EmptyResultSet
         return root.as_sql(compiler, connection)
 
 