diff --git a/galleries/examples/lines_bars_and_markers/broken_barh.py b/galleries/examples/lines_bars_and_markers/broken_barh.py
--- a/galleries/examples/lines_bars_and_markers/broken_barh.py
+++ b/galleries/examples/lines_bars_and_markers/broken_barh.py
@@ -18,13 +18,13 @@
 network = np.column_stack([10*np.random.random(10), np.full(10, 0.05)])
 
 fig, ax = plt.subplots()
-# broken_barh(xranges, (ymin, height))
-ax.broken_barh(cpu_1, (-0.2, 0.4))
-ax.broken_barh(cpu_2, (0.8, 0.4))
-ax.broken_barh(cpu_3, (1.8, 0.4))
-ax.broken_barh(cpu_4, (2.8, 0.4))
-ax.broken_barh(disk, (3.8, 0.4), color="tab:orange")
-ax.broken_barh(network, (4.8, 0.4), color="tab:green")
+# broken_barh(xranges, (ypos, height))
+ax.broken_barh(cpu_1, (0, 0.4), align="center")
+ax.broken_barh(cpu_2, (1, 0.4), align="center")
+ax.broken_barh(cpu_3, (2, 0.4), align="center")
+ax.broken_barh(cpu_4, (3, 0.4), align="center")
+ax.broken_barh(disk, (4, 0.4), align="center", color="tab:orange")
+ax.broken_barh(network, (5, 0.4), align="center", color="tab:green")
 ax.set_xlim(0, 10)
 ax.set_yticks(range(6),
               labels=["CPU 1", "CPU 2", "CPU 3", "CPU 4", "disk", "network"])