diff --git a/sklearn/metrics/_classification.py b/sklearn/metrics/_classification.py
--- a/sklearn/metrics/_classification.py
+++ b/sklearn/metrics/_classification.py
@@ -3789,24 +3789,12 @@
         labels=labels,
     )
 
-    # Proportion of labels in the dataset
-    weights = _check_sample_weight(sample_weight, y_true)
-
-    # If labels is passed, augment y_true to ensure that all labels are represented
-    # Use 0 weight for the new samples to not affect the counts
-    y_true_, weights_ = (
-        (
-            np.concatenate([y_true, labels]),
-            np.concatenate([weights, np.zeros_like(weights, shape=len(labels))]),
-        )
-        if labels is not None
-        else (y_true, weights)
+    y_pred = check_array(y_pred, ensure_2d=False, dtype="numeric")
+    transformed_labels, _ = _validate_multiclass_probabilistic_prediction(
+        y_true, y_pred, sample_weight, labels
     )
-
-    _, y_value_indices = np.unique(y_true_, return_inverse=True)
-    counts = np.bincount(y_value_indices, weights=weights_)
-    y_prob = counts / weights.sum()
-    y_pred_null = np.tile(y_prob, (len(y_true), 1))
+    y_pred_null = np.average(transformed_labels, axis=0, weights=sample_weight)
+    y_pred_null = np.tile(y_pred_null, (len(y_true), 1))
 
     # log loss of the null model
     denominator = log_loss(
