diff --git a/django/db/backends/base/creation.py b/django/db/backends/base/creation.py
--- a/django/db/backends/base/creation.py
+++ b/django/db/backends/base/creation.py
@@ -21,6 +21,8 @@
     Encapsulate backend-specific differences pertaining to creation and
     destruction of the test database.
     """
+
+    destroy_test_db_connection_close_method = None
 
     def __init__(self, connection):
         self.connection = connection
