diff --git a/examples/neighbors/approximate_nearest_neighbors.py b/examples/neighbors/approximate_nearest_neighbors.py
--- a/examples/neighbors/approximate_nearest_neighbors.py
+++ b/examples/neighbors/approximate_nearest_neighbors.py
@@ -130,11 +130,11 @@
 n_neighbors = int(3.0 * perplexity + 1) + 1
 
 tsne_params = dict(
-    init="random",  # pca not supported for sparse matrices
+    init="random",  # pca cannot be used with precomputed distances
     perplexity=perplexity,
     method="barnes_hut",
     random_state=42,
-    n_iter=n_iter,
+    max_iter=max_iter,
     learning_rate="auto",
 )
 
