diff --git a/sklearn/preprocessing/_encoders.py b/sklearn/preprocessing/_encoders.py
--- a/sklearn/preprocessing/_encoders.py
+++ b/sklearn/preprocessing/_encoders.py
@@ -245,14 +245,20 @@ def _transform(
             # already called above.
             X_int[:, i] = _encode(Xi, uniques=self.categories_[i], check_unknown=False)
         if columns_with_unknown:
-            warnings.warn(
-                (
+            if handle_unknown == "infrequent_if_exist":
+                msg = (
+                    "Found unknown categories in columns "
+                    f"{columns_with_unknown} during transform. These "
+                    "unknown categories will be encoded as the "
+                    "infrequent category."
+                )
+            else:
+                msg = (
                     "Found unknown categories in columns "
                     f"{columns_with_unknown} during transform. These "
                     "unknown categories will be encoded as all zeros"
-                ),
-                UserWarning,
-            )
+                )
+            warnings.warn(msg, UserWarning)
 
         self._map_infrequent_categories(X_int, X_mask, ignore_category_indices)
         return X_int, X_mask
@@ -436,7 +442,7 @@ def _map_infrequent_categories(self, X_int, X_mask, ignore_category_indices):
                 continue
 
             X_int[~X_mask[:, col_idx], col_idx] = infrequent_idx[0]
-            if self.handle_unknown == "infrequent_if_exist":
+            if self.handle_unknown in ("infrequent_if_exist", "warn"):
                 # All the unknown values are now mapped to the
                 # infrequent_idx[0], which makes the unknown values valid
                 # This is needed in `transform` when the encoding is formed