diff --git a/lib/matplotlib/axes/_base.py b/lib/matplotlib/axes/_base.py
--- a/lib/matplotlib/axes/_base.py
+++ b/lib/matplotlib/axes/_base.py
@@ -4596,7 +4596,7 @@ def _make_twin_axes(self, *args, **kwargs):
         self._twinned_axes.join(self, twin)
         return twin
 
-    def twinx(self, **kwargs):
+    def twinx(self, axes_class=None, **kwargs):
         """
         Create a twin Axes sharing the xaxis.
 
@@ -4608,6 +4608,11 @@ def twinx(self, **kwargs):
 
         Parameters
         ----------
+        axes_class : subclass type of `~.axes.Axes`, optional
+            The `.axes.Axes` subclass that is instantiated.  This parameter
+            is incompatible with *projection* and *polar*.  See
+            :ref:`axisartist_users-guide-index` for examples.
+
         kwargs : dict
             The keyword arguments passed to ``add_subplot()`` or ``add_axes()``.
 
@@ -4621,9 +4626,9 @@ def twinx(self, **kwargs):
         For those who are 'picking' artists while using twinx, pick
         events are only called for the artists in the top-most Axes.
         """
-        if not {"projection", "polar", "axes_class"}.intersection(kwargs):
-            kwargs["axes_class"] = type(self)
-        ax2 = self._make_twin_axes(sharex=self, axes_class=type(self), **kwargs)
+        if axes_class:
+            kwargs["axes_class"] = axes_class
+        ax2 = self._make_twin_axes(sharex=self, **kwargs)
         ax2.yaxis.tick_right()
         ax2.yaxis.set_label_position('right')
         ax2.yaxis.set_offset_position('right')
@@ -4634,7 +4639,7 @@ def twinx(self, **kwargs):
         ax2.xaxis.units = self.xaxis.units
         return ax2
 
-    def twiny(self, **kwargs):
+    def twiny(self, axes_class=None, **kwargs):
         """
         Create a twin Axes sharing the yaxis.
 
@@ -4646,6 +4651,11 @@ def twiny(self, **kwargs):
 
         Parameters
         ----------
+        axes_class : subclass type of `~.axes.Axes`, optional
+            The `.axes.Axes` subclass that is instantiated.  This parameter
+            is incompatible with *projection* and *polar*.  See
+            :ref:`axisartist_users-guide-index` for examples.
+
         kwargs : dict
             The keyword arguments passed to ``add_subplot()`` or ``add_axes()``.
 
@@ -4659,9 +4669,9 @@ def twiny(self, **kwargs):
         For those who are 'picking' artists while using twiny, pick
         events are only called for the artists in the top-most Axes.
         """
-        if not {"projection", "polar", "axes_class"}.intersection(kwargs):
-            kwargs["axes_class"] = type(self)
-        ax2 = self._make_twin_axes(sharey=self, axes_class=type(self), **kwargs)
+        if axes_class:
+            kwargs["axes_class"] = axes_class
+        ax2 = self._make_twin_axes(sharey=self, **kwargs)
         ax2.xaxis.tick_top()
         ax2.xaxis.set_label_position('top')
         ax2.set_autoscaley_on(self.get_autoscaley_on())