diff --git a/sympy/core/mul.py b/sympy/core/mul.py
--- a/sympy/core/mul.py
+++ b/sympy/core/mul.py
@@ -1480,12 +1480,11 @@ def _eval_is_integer(self):
         if len(denominators) == 1:
             d = denominators[0]
             if d.is_Integer:
-                t = trailing(d.p)
-                is_power_of_two = (d == 2**t)
+                is_power_of_two = d.p & (d.p - 1) == 0
                 # if minimal power of 2 in num vs den is not
                 # negative then we have an integer
                 if is_power_of_two and (Add(*[i.as_base_exp()[1] for i in
-                        numerators if i.is_even]) - t
+                        numerators if i.is_even]) - trailing(d.p)
                         ).is_nonnegative:
                     return True
         if len(numerators) == 1:
