diff --git a/sklearn/manifold/_mds.py b/sklearn/manifold/_mds.py
--- a/sklearn/manifold/_mds.py
+++ b/sklearn/manifold/_mds.py
@@ -275,14 +275,18 @@ def smacof(
         Whether or not to return the number of iterations.
 
     normalized_stress : bool or "auto" default="auto"
-        Whether use and return normed stress value (Stress-1) instead of raw
-        stress calculated by default. Only supported in non-metric MDS.
+        Whether to return normalized stress value (Stress-1) instead of raw
+        stress. By default, metric MDS returns raw stress while non-metric MDS
+        returns normalized stress.
 
         .. versionadded:: 1.2
 
         .. versionchanged:: 1.4
            The default value changed from `False` to `"auto"` in version 1.4.
 
+        .. versionchanged:: 1.7
+           Normalized stress is now supported for metric MDS as well.
+
     Returns
     -------
     X : ndarray of shape (n_samples, n_components)
@@ -291,7 +295,7 @@ def smacof(
     stress : float
         The final value of the stress (sum of squared distance of the
         disparities and the distances for all constrained points).
-        If `normalized_stress=True`, and `metric=False` returns Stress-1.
+        If `normalized_stress=True`, returns Stress-1.
         A value of 0 indicates "perfect" fit, 0.025 excellent, 0.05 good,
         0.1 fair, and 0.2 poor [1]_.
 
@@ -318,12 +322,12 @@ def smacof(
     >>> X = np.array([[0, 1, 2], [1, 0, 3],[2, 3, 0]])
     >>> dissimilarities = euclidean_distances(X)
     >>> mds_result, stress = smacof(dissimilarities, n_components=2, random_state=42)
-    >>> mds_result
-    array([[ 0.05... -1.07... ],
-           [ 1.74..., -0.75...],
-           [-1.79...,  1.83...]])
-    >>> stress
-    np.float64(0.0012...)
+    >>> np.round(mds_result, 5)
+    array([[ 0.05352, -1.07253],
+           [ 1.74231, -0.75675],
+           [-1.79583,  1.82928]])
+    >>> np.round(stress, 5).item()
+    0.00128
     """
 
     dissimilarities = check_array(dissimilarities)
@@ -332,11 +336,6 @@ def smacof(
     if normalized_stress == "auto":
         normalized_stress = not metric
 
-    if normalized_stress and metric:
-        raise ValueError(
-            "Normalized stress is not supported for metric MDS. Either set"
-            " `normalized_stress=False` or use `metric=False`."
-        )
     if hasattr(init, "__array__"):
         init = np.asarray(init).copy()
         if not n_init == 1:
@@ -449,14 +449,18 @@ class MDS(BaseEstimator):
             ``fit_transform``.
 
     normalized_stress : bool or "auto" default="auto"
-        Whether use and return normed stress value (Stress-1) instead of raw
-        stress calculated by default. Only supported in non-metric MDS.
+        Whether use and return normalized stress value (Stress-1) instead of raw
+        stress. By default, metric MDS uses raw stress while non-metric MDS uses
+        normalized stress.
 
         .. versionadded:: 1.2
 
         .. versionchanged:: 1.4
            The default value changed from `False` to `"auto"` in version 1.4.
 
+        .. versionchanged:: 1.7
+           Normalized stress is now supported for metric MDS as well.
+
     Attributes
     ----------
     embedding_ : ndarray of shape (n_samples, n_components)
@@ -465,7 +469,7 @@ class MDS(BaseEstimator):
     stress_ : float
         The final value of the stress (sum of squared distance of the
         disparities and the distances for all constrained points).
-        If `normalized_stress=True`, and `metric=False` returns Stress-1.
+        If `normalized_stress=True`, returns Stress-1.
         A value of 0 indicates "perfect" fit, 0.025 excellent, 0.05 good,
         0.1 fair, and 0.2 poor [1]_.
 