diff --git a/django/db/models/sql/query.py b/django/db/models/sql/query.py
--- a/django/db/models/sql/query.py
+++ b/django/db/models/sql/query.py
@@ -1214,9 +1214,19 @@
         if "%" in alias:
             if "aggregate" in {frame.function for frame in inspect.stack()}:
                 stacklevel = 5
+            else:
+                # annotate() and alias().
+                stacklevel = 6
+            warnings.warn(
+                "Using percent signs in a column alias is deprecated.",
+                stacklevel=stacklevel,
+                category=RemovedInDjango70Warning,
+            )
         if FORBIDDEN_ALIAS_PATTERN.search(alias):
             raise ValueError(
                 "Column aliases cannot contain whitespace characters, quotation marks, "
+                # RemovedInDjango70Warning: When the deprecation ends, replace with:
+                # "semicolons, percent signs, or SQL comments."
                 "semicolons, or SQL comments."
             )
 
