diff --git a/astropy/table/table.py b/astropy/table/table.py
--- a/astropy/table/table.py
+++ b/astropy/table/table.py
@@ -2781,6 +2781,7 @@
         """
         # If the table has been sliced then each index will have original=False
         # indicating that the data are a sliced reference (not from the original table).
+        sliced =
 
         keep_mask = np.ones(len(self), dtype=bool)
         keep_mask[row_specifier] = False
