diff --git a/xarray/coding/frequencies.py b/xarray/coding/frequencies.py
--- a/xarray/coding/frequencies.py
+++ b/xarray/coding/frequencies.py
@@ -48,6 +48,7 @@
 from xarray.coding.cftime_offsets import _MONTH_ABBREVIATIONS, _legacy_to_new_freq
 from xarray.coding.cftimeindex import CFTimeIndex
 from xarray.core.common import _contains_datetime_like_objects
+from xarray.core.dtypes import _is_numpy_subdtype
 
 _ONE_MICRO = 1
 _ONE_MILLI = _ONE_MICRO * 1000

