diff --git a/sklearn/ensemble/_bagging.py b/sklearn/ensemble/_bagging.py
--- a/sklearn/ensemble/_bagging.py
+++ b/sklearn/ensemble/_bagging.py
@@ -519,20 +522,7 @@
         if max_samples is None:
             max_samples = self.max_samples
 
-        if not isinstance(max_samples, numbers.Integral):
-            if sample_weight is None:
-                max_samples = max(int(max_samples * X.shape[0]), 1)
-            else:
-                sw_sum = np.sum(sample_weight)
-                if sw_sum <= 1:
-                    raise ValueError(
-                        f"The total sum of sample weights is {sw_sum}, which prevents "
-                        "resampling with a fractional value for max_samples="
-                        f"{max_samples}. Either pass max_samples as an integer or "
-                        "use a larger sample_weight."
-                    )
-                max_samples = max(int(max_samples * sw_sum), 1)
-
+        max_samples = _get_n_samples_bootstrap(X.shape[0], max_samples, sample_weight)
         if not self.bootstrap and max_samples > X.shape[0]:
             raise ValueError(
                 f"Effective max_samples={max_samples} must be <= n_samples="
