diff --git a/sphinx/config.py b/sphinx/config.py
--- a/sphinx/config.py
+++ b/sphinx/config.py
@@ -895,6 +895,11 @@
     return changed
 
 
+def deprecate_source_encoding(_app: Sphinx, config: Config) -> None:
+    """Warn on non-UTF 8 source_encoding."""
+    # RemovedInSphinx10Warning
+
+
 def setup(app: Sphinx) -> ExtensionMetadata:
     app.connect('config-inited', deprecate_source_encoding, priority=790)
     app.connect('config-inited', convert_source_suffix, priority=800)
