diff --git a/lib/matplotlib/axes/_axes.py b/lib/matplotlib/axes/_axes.py
--- a/lib/matplotlib/axes/_axes.py
+++ b/lib/matplotlib/axes/_axes.py
@@ -3761,7 +3761,7 @@
             # platforms), we select with `err==err` (which is False for nan).
             # Also, since datetime.timedelta cannot be compared with 0,
             # we compare with the negative error instead.
-            if
+            if np.any((check := err[err == err]) < -check):
                 raise ValueError(
                     f"'{dep_axis}err' must not contain negative values")
             # This is like
