diff --git a/astropy/io/fits/fitsrec.py b/astropy/io/fits/fitsrec.py
--- a/astropy/io/fits/fitsrec.py
+++ b/astropy/io/fits/fitsrec.py
@@ -478,6 +478,15 @@
         return data
 
     def __repr__(self):
+        # recarray.__repr__ hard-codes the name of the class, so we overwrite.
+        # The following is mostly a straight copy except for the name change
+        # and for treating using str to typeset integer -- the latter to fix
+        # the case where the integer columns are scaled (see gh-17583). Also,
+        # removed a branch for "if the user is playing strange game with dtypes".
+        #
+        # FIXME: recarray removes the "numpy.record" mention in the dtype repr,
+        # we could do the same in a future version
+
         repr_dtype = self.dtype
         # if repr_dtype.type is np.record:
         #     repr_dtype = np.dtype((np.void, repr_dtype))
@@ -485,7 +494,10 @@
         fmt = "FITS_rec(%s,%sdtype=%s)"
         # get data/shape string. logic taken from numeric.array_repr
         if self.size > 0 or self.shape == (0,):
-            lst =
+            lst = np.array2string(
+                self, separator=", ", prefix=prefix, suffix=",", formatter=dict(int=str)
+            )
+        else:
 
     def __getattribute__(self, attr):
         # First, see if ndarray has this attr, and return it if so. Note that
