diff --git a/sympy/vector/functions.py b/sympy/vector/functions.py
--- a/sympy/vector/functions.py
+++ b/sympy/vector/functions.py
@@ -70,11 +70,11 @@
             raise ValueError("system2 should not be provided for \
                                 Vectors")
         # Given expr is a Vector
+        subs_dict = {}
         if variables:
             # If variables attribute is True, substitute
             # the coordinate variables in the Vector
             system_list = {x.system for x in expr.atoms(BaseScalar, BaseVector)} - {system}
-            subs_dict = {}
             for f in system_list:
                 subs_dict.update(f.scalar_map(system))
             expr = expr.subs(subs_dict)
@@ -83,10 +83,11 @@
         parts = expr.separate()
         for x in parts:
             if x != system:
-                temp = system.rotation_matrix(x) * parts[x].to_matrix(x)
+                temp = system.change_of_basis_matrix_from(x) * parts[x].to_matrix(x)
                 outvec += matrix_to_vector(temp, system)
             else:
                 outvec += parts[x]
+        outvec = outvec.subs(subs_dict)
         return outvec
 
     elif isinstance(expr, Dyadic):
