diff --git a/astropy/io/fits/connect.py b/astropy/io/fits/connect.py
--- a/astropy/io/fits/connect.py
+++ b/astropy/io/fits/connect.py
@@ -432,7 +432,7 @@ def _encode_mixins(tbl):
     return encode_tbl
 
 
-def write_table_fits(input, output, overwrite=False, append=False):
+def write_table_fits(input, output, overwrite=False, append=False, name=None):
     """
     Write a Table object to a FITS file.
 
@@ -446,11 +446,14 @@ def write_table_fits(input, output, overwrite=False, append=False):
         Whether to overwrite any existing file without warning.
     append : bool
         Whether to append the table to an existing file
+    name : str
+        Name to be populated in ``EXTNAME`` keyword.
+
     """
     # Encode any mixin columns into standard Columns.
     input = _encode_mixins(input)
 
-    table_hdu = table_to_hdu(input, character_as_bytes=True)
+    table_hdu = table_to_hdu(input, character_as_bytes=True, name=name)
 
     # Check if output file already exists
     if isinstance(output, (str, os.PathLike)) and os.path.exists(output):
diff --git a/astropy/io/fits/convenience.py b/astropy/io/fits/convenience.py
--- a/astropy/io/fits/convenience.py
+++ b/astropy/io/fits/convenience.py
@@ -466,7 +466,7 @@ def writeto(
     )
 
 
-def table_to_hdu(table, character_as_bytes=False):
+def table_to_hdu(table, character_as_bytes=False, name=None):
     """
     Convert an `~astropy.table.Table` object to a FITS
     `~astropy.io.fits.BinTableHDU`.
@@ -479,6 +479,8 @@ def table_to_hdu(table, character_as_bytes=False):
         Whether to return bytes for string columns when accessed from the HDU.
         By default this is `False` and (unicode) strings are returned, but for
         large tables this may use up a lot of memory.
+    name : str
+        Name to be populated in ``EXTNAME`` keyword.
 
     Returns
     -------
@@ -560,6 +562,9 @@ def table_to_hdu(table, character_as_bytes=False):
             tarray, header=hdr, character_as_bytes=character_as_bytes
         )
 
+    if name is not None:
+        table_hdu.name = name
+
     # Set units and format display for output HDU
     for col in table_hdu.columns:
         if table[col.name].info.format is not None: