diff --git a/sphinx/writers/latex.py b/sphinx/writers/latex.py
--- a/sphinx/writers/latex.py
+++ b/sphinx/writers/latex.py
@@ -1367,19 +1367,25 @@ def visit_entry(self, node: Element) -> None:
                 r'\sphinxmultirow{%d}{%d}{%%' % (cell.height, cell.cell_id) + CR
             )
             context = '}%' + CR + context
-        if cell.width > 1 or cell.height > 1:
-            self.body.append(
-                r'\begin{varwidth}[t]{\sphinxcolwidth{%d}{%d}}'
-                % (cell.width, self.table.colcount)
-                + CR
-            )
-            context = (
-                r'\par' + CR + r'\vskip-\baselineskip'
-                r'\vbox{\hbox{\strut}}\end{varwidth}%' + CR + context
-            )
-            self.table.entry_needs_linetrimming = 1
-        if len(list(node.findall(nodes.paragraph))) >= 2:
-            self.table.has_oldproblematic = True
+        # 8.3.0 wraps ALL cells contents in "varwidth".  This fixes a
+        # number of issues and allows more usage of tabulary.
+        #
+        # "varwidth" usage allows a *tight fit* to multiple paragraphs,
+        # line blocks, bullet lists, enumerated lists; it is less
+        # successful at finding a tight fit for object descriptions or
+        # admonitions: the table will then probably occupy full-width, and
+        # columns containing such cells will auto-divide the total width
+        # equally.
+        #
+        # "\sphinxcolwidth" has an appropriate definition in
+        # sphinxlatextables.sty which in particular takes into account
+        # tabulary "two-pass" system.
+        self.body.append(
+            r'\begin{varwidth}[t]{\sphinxcolwidth{%d}{%d}}'
+            % (cell.width, self.table.colcount)
+            + CR
+        )
+        context = r'\sphinxbeforeendvarwidth' + CR + r'\end{varwidth}%' + CR + context
         if (
             isinstance(node.parent.parent, nodes.thead)
             or (cell.col in self.table.stubs)
@@ -1392,8 +1398,7 @@ def visit_entry(self, node: Element) -> None:
                 pass
             else:
                 self.body.append(r'\sphinxstyletheadfamily ')
-        if self.table.entry_needs_linetrimming:
-            self.pushbody([])
+        self.pushbody([])
         self.context.append(context)
 
     def depart_entry(self, node: Element) -> None: