diff --git a/django/core/mail/__init__.py b/django/core/mail/__init__.py
--- a/django/core/mail/__init__.py
+++ b/django/core/mail/__init__.py
@@ -109,8 +109,22 @@
     return mail.send()
 
 
+@deprecate_posargs(
+    RemovedInDjango70Warning,
+    [
+        "fail_silently",
+        "auth_user",
+        "auth_password",
+        "connection",
+    ],
+)
 def send_mass_mail(
-    datatuple, fail_silently=False, auth_user=None, auth_password=None, connection=None
+    datatuple,
+    *,
+    fail_silently=False,
+    auth_user=None,
+    auth_password=None,
+    connection=None,
 ):
     """
     Given a datatuple of (subject, message, from_email, recipient_list), send
