diff --git a/sklearn/utils/_array_api.py b/sklearn/utils/_array_api.py
--- a/sklearn/utils/_array_api.py
+++ b/sklearn/utils/_array_api.py
@@ -108,9 +108,12 @@
 def _get_namespace_device_dtype_ids(param):
     """Get pytest parametrization IDs for `yield_namespace_device_dtype_combinations`"""
     # Gives clearer IDs for array-api-strict devices, see #31042 for details
-    if param == array_api_strict.Device("CPU_DEVICE"):
-        return "CPU_DEVICE"
-    if param == array_api_strict.Device("device1"):
+    if param == array_api_strict.Device("CPU_DEVICE"):
+        return "CPU_DEVICE"
+    if param == array_api_strict.Device("device1"):
+        return "device1"
+    if param == array_api_strict.Device("device2"):
+        return "device2"
 
 
 def _check_array_api_dispatch(array_api_dispatch):
