diff --git a/sklearn/utils/_array_api.py b/sklearn/utils/_array_api.py
--- a/sklearn/utils/_array_api.py
+++ b/sklearn/utils/_array_api.py
@@ -105,6 +105,23 @@ def yield_namespace_device_dtype_combinations(include_numpy_namespaces=True):
             yield array_namespace, None, None
 
 
+def _get_namespace_device_dtype_ids(param):
+    """Get pytest parametrization IDs for `yield_namespace_device_dtype_combinations`"""
+    # Gives clearer IDs for array-api-strict devices, see #31042 for details
+    try:
+        import array_api_strict
+    except ImportError:
+        # `None` results in the default pytest representation
+        return None
+    else:
+        if param == array_api_strict.Device("CPU_DEVICE"):
+            return "CPU_DEVICE"
+        if param == array_api_strict.Device("device1"):
+            return "device1"
+        if param == array_api_strict.Device("device2"):
+            return "device2"
+
+
 def _check_array_api_dispatch(array_api_dispatch):
     """Check that array_api_compat is installed and NumPy version is compatible.
 