diff --git a/astropy/units/quantity_helper/function_helpers.py b/astropy/units/quantity_helper/function_helpers.py
--- a/astropy/units/quantity_helper/function_helpers.py
+++ b/astropy/units/quantity_helper/function_helpers.py
@@ -36,7 +36,6 @@
 
 import functools
 import operator
-from collections import Counter
 
 import numpy as np
 from numpy.lib import recfunctions as rfn
@@ -539,46 +538,7 @@ def arange(start_or_stop, /, stop=None, step=1, *, dtype=None):
         return arange_impl(start_or_stop, stop=stop, step=step, dtype=dtype)
 
 
-Missing = object()
-
-
-def unwrap_arange_args_v1(
-    *args, start_=Missing, stop_=Missing, step_=Missing, dtype_=Missing
-):
-    n_args = len(args)
-    n_missing = Counter((start_, stop_, step_, dtype_))[Missing]
-
-    assert n_args <= 4
-    assert len(args) == n_missing
-
-    if dtype_ is Missing:
-        dtype = None
-    else:
-        dtype = dtype_
-
-    # bind positional arguments to their meaningful names
-    # following the (complex) logic of np.arange
-    match args:
-        case (pos1,):
-            assert stop_ is Missing or start_ is Missing
-            if stop_ is Missing:
-                stop = pos1
-            elif start_ is Missing:
-                start = pos1
-        case start, stop, *rest:
-            if start is not Missing and stop is Missing:
-                start, stop = stop, start
-            match rest:
-                # rebind step and dtype if possible
-                case (step,):
-                    pass
-                case step, dtype:
-                    assert dtype_ is Missing
-
-    return start, stop, step, dtype
-
-
-def unwrap_arange_args_v2(*, start_or_stop, stop_, step_):
+def unwrap_arange_args(*, start_or_stop, stop_, step_):
     # handle the perilous task of disentangling original arguments
     # This isn't trivial because start_or_stop may actually bind to two
     # different inputs, as the name suggests.
@@ -601,26 +561,19 @@ def unwrap_arange_args_v2(*, start_or_stop, stop_, step_):
 def wrap_arange_args(*, start, stop, step, expected_out_unit):
     # do the reverse operation than unwrap_arange_args
     # this is needed because start_or_stop *must* be passed as positional
-    # (starting in numpy 2.4)
 
     # purely defensive programming
     assert stop is not None, "Please report this."
 
-    match start, stop, step:
-        case (None, _, 1):
-            qty_args = (stop,)
-            kwargs = {}
-        case (_, _, 1):
-            qty_args = (start, stop)
-            kwargs = {}
-        case (None, _, _):
+    match start, stop:
+        case (None, _):
             qty_args = (stop,)
-            kwargs = {"step": step}
         case _:
             qty_args = (start, stop)
-            kwargs = {"step": step}
 
-    # reverse positional arguments so `stop`` always comes first
+    kwargs = {} if step == 1 else {"step": step}
+
+    # reverse positional arguments so `stop` always comes first
     # this is done to ensure that the arrays are first converted to the
     # expected unit, which we guarantee should be stop's
     args_rev, out_unit = _quantities2arrays(*reversed(qty_args))
@@ -639,10 +592,17 @@ def wrap_arange_args(*, start, stop, step, expected_out_unit):
 
 def arange_impl(start_or_stop, /, *, stop, step, dtype, device=None):
     # Because this wrapper requires exceptional amounts of additional logic
-    # to decode/encode its complicated signature, we'll sprinkle a few
+    # to unwrap/wrap its complicated signature, we'll sprinkle a few
     # sanity checks in the form of `assert` statements, which should help making
     # heads or tails of what's happening in the event of an unexpected exception.
-    start, stop, step = unwrap_arange_args_v2(
+    #
+    # also note that we intentionally choose to match numpy.arange's signature
+    # at typecheck time, as opposed to its actual runtime signature, which is
+    # even richer (as of numpy 2.4). For instance, this means we don't support
+    # `start` being passed as keyword, or `dtype` being passed as positional.
+    # This is done to improve the overall stability and maintainability of this
+    # complicated wrapper function.
+    start, stop, step = unwrap_arange_args(
         start_or_stop=start_or_stop, stop_=stop, step_=step
     )
     out_unit = getattr(stop, "unit", UNIT_FROM_LIKE_ARG)
