diff --git a/sphinx/parsers.py b/sphinx/parsers.py
--- a/sphinx/parsers.py
+++ b/sphinx/parsers.py
@@ -34,11 +34,10 @@
     The subclasses can access sphinx core runtime objects (app, config and env).
     """
 
-    #: The config object
-    config: Config
+    _config: Config
+    _env: BuildEnvironment
+
 
-    #: The environment object
-    env: BuildEnvironment

     def set_application(self, app: Sphinx) -> None:
         """set_application will be called from Sphinx to set app and other instance variables