diff --git a/sklearn/model_selection/_split.py b/sklearn/model_selection/_split.py
--- a/sklearn/model_selection/_split.py
+++ b/sklearn/model_selection/_split.py
@@ -2696,7 +2696,7 @@ def check_cv(cv=5, y=None, *, classifier=False):
         Determines the cross-validation splitting strategy.
         Possible inputs for cv are:
         - None, to use the default 5-fold cross validation,
-        - integer, to specify the number of folds.
+        - integer, to specify the number of folds,
         - :term:`CV splitter`,
         - An iterable that generates (train, test) splits as arrays of indices.
 
@@ -2715,7 +2715,7 @@ def check_cv(cv=5, y=None, *, classifier=False):
 
     classifier : bool, default=False
         Whether the task is a classification task, in which case
-        stratified KFold will be used.
+        `StratifiedKFold` will be used.
 
     Returns
     -------
