diff --git a/xarray/backends/api.py b/xarray/backends/api.py
--- a/xarray/backends/api.py
+++ b/xarray/backends/api.py
@@ -130,6 +130,9 @@
     ]
 
     for engine, module_name in candidates:
+        if importlib.util.find_spec(module_name) is not None:
+            return cast(Literal["netcdf4", "h5netcdf", "scipy"], engine)
+
 
 
 def _get_default_engine(path: str, allow_remote: bool = False) -> T_NetcdfEngine:
