diff --git a/astropy/modeling/utils.py b/astropy/modeling/utils.py
--- a/astropy/modeling/utils.py
+++ b/astropy/modeling/utils.py
@@ -314,3 +314,24 @@ def add(self, operator_name, operator):
         self._set_value(key, operator)
 
         return key
+
+
+@overload
+def quantity_asanyarray(a: Sequence[int]) -> NDArray[np.integer]: ...
+@overload
+def quantity_asanyarray(a: Sequence[int], dtype: DType) -> NDArray[DType]: ...
+@overload
+def quantity_asanyarray(a: Sequence[u.Quantity]) -> u.Quantity: ...
+def quantity_asanyarray(
+    a: Sequence[int] | Sequence[u.Quantity], dtype: DType | None = None
+) -> NDArray[np.integer] | NDArray[DType] | u.Quantity:
+    if (
+        not isinstance(a, np.ndarray)
+        and not np.isscalar(a)
+        and any(isinstance(x, u.Quantity) for x in a)
+    ):
+        return u.Quantity(a, dtype=dtype)
+    else:
+        # skip over some dtype deprecation.
+        dtype = np.float64 if dtype is np.inexact else dtype
+        return np.asanyarray(a, dtype=dtype)
