diff --git a/sklearn/inspection/_plot/decision_boundary.py b/sklearn/inspection/_plot/decision_boundary.py
--- a/sklearn/inspection/_plot/decision_boundary.py
+++ b/sklearn/inspection/_plot/decision_boundary.py
@@ -267,7 +267,7 @@ def plot(self, plot_method="contourf", ax=None, xlabel=None, ylabel=None, **kwar
                 else:
                     cmap = plt.get_cmap(cmap, self.n_classes)
                     if not hasattr(cmap, "colors"):
-                        # For LinearSegmentedColormap
+                        # Get `LinearSegmentedColormap` for non-qualitative cmaps
                         colors = cmap(np.linspace(0, 1, self.n_classes))
                     else:
                         colors = cmap.colors
