diff --git a/sklearn/inspection/_plot/decision_boundary.py b/sklearn/inspection/_plot/decision_boundary.py
--- a/sklearn/inspection/_plot/decision_boundary.py
+++ b/sklearn/inspection/_plot/decision_boundary.py
@@ -300,7 +300,7 @@ def plot(self, plot_method="contourf", ax=None, xlabel=None, ylabel=None, **kwar
                     if self.response.ndim == 2:  # predict
                         response = np.ma.array(
                             (self.response == class_idx),
-                            mask=~(self.response == class_idx),
+                            mask=(self.response != class_idx),
                         )
 
                     else:  # predict_proba or decision_function
