diff --git a/django/contrib/admin/options.py b/django/contrib/admin/options.py
--- a/django/contrib/admin/options.py
+++ b/django/contrib/admin/options.py
@@ -2050,8 +2050,6 @@ def changelist_view(self, request, extra_context=None):
             # me back" button on the action confirmation page.
             return HttpResponseRedirect(request.get_full_path())
 
-        formset = None
-
         # Handle POSTed bulk-edit data.
         if request.method == "POST" and cl.list_editable and "_save" in request.POST:
             if not self.has_change_permission(request):
