diff --git a/django/utils/http.py b/django/utils/http.py
--- a/django/utils/http.py
+++ b/django/utils/http.py
@@ -323,15 +323,15 @@ def escape_leading_slashes(url):
 
     If `line` is longer than `max_length`, `ValueError` is raised.
     """
-    if max_length is not None and line and len(line) > max_length:
-        raise ValueError("Unable to parse header parameters (value too long).")
+    if not line:
+        return "", {}
 
-    m = Message()
-    m["content-type"] = line
-    params = m.get_params()
+    if max_length is not None and len(line) > max_length:
+        raise ValueError("Unable to parse header parameters (value too long).")
 
+    parts = _parseparam(";" + line)
+    key = parts.__next__().lower()
     pdict = {}
-    key = params.pop(0)[0].lower()
     for name, value in params:
         if not name:
             continue