diff --git a/sphinx/ext/napoleon/docstring.py b/sphinx/ext/napoleon/docstring.py
--- a/sphinx/ext/napoleon/docstring.py
+++ b/sphinx/ext/napoleon/docstring.py
@@ -948,9 +948,8 @@
         lines: list[str] = []
         for _name, _type, _desc in self._consume_fields(parse_type=False):
             lines.append(f'.. method:: {_name}')
-            if self._opt:
-                if 'no-index' in self._opt or 'noindex' in self._opt:
-                    lines.append('   :no-index:')
+            if self._no_index:
+                lines.append('   :no-index:')
             if _desc:
                 lines.extend(['', *self._indent(_desc, 3)])
             lines.append('')
