diff --git a/sympy/functions/elementary/piecewise.py b/sympy/functions/elementary/piecewise.py
--- a/sympy/functions/elementary/piecewise.py
+++ b/sympy/functions/elementary/piecewise.py
@@ -577,8 +577,9 @@ def handler(ipw):
                     return Undefined
                 # TODO simplify hi <= upto
                 return Piecewise((sum, hi <= upto), (Undefined, True))
-            sum += abei[i][-2]._eval_interval(x, a, b)
+            k = abei[i][-2]._eval_interval(x, a, b)
+            sum += k
             upto = b
         return sum
 
     def _intervals(self, sym, err_on_Eq=False):

