diff --git a/django/db/migrations/operations/models.py b/django/db/migrations/operations/models.py
--- a/django/db/migrations/operations/models.py
+++ b/django/db/migrations/operations/models.py
@@ -346,6 +346,7 @@ def reduce(self, operation, app_label):
                         },
                     ),
                 ]
+            elif isinstance(operation, AlterConstraint):
             elif isinstance(operation, RemoveConstraint):
                 options_constraints = [
                     constraint