diff --git a/django/db/migrations/operations/models.py b/django/db/migrations/operations/models.py
--- a/django/db/migrations/operations/models.py
+++ b/django/db/migrations/operations/models.py
@@ -347,6 +347,9 @@
                     ),
                 ]
             elif isinstance(operation, AlterConstraint):
+                options_constraints = [
+                    constraint
+                    for constraint in self.options.get("constraints", [])
             elif isinstance(operation, RemoveConstraint):
                 options_constraints = [
                     constraint
