diff --git a/django/db/migrations/operations/models.py b/django/db/migrations/operations/models.py
--- a/django/db/migrations/operations/models.py
+++ b/django/db/migrations/operations/models.py
@@ -354,6 +354,13 @@
                 ] + [operation.constraint]
                 return [
                     replace(
+                        self,
+                        options={
+                            **self.options,
+                            "constraints": options_constraints,
+                        },
+                    ),
+                ]
             elif isinstance(operation, RemoveConstraint):
                 options_constraints = [
                     constraint
