diff --git a/django/db/migrations/operations/models.py b/django/db/migrations/operations/models.py
--- a/django/db/migrations/operations/models.py
+++ b/django/db/migrations/operations/models.py
@@ -346,6 +346,21 @@ def reduce(self, operation, app_label):
                         },
                     ),
                 ]
+            elif isinstance(operation, AlterConstraint):
+                options_constraints = [
+                    constraint
+                    for constraint in self.options.get("constraints", [])
+                    if constraint.name != operation.name
+                ] + [operation.constraint]
+                return [
+                    replace(
+                        self,
+                        options={
+                            **self.options,
+                            "constraints": options_constraints,
+                        },
+                    ),
+                ]
             elif isinstance(operation, RemoveConstraint):
                 options_constraints = [
                     constraint