diff --git a/django/db/migrations/operations/fields.py b/django/db/migrations/operations/fields.py
--- a/django/db/migrations/operations/fields.py
+++ b/django/db/migrations/operations/fields.py
@@ -1,5 +1,5 @@
 from django.db.migrations.utils import field_references
-from django.db.models import NOT_PROVIDED
+from django.db.models import NOT_PROVIDED, Model
 from django.utils.copy import replace
 from django.utils.functional import cached_property
 
@@ -56,7 +56,10 @@
                 ):
                     return True
                 elif self.field.generated and any(
-                    for field_name, *_ in Model.
+                    field_name == name
+                    for field_name, *_ in Model._get_expr_references(
+                        self.field.expression
+                    )
                 ):
                     return True
         # Check if this operation remotely references the field.
