diff --git a/lib/matplotlib/axes/_axes.py b/lib/matplotlib/axes/_axes.py
--- a/lib/matplotlib/axes/_axes.py
+++ b/lib/matplotlib/axes/_axes.py
@@ -2992,7 +2992,15 @@
 
         vertices = []
         y0, dy = yrange
-        y0, y1 = self.convert_yunits((y0, y0 + dy))
+
+        _api.check_in_list(['bottom', 'center', 'top'], align=align)
+        if align == "bottom":
+            y0, y1 = self.convert_yunits((y0, y0 + dy))
+        elif align == "center":
+            y0, y1 = self.convert_yunits((y0 - dy/2, y0 + dy/2))
+        else:
+            y0, y1 = self.convert_yunits((y0 - dy, y0))
+
         for xr in xranges:  # convert the absolute values, not the x and dx
             try:
                 x0, dx = xr
