diff --git a/django/db/models/fields/related.py b/django/db/models/fields/related.py
--- a/django/db/models/fields/related.py
+++ b/django/db/models/fields/related.py
@@ -626,9 +626,10 @@ def _check_to_fields_composite_pk(self):
                 if isinstance(field, CompositePrimaryKey):
                     errors.append(
                         checks.Error(
-                            "Field defines a relation to the CompositePrimaryKey of "
-                            f"model {self.remote_field.model._meta.object_name!r} "
-                            "which is not supported.",
+                            "Field defines a relation involving model "
+                            f"{self.remote_field.model._meta.object_name!r} which has "
+                            "a CompositePrimaryKey and such relations are not "
+                            "supported.",
                             obj=self,
                             id="fields.E347",
                         )
@@ -1538,20 +1539,24 @@ def _check_relationship_model(self, from_model=None, **kwargs):
                 to_model_name = to_model
             else:
                 to_model_name = to_model._meta.object_name
-            if (
-                self.remote_field.through_fields is None
-                and not isinstance(to_model, str)
-                and isinstance(to_model._meta.pk, CompositePrimaryKey)
+            if self.remote_field.through_fields is None and not isinstance(
+                to_model, str
             ):
-                errors.append(
-                    checks.Error(
-                        "Field defines a relation to the CompositePrimaryKey of model "
-                        f"{self.remote_field.model._meta.object_name!r} which is not "
-                        "supported.",
-                        obj=self,
-                        id="fields.E347",
+                model_name = None
+                if isinstance(to_model._meta.pk, CompositePrimaryKey):
+                    model_name = self.remote_field.model._meta.object_name
+                elif isinstance(from_model._meta.pk, CompositePrimaryKey):
+                    model_name = from_model_name
+                if model_name:
+                    errors.append(
+                        checks.Error(
+                            f"Field defines a relation involving model {model_name!r} "
+                            "which has a CompositePrimaryKey and such relations are "
+                            "not supported.",
+                            obj=self,
+                            id="fields.E347",
+                        )
                     )
-                )
             relationship_model_name = self.remote_field.through._meta.object_name
             self_referential = from_model == to_model
             # Count foreign keys in intermediate model