diff --git a/xarray/core/indexing.py b/xarray/core/indexing.py
--- a/xarray/core/indexing.py
+++ b/xarray/core/indexing.py
@@ -341,6 +341,20 @@
     >>> slice_slice_by_array(slice(None, None, -1), np.array([2, 4, 7]), 20)
     array([17, 15, 12])
     """
+    # to get a concrete slice, limited to the size of the array
+    normalized_slice = normalize_slice(old_slice, size)
+
+    size_after_slice = len(range(*normalized_slice.indices(size)))
+    normalized_array = normalize_array(array, size_after_slice)
+
+    new_indexer = normalized_array * normalized_slice.step + normalized_slice.start
+
+    if np.any(new_indexer >= size):
+        raise IndexError("indices out of bounds")  # TODO: more helpful error message
+
+    return new_indexer
+
+
 def _index_indexer_1d(old_indexer, applied_indexer, size: int):
     if isinstance(applied_indexer, slice) and applied_indexer == slice(None):
         # shortcut for the usual case